// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/record/aws/EndPoint.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndPoint = void 0;
const ClassFor_1 = require("../../../../data/ClassFor");
const JsonApiRecordBase_1 = require("../../../../jsonApi/JsonApiRecordBase");
const HashCodeUtil_1 = require("../../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._aBooleanIsMutated = "_aBooleanIsMutated";
        this._aNumberIsMutated = "_aNumberIsMutated";
        this._aStringIsMutated = "_aStringIsMutated";
        this._routesIsMutated = "_routesIsMutated";
        this.aBoolean = "aBoolean";
        this.aNumber = "aNumber";
        this.aString = "aString";
        this.jsonType = "jsonType";
        this.routes = "routes";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._aBooleanIsMutated = "_aBooleanIsMutated";
        this._aNumberIsMutated = "_aNumberIsMutated";
        this._aStringIsMutated = "_aStringIsMutated";
        this._routesIsMutated = "_routesIsMutated";
        this.aBoolean = "aBoolean";
        this.aNumber = "aNumber";
        this.aString = "aString";
        this.jsonType = "jsonType";
        this.routes = "routes";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.record.aws.EndPoint";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class EndPoint extends JsonApiRecordBase_1.JsonApiRecordBase {
    constructor(p) {
        super(p);
        this._aBooleanIsMutated = false;
        this._aNumberIsMutated = false;
        this._aStringIsMutated = false;
        this._routesIsMutated = false;
        this.aBoolean = false;
        this.aNumber = 0;
        this.aString = null;
        this.jsonType = "com.salesstryke.entity.classes.record.aws.EndPoint";
        this.routes = Array();
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.record.aws.EndPoint");
    }
    /**
     * The name of an array of strings value.
     */
    addRoute(item) {
        this.routes.push(item);
        return (this);
    }
    /**
     *
     * @returns boolean
     */
    _getIsABooleanMutated() {
        return (this._aBooleanIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsANumberMutated() {
        return (this._aNumberIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAStringMutated() {
        return (this._aStringIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRoutesMutated() {
        return (this._routesIsMutated);
    }
    /**
     * The name of a boolean value.
     * @returns boolean
     */
    getABoolean() {
        return (this.aBoolean);
    }
    /**
     * The name of a number value.
     * @returns number
     */
    getANumber() {
        return (this.aNumber);
    }
    /**
     * The name of a string value.
     * @returns string | null
     */
    getAString() {
        return (this.aString);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The name of an array of strings value.
     * @returns Array<string>
     */
    getRoutes() {
        return (this.routes);
    }
    /**
* ```
     *
* ```
     */
    _setIsABooleanMutated(_aBooleanIsMutated) {
        this._aBooleanIsMutated = _aBooleanIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsANumberMutated(_aNumberIsMutated) {
        this._aNumberIsMutated = _aNumberIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAStringMutated(_aStringIsMutated) {
        this._aStringIsMutated = _aStringIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRoutesMutated(_routesIsMutated) {
        this._routesIsMutated = _routesIsMutated;
        return (this);
    }
    /**
* ```
     * The name of a boolean value.
* ```
     */
    setABoolean(aBoolean) {
        if (this.aBoolean !== aBoolean) {
            this._setIsABooleanMutated(true);
        }
        this.aBoolean = aBoolean;
        return (this);
    }
    /**
* ```
     * The name of a number value.
* ```
     */
    setANumber(aNumber) {
        if (this.aNumber !== aNumber) {
            this._setIsANumberMutated(true);
        }
        this.aNumber = aNumber;
        return (this);
    }
    /**
* ```
     * The name of a string value.
* ```
     */
    setAString(aString) {
        if (this.aString !== aString) {
            this._setIsAStringMutated(true);
        }
        this.aString = aString;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The name of an array of strings value.
* ```
     */
    setRoutes(routes) {
        if (this.routes !== routes) {
            this._setIsRoutesMutated(true);
        }
        this.routes = routes;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: 'aBoolean',
            value: this.aBoolean
        });
        this._setCurrentState2({
            attributeName: 'aNumber',
            value: this.aNumber
        });
        this._setCurrentState2({
            attributeName: 'aString',
            value: this.aString
        });
        this._setCurrentState2({
            attributeName: 'routes',
            value: this.routes
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: 'aBoolean',
            value: this.aBoolean
        });
        this._setInitializedState2({
            attributeName: 'aNumber',
            value: this.aNumber
        });
        this._setInitializedState2({
            attributeName: 'aString',
            value: this.aString
        });
        this._setInitializedState2({
            attributeName: 'routes',
            value: this.routes
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.aBoolean);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.aNumber);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.aString);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.routes);
        return (hashCode_n);
    }
}
exports.EndPoint = EndPoint;
EndPoint.metaDataForEndPoint = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.record.aws.EndPoint", EndPoint);
