// @salesstryke/mobile-api/src/com/salesstryke/entity/model/record/aws/EndPointEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndPointEntityModel = void 0;
const EntityBase_1 = require("../../../../data/entity/EntityBase");
const EntityType_1 = require("../../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../../utils/DataModelAttributeHelperUtil");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.record.aws.EndPoint';
class EndPointEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        let entityModel = this.getEntityModel();
        entityModel
            .setEntityType(EntityType_1.EntityType.RECORD)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('8f0fe440-ee38-4f21-8fc7-a0369cced799');
        // boolean
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRecordBoolean('aBoolean', 'b6f7381e-e409-400b-bb3b-76c6d1c84193', 'The name of a boolean value.'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'dedb2dd9-324b-43f4-987a-6ef50595f4c0'));
        // number
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRecordNumber("aNumber", 'e3a48730-b83c-4b5d-9736-4637a4ebe6cb', // uuid
        "The name of a number value."));
        // string
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRecordString('aString', '4885a21c-76f3-4712-ac7b-91ee1b0a4daf', // uuid
        'The name of a string value.'));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRecordArrayOfStrings({
            attributeName: 'routes',
            comment: 'The name of an array of strings value.',
            typeScriptAddItemMethodName: 'addRoute',
            uuid: '69d3c41b-657f-4d9e-8bd6-db4c8b0d3ced'
        }));
    }
    static getInstance() {
        return (EndPointEntityModel.INSTANCE);
    }
}
exports.EndPointEntityModel = EndPointEntityModel;
EndPointEntityModel.INSTANCE = new EndPointEntityModel();
