// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/CartProductRule.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CartProductRule = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistenceType_1 = require("../../../data/PersistenceType");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._activeEndDateIsMutated = "_activeEndDateIsMutated";
        this._activeStartDateIsMutated = "_activeStartDateIsMutated";
        this._byRuleQuantityIsMutated = "_byRuleQuantityIsMutated";
        this._cartItemIsMutated = "_cartItemIsMutated";
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._productsIsMutated = "_productsIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._ruleOrderIndexIsMutated = "_ruleOrderIndexIsMutated";
        this._rulesBuilderjsonIsMutated = "_rulesBuilderjsonIsMutated";
        this._rulesIsMutated = "_rulesIsMutated";
        this.activeEndDate = "activeEndDate";
        this.activeStartDate = "activeStartDate";
        this.byRuleQuantity = "byRuleQuantity";
        this.cartItem = "cartItem";
        this.createdDate = "createdDate";
        this.description = "description";
        this.id = "id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.products = "products";
        this.ruleOrderIndex = "ruleOrderIndex";
        this.rules = "rules";
        this.rulesBuilderjson = "rulesBuilderjson";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._activeEndDateIsMutated = "_activeEndDateIsMutated";
        this._activeStartDateIsMutated = "_activeStartDateIsMutated";
        this._byRuleQuantityIsMutated = "_byRuleQuantityIsMutated";
        this._cartItemIsMutated = "_cartItemIsMutated";
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._productsIsMutated = "_productsIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._ruleOrderIndexIsMutated = "_ruleOrderIndexIsMutated";
        this._rulesBuilderjsonIsMutated = "_rulesBuilderjsonIsMutated";
        this._rulesIsMutated = "_rulesIsMutated";
        this.activeEndDate = "activeEndDate";
        this.activeStartDate = "activeStartDate";
        this.byRuleQuantity = "byRuleQuantity";
        this.cartItem = "cartItem";
        this.createdDate = "createdDate";
        this.description = "description";
        this._id = "_id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.products = "products";
        this.ruleOrderIndex = "ruleOrderIndex";
        this.rules = "rules";
        this.rulesBuilderjson = "rulesBuilderjson";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.CartProductRule";
        this.mongDataModeName = "CartProductRule";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class CartProductRule extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._activeEndDateIsMutated = false;
        this._activeStartDateIsMutated = false;
        this._byRuleQuantityIsMutated = false;
        this._cartItemIsMutated = false;
        this._descriptionIsMutated = false;
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._nameIsMutated = false;
        this._productsIsMutated = false;
        this._ruleOrderIndexIsMutated = false;
        this._rulesBuilderjsonIsMutated = false;
        this._rulesIsMutated = false;
        this.activeEndDate = null;
        this.activeStartDate = null;
        this.byRuleQuantity = 0;
        this.cartItem = null;
        this.description = null;
        this.id = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.CartProductRule";
        this.name = null;
        this.products = new Array();
        this.ruleOrderIndex = 0;
        this.rules = null;
        this.rulesBuilderjson = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._activeEndDateIsMutated = false;
        this._activeStartDateIsMutated = false;
        this._byRuleQuantityIsMutated = false;
        this._cartItemIsMutated = false;
        this._descriptionIsMutated = false;
        this._nameIsMutated = false;
        this._productsIsMutated = false;
        this._ruleOrderIndexIsMutated = false;
        this._rulesIsMutated = false;
        this._rulesBuilderjsonIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.CartProductRule");
    }
    /**
     * List of product.
     *
     * Default PersistenceType : PersistenceType.ATTACH
     */
    addProduct(product, persistanceType = PersistenceType_1.PersistenceType.ATTACH) {
        if (this._getIsTrackingMutations()) {
            product._trackMutations();
        }
        product._setRelationshipPersistableType(persistanceType.getId());
        this.products.push(product);
        return (this);
    }
    /**
     * List of product.
     * If the Entity ID is null, will not be added.
     * Default PersistenceType : PersistenceType.DETACH
     */
    removeProduct(product, persistanceType = PersistenceType_1.PersistenceType.DETACH) {
        if (this._getIsTrackingMutations()) {
            product._trackMutations();
        }
        if (persistanceType === PersistenceType_1.PersistenceType.NONE) {
            let index = this.products.findIndex(obj => obj === product);
            if (index !== -1) {
                this.products.splice(index, 1);
            }
            product._setRelationshipPersistableType(PersistenceType_1.PersistenceType.NONE.getId());
            return (this);
        }
        product._setRelationshipPersistableType(persistanceType.getId());
        this.products.push(product);
        return (this);
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveEndDateMutated() {
        return (this._activeEndDateIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveStartDateMutated() {
        return (this._activeStartDateIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsByRuleQuantityMutated() {
        return (this._byRuleQuantityIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsCartItemMutated() {
        return (this._cartItemIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDescriptionMutated() {
        return (this._descriptionIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductsMutated() {
        return (this._productsIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRuleOrderIndexMutated() {
        return (this._ruleOrderIndexIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRulesBuilderjsonMutated() {
        return (this._rulesBuilderjsonIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRulesMutated() {
        return (this._rulesIsMutated);
    }
    /**
     * Active end date.
     * @returns Date | null
     */
    getActiveEndDate() {
        return (this.activeEndDate);
    }
    /**
     * Active start date.
     * @returns Date | null
     */
    getActiveStartDate() {
        return (this.activeStartDate);
    }
    /**
     * quantity by rule.
     * @returns number
     */
    getByRuleQuantity() {
        return (this.byRuleQuantity);
    }
    /**
     * The cart item.
     * @returns CartItem | null
     */
    getCartItem() {
        return (this.cartItem);
    }
    /**
     * The description.
     * @returns string | null
     */
    getDescription() {
        return (this.description);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The name.
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
     * List of product.
     * @returns Array<Product>
     */
    getProducts() {
        return (this.products);
    }
    /**
     * Index of the rule
     * @returns number
     */
    getRuleOrderIndex() {
        return (this.ruleOrderIndex);
    }
    /**
     * Rules.
     * @returns string | null
     */
    getRules() {
        return (this.rules);
    }
    /**
     * Rules builder json.
     * @returns string | null
     */
    getRulesBuilderjson() {
        return (this.rulesBuilderjson);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveEndDateMutated(_activeEndDateIsMutated) {
        this._activeEndDateIsMutated = _activeEndDateIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveStartDateMutated(_activeStartDateIsMutated) {
        this._activeStartDateIsMutated = _activeStartDateIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsByRuleQuantityMutated(_byRuleQuantityIsMutated) {
        this._byRuleQuantityIsMutated = _byRuleQuantityIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsCartItemMutated(_cartItemIsMutated) {
        this._cartItemIsMutated = _cartItemIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDescriptionMutated(_descriptionIsMutated) {
        this._descriptionIsMutated = _descriptionIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductsMutated(_productsIsMutated) {
        this._productsIsMutated = _productsIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRuleOrderIndexMutated(_ruleOrderIndexIsMutated) {
        this._ruleOrderIndexIsMutated = _ruleOrderIndexIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRulesBuilderjsonMutated(_rulesBuilderjsonIsMutated) {
        this._rulesBuilderjsonIsMutated = _rulesBuilderjsonIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRulesMutated(_rulesIsMutated) {
        this._rulesIsMutated = _rulesIsMutated;
        return (this);
    }
    /**
* ```
     * Active end date.
* ```
     */
    setActiveEndDate(activeEndDate) {
        if (!this._getIsMutateSuppressed() && this.activeEndDate !== activeEndDate) {
            this._setIsActiveEndDateMutated(true);
        }
        this.activeEndDate = activeEndDate;
        return (this);
    }
    /**
* ```
     * Active start date.
* ```
     */
    setActiveStartDate(activeStartDate) {
        if (!this._getIsMutateSuppressed() && this.activeStartDate !== activeStartDate) {
            this._setIsActiveStartDateMutated(true);
        }
        this.activeStartDate = activeStartDate;
        return (this);
    }
    /**
* ```
     * quantity by rule.
* ```
     */
    setByRuleQuantity(byRuleQuantity) {
        if (!this._getIsMutateSuppressed() && this.byRuleQuantity !== byRuleQuantity) {
            this._setIsByRuleQuantityMutated(true);
        }
        this.byRuleQuantity = byRuleQuantity;
        return (this);
    }
    /**
* ```
     * The cart item.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setCartItem(cartItem, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.cartItem !== cartItem) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(cartItem, persistanceType);
            this._setIsCartItemMutated(true);
        }
        this.cartItem = cartItem;
        return (this);
    }
    /**
* ```
     * The description.
* ```
     */
    setDescription(description) {
        if (!this._getIsMutateSuppressed() && this.description !== description) {
            this._setIsDescriptionMutated(true);
        }
        this.description = description;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The name.
* ```
     */
    setName(name) {
        if (!this._getIsMutateSuppressed() && this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    /**
* ```
     * List of product.
* ```
     */
    setProducts(products) {
        if (!this._getIsMutateSuppressed() && this.products !== products) {
            this._setIsProductsMutated(true);
        }
        this.products = products;
        return (this);
    }
    /**
* ```
     * Index of the rule
* ```
     */
    setRuleOrderIndex(ruleOrderIndex) {
        if (!this._getIsMutateSuppressed() && this.ruleOrderIndex !== ruleOrderIndex) {
            this._setIsRuleOrderIndexMutated(true);
        }
        this.ruleOrderIndex = ruleOrderIndex;
        return (this);
    }
    /**
* ```
     * Rules.
* ```
     */
    setRules(rules) {
        if (!this._getIsMutateSuppressed() && this.rules !== rules) {
            this._setIsRulesMutated(true);
        }
        this.rules = rules;
        return (this);
    }
    /**
* ```
     * Rules builder json.
* ```
     */
    setRulesBuilderjson(rulesBuilderjson) {
        if (!this._getIsMutateSuppressed() && this.rulesBuilderjson !== rulesBuilderjson) {
            this._setIsRulesBuilderjsonMutated(true);
        }
        this.rulesBuilderjson = rulesBuilderjson;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'activeEndDate',
            value: this.activeEndDate
        });
        this._setCurrentState2({
            attributeName: 'activeStartDate',
            value: this.activeStartDate
        });
        this._setCurrentState2({
            attributeName: 'byRuleQuantity',
            value: this.byRuleQuantity
        });
        this._setCurrentState2({
            attributeName: 'cartItem',
            value: this.cartItem
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'description',
            value: this.description
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._setCurrentState2({
            attributeName: 'products',
            value: this.products
        });
        this._setCurrentState2({
            attributeName: 'ruleOrderIndex',
            value: this.ruleOrderIndex
        });
        this._setCurrentState2({
            attributeName: 'rules',
            value: this.rules
        });
        this._setCurrentState2({
            attributeName: 'rulesBuilderjson',
            value: this.rulesBuilderjson
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'activeEndDate',
            value: this.activeEndDate
        });
        this._setInitializedState2({
            attributeName: 'activeStartDate',
            value: this.activeStartDate
        });
        this._setInitializedState2({
            attributeName: 'byRuleQuantity',
            value: this.byRuleQuantity
        });
        this._setInitializedState2({
            attributeName: 'cartItem',
            value: this.cartItem
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'description',
            value: this.description
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        this._setInitializedState2({
            attributeName: 'products',
            value: this.products
        });
        this._setInitializedState2({
            attributeName: 'ruleOrderIndex',
            value: this.ruleOrderIndex
        });
        this._setInitializedState2({
            attributeName: 'rules',
            value: this.rules
        });
        this._setInitializedState2({
            attributeName: 'rulesBuilderjson',
            value: this.rulesBuilderjson
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.activeEndDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.activeStartDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.byRuleQuantity);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.cartItem);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.description);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.products);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.ruleOrderIndex);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.rules);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.rulesBuilderjson);
        return (hashCode_n);
    }
}
exports.CartProductRule = CartProductRule;
CartProductRule.metaDataForCartProductRule = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.CartProductRule", CartProductRule);
