// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/CnsCampaignListItem.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsCampaignListItem = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._awsDataIsMutated = "_awsDataIsMutated";
        this._cnsCampaignListIsMutated = "_cnsCampaignListIsMutated";
        this._customerIdIsMutated = "_customerIdIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this.awsData = "awsData";
        this.cnsCampaignList = "cnsCampaignList";
        this.createdDate = "createdDate";
        this.customerId = "customerId";
        this.id = "id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._awsDataIsMutated = "_awsDataIsMutated";
        this._cnsCampaignListIsMutated = "_cnsCampaignListIsMutated";
        this._customerIdIsMutated = "_customerIdIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this.awsData = "awsData";
        this.cnsCampaignList = "cnsCampaignList";
        this.createdDate = "createdDate";
        this.customerId = "customerId";
        this._id = "_id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.CnsCampaignListItem";
        this.mongDataModeName = "CnsCampaignListItem";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class CnsCampaignListItem extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._awsDataIsMutated = false;
        this._cnsCampaignListIsMutated = false;
        this._customerIdIsMutated = false;
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this.awsData = null;
        this.cnsCampaignList = null;
        this.customerId = null;
        this.id = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.CnsCampaignListItem";
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._awsDataIsMutated = false;
        this._cnsCampaignListIsMutated = false;
        this._customerIdIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.CnsCampaignListItem");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsAwsDataMutated() {
        return (this._awsDataIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsCnsCampaignListMutated() {
        return (this._cnsCampaignListIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsCustomerIdMutated() {
        return (this._customerIdIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     * AwsData for this item, such as recorded id in aws for the customerId.
     * @returns string | null
     */
    getAwsData() {
        return (this.awsData);
    }
    /**
     * The campaign list this item belong to.
     * @returns CnsCampaignList | null
     */
    getCnsCampaignList() {
        return (this.cnsCampaignList);
    }
    /**
     * The CustomerId.
     * @returns string | null
     */
    getCustomerId() {
        return (this.customerId);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAwsDataMutated(_awsDataIsMutated) {
        this._awsDataIsMutated = _awsDataIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsCnsCampaignListMutated(_cnsCampaignListIsMutated) {
        this._cnsCampaignListIsMutated = _cnsCampaignListIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsCustomerIdMutated(_customerIdIsMutated) {
        this._customerIdIsMutated = _customerIdIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     * AwsData for this item, such as recorded id in aws for the customerId.
* ```
     */
    setAwsData(awsData) {
        if (!this._getIsMutateSuppressed() && this.awsData !== awsData) {
            this._setIsAwsDataMutated(true);
        }
        this.awsData = awsData;
        return (this);
    }
    /**
* ```
     * The campaign list this item belong to.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setCnsCampaignList(cnsCampaignList, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.cnsCampaignList !== cnsCampaignList) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(cnsCampaignList, persistanceType);
            this._setIsCnsCampaignListMutated(true);
        }
        this.cnsCampaignList = cnsCampaignList;
        return (this);
    }
    /**
* ```
     * The CustomerId.
* ```
     */
    setCustomerId(customerId) {
        if (!this._getIsMutateSuppressed() && this.customerId !== customerId) {
            this._setIsCustomerIdMutated(true);
        }
        this.customerId = customerId;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'awsData',
            value: this.awsData
        });
        this._setCurrentState2({
            attributeName: 'cnsCampaignList',
            value: this.cnsCampaignList
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'customerId',
            value: this.customerId
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'awsData',
            value: this.awsData
        });
        this._setInitializedState2({
            attributeName: 'cnsCampaignList',
            value: this.cnsCampaignList
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'customerId',
            value: this.customerId
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.awsData);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.cnsCampaignList);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.customerId);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        return (hashCode_n);
    }
}
exports.CnsCampaignListItem = CnsCampaignListItem;
CnsCampaignListItem.metaDataForCnsCampaignListItem = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.CnsCampaignListItem", CnsCampaignListItem);
