// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/CnsSync.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsSync = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._appCnsSyncStatusTypeIsMutated = "_appCnsSyncStatusTypeIsMutated";
        this._emailCnsSyncStatusTypeIsMutated = "_emailCnsSyncStatusTypeIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataFileUploadSyncIsMutated = "_importDataFileUploadSyncIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._organizationIsMutated = "_organizationIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._textCnsSyncStatusTypeIsMutated = "_textCnsSyncStatusTypeIsMutated";
        this.appCnsSyncStatusType = "appCnsSyncStatusType";
        this.createdDate = "createdDate";
        this.emailCnsSyncStatusType = "emailCnsSyncStatusType";
        this.id = "id";
        this.importDataFileUploadSync = "importDataFileUploadSync";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.organization = "organization";
        this.textCnsSyncStatusType = "textCnsSyncStatusType";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._appCnsSyncStatusTypeIsMutated = "_appCnsSyncStatusTypeIsMutated";
        this._emailCnsSyncStatusTypeIsMutated = "_emailCnsSyncStatusTypeIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataFileUploadSyncIsMutated = "_importDataFileUploadSyncIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._organizationIsMutated = "_organizationIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._textCnsSyncStatusTypeIsMutated = "_textCnsSyncStatusTypeIsMutated";
        this.appCnsSyncStatusType = "appCnsSyncStatusType";
        this.createdDate = "createdDate";
        this.emailCnsSyncStatusType = "emailCnsSyncStatusType";
        this._id = "_id";
        this.importDataFileUploadSync = "importDataFileUploadSync";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.organization = "organization";
        this.textCnsSyncStatusType = "textCnsSyncStatusType";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.CnsSync";
        this.mongDataModeName = "CnsSync";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class CnsSync extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._appCnsSyncStatusTypeIsMutated = false;
        this._emailCnsSyncStatusTypeIsMutated = false;
        this._idSetBySystem = false;
        this._importDataFileUploadSyncIsMutated = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._organizationIsMutated = false;
        this._textCnsSyncStatusTypeIsMutated = false;
        this.appCnsSyncStatusType = null;
        this.emailCnsSyncStatusType = null;
        this.id = null;
        this.importDataFileUploadSync = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.CnsSync";
        this.organization = null;
        this.textCnsSyncStatusType = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._appCnsSyncStatusTypeIsMutated = false;
        this._emailCnsSyncStatusTypeIsMutated = false;
        this._importDataFileUploadSyncIsMutated = false;
        this._organizationIsMutated = false;
        this._textCnsSyncStatusTypeIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.CnsSync");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsAppCnsSyncStatusTypeMutated() {
        return (this._appCnsSyncStatusTypeIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsEmailCnsSyncStatusTypeMutated() {
        return (this._emailCnsSyncStatusTypeIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportDataFileUploadSyncMutated() {
        return (this._importDataFileUploadSyncIsMutated);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsOrganizationMutated() {
        return (this._organizationIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsTextCnsSyncStatusTypeMutated() {
        return (this._textCnsSyncStatusTypeIsMutated);
    }
    /**
     * The APP CnsSyncStatusType.
     * @returns CnsSyncStatusType | null
     */
    getAppCnsSyncStatusType() {
        return (this.appCnsSyncStatusType);
    }
    /**
     * The EMAIL CnsSyncStatusType.
     * @returns CnsSyncStatusType | null
     */
    getEmailCnsSyncStatusType() {
        return (this.emailCnsSyncStatusType);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * The ImportDataFileUploadSync.
     * @returns ImportDataFileUpload | null
     */
    getImportDataFileUploadSync() {
        return (this.importDataFileUploadSync);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The organization.
     * @returns Organization | null
     */
    getOrganization() {
        return (this.organization);
    }
    /**
     * The EMAIL CnsSyncStatusType.
     * @returns CnsSyncStatusType | null
     */
    getTextCnsSyncStatusType() {
        return (this.textCnsSyncStatusType);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAppCnsSyncStatusTypeMutated(_appCnsSyncStatusTypeIsMutated) {
        this._appCnsSyncStatusTypeIsMutated = _appCnsSyncStatusTypeIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsEmailCnsSyncStatusTypeMutated(_emailCnsSyncStatusTypeIsMutated) {
        this._emailCnsSyncStatusTypeIsMutated = _emailCnsSyncStatusTypeIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportDataFileUploadSyncMutated(_importDataFileUploadSyncIsMutated) {
        this._importDataFileUploadSyncIsMutated = _importDataFileUploadSyncIsMutated;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsOrganizationMutated(_organizationIsMutated) {
        this._organizationIsMutated = _organizationIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsTextCnsSyncStatusTypeMutated(_textCnsSyncStatusTypeIsMutated) {
        this._textCnsSyncStatusTypeIsMutated = _textCnsSyncStatusTypeIsMutated;
        return (this);
    }
    /**
* ```
     * The APP CnsSyncStatusType.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setAppCnsSyncStatusType(appCnsSyncStatusType, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.appCnsSyncStatusType !== appCnsSyncStatusType) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(appCnsSyncStatusType, persistanceType);
            this._setIsAppCnsSyncStatusTypeMutated(true);
        }
        this.appCnsSyncStatusType = appCnsSyncStatusType;
        return (this);
    }
    /**
* ```
     * The EMAIL CnsSyncStatusType.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setEmailCnsSyncStatusType(emailCnsSyncStatusType, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.emailCnsSyncStatusType !== emailCnsSyncStatusType) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(emailCnsSyncStatusType, persistanceType);
            this._setIsEmailCnsSyncStatusTypeMutated(true);
        }
        this.emailCnsSyncStatusType = emailCnsSyncStatusType;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     * The ImportDataFileUploadSync.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setImportDataFileUploadSync(importDataFileUploadSync, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.importDataFileUploadSync !== importDataFileUploadSync) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(importDataFileUploadSync, persistanceType);
            this._setIsImportDataFileUploadSyncMutated(true);
        }
        this.importDataFileUploadSync = importDataFileUploadSync;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The organization.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setOrganization(organization, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.organization !== organization) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(organization, persistanceType);
            this._setIsOrganizationMutated(true);
        }
        this.organization = organization;
        return (this);
    }
    /**
* ```
     * The EMAIL CnsSyncStatusType.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setTextCnsSyncStatusType(textCnsSyncStatusType, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.textCnsSyncStatusType !== textCnsSyncStatusType) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(textCnsSyncStatusType, persistanceType);
            this._setIsTextCnsSyncStatusTypeMutated(true);
        }
        this.textCnsSyncStatusType = textCnsSyncStatusType;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'appCnsSyncStatusType',
            value: this.appCnsSyncStatusType
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'emailCnsSyncStatusType',
            value: this.emailCnsSyncStatusType
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'importDataFileUploadSync',
            value: this.importDataFileUploadSync
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setCurrentState2({
            attributeName: 'textCnsSyncStatusType',
            value: this.textCnsSyncStatusType
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'appCnsSyncStatusType',
            value: this.appCnsSyncStatusType
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'emailCnsSyncStatusType',
            value: this.emailCnsSyncStatusType
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'importDataFileUploadSync',
            value: this.importDataFileUploadSync
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setInitializedState2({
            attributeName: 'textCnsSyncStatusType',
            value: this.textCnsSyncStatusType
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.appCnsSyncStatusType);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.emailCnsSyncStatusType);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importDataFileUploadSync);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.organization);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.textCnsSyncStatusType);
        return (hashCode_n);
    }
}
exports.CnsSync = CnsSync;
CnsSync.metaDataForCnsSync = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.CnsSync", CnsSync);
