// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/DataArray.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataArray = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const EntityUtils_1 = require("../../../utils/entity/EntityUtils");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._booleanArrayIsMutated = "_booleanArrayIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._numberArrayIsMutated = "_numberArrayIsMutated";
        this._primitiveArrayIsMutated = "_primitiveArrayIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._stringArrayIsMutated = "_stringArrayIsMutated";
        this.booleanArray = "booleanArray";
        this.createdDate = "createdDate";
        this.id = "id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.numberArray = "numberArray";
        this.primitiveArray = "primitiveArray";
        this.stringArray = "stringArray";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._booleanArrayIsMutated = "_booleanArrayIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._numberArrayIsMutated = "_numberArrayIsMutated";
        this._primitiveArrayIsMutated = "_primitiveArrayIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._stringArrayIsMutated = "_stringArrayIsMutated";
        this.booleanArray = "booleanArray";
        this.createdDate = "createdDate";
        this._id = "_id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.numberArray = "numberArray";
        this.primitiveArray = "primitiveArray";
        this.stringArray = "stringArray";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.DataArray";
        this.mongDataModeName = "DataArray";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class DataArray extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._booleanArrayIsMutated = false;
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._numberArrayIsMutated = false;
        this._primitiveArrayIsMutated = false;
        this._stringArrayIsMutated = false;
        this.booleanArray = null;
        this.id = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.DataArray";
        this.numberArray = null;
        this.primitiveArray = null;
        this.stringArray = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._booleanArrayIsMutated = false;
        this._numberArrayIsMutated = false;
        this._primitiveArrayIsMutated = false;
        this._stringArrayIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.DataArray");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanArrayMutated() {
        return (this._booleanArrayIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberArrayMutated() {
        return (this._numberArrayIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsPrimitiveArrayMutated() {
        return (this._primitiveArrayIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringArrayMutated() {
        return (this._stringArrayIsMutated);
    }
    /**
     * Aarray of boolean values.
     * @returns TBooleanArray
     */
    getBooleanArray() {
        return EntityUtils_1.EntityUtils.jsonStringGetter({ value: this.booleanArray || '' });
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * Aarray of number values.
     * @returns TNumberArray
     */
    getNumberArray() {
        return EntityUtils_1.EntityUtils.jsonStringGetter({ value: this.numberArray || '' });
    }
    /**
     * Aarray of primitive values.
     * @returns TPrimitiveArray
     */
    getPrimitiveArray() {
        return EntityUtils_1.EntityUtils.jsonStringGetter({ value: this.primitiveArray || '' });
    }
    /**
     * Aarray of string values.
     * @returns TStringArray
     */
    getStringArray() {
        return EntityUtils_1.EntityUtils.jsonStringGetter({ value: this.stringArray || '' });
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanArrayMutated(_booleanArrayIsMutated) {
        this._booleanArrayIsMutated = _booleanArrayIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberArrayMutated(_numberArrayIsMutated) {
        this._numberArrayIsMutated = _numberArrayIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsPrimitiveArrayMutated(_primitiveArrayIsMutated) {
        this._primitiveArrayIsMutated = _primitiveArrayIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringArrayMutated(_stringArrayIsMutated) {
        this._stringArrayIsMutated = _stringArrayIsMutated;
        return (this);
    }
    /**
* ```
     * Aarray of boolean values.
* ```
     */
    setBooleanArray(booleanArray) {
        if (!this._getIsMutateSuppressed() && this.booleanArray !== EntityUtils_1.EntityUtils.jsonStringSetter({ value: booleanArray })) {
            this._setIsBooleanArrayMutated(true);
        }
        this.booleanArray = EntityUtils_1.EntityUtils.jsonStringSetter({ value: booleanArray });
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * Aarray of number values.
* ```
     */
    setNumberArray(numberArray) {
        if (!this._getIsMutateSuppressed() && this.numberArray !== EntityUtils_1.EntityUtils.jsonStringSetter({ value: numberArray })) {
            this._setIsNumberArrayMutated(true);
        }
        this.numberArray = EntityUtils_1.EntityUtils.jsonStringSetter({ value: numberArray });
        return (this);
    }
    /**
* ```
     * Aarray of primitive values.
* ```
     */
    setPrimitiveArray(primitiveArray) {
        if (!this._getIsMutateSuppressed() && this.primitiveArray !== EntityUtils_1.EntityUtils.jsonStringSetter({ value: primitiveArray })) {
            this._setIsPrimitiveArrayMutated(true);
        }
        this.primitiveArray = EntityUtils_1.EntityUtils.jsonStringSetter({ value: primitiveArray });
        return (this);
    }
    /**
* ```
     * Aarray of string values.
* ```
     */
    setStringArray(stringArray) {
        if (!this._getIsMutateSuppressed() && this.stringArray !== EntityUtils_1.EntityUtils.jsonStringSetter({ value: stringArray })) {
            this._setIsStringArrayMutated(true);
        }
        this.stringArray = EntityUtils_1.EntityUtils.jsonStringSetter({ value: stringArray });
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'booleanArray',
            value: this.booleanArray
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'numberArray',
            value: this.numberArray
        });
        this._setCurrentState2({
            attributeName: 'primitiveArray',
            value: this.primitiveArray
        });
        this._setCurrentState2({
            attributeName: 'stringArray',
            value: this.stringArray
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'booleanArray',
            value: this.booleanArray
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'numberArray',
            value: this.numberArray
        });
        this._setInitializedState2({
            attributeName: 'primitiveArray',
            value: this.primitiveArray
        });
        this._setInitializedState2({
            attributeName: 'stringArray',
            value: this.stringArray
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanArray);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberArray);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.primitiveArray);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringArray);
        return (hashCode_n);
    }
}
exports.DataArray = DataArray;
DataArray.metaDataForDataArray = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.DataArray", DataArray);
