// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/ImportDataFileUpload.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataFileUpload = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._entityState = "_entityState";
        this._fileNameIsMutated = "_fileNameIsMutated";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataMapIsMutated = "_importDataMapIsMutated";
        this._importStatusIsMutated = "_importStatusIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._presignedUrlIsMutated = "_presignedUrlIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._s3BucketNameIsMutated = "_s3BucketNameIsMutated";
        this.createdDate = "createdDate";
        this.fileName = "fileName";
        this.id = "id";
        this.importDataMap = "importDataMap";
        this.importStatus = "importStatus";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.organization = "organization";
        this.presignedUrl = "presignedUrl";
        this.s3BucketName = "s3BucketName";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._entityState = "_entityState";
        this._fileNameIsMutated = "_fileNameIsMutated";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataMapIsMutated = "_importDataMapIsMutated";
        this._importStatusIsMutated = "_importStatusIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._presignedUrlIsMutated = "_presignedUrlIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._s3BucketNameIsMutated = "_s3BucketNameIsMutated";
        this.createdDate = "createdDate";
        this.fileName = "fileName";
        this._id = "_id";
        this.importDataMap = "importDataMap";
        this.importStatus = "importStatus";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.organization = "organization";
        this.presignedUrl = "presignedUrl";
        this.s3BucketName = "s3BucketName";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.ImportDataFileUpload";
        this.mongDataModeName = "ImportDataFileUpload";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class ImportDataFileUpload extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._fileNameIsMutated = false;
        this._idSetBySystem = false;
        this._importDataMapIsMutated = false;
        this._importStatusIsMutated = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._nameIsMutated = false;
        this._organizationIsMutated = false;
        this._presignedUrlIsMutated = false;
        this._s3BucketNameIsMutated = false;
        this.fileName = null;
        this.id = null;
        this.importDataMap = null;
        this.importStatus = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.ImportDataFileUpload";
        this.name = null;
        this.organization = null;
        this.presignedUrl = null;
        this.s3BucketName = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._fileNameIsMutated = false;
        this._importDataMapIsMutated = false;
        this._importStatusIsMutated = false;
        this._nameIsMutated = false;
        this._organizationIsMutated = false;
        this._presignedUrlIsMutated = false;
        this._s3BucketNameIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.ImportDataFileUpload");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsFileNameMutated() {
        return (this._fileNameIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportDataMapMutated() {
        return (this._importDataMapIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportStatusMutated() {
        return (this._importStatusIsMutated);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsOrganizationMutated() {
        return (this._organizationIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsPresignedUrlMutated() {
        return (this._presignedUrlIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsS3BucketNameMutated() {
        return (this._s3BucketNameIsMutated);
    }
    /**
     * The import file name.
     * @returns string | null
     */
    getFileName() {
        return (this.fileName);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * The ImportDataMap for the data.
     * @returns ImportDataMap | null
     */
    getImportDataMap() {
        return (this.importDataMap);
    }
    /**
     * The import status.
     * @returns string | null
     */
    getImportStatus() {
        return (this.importStatus);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The original file name.
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
     * The organization.
     * @returns Organization | null
     */
    getOrganization() {
        return (this.organization);
    }
    /**
     * The pre signed url for s3.
     * @returns string | null
     */
    getPresignedUrl() {
        return (this.presignedUrl);
    }
    /**
     * The S3 Bucket name.
     * @returns string | null
     */
    getS3BucketName() {
        return (this.s3BucketName);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsFileNameMutated(_fileNameIsMutated) {
        this._fileNameIsMutated = _fileNameIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportDataMapMutated(_importDataMapIsMutated) {
        this._importDataMapIsMutated = _importDataMapIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportStatusMutated(_importStatusIsMutated) {
        this._importStatusIsMutated = _importStatusIsMutated;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsOrganizationMutated(_organizationIsMutated) {
        this._organizationIsMutated = _organizationIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsPresignedUrlMutated(_presignedUrlIsMutated) {
        this._presignedUrlIsMutated = _presignedUrlIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsS3BucketNameMutated(_s3BucketNameIsMutated) {
        this._s3BucketNameIsMutated = _s3BucketNameIsMutated;
        return (this);
    }
    /**
* ```
     * The import file name.
* ```
     */
    setFileName(fileName) {
        if (!this._getIsMutateSuppressed() && this.fileName !== fileName) {
            this._setIsFileNameMutated(true);
        }
        this.fileName = fileName;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     * The ImportDataMap for the data.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setImportDataMap(importDataMap, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.importDataMap !== importDataMap) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(importDataMap, persistanceType);
            this._setIsImportDataMapMutated(true);
        }
        this.importDataMap = importDataMap;
        return (this);
    }
    /**
* ```
     * The import status.
* ```
     */
    setImportStatus(importStatus) {
        if (!this._getIsMutateSuppressed() && this.importStatus !== importStatus) {
            this._setIsImportStatusMutated(true);
        }
        this.importStatus = importStatus;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The original file name.
* ```
     */
    setName(name) {
        if (!this._getIsMutateSuppressed() && this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    /**
* ```
     * The organization.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setOrganization(organization, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.organization !== organization) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(organization, persistanceType);
            this._setIsOrganizationMutated(true);
        }
        this.organization = organization;
        return (this);
    }
    /**
* ```
     * The pre signed url for s3.
* ```
     */
    setPresignedUrl(presignedUrl) {
        if (!this._getIsMutateSuppressed() && this.presignedUrl !== presignedUrl) {
            this._setIsPresignedUrlMutated(true);
        }
        this.presignedUrl = presignedUrl;
        return (this);
    }
    /**
* ```
     * The S3 Bucket name.
* ```
     */
    setS3BucketName(s3BucketName) {
        if (!this._getIsMutateSuppressed() && this.s3BucketName !== s3BucketName) {
            this._setIsS3BucketNameMutated(true);
        }
        this.s3BucketName = s3BucketName;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'fileName',
            value: this.fileName
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'importDataMap',
            value: this.importDataMap
        });
        this._setCurrentState2({
            attributeName: 'importStatus',
            value: this.importStatus
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._setCurrentState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setCurrentState2({
            attributeName: 'presignedUrl',
            value: this.presignedUrl
        });
        this._setCurrentState2({
            attributeName: 's3BucketName',
            value: this.s3BucketName
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'fileName',
            value: this.fileName
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'importDataMap',
            value: this.importDataMap
        });
        this._setInitializedState2({
            attributeName: 'importStatus',
            value: this.importStatus
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        this._setInitializedState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setInitializedState2({
            attributeName: 'presignedUrl',
            value: this.presignedUrl
        });
        this._setInitializedState2({
            attributeName: 's3BucketName',
            value: this.s3BucketName
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.fileName);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importDataMap);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importStatus);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.organization);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.presignedUrl);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.s3BucketName);
        return (hashCode_n);
    }
}
exports.ImportDataFileUpload = ImportDataFileUpload;
ImportDataFileUpload.metaDataForImportDataFileUpload = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.ImportDataFileUpload", ImportDataFileUpload);
