// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/ImportData.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportData = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._booleanIndexedVar1IsMutated = "_booleanIndexedVar1IsMutated";
        this._booleanIndexedVar2IsMutated = "_booleanIndexedVar2IsMutated";
        this._booleanIndexedVar3IsMutated = "_booleanIndexedVar3IsMutated";
        this._booleanIndexedVar4IsMutated = "_booleanIndexedVar4IsMutated";
        this._booleanIndexedVar5IsMutated = "_booleanIndexedVar5IsMutated";
        this._booleanIndexedVar6IsMutated = "_booleanIndexedVar6IsMutated";
        this._booleanIndexedVar7IsMutated = "_booleanIndexedVar7IsMutated";
        this._booleanIndexedVar8IsMutated = "_booleanIndexedVar8IsMutated";
        this._booleanVar10IsMutated = "_booleanVar10IsMutated";
        this._booleanVar11IsMutated = "_booleanVar11IsMutated";
        this._booleanVar12IsMutated = "_booleanVar12IsMutated";
        this._booleanVar13IsMutated = "_booleanVar13IsMutated";
        this._booleanVar14IsMutated = "_booleanVar14IsMutated";
        this._booleanVar15IsMutated = "_booleanVar15IsMutated";
        this._booleanVar16IsMutated = "_booleanVar16IsMutated";
        this._booleanVar1IsMutated = "_booleanVar1IsMutated";
        this._booleanVar2IsMutated = "_booleanVar2IsMutated";
        this._booleanVar3IsMutated = "_booleanVar3IsMutated";
        this._booleanVar4IsMutated = "_booleanVar4IsMutated";
        this._booleanVar5IsMutated = "_booleanVar5IsMutated";
        this._booleanVar6IsMutated = "_booleanVar6IsMutated";
        this._booleanVar7IsMutated = "_booleanVar7IsMutated";
        this._booleanVar8IsMutated = "_booleanVar8IsMutated";
        this._booleanVar9IsMutated = "_booleanVar9IsMutated";
        this._dateIndexedVar1IsMutated = "_dateIndexedVar1IsMutated";
        this._dateIndexedVar2IsMutated = "_dateIndexedVar2IsMutated";
        this._dateIndexedVar3IsMutated = "_dateIndexedVar3IsMutated";
        this._dateIndexedVar4IsMutated = "_dateIndexedVar4IsMutated";
        this._dateIndexedVar5IsMutated = "_dateIndexedVar5IsMutated";
        this._dateIndexedVar6IsMutated = "_dateIndexedVar6IsMutated";
        this._dateIndexedVar7IsMutated = "_dateIndexedVar7IsMutated";
        this._dateIndexedVar8IsMutated = "_dateIndexedVar8IsMutated";
        this._dateVar1IsMutated = "_dateVar1IsMutated";
        this._dateVar2IsMutated = "_dateVar2IsMutated";
        this._dateVar3IsMutated = "_dateVar3IsMutated";
        this._dateVar4IsMutated = "_dateVar4IsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataFileUploadIsMutated = "_importDataFileUploadIsMutated";
        this._importDataMapIsMutated = "_importDataMapIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._numberIndexedVar1IsMutated = "_numberIndexedVar1IsMutated";
        this._numberIndexedVar2IsMutated = "_numberIndexedVar2IsMutated";
        this._numberIndexedVar3IsMutated = "_numberIndexedVar3IsMutated";
        this._numberIndexedVar4IsMutated = "_numberIndexedVar4IsMutated";
        this._numberIndexedVar5IsMutated = "_numberIndexedVar5IsMutated";
        this._numberIndexedVar6IsMutated = "_numberIndexedVar6IsMutated";
        this._numberIndexedVar7IsMutated = "_numberIndexedVar7IsMutated";
        this._numberIndexedVar8IsMutated = "_numberIndexedVar8IsMutated";
        this._numberVar10IsMutated = "_numberVar10IsMutated";
        this._numberVar11IsMutated = "_numberVar11IsMutated";
        this._numberVar12IsMutated = "_numberVar12IsMutated";
        this._numberVar13IsMutated = "_numberVar13IsMutated";
        this._numberVar14IsMutated = "_numberVar14IsMutated";
        this._numberVar15IsMutated = "_numberVar15IsMutated";
        this._numberVar16IsMutated = "_numberVar16IsMutated";
        this._numberVar1IsMutated = "_numberVar1IsMutated";
        this._numberVar2IsMutated = "_numberVar2IsMutated";
        this._numberVar3IsMutated = "_numberVar3IsMutated";
        this._numberVar4IsMutated = "_numberVar4IsMutated";
        this._numberVar5IsMutated = "_numberVar5IsMutated";
        this._numberVar6IsMutated = "_numberVar6IsMutated";
        this._numberVar7IsMutated = "_numberVar7IsMutated";
        this._numberVar8IsMutated = "_numberVar8IsMutated";
        this._numberVar9IsMutated = "_numberVar9IsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._stringIndexedVar10IsMutated = "_stringIndexedVar10IsMutated";
        this._stringIndexedVar11IsMutated = "_stringIndexedVar11IsMutated";
        this._stringIndexedVar12IsMutated = "_stringIndexedVar12IsMutated";
        this._stringIndexedVar13IsMutated = "_stringIndexedVar13IsMutated";
        this._stringIndexedVar14IsMutated = "_stringIndexedVar14IsMutated";
        this._stringIndexedVar15IsMutated = "_stringIndexedVar15IsMutated";
        this._stringIndexedVar16IsMutated = "_stringIndexedVar16IsMutated";
        this._stringIndexedVar1IsMutated = "_stringIndexedVar1IsMutated";
        this._stringIndexedVar2IsMutated = "_stringIndexedVar2IsMutated";
        this._stringIndexedVar3IsMutated = "_stringIndexedVar3IsMutated";
        this._stringIndexedVar4IsMutated = "_stringIndexedVar4IsMutated";
        this._stringIndexedVar5IsMutated = "_stringIndexedVar5IsMutated";
        this._stringIndexedVar6IsMutated = "_stringIndexedVar6IsMutated";
        this._stringIndexedVar7IsMutated = "_stringIndexedVar7IsMutated";
        this._stringIndexedVar8IsMutated = "_stringIndexedVar8IsMutated";
        this._stringIndexedVar9IsMutated = "_stringIndexedVar9IsMutated";
        this._stringVar10IsMutated = "_stringVar10IsMutated";
        this._stringVar11IsMutated = "_stringVar11IsMutated";
        this._stringVar12IsMutated = "_stringVar12IsMutated";
        this._stringVar13IsMutated = "_stringVar13IsMutated";
        this._stringVar14IsMutated = "_stringVar14IsMutated";
        this._stringVar15IsMutated = "_stringVar15IsMutated";
        this._stringVar16IsMutated = "_stringVar16IsMutated";
        this._stringVar17IsMutated = "_stringVar17IsMutated";
        this._stringVar18IsMutated = "_stringVar18IsMutated";
        this._stringVar19IsMutated = "_stringVar19IsMutated";
        this._stringVar1IsMutated = "_stringVar1IsMutated";
        this._stringVar20IsMutated = "_stringVar20IsMutated";
        this._stringVar21IsMutated = "_stringVar21IsMutated";
        this._stringVar22IsMutated = "_stringVar22IsMutated";
        this._stringVar23IsMutated = "_stringVar23IsMutated";
        this._stringVar24IsMutated = "_stringVar24IsMutated";
        this._stringVar2IsMutated = "_stringVar2IsMutated";
        this._stringVar3IsMutated = "_stringVar3IsMutated";
        this._stringVar4IsMutated = "_stringVar4IsMutated";
        this._stringVar5IsMutated = "_stringVar5IsMutated";
        this._stringVar6IsMutated = "_stringVar6IsMutated";
        this._stringVar7IsMutated = "_stringVar7IsMutated";
        this._stringVar8IsMutated = "_stringVar8IsMutated";
        this._stringVar9IsMutated = "_stringVar9IsMutated";
        this._workspaceIsMutated = "_workspaceIsMutated";
        this.booleanIndexedVar1 = "booleanIndexedVar1";
        this.booleanIndexedVar2 = "booleanIndexedVar2";
        this.booleanIndexedVar3 = "booleanIndexedVar3";
        this.booleanIndexedVar4 = "booleanIndexedVar4";
        this.booleanIndexedVar5 = "booleanIndexedVar5";
        this.booleanIndexedVar6 = "booleanIndexedVar6";
        this.booleanIndexedVar7 = "booleanIndexedVar7";
        this.booleanIndexedVar8 = "booleanIndexedVar8";
        this.booleanVar1 = "booleanVar1";
        this.booleanVar10 = "booleanVar10";
        this.booleanVar11 = "booleanVar11";
        this.booleanVar12 = "booleanVar12";
        this.booleanVar13 = "booleanVar13";
        this.booleanVar14 = "booleanVar14";
        this.booleanVar15 = "booleanVar15";
        this.booleanVar16 = "booleanVar16";
        this.booleanVar2 = "booleanVar2";
        this.booleanVar3 = "booleanVar3";
        this.booleanVar4 = "booleanVar4";
        this.booleanVar5 = "booleanVar5";
        this.booleanVar6 = "booleanVar6";
        this.booleanVar7 = "booleanVar7";
        this.booleanVar8 = "booleanVar8";
        this.booleanVar9 = "booleanVar9";
        this.createdDate = "createdDate";
        this.dateIndexedVar1 = "dateIndexedVar1";
        this.dateIndexedVar2 = "dateIndexedVar2";
        this.dateIndexedVar3 = "dateIndexedVar3";
        this.dateIndexedVar4 = "dateIndexedVar4";
        this.dateIndexedVar5 = "dateIndexedVar5";
        this.dateIndexedVar6 = "dateIndexedVar6";
        this.dateIndexedVar7 = "dateIndexedVar7";
        this.dateIndexedVar8 = "dateIndexedVar8";
        this.dateVar1 = "dateVar1";
        this.dateVar2 = "dateVar2";
        this.dateVar3 = "dateVar3";
        this.dateVar4 = "dateVar4";
        this.id = "id";
        this.importDataFileUpload = "importDataFileUpload";
        this.importDataMap = "importDataMap";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.numberIndexedVar1 = "numberIndexedVar1";
        this.numberIndexedVar2 = "numberIndexedVar2";
        this.numberIndexedVar3 = "numberIndexedVar3";
        this.numberIndexedVar4 = "numberIndexedVar4";
        this.numberIndexedVar5 = "numberIndexedVar5";
        this.numberIndexedVar6 = "numberIndexedVar6";
        this.numberIndexedVar7 = "numberIndexedVar7";
        this.numberIndexedVar8 = "numberIndexedVar8";
        this.numberVar1 = "numberVar1";
        this.numberVar10 = "numberVar10";
        this.numberVar11 = "numberVar11";
        this.numberVar12 = "numberVar12";
        this.numberVar13 = "numberVar13";
        this.numberVar14 = "numberVar14";
        this.numberVar15 = "numberVar15";
        this.numberVar16 = "numberVar16";
        this.numberVar2 = "numberVar2";
        this.numberVar3 = "numberVar3";
        this.numberVar4 = "numberVar4";
        this.numberVar5 = "numberVar5";
        this.numberVar6 = "numberVar6";
        this.numberVar7 = "numberVar7";
        this.numberVar8 = "numberVar8";
        this.numberVar9 = "numberVar9";
        this.organization = "organization";
        this.stringIndexedVar1 = "stringIndexedVar1";
        this.stringIndexedVar10 = "stringIndexedVar10";
        this.stringIndexedVar11 = "stringIndexedVar11";
        this.stringIndexedVar12 = "stringIndexedVar12";
        this.stringIndexedVar13 = "stringIndexedVar13";
        this.stringIndexedVar14 = "stringIndexedVar14";
        this.stringIndexedVar15 = "stringIndexedVar15";
        this.stringIndexedVar16 = "stringIndexedVar16";
        this.stringIndexedVar2 = "stringIndexedVar2";
        this.stringIndexedVar3 = "stringIndexedVar3";
        this.stringIndexedVar4 = "stringIndexedVar4";
        this.stringIndexedVar5 = "stringIndexedVar5";
        this.stringIndexedVar6 = "stringIndexedVar6";
        this.stringIndexedVar7 = "stringIndexedVar7";
        this.stringIndexedVar8 = "stringIndexedVar8";
        this.stringIndexedVar9 = "stringIndexedVar9";
        this.stringVar1 = "stringVar1";
        this.stringVar10 = "stringVar10";
        this.stringVar11 = "stringVar11";
        this.stringVar12 = "stringVar12";
        this.stringVar13 = "stringVar13";
        this.stringVar14 = "stringVar14";
        this.stringVar15 = "stringVar15";
        this.stringVar16 = "stringVar16";
        this.stringVar17 = "stringVar17";
        this.stringVar18 = "stringVar18";
        this.stringVar19 = "stringVar19";
        this.stringVar2 = "stringVar2";
        this.stringVar20 = "stringVar20";
        this.stringVar21 = "stringVar21";
        this.stringVar22 = "stringVar22";
        this.stringVar23 = "stringVar23";
        this.stringVar24 = "stringVar24";
        this.stringVar3 = "stringVar3";
        this.stringVar4 = "stringVar4";
        this.stringVar5 = "stringVar5";
        this.stringVar6 = "stringVar6";
        this.stringVar7 = "stringVar7";
        this.stringVar8 = "stringVar8";
        this.stringVar9 = "stringVar9";
        this.workspace = "workspace";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._booleanIndexedVar1IsMutated = "_booleanIndexedVar1IsMutated";
        this._booleanIndexedVar2IsMutated = "_booleanIndexedVar2IsMutated";
        this._booleanIndexedVar3IsMutated = "_booleanIndexedVar3IsMutated";
        this._booleanIndexedVar4IsMutated = "_booleanIndexedVar4IsMutated";
        this._booleanIndexedVar5IsMutated = "_booleanIndexedVar5IsMutated";
        this._booleanIndexedVar6IsMutated = "_booleanIndexedVar6IsMutated";
        this._booleanIndexedVar7IsMutated = "_booleanIndexedVar7IsMutated";
        this._booleanIndexedVar8IsMutated = "_booleanIndexedVar8IsMutated";
        this._booleanVar10IsMutated = "_booleanVar10IsMutated";
        this._booleanVar11IsMutated = "_booleanVar11IsMutated";
        this._booleanVar12IsMutated = "_booleanVar12IsMutated";
        this._booleanVar13IsMutated = "_booleanVar13IsMutated";
        this._booleanVar14IsMutated = "_booleanVar14IsMutated";
        this._booleanVar15IsMutated = "_booleanVar15IsMutated";
        this._booleanVar16IsMutated = "_booleanVar16IsMutated";
        this._booleanVar1IsMutated = "_booleanVar1IsMutated";
        this._booleanVar2IsMutated = "_booleanVar2IsMutated";
        this._booleanVar3IsMutated = "_booleanVar3IsMutated";
        this._booleanVar4IsMutated = "_booleanVar4IsMutated";
        this._booleanVar5IsMutated = "_booleanVar5IsMutated";
        this._booleanVar6IsMutated = "_booleanVar6IsMutated";
        this._booleanVar7IsMutated = "_booleanVar7IsMutated";
        this._booleanVar8IsMutated = "_booleanVar8IsMutated";
        this._booleanVar9IsMutated = "_booleanVar9IsMutated";
        this._dateIndexedVar1IsMutated = "_dateIndexedVar1IsMutated";
        this._dateIndexedVar2IsMutated = "_dateIndexedVar2IsMutated";
        this._dateIndexedVar3IsMutated = "_dateIndexedVar3IsMutated";
        this._dateIndexedVar4IsMutated = "_dateIndexedVar4IsMutated";
        this._dateIndexedVar5IsMutated = "_dateIndexedVar5IsMutated";
        this._dateIndexedVar6IsMutated = "_dateIndexedVar6IsMutated";
        this._dateIndexedVar7IsMutated = "_dateIndexedVar7IsMutated";
        this._dateIndexedVar8IsMutated = "_dateIndexedVar8IsMutated";
        this._dateVar1IsMutated = "_dateVar1IsMutated";
        this._dateVar2IsMutated = "_dateVar2IsMutated";
        this._dateVar3IsMutated = "_dateVar3IsMutated";
        this._dateVar4IsMutated = "_dateVar4IsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataFileUploadIsMutated = "_importDataFileUploadIsMutated";
        this._importDataMapIsMutated = "_importDataMapIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._numberIndexedVar1IsMutated = "_numberIndexedVar1IsMutated";
        this._numberIndexedVar2IsMutated = "_numberIndexedVar2IsMutated";
        this._numberIndexedVar3IsMutated = "_numberIndexedVar3IsMutated";
        this._numberIndexedVar4IsMutated = "_numberIndexedVar4IsMutated";
        this._numberIndexedVar5IsMutated = "_numberIndexedVar5IsMutated";
        this._numberIndexedVar6IsMutated = "_numberIndexedVar6IsMutated";
        this._numberIndexedVar7IsMutated = "_numberIndexedVar7IsMutated";
        this._numberIndexedVar8IsMutated = "_numberIndexedVar8IsMutated";
        this._numberVar10IsMutated = "_numberVar10IsMutated";
        this._numberVar11IsMutated = "_numberVar11IsMutated";
        this._numberVar12IsMutated = "_numberVar12IsMutated";
        this._numberVar13IsMutated = "_numberVar13IsMutated";
        this._numberVar14IsMutated = "_numberVar14IsMutated";
        this._numberVar15IsMutated = "_numberVar15IsMutated";
        this._numberVar16IsMutated = "_numberVar16IsMutated";
        this._numberVar1IsMutated = "_numberVar1IsMutated";
        this._numberVar2IsMutated = "_numberVar2IsMutated";
        this._numberVar3IsMutated = "_numberVar3IsMutated";
        this._numberVar4IsMutated = "_numberVar4IsMutated";
        this._numberVar5IsMutated = "_numberVar5IsMutated";
        this._numberVar6IsMutated = "_numberVar6IsMutated";
        this._numberVar7IsMutated = "_numberVar7IsMutated";
        this._numberVar8IsMutated = "_numberVar8IsMutated";
        this._numberVar9IsMutated = "_numberVar9IsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._stringIndexedVar10IsMutated = "_stringIndexedVar10IsMutated";
        this._stringIndexedVar11IsMutated = "_stringIndexedVar11IsMutated";
        this._stringIndexedVar12IsMutated = "_stringIndexedVar12IsMutated";
        this._stringIndexedVar13IsMutated = "_stringIndexedVar13IsMutated";
        this._stringIndexedVar14IsMutated = "_stringIndexedVar14IsMutated";
        this._stringIndexedVar15IsMutated = "_stringIndexedVar15IsMutated";
        this._stringIndexedVar16IsMutated = "_stringIndexedVar16IsMutated";
        this._stringIndexedVar1IsMutated = "_stringIndexedVar1IsMutated";
        this._stringIndexedVar2IsMutated = "_stringIndexedVar2IsMutated";
        this._stringIndexedVar3IsMutated = "_stringIndexedVar3IsMutated";
        this._stringIndexedVar4IsMutated = "_stringIndexedVar4IsMutated";
        this._stringIndexedVar5IsMutated = "_stringIndexedVar5IsMutated";
        this._stringIndexedVar6IsMutated = "_stringIndexedVar6IsMutated";
        this._stringIndexedVar7IsMutated = "_stringIndexedVar7IsMutated";
        this._stringIndexedVar8IsMutated = "_stringIndexedVar8IsMutated";
        this._stringIndexedVar9IsMutated = "_stringIndexedVar9IsMutated";
        this._stringVar10IsMutated = "_stringVar10IsMutated";
        this._stringVar11IsMutated = "_stringVar11IsMutated";
        this._stringVar12IsMutated = "_stringVar12IsMutated";
        this._stringVar13IsMutated = "_stringVar13IsMutated";
        this._stringVar14IsMutated = "_stringVar14IsMutated";
        this._stringVar15IsMutated = "_stringVar15IsMutated";
        this._stringVar16IsMutated = "_stringVar16IsMutated";
        this._stringVar17IsMutated = "_stringVar17IsMutated";
        this._stringVar18IsMutated = "_stringVar18IsMutated";
        this._stringVar19IsMutated = "_stringVar19IsMutated";
        this._stringVar1IsMutated = "_stringVar1IsMutated";
        this._stringVar20IsMutated = "_stringVar20IsMutated";
        this._stringVar21IsMutated = "_stringVar21IsMutated";
        this._stringVar22IsMutated = "_stringVar22IsMutated";
        this._stringVar23IsMutated = "_stringVar23IsMutated";
        this._stringVar24IsMutated = "_stringVar24IsMutated";
        this._stringVar2IsMutated = "_stringVar2IsMutated";
        this._stringVar3IsMutated = "_stringVar3IsMutated";
        this._stringVar4IsMutated = "_stringVar4IsMutated";
        this._stringVar5IsMutated = "_stringVar5IsMutated";
        this._stringVar6IsMutated = "_stringVar6IsMutated";
        this._stringVar7IsMutated = "_stringVar7IsMutated";
        this._stringVar8IsMutated = "_stringVar8IsMutated";
        this._stringVar9IsMutated = "_stringVar9IsMutated";
        this._workspaceIsMutated = "_workspaceIsMutated";
        this.booleanIndexedVar1 = "booleanIndexedVar1";
        this.booleanIndexedVar2 = "booleanIndexedVar2";
        this.booleanIndexedVar3 = "booleanIndexedVar3";
        this.booleanIndexedVar4 = "booleanIndexedVar4";
        this.booleanIndexedVar5 = "booleanIndexedVar5";
        this.booleanIndexedVar6 = "booleanIndexedVar6";
        this.booleanIndexedVar7 = "booleanIndexedVar7";
        this.booleanIndexedVar8 = "booleanIndexedVar8";
        this.booleanVar1 = "booleanVar1";
        this.booleanVar10 = "booleanVar10";
        this.booleanVar11 = "booleanVar11";
        this.booleanVar12 = "booleanVar12";
        this.booleanVar13 = "booleanVar13";
        this.booleanVar14 = "booleanVar14";
        this.booleanVar15 = "booleanVar15";
        this.booleanVar16 = "booleanVar16";
        this.booleanVar2 = "booleanVar2";
        this.booleanVar3 = "booleanVar3";
        this.booleanVar4 = "booleanVar4";
        this.booleanVar5 = "booleanVar5";
        this.booleanVar6 = "booleanVar6";
        this.booleanVar7 = "booleanVar7";
        this.booleanVar8 = "booleanVar8";
        this.booleanVar9 = "booleanVar9";
        this.createdDate = "createdDate";
        this.dateIndexedVar1 = "dateIndexedVar1";
        this.dateIndexedVar2 = "dateIndexedVar2";
        this.dateIndexedVar3 = "dateIndexedVar3";
        this.dateIndexedVar4 = "dateIndexedVar4";
        this.dateIndexedVar5 = "dateIndexedVar5";
        this.dateIndexedVar6 = "dateIndexedVar6";
        this.dateIndexedVar7 = "dateIndexedVar7";
        this.dateIndexedVar8 = "dateIndexedVar8";
        this.dateVar1 = "dateVar1";
        this.dateVar2 = "dateVar2";
        this.dateVar3 = "dateVar3";
        this.dateVar4 = "dateVar4";
        this._id = "_id";
        this.importDataFileUpload = "importDataFileUpload";
        this.importDataMap = "importDataMap";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.numberIndexedVar1 = "numberIndexedVar1";
        this.numberIndexedVar2 = "numberIndexedVar2";
        this.numberIndexedVar3 = "numberIndexedVar3";
        this.numberIndexedVar4 = "numberIndexedVar4";
        this.numberIndexedVar5 = "numberIndexedVar5";
        this.numberIndexedVar6 = "numberIndexedVar6";
        this.numberIndexedVar7 = "numberIndexedVar7";
        this.numberIndexedVar8 = "numberIndexedVar8";
        this.numberVar1 = "numberVar1";
        this.numberVar10 = "numberVar10";
        this.numberVar11 = "numberVar11";
        this.numberVar12 = "numberVar12";
        this.numberVar13 = "numberVar13";
        this.numberVar14 = "numberVar14";
        this.numberVar15 = "numberVar15";
        this.numberVar16 = "numberVar16";
        this.numberVar2 = "numberVar2";
        this.numberVar3 = "numberVar3";
        this.numberVar4 = "numberVar4";
        this.numberVar5 = "numberVar5";
        this.numberVar6 = "numberVar6";
        this.numberVar7 = "numberVar7";
        this.numberVar8 = "numberVar8";
        this.numberVar9 = "numberVar9";
        this.organization = "organization";
        this.stringIndexedVar1 = "stringIndexedVar1";
        this.stringIndexedVar10 = "stringIndexedVar10";
        this.stringIndexedVar11 = "stringIndexedVar11";
        this.stringIndexedVar12 = "stringIndexedVar12";
        this.stringIndexedVar13 = "stringIndexedVar13";
        this.stringIndexedVar14 = "stringIndexedVar14";
        this.stringIndexedVar15 = "stringIndexedVar15";
        this.stringIndexedVar16 = "stringIndexedVar16";
        this.stringIndexedVar2 = "stringIndexedVar2";
        this.stringIndexedVar3 = "stringIndexedVar3";
        this.stringIndexedVar4 = "stringIndexedVar4";
        this.stringIndexedVar5 = "stringIndexedVar5";
        this.stringIndexedVar6 = "stringIndexedVar6";
        this.stringIndexedVar7 = "stringIndexedVar7";
        this.stringIndexedVar8 = "stringIndexedVar8";
        this.stringIndexedVar9 = "stringIndexedVar9";
        this.stringVar1 = "stringVar1";
        this.stringVar10 = "stringVar10";
        this.stringVar11 = "stringVar11";
        this.stringVar12 = "stringVar12";
        this.stringVar13 = "stringVar13";
        this.stringVar14 = "stringVar14";
        this.stringVar15 = "stringVar15";
        this.stringVar16 = "stringVar16";
        this.stringVar17 = "stringVar17";
        this.stringVar18 = "stringVar18";
        this.stringVar19 = "stringVar19";
        this.stringVar2 = "stringVar2";
        this.stringVar20 = "stringVar20";
        this.stringVar21 = "stringVar21";
        this.stringVar22 = "stringVar22";
        this.stringVar23 = "stringVar23";
        this.stringVar24 = "stringVar24";
        this.stringVar3 = "stringVar3";
        this.stringVar4 = "stringVar4";
        this.stringVar5 = "stringVar5";
        this.stringVar6 = "stringVar6";
        this.stringVar7 = "stringVar7";
        this.stringVar8 = "stringVar8";
        this.stringVar9 = "stringVar9";
        this.workspace = "workspace";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.ImportData";
        this.mongDataModeName = "ImportData";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class ImportData extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._booleanIndexedVar1IsMutated = false;
        this._booleanIndexedVar2IsMutated = false;
        this._booleanIndexedVar3IsMutated = false;
        this._booleanIndexedVar4IsMutated = false;
        this._booleanIndexedVar5IsMutated = false;
        this._booleanIndexedVar6IsMutated = false;
        this._booleanIndexedVar7IsMutated = false;
        this._booleanIndexedVar8IsMutated = false;
        this._booleanVar10IsMutated = false;
        this._booleanVar11IsMutated = false;
        this._booleanVar12IsMutated = false;
        this._booleanVar13IsMutated = false;
        this._booleanVar14IsMutated = false;
        this._booleanVar15IsMutated = false;
        this._booleanVar16IsMutated = false;
        this._booleanVar1IsMutated = false;
        this._booleanVar2IsMutated = false;
        this._booleanVar3IsMutated = false;
        this._booleanVar4IsMutated = false;
        this._booleanVar5IsMutated = false;
        this._booleanVar6IsMutated = false;
        this._booleanVar7IsMutated = false;
        this._booleanVar8IsMutated = false;
        this._booleanVar9IsMutated = false;
        this._dateIndexedVar1IsMutated = false;
        this._dateIndexedVar2IsMutated = false;
        this._dateIndexedVar3IsMutated = false;
        this._dateIndexedVar4IsMutated = false;
        this._dateIndexedVar5IsMutated = false;
        this._dateIndexedVar6IsMutated = false;
        this._dateIndexedVar7IsMutated = false;
        this._dateIndexedVar8IsMutated = false;
        this._dateVar1IsMutated = false;
        this._dateVar2IsMutated = false;
        this._dateVar3IsMutated = false;
        this._dateVar4IsMutated = false;
        this._idSetBySystem = false;
        this._importDataFileUploadIsMutated = false;
        this._importDataMapIsMutated = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._numberIndexedVar1IsMutated = false;
        this._numberIndexedVar2IsMutated = false;
        this._numberIndexedVar3IsMutated = false;
        this._numberIndexedVar4IsMutated = false;
        this._numberIndexedVar5IsMutated = false;
        this._numberIndexedVar6IsMutated = false;
        this._numberIndexedVar7IsMutated = false;
        this._numberIndexedVar8IsMutated = false;
        this._numberVar10IsMutated = false;
        this._numberVar11IsMutated = false;
        this._numberVar12IsMutated = false;
        this._numberVar13IsMutated = false;
        this._numberVar14IsMutated = false;
        this._numberVar15IsMutated = false;
        this._numberVar16IsMutated = false;
        this._numberVar1IsMutated = false;
        this._numberVar2IsMutated = false;
        this._numberVar3IsMutated = false;
        this._numberVar4IsMutated = false;
        this._numberVar5IsMutated = false;
        this._numberVar6IsMutated = false;
        this._numberVar7IsMutated = false;
        this._numberVar8IsMutated = false;
        this._numberVar9IsMutated = false;
        this._organizationIsMutated = false;
        this._stringIndexedVar10IsMutated = false;
        this._stringIndexedVar11IsMutated = false;
        this._stringIndexedVar12IsMutated = false;
        this._stringIndexedVar13IsMutated = false;
        this._stringIndexedVar14IsMutated = false;
        this._stringIndexedVar15IsMutated = false;
        this._stringIndexedVar16IsMutated = false;
        this._stringIndexedVar1IsMutated = false;
        this._stringIndexedVar2IsMutated = false;
        this._stringIndexedVar3IsMutated = false;
        this._stringIndexedVar4IsMutated = false;
        this._stringIndexedVar5IsMutated = false;
        this._stringIndexedVar6IsMutated = false;
        this._stringIndexedVar7IsMutated = false;
        this._stringIndexedVar8IsMutated = false;
        this._stringIndexedVar9IsMutated = false;
        this._stringVar10IsMutated = false;
        this._stringVar11IsMutated = false;
        this._stringVar12IsMutated = false;
        this._stringVar13IsMutated = false;
        this._stringVar14IsMutated = false;
        this._stringVar15IsMutated = false;
        this._stringVar16IsMutated = false;
        this._stringVar17IsMutated = false;
        this._stringVar18IsMutated = false;
        this._stringVar19IsMutated = false;
        this._stringVar1IsMutated = false;
        this._stringVar20IsMutated = false;
        this._stringVar21IsMutated = false;
        this._stringVar22IsMutated = false;
        this._stringVar23IsMutated = false;
        this._stringVar24IsMutated = false;
        this._stringVar2IsMutated = false;
        this._stringVar3IsMutated = false;
        this._stringVar4IsMutated = false;
        this._stringVar5IsMutated = false;
        this._stringVar6IsMutated = false;
        this._stringVar7IsMutated = false;
        this._stringVar8IsMutated = false;
        this._stringVar9IsMutated = false;
        this._workspaceIsMutated = false;
        this.booleanIndexedVar1 = undefined;
        this.booleanIndexedVar2 = undefined;
        this.booleanIndexedVar3 = undefined;
        this.booleanIndexedVar4 = undefined;
        this.booleanIndexedVar5 = undefined;
        this.booleanIndexedVar6 = undefined;
        this.booleanIndexedVar7 = undefined;
        this.booleanIndexedVar8 = undefined;
        this.booleanVar1 = undefined;
        this.booleanVar10 = undefined;
        this.booleanVar11 = undefined;
        this.booleanVar12 = undefined;
        this.booleanVar13 = undefined;
        this.booleanVar14 = undefined;
        this.booleanVar15 = undefined;
        this.booleanVar16 = undefined;
        this.booleanVar2 = undefined;
        this.booleanVar3 = undefined;
        this.booleanVar4 = undefined;
        this.booleanVar5 = undefined;
        this.booleanVar6 = undefined;
        this.booleanVar7 = undefined;
        this.booleanVar8 = undefined;
        this.booleanVar9 = undefined;
        this.dateIndexedVar1 = null;
        this.dateIndexedVar2 = null;
        this.dateIndexedVar3 = null;
        this.dateIndexedVar4 = null;
        this.dateIndexedVar5 = null;
        this.dateIndexedVar6 = null;
        this.dateIndexedVar7 = null;
        this.dateIndexedVar8 = null;
        this.dateVar1 = null;
        this.dateVar2 = null;
        this.dateVar3 = null;
        this.dateVar4 = null;
        this.id = null;
        this.importDataFileUpload = null;
        this.importDataMap = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.ImportData";
        this.numberIndexedVar1 = 0;
        this.numberIndexedVar2 = 0;
        this.numberIndexedVar3 = 0;
        this.numberIndexedVar4 = 0;
        this.numberIndexedVar5 = 0;
        this.numberIndexedVar6 = 0;
        this.numberIndexedVar7 = 0;
        this.numberIndexedVar8 = 0;
        this.numberVar1 = 0;
        this.numberVar10 = 0;
        this.numberVar11 = 0;
        this.numberVar12 = 0;
        this.numberVar13 = 0;
        this.numberVar14 = 0;
        this.numberVar15 = 0;
        this.numberVar16 = 0;
        this.numberVar2 = 0;
        this.numberVar3 = 0;
        this.numberVar4 = 0;
        this.numberVar5 = 0;
        this.numberVar6 = 0;
        this.numberVar7 = 0;
        this.numberVar8 = 0;
        this.numberVar9 = 0;
        this.organization = null;
        this.stringIndexedVar1 = null;
        this.stringIndexedVar10 = null;
        this.stringIndexedVar11 = null;
        this.stringIndexedVar12 = null;
        this.stringIndexedVar13 = null;
        this.stringIndexedVar14 = null;
        this.stringIndexedVar15 = null;
        this.stringIndexedVar16 = null;
        this.stringIndexedVar2 = null;
        this.stringIndexedVar3 = null;
        this.stringIndexedVar4 = null;
        this.stringIndexedVar5 = null;
        this.stringIndexedVar6 = null;
        this.stringIndexedVar7 = null;
        this.stringIndexedVar8 = null;
        this.stringIndexedVar9 = null;
        this.stringVar1 = null;
        this.stringVar10 = null;
        this.stringVar11 = null;
        this.stringVar12 = null;
        this.stringVar13 = null;
        this.stringVar14 = null;
        this.stringVar15 = null;
        this.stringVar16 = null;
        this.stringVar17 = null;
        this.stringVar18 = null;
        this.stringVar19 = null;
        this.stringVar2 = null;
        this.stringVar20 = null;
        this.stringVar21 = null;
        this.stringVar22 = null;
        this.stringVar23 = null;
        this.stringVar24 = null;
        this.stringVar3 = null;
        this.stringVar4 = null;
        this.stringVar5 = null;
        this.stringVar6 = null;
        this.stringVar7 = null;
        this.stringVar8 = null;
        this.stringVar9 = null;
        this.workspace = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._booleanIndexedVar1IsMutated = false;
        this._booleanIndexedVar2IsMutated = false;
        this._booleanIndexedVar3IsMutated = false;
        this._booleanIndexedVar4IsMutated = false;
        this._booleanIndexedVar5IsMutated = false;
        this._booleanIndexedVar6IsMutated = false;
        this._booleanIndexedVar7IsMutated = false;
        this._booleanIndexedVar8IsMutated = false;
        this._booleanVar1IsMutated = false;
        this._booleanVar10IsMutated = false;
        this._booleanVar11IsMutated = false;
        this._booleanVar12IsMutated = false;
        this._booleanVar13IsMutated = false;
        this._booleanVar14IsMutated = false;
        this._booleanVar15IsMutated = false;
        this._booleanVar16IsMutated = false;
        this._booleanVar2IsMutated = false;
        this._booleanVar3IsMutated = false;
        this._booleanVar4IsMutated = false;
        this._booleanVar5IsMutated = false;
        this._booleanVar6IsMutated = false;
        this._booleanVar7IsMutated = false;
        this._booleanVar8IsMutated = false;
        this._booleanVar9IsMutated = false;
        this._dateIndexedVar1IsMutated = false;
        this._dateIndexedVar2IsMutated = false;
        this._dateIndexedVar3IsMutated = false;
        this._dateIndexedVar4IsMutated = false;
        this._dateIndexedVar5IsMutated = false;
        this._dateIndexedVar6IsMutated = false;
        this._dateIndexedVar7IsMutated = false;
        this._dateIndexedVar8IsMutated = false;
        this._dateVar1IsMutated = false;
        this._dateVar2IsMutated = false;
        this._dateVar3IsMutated = false;
        this._dateVar4IsMutated = false;
        this._importDataFileUploadIsMutated = false;
        this._importDataMapIsMutated = false;
        this._numberIndexedVar1IsMutated = false;
        this._numberIndexedVar2IsMutated = false;
        this._numberIndexedVar3IsMutated = false;
        this._numberIndexedVar4IsMutated = false;
        this._numberIndexedVar5IsMutated = false;
        this._numberIndexedVar6IsMutated = false;
        this._numberIndexedVar7IsMutated = false;
        this._numberIndexedVar8IsMutated = false;
        this._numberVar1IsMutated = false;
        this._numberVar10IsMutated = false;
        this._numberVar11IsMutated = false;
        this._numberVar12IsMutated = false;
        this._numberVar13IsMutated = false;
        this._numberVar14IsMutated = false;
        this._numberVar15IsMutated = false;
        this._numberVar16IsMutated = false;
        this._numberVar2IsMutated = false;
        this._numberVar3IsMutated = false;
        this._numberVar4IsMutated = false;
        this._numberVar5IsMutated = false;
        this._numberVar6IsMutated = false;
        this._numberVar7IsMutated = false;
        this._numberVar8IsMutated = false;
        this._numberVar9IsMutated = false;
        this._organizationIsMutated = false;
        this._stringIndexedVar1IsMutated = false;
        this._stringIndexedVar10IsMutated = false;
        this._stringIndexedVar11IsMutated = false;
        this._stringIndexedVar12IsMutated = false;
        this._stringIndexedVar13IsMutated = false;
        this._stringIndexedVar14IsMutated = false;
        this._stringIndexedVar15IsMutated = false;
        this._stringIndexedVar16IsMutated = false;
        this._stringIndexedVar2IsMutated = false;
        this._stringIndexedVar3IsMutated = false;
        this._stringIndexedVar4IsMutated = false;
        this._stringIndexedVar5IsMutated = false;
        this._stringIndexedVar6IsMutated = false;
        this._stringIndexedVar7IsMutated = false;
        this._stringIndexedVar8IsMutated = false;
        this._stringIndexedVar9IsMutated = false;
        this._stringVar1IsMutated = false;
        this._stringVar10IsMutated = false;
        this._stringVar11IsMutated = false;
        this._stringVar12IsMutated = false;
        this._stringVar13IsMutated = false;
        this._stringVar14IsMutated = false;
        this._stringVar15IsMutated = false;
        this._stringVar16IsMutated = false;
        this._stringVar17IsMutated = false;
        this._stringVar18IsMutated = false;
        this._stringVar19IsMutated = false;
        this._stringVar2IsMutated = false;
        this._stringVar20IsMutated = false;
        this._stringVar21IsMutated = false;
        this._stringVar22IsMutated = false;
        this._stringVar23IsMutated = false;
        this._stringVar24IsMutated = false;
        this._stringVar3IsMutated = false;
        this._stringVar4IsMutated = false;
        this._stringVar5IsMutated = false;
        this._stringVar6IsMutated = false;
        this._stringVar7IsMutated = false;
        this._stringVar8IsMutated = false;
        this._stringVar9IsMutated = false;
        this._workspaceIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.ImportData");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar1Mutated() {
        return (this._booleanIndexedVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar2Mutated() {
        return (this._booleanIndexedVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar3Mutated() {
        return (this._booleanIndexedVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar4Mutated() {
        return (this._booleanIndexedVar4IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar5Mutated() {
        return (this._booleanIndexedVar5IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar6Mutated() {
        return (this._booleanIndexedVar6IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar7Mutated() {
        return (this._booleanIndexedVar7IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanIndexedVar8Mutated() {
        return (this._booleanIndexedVar8IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar10Mutated() {
        return (this._booleanVar10IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar11Mutated() {
        return (this._booleanVar11IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar12Mutated() {
        return (this._booleanVar12IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar13Mutated() {
        return (this._booleanVar13IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar14Mutated() {
        return (this._booleanVar14IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar15Mutated() {
        return (this._booleanVar15IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar16Mutated() {
        return (this._booleanVar16IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar1Mutated() {
        return (this._booleanVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar2Mutated() {
        return (this._booleanVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar3Mutated() {
        return (this._booleanVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar4Mutated() {
        return (this._booleanVar4IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar5Mutated() {
        return (this._booleanVar5IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar6Mutated() {
        return (this._booleanVar6IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar7Mutated() {
        return (this._booleanVar7IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar8Mutated() {
        return (this._booleanVar8IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsBooleanVar9Mutated() {
        return (this._booleanVar9IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar1Mutated() {
        return (this._dateIndexedVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar2Mutated() {
        return (this._dateIndexedVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar3Mutated() {
        return (this._dateIndexedVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar4Mutated() {
        return (this._dateIndexedVar4IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar5Mutated() {
        return (this._dateIndexedVar5IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar6Mutated() {
        return (this._dateIndexedVar6IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar7Mutated() {
        return (this._dateIndexedVar7IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateIndexedVar8Mutated() {
        return (this._dateIndexedVar8IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateVar1Mutated() {
        return (this._dateVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateVar2Mutated() {
        return (this._dateVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateVar3Mutated() {
        return (this._dateVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDateVar4Mutated() {
        return (this._dateVar4IsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportDataFileUploadMutated() {
        return (this._importDataFileUploadIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportDataMapMutated() {
        return (this._importDataMapIsMutated);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar1Mutated() {
        return (this._numberIndexedVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar2Mutated() {
        return (this._numberIndexedVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar3Mutated() {
        return (this._numberIndexedVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar4Mutated() {
        return (this._numberIndexedVar4IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar5Mutated() {
        return (this._numberIndexedVar5IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar6Mutated() {
        return (this._numberIndexedVar6IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar7Mutated() {
        return (this._numberIndexedVar7IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberIndexedVar8Mutated() {
        return (this._numberIndexedVar8IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar10Mutated() {
        return (this._numberVar10IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar11Mutated() {
        return (this._numberVar11IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar12Mutated() {
        return (this._numberVar12IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar13Mutated() {
        return (this._numberVar13IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar14Mutated() {
        return (this._numberVar14IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar15Mutated() {
        return (this._numberVar15IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar16Mutated() {
        return (this._numberVar16IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar1Mutated() {
        return (this._numberVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar2Mutated() {
        return (this._numberVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar3Mutated() {
        return (this._numberVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar4Mutated() {
        return (this._numberVar4IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar5Mutated() {
        return (this._numberVar5IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar6Mutated() {
        return (this._numberVar6IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar7Mutated() {
        return (this._numberVar7IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar8Mutated() {
        return (this._numberVar8IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNumberVar9Mutated() {
        return (this._numberVar9IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsOrganizationMutated() {
        return (this._organizationIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar10Mutated() {
        return (this._stringIndexedVar10IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar11Mutated() {
        return (this._stringIndexedVar11IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar12Mutated() {
        return (this._stringIndexedVar12IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar13Mutated() {
        return (this._stringIndexedVar13IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar14Mutated() {
        return (this._stringIndexedVar14IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar15Mutated() {
        return (this._stringIndexedVar15IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar16Mutated() {
        return (this._stringIndexedVar16IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar1Mutated() {
        return (this._stringIndexedVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar2Mutated() {
        return (this._stringIndexedVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar3Mutated() {
        return (this._stringIndexedVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar4Mutated() {
        return (this._stringIndexedVar4IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar5Mutated() {
        return (this._stringIndexedVar5IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar6Mutated() {
        return (this._stringIndexedVar6IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar7Mutated() {
        return (this._stringIndexedVar7IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar8Mutated() {
        return (this._stringIndexedVar8IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringIndexedVar9Mutated() {
        return (this._stringIndexedVar9IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar10Mutated() {
        return (this._stringVar10IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar11Mutated() {
        return (this._stringVar11IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar12Mutated() {
        return (this._stringVar12IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar13Mutated() {
        return (this._stringVar13IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar14Mutated() {
        return (this._stringVar14IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar15Mutated() {
        return (this._stringVar15IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar16Mutated() {
        return (this._stringVar16IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar17Mutated() {
        return (this._stringVar17IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar18Mutated() {
        return (this._stringVar18IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar19Mutated() {
        return (this._stringVar19IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar1Mutated() {
        return (this._stringVar1IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar20Mutated() {
        return (this._stringVar20IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar21Mutated() {
        return (this._stringVar21IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar22Mutated() {
        return (this._stringVar22IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar23Mutated() {
        return (this._stringVar23IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar24Mutated() {
        return (this._stringVar24IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar2Mutated() {
        return (this._stringVar2IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar3Mutated() {
        return (this._stringVar3IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar4Mutated() {
        return (this._stringVar4IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar5Mutated() {
        return (this._stringVar5IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar6Mutated() {
        return (this._stringVar6IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar7Mutated() {
        return (this._stringVar7IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar8Mutated() {
        return (this._stringVar8IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsStringVar9Mutated() {
        return (this._stringVar9IsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsWorkspaceMutated() {
        return (this._workspaceIsMutated);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar1() {
        return ((this.booleanIndexedVar1 === undefined) ? false : this.booleanIndexedVar1);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar2() {
        return ((this.booleanIndexedVar2 === undefined) ? false : this.booleanIndexedVar2);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar3() {
        return ((this.booleanIndexedVar3 === undefined) ? false : this.booleanIndexedVar3);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar4() {
        return ((this.booleanIndexedVar4 === undefined) ? false : this.booleanIndexedVar4);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar5() {
        return ((this.booleanIndexedVar5 === undefined) ? false : this.booleanIndexedVar5);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar6() {
        return ((this.booleanIndexedVar6 === undefined) ? false : this.booleanIndexedVar6);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar7() {
        return ((this.booleanIndexedVar7 === undefined) ? false : this.booleanIndexedVar7);
    }
    /**
     * Indexed boolean variable name.
     * @returns boolean
     */
    getBooleanIndexedVar8() {
        return ((this.booleanIndexedVar8 === undefined) ? false : this.booleanIndexedVar8);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar1() {
        return ((this.booleanVar1 === undefined) ? false : this.booleanVar1);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar10() {
        return ((this.booleanVar10 === undefined) ? false : this.booleanVar10);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar11() {
        return ((this.booleanVar11 === undefined) ? false : this.booleanVar11);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar12() {
        return ((this.booleanVar12 === undefined) ? false : this.booleanVar12);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar13() {
        return ((this.booleanVar13 === undefined) ? false : this.booleanVar13);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar14() {
        return ((this.booleanVar14 === undefined) ? false : this.booleanVar14);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar15() {
        return ((this.booleanVar15 === undefined) ? false : this.booleanVar15);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar16() {
        return ((this.booleanVar16 === undefined) ? false : this.booleanVar16);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar2() {
        return ((this.booleanVar2 === undefined) ? false : this.booleanVar2);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar3() {
        return ((this.booleanVar3 === undefined) ? false : this.booleanVar3);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar4() {
        return ((this.booleanVar4 === undefined) ? false : this.booleanVar4);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar5() {
        return ((this.booleanVar5 === undefined) ? false : this.booleanVar5);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar6() {
        return ((this.booleanVar6 === undefined) ? false : this.booleanVar6);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar7() {
        return ((this.booleanVar7 === undefined) ? false : this.booleanVar7);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar8() {
        return ((this.booleanVar8 === undefined) ? false : this.booleanVar8);
    }
    /**
     * Boolean variable name.
     * @returns boolean
     */
    getBooleanVar9() {
        return ((this.booleanVar9 === undefined) ? false : this.booleanVar9);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar1() {
        return (this.dateIndexedVar1);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar2() {
        return (this.dateIndexedVar2);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar3() {
        return (this.dateIndexedVar3);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar4() {
        return (this.dateIndexedVar4);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar5() {
        return (this.dateIndexedVar5);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar6() {
        return (this.dateIndexedVar6);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar7() {
        return (this.dateIndexedVar7);
    }
    /**
     * Indexed date variable name
     * @returns Date | null
     */
    getDateIndexedVar8() {
        return (this.dateIndexedVar8);
    }
    /**
     * Date variable name
     * @returns Date | null
     */
    getDateVar1() {
        return (this.dateVar1);
    }
    /**
     * Date variable name
     * @returns Date | null
     */
    getDateVar2() {
        return (this.dateVar2);
    }
    /**
     * Date variable name
     * @returns Date | null
     */
    getDateVar3() {
        return (this.dateVar3);
    }
    /**
     * Date variable name
     * @returns Date | null
     */
    getDateVar4() {
        return (this.dateVar4);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * ImportDataFileUpload OID.
     * @returns string | null
     */
    getImportDataFileUpload() {
        return (this.importDataFileUpload);
    }
    /**
     * ImportDataMap OID.
     * @returns string | null
     */
    getImportDataMap() {
        return (this.importDataMap);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar1() {
        return (this.numberIndexedVar1);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar2() {
        return (this.numberIndexedVar2);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar3() {
        return (this.numberIndexedVar3);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar4() {
        return (this.numberIndexedVar4);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar5() {
        return (this.numberIndexedVar5);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar6() {
        return (this.numberIndexedVar6);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar7() {
        return (this.numberIndexedVar7);
    }
    /**
     * Indexed number variable name
     * @returns number
     */
    getNumberIndexedVar8() {
        return (this.numberIndexedVar8);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar1() {
        return (this.numberVar1);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar10() {
        return (this.numberVar10);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar11() {
        return (this.numberVar11);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar12() {
        return (this.numberVar12);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar13() {
        return (this.numberVar13);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar14() {
        return (this.numberVar14);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar15() {
        return (this.numberVar15);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar16() {
        return (this.numberVar16);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar2() {
        return (this.numberVar2);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar3() {
        return (this.numberVar3);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar4() {
        return (this.numberVar4);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar5() {
        return (this.numberVar5);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar6() {
        return (this.numberVar6);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar7() {
        return (this.numberVar7);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar8() {
        return (this.numberVar8);
    }
    /**
     * Number variable name
     * @returns number
     */
    getNumberVar9() {
        return (this.numberVar9);
    }
    /**
     * The organization.
     * @returns Organization | null
     */
    getOrganization() {
        return (this.organization);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar1() {
        return (this.stringIndexedVar1);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar10() {
        return (this.stringIndexedVar10);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar11() {
        return (this.stringIndexedVar11);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar12() {
        return (this.stringIndexedVar12);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar13() {
        return (this.stringIndexedVar13);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar14() {
        return (this.stringIndexedVar14);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar15() {
        return (this.stringIndexedVar15);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar16() {
        return (this.stringIndexedVar16);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar2() {
        return (this.stringIndexedVar2);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar3() {
        return (this.stringIndexedVar3);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar4() {
        return (this.stringIndexedVar4);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar5() {
        return (this.stringIndexedVar5);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar6() {
        return (this.stringIndexedVar6);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar7() {
        return (this.stringIndexedVar7);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar8() {
        return (this.stringIndexedVar8);
    }
    /**
     * Indexed string variable name.
     * @returns string | null
     */
    getStringIndexedVar9() {
        return (this.stringIndexedVar9);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar1() {
        return (this.stringVar1);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar10() {
        return (this.stringVar10);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar11() {
        return (this.stringVar11);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar12() {
        return (this.stringVar12);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar13() {
        return (this.stringVar13);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar14() {
        return (this.stringVar14);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar15() {
        return (this.stringVar15);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar16() {
        return (this.stringVar16);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar17() {
        return (this.stringVar17);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar18() {
        return (this.stringVar18);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar19() {
        return (this.stringVar19);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar2() {
        return (this.stringVar2);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar20() {
        return (this.stringVar20);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar21() {
        return (this.stringVar21);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar22() {
        return (this.stringVar22);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar23() {
        return (this.stringVar23);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar24() {
        return (this.stringVar24);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar3() {
        return (this.stringVar3);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar4() {
        return (this.stringVar4);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar5() {
        return (this.stringVar5);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar6() {
        return (this.stringVar6);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar7() {
        return (this.stringVar7);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar8() {
        return (this.stringVar8);
    }
    /**
     * String variable name.
     * @returns string | null
     */
    getStringVar9() {
        return (this.stringVar9);
    }
    /**
     * The workspace.
     * @returns Workspace | null
     */
    getWorkspace() {
        return (this.workspace);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar1Mutated(_booleanIndexedVar1IsMutated) {
        this._booleanIndexedVar1IsMutated = _booleanIndexedVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar2Mutated(_booleanIndexedVar2IsMutated) {
        this._booleanIndexedVar2IsMutated = _booleanIndexedVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar3Mutated(_booleanIndexedVar3IsMutated) {
        this._booleanIndexedVar3IsMutated = _booleanIndexedVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar4Mutated(_booleanIndexedVar4IsMutated) {
        this._booleanIndexedVar4IsMutated = _booleanIndexedVar4IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar5Mutated(_booleanIndexedVar5IsMutated) {
        this._booleanIndexedVar5IsMutated = _booleanIndexedVar5IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar6Mutated(_booleanIndexedVar6IsMutated) {
        this._booleanIndexedVar6IsMutated = _booleanIndexedVar6IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar7Mutated(_booleanIndexedVar7IsMutated) {
        this._booleanIndexedVar7IsMutated = _booleanIndexedVar7IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanIndexedVar8Mutated(_booleanIndexedVar8IsMutated) {
        this._booleanIndexedVar8IsMutated = _booleanIndexedVar8IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar10Mutated(_booleanVar10IsMutated) {
        this._booleanVar10IsMutated = _booleanVar10IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar11Mutated(_booleanVar11IsMutated) {
        this._booleanVar11IsMutated = _booleanVar11IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar12Mutated(_booleanVar12IsMutated) {
        this._booleanVar12IsMutated = _booleanVar12IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar13Mutated(_booleanVar13IsMutated) {
        this._booleanVar13IsMutated = _booleanVar13IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar14Mutated(_booleanVar14IsMutated) {
        this._booleanVar14IsMutated = _booleanVar14IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar15Mutated(_booleanVar15IsMutated) {
        this._booleanVar15IsMutated = _booleanVar15IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar16Mutated(_booleanVar16IsMutated) {
        this._booleanVar16IsMutated = _booleanVar16IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar1Mutated(_booleanVar1IsMutated) {
        this._booleanVar1IsMutated = _booleanVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar2Mutated(_booleanVar2IsMutated) {
        this._booleanVar2IsMutated = _booleanVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar3Mutated(_booleanVar3IsMutated) {
        this._booleanVar3IsMutated = _booleanVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar4Mutated(_booleanVar4IsMutated) {
        this._booleanVar4IsMutated = _booleanVar4IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar5Mutated(_booleanVar5IsMutated) {
        this._booleanVar5IsMutated = _booleanVar5IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar6Mutated(_booleanVar6IsMutated) {
        this._booleanVar6IsMutated = _booleanVar6IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar7Mutated(_booleanVar7IsMutated) {
        this._booleanVar7IsMutated = _booleanVar7IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar8Mutated(_booleanVar8IsMutated) {
        this._booleanVar8IsMutated = _booleanVar8IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsBooleanVar9Mutated(_booleanVar9IsMutated) {
        this._booleanVar9IsMutated = _booleanVar9IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar1Mutated(_dateIndexedVar1IsMutated) {
        this._dateIndexedVar1IsMutated = _dateIndexedVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar2Mutated(_dateIndexedVar2IsMutated) {
        this._dateIndexedVar2IsMutated = _dateIndexedVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar3Mutated(_dateIndexedVar3IsMutated) {
        this._dateIndexedVar3IsMutated = _dateIndexedVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar4Mutated(_dateIndexedVar4IsMutated) {
        this._dateIndexedVar4IsMutated = _dateIndexedVar4IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar5Mutated(_dateIndexedVar5IsMutated) {
        this._dateIndexedVar5IsMutated = _dateIndexedVar5IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar6Mutated(_dateIndexedVar6IsMutated) {
        this._dateIndexedVar6IsMutated = _dateIndexedVar6IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar7Mutated(_dateIndexedVar7IsMutated) {
        this._dateIndexedVar7IsMutated = _dateIndexedVar7IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateIndexedVar8Mutated(_dateIndexedVar8IsMutated) {
        this._dateIndexedVar8IsMutated = _dateIndexedVar8IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateVar1Mutated(_dateVar1IsMutated) {
        this._dateVar1IsMutated = _dateVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateVar2Mutated(_dateVar2IsMutated) {
        this._dateVar2IsMutated = _dateVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateVar3Mutated(_dateVar3IsMutated) {
        this._dateVar3IsMutated = _dateVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDateVar4Mutated(_dateVar4IsMutated) {
        this._dateVar4IsMutated = _dateVar4IsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportDataFileUploadMutated(_importDataFileUploadIsMutated) {
        this._importDataFileUploadIsMutated = _importDataFileUploadIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportDataMapMutated(_importDataMapIsMutated) {
        this._importDataMapIsMutated = _importDataMapIsMutated;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar1Mutated(_numberIndexedVar1IsMutated) {
        this._numberIndexedVar1IsMutated = _numberIndexedVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar2Mutated(_numberIndexedVar2IsMutated) {
        this._numberIndexedVar2IsMutated = _numberIndexedVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar3Mutated(_numberIndexedVar3IsMutated) {
        this._numberIndexedVar3IsMutated = _numberIndexedVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar4Mutated(_numberIndexedVar4IsMutated) {
        this._numberIndexedVar4IsMutated = _numberIndexedVar4IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar5Mutated(_numberIndexedVar5IsMutated) {
        this._numberIndexedVar5IsMutated = _numberIndexedVar5IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar6Mutated(_numberIndexedVar6IsMutated) {
        this._numberIndexedVar6IsMutated = _numberIndexedVar6IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar7Mutated(_numberIndexedVar7IsMutated) {
        this._numberIndexedVar7IsMutated = _numberIndexedVar7IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberIndexedVar8Mutated(_numberIndexedVar8IsMutated) {
        this._numberIndexedVar8IsMutated = _numberIndexedVar8IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar10Mutated(_numberVar10IsMutated) {
        this._numberVar10IsMutated = _numberVar10IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar11Mutated(_numberVar11IsMutated) {
        this._numberVar11IsMutated = _numberVar11IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar12Mutated(_numberVar12IsMutated) {
        this._numberVar12IsMutated = _numberVar12IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar13Mutated(_numberVar13IsMutated) {
        this._numberVar13IsMutated = _numberVar13IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar14Mutated(_numberVar14IsMutated) {
        this._numberVar14IsMutated = _numberVar14IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar15Mutated(_numberVar15IsMutated) {
        this._numberVar15IsMutated = _numberVar15IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar16Mutated(_numberVar16IsMutated) {
        this._numberVar16IsMutated = _numberVar16IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar1Mutated(_numberVar1IsMutated) {
        this._numberVar1IsMutated = _numberVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar2Mutated(_numberVar2IsMutated) {
        this._numberVar2IsMutated = _numberVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar3Mutated(_numberVar3IsMutated) {
        this._numberVar3IsMutated = _numberVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar4Mutated(_numberVar4IsMutated) {
        this._numberVar4IsMutated = _numberVar4IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar5Mutated(_numberVar5IsMutated) {
        this._numberVar5IsMutated = _numberVar5IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar6Mutated(_numberVar6IsMutated) {
        this._numberVar6IsMutated = _numberVar6IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar7Mutated(_numberVar7IsMutated) {
        this._numberVar7IsMutated = _numberVar7IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar8Mutated(_numberVar8IsMutated) {
        this._numberVar8IsMutated = _numberVar8IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNumberVar9Mutated(_numberVar9IsMutated) {
        this._numberVar9IsMutated = _numberVar9IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsOrganizationMutated(_organizationIsMutated) {
        this._organizationIsMutated = _organizationIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar10Mutated(_stringIndexedVar10IsMutated) {
        this._stringIndexedVar10IsMutated = _stringIndexedVar10IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar11Mutated(_stringIndexedVar11IsMutated) {
        this._stringIndexedVar11IsMutated = _stringIndexedVar11IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar12Mutated(_stringIndexedVar12IsMutated) {
        this._stringIndexedVar12IsMutated = _stringIndexedVar12IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar13Mutated(_stringIndexedVar13IsMutated) {
        this._stringIndexedVar13IsMutated = _stringIndexedVar13IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar14Mutated(_stringIndexedVar14IsMutated) {
        this._stringIndexedVar14IsMutated = _stringIndexedVar14IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar15Mutated(_stringIndexedVar15IsMutated) {
        this._stringIndexedVar15IsMutated = _stringIndexedVar15IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar16Mutated(_stringIndexedVar16IsMutated) {
        this._stringIndexedVar16IsMutated = _stringIndexedVar16IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar1Mutated(_stringIndexedVar1IsMutated) {
        this._stringIndexedVar1IsMutated = _stringIndexedVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar2Mutated(_stringIndexedVar2IsMutated) {
        this._stringIndexedVar2IsMutated = _stringIndexedVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar3Mutated(_stringIndexedVar3IsMutated) {
        this._stringIndexedVar3IsMutated = _stringIndexedVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar4Mutated(_stringIndexedVar4IsMutated) {
        this._stringIndexedVar4IsMutated = _stringIndexedVar4IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar5Mutated(_stringIndexedVar5IsMutated) {
        this._stringIndexedVar5IsMutated = _stringIndexedVar5IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar6Mutated(_stringIndexedVar6IsMutated) {
        this._stringIndexedVar6IsMutated = _stringIndexedVar6IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar7Mutated(_stringIndexedVar7IsMutated) {
        this._stringIndexedVar7IsMutated = _stringIndexedVar7IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar8Mutated(_stringIndexedVar8IsMutated) {
        this._stringIndexedVar8IsMutated = _stringIndexedVar8IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringIndexedVar9Mutated(_stringIndexedVar9IsMutated) {
        this._stringIndexedVar9IsMutated = _stringIndexedVar9IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar10Mutated(_stringVar10IsMutated) {
        this._stringVar10IsMutated = _stringVar10IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar11Mutated(_stringVar11IsMutated) {
        this._stringVar11IsMutated = _stringVar11IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar12Mutated(_stringVar12IsMutated) {
        this._stringVar12IsMutated = _stringVar12IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar13Mutated(_stringVar13IsMutated) {
        this._stringVar13IsMutated = _stringVar13IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar14Mutated(_stringVar14IsMutated) {
        this._stringVar14IsMutated = _stringVar14IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar15Mutated(_stringVar15IsMutated) {
        this._stringVar15IsMutated = _stringVar15IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar16Mutated(_stringVar16IsMutated) {
        this._stringVar16IsMutated = _stringVar16IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar17Mutated(_stringVar17IsMutated) {
        this._stringVar17IsMutated = _stringVar17IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar18Mutated(_stringVar18IsMutated) {
        this._stringVar18IsMutated = _stringVar18IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar19Mutated(_stringVar19IsMutated) {
        this._stringVar19IsMutated = _stringVar19IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar1Mutated(_stringVar1IsMutated) {
        this._stringVar1IsMutated = _stringVar1IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar20Mutated(_stringVar20IsMutated) {
        this._stringVar20IsMutated = _stringVar20IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar21Mutated(_stringVar21IsMutated) {
        this._stringVar21IsMutated = _stringVar21IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar22Mutated(_stringVar22IsMutated) {
        this._stringVar22IsMutated = _stringVar22IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar23Mutated(_stringVar23IsMutated) {
        this._stringVar23IsMutated = _stringVar23IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar24Mutated(_stringVar24IsMutated) {
        this._stringVar24IsMutated = _stringVar24IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar2Mutated(_stringVar2IsMutated) {
        this._stringVar2IsMutated = _stringVar2IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar3Mutated(_stringVar3IsMutated) {
        this._stringVar3IsMutated = _stringVar3IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar4Mutated(_stringVar4IsMutated) {
        this._stringVar4IsMutated = _stringVar4IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar5Mutated(_stringVar5IsMutated) {
        this._stringVar5IsMutated = _stringVar5IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar6Mutated(_stringVar6IsMutated) {
        this._stringVar6IsMutated = _stringVar6IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar7Mutated(_stringVar7IsMutated) {
        this._stringVar7IsMutated = _stringVar7IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar8Mutated(_stringVar8IsMutated) {
        this._stringVar8IsMutated = _stringVar8IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsStringVar9Mutated(_stringVar9IsMutated) {
        this._stringVar9IsMutated = _stringVar9IsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsWorkspaceMutated(_workspaceIsMutated) {
        this._workspaceIsMutated = _workspaceIsMutated;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar1(booleanIndexedVar1) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar1 !== booleanIndexedVar1) {
            this._setIsBooleanIndexedVar1Mutated(true);
        }
        this.booleanIndexedVar1 = booleanIndexedVar1;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar2(booleanIndexedVar2) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar2 !== booleanIndexedVar2) {
            this._setIsBooleanIndexedVar2Mutated(true);
        }
        this.booleanIndexedVar2 = booleanIndexedVar2;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar3(booleanIndexedVar3) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar3 !== booleanIndexedVar3) {
            this._setIsBooleanIndexedVar3Mutated(true);
        }
        this.booleanIndexedVar3 = booleanIndexedVar3;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar4(booleanIndexedVar4) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar4 !== booleanIndexedVar4) {
            this._setIsBooleanIndexedVar4Mutated(true);
        }
        this.booleanIndexedVar4 = booleanIndexedVar4;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar5(booleanIndexedVar5) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar5 !== booleanIndexedVar5) {
            this._setIsBooleanIndexedVar5Mutated(true);
        }
        this.booleanIndexedVar5 = booleanIndexedVar5;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar6(booleanIndexedVar6) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar6 !== booleanIndexedVar6) {
            this._setIsBooleanIndexedVar6Mutated(true);
        }
        this.booleanIndexedVar6 = booleanIndexedVar6;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar7(booleanIndexedVar7) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar7 !== booleanIndexedVar7) {
            this._setIsBooleanIndexedVar7Mutated(true);
        }
        this.booleanIndexedVar7 = booleanIndexedVar7;
        return (this);
    }
    /**
* ```
     * Indexed boolean variable name.
* ```
     */
    setBooleanIndexedVar8(booleanIndexedVar8) {
        if (!this._getIsMutateSuppressed() && this.booleanIndexedVar8 !== booleanIndexedVar8) {
            this._setIsBooleanIndexedVar8Mutated(true);
        }
        this.booleanIndexedVar8 = booleanIndexedVar8;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar1(booleanVar1) {
        if (!this._getIsMutateSuppressed() && this.booleanVar1 !== booleanVar1) {
            this._setIsBooleanVar1Mutated(true);
        }
        this.booleanVar1 = booleanVar1;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar10(booleanVar10) {
        if (!this._getIsMutateSuppressed() && this.booleanVar10 !== booleanVar10) {
            this._setIsBooleanVar10Mutated(true);
        }
        this.booleanVar10 = booleanVar10;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar11(booleanVar11) {
        if (!this._getIsMutateSuppressed() && this.booleanVar11 !== booleanVar11) {
            this._setIsBooleanVar11Mutated(true);
        }
        this.booleanVar11 = booleanVar11;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar12(booleanVar12) {
        if (!this._getIsMutateSuppressed() && this.booleanVar12 !== booleanVar12) {
            this._setIsBooleanVar12Mutated(true);
        }
        this.booleanVar12 = booleanVar12;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar13(booleanVar13) {
        if (!this._getIsMutateSuppressed() && this.booleanVar13 !== booleanVar13) {
            this._setIsBooleanVar13Mutated(true);
        }
        this.booleanVar13 = booleanVar13;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar14(booleanVar14) {
        if (!this._getIsMutateSuppressed() && this.booleanVar14 !== booleanVar14) {
            this._setIsBooleanVar14Mutated(true);
        }
        this.booleanVar14 = booleanVar14;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar15(booleanVar15) {
        if (!this._getIsMutateSuppressed() && this.booleanVar15 !== booleanVar15) {
            this._setIsBooleanVar15Mutated(true);
        }
        this.booleanVar15 = booleanVar15;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar16(booleanVar16) {
        if (!this._getIsMutateSuppressed() && this.booleanVar16 !== booleanVar16) {
            this._setIsBooleanVar16Mutated(true);
        }
        this.booleanVar16 = booleanVar16;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar2(booleanVar2) {
        if (!this._getIsMutateSuppressed() && this.booleanVar2 !== booleanVar2) {
            this._setIsBooleanVar2Mutated(true);
        }
        this.booleanVar2 = booleanVar2;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar3(booleanVar3) {
        if (!this._getIsMutateSuppressed() && this.booleanVar3 !== booleanVar3) {
            this._setIsBooleanVar3Mutated(true);
        }
        this.booleanVar3 = booleanVar3;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar4(booleanVar4) {
        if (!this._getIsMutateSuppressed() && this.booleanVar4 !== booleanVar4) {
            this._setIsBooleanVar4Mutated(true);
        }
        this.booleanVar4 = booleanVar4;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar5(booleanVar5) {
        if (!this._getIsMutateSuppressed() && this.booleanVar5 !== booleanVar5) {
            this._setIsBooleanVar5Mutated(true);
        }
        this.booleanVar5 = booleanVar5;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar6(booleanVar6) {
        if (!this._getIsMutateSuppressed() && this.booleanVar6 !== booleanVar6) {
            this._setIsBooleanVar6Mutated(true);
        }
        this.booleanVar6 = booleanVar6;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar7(booleanVar7) {
        if (!this._getIsMutateSuppressed() && this.booleanVar7 !== booleanVar7) {
            this._setIsBooleanVar7Mutated(true);
        }
        this.booleanVar7 = booleanVar7;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar8(booleanVar8) {
        if (!this._getIsMutateSuppressed() && this.booleanVar8 !== booleanVar8) {
            this._setIsBooleanVar8Mutated(true);
        }
        this.booleanVar8 = booleanVar8;
        return (this);
    }
    /**
* ```
     * Boolean variable name.
* ```
     */
    setBooleanVar9(booleanVar9) {
        if (!this._getIsMutateSuppressed() && this.booleanVar9 !== booleanVar9) {
            this._setIsBooleanVar9Mutated(true);
        }
        this.booleanVar9 = booleanVar9;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar1(dateIndexedVar1) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar1 !== dateIndexedVar1) {
            this._setIsDateIndexedVar1Mutated(true);
        }
        this.dateIndexedVar1 = dateIndexedVar1;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar2(dateIndexedVar2) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar2 !== dateIndexedVar2) {
            this._setIsDateIndexedVar2Mutated(true);
        }
        this.dateIndexedVar2 = dateIndexedVar2;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar3(dateIndexedVar3) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar3 !== dateIndexedVar3) {
            this._setIsDateIndexedVar3Mutated(true);
        }
        this.dateIndexedVar3 = dateIndexedVar3;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar4(dateIndexedVar4) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar4 !== dateIndexedVar4) {
            this._setIsDateIndexedVar4Mutated(true);
        }
        this.dateIndexedVar4 = dateIndexedVar4;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar5(dateIndexedVar5) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar5 !== dateIndexedVar5) {
            this._setIsDateIndexedVar5Mutated(true);
        }
        this.dateIndexedVar5 = dateIndexedVar5;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar6(dateIndexedVar6) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar6 !== dateIndexedVar6) {
            this._setIsDateIndexedVar6Mutated(true);
        }
        this.dateIndexedVar6 = dateIndexedVar6;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar7(dateIndexedVar7) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar7 !== dateIndexedVar7) {
            this._setIsDateIndexedVar7Mutated(true);
        }
        this.dateIndexedVar7 = dateIndexedVar7;
        return (this);
    }
    /**
* ```
     * Indexed date variable name
* ```
     */
    setDateIndexedVar8(dateIndexedVar8) {
        if (!this._getIsMutateSuppressed() && this.dateIndexedVar8 !== dateIndexedVar8) {
            this._setIsDateIndexedVar8Mutated(true);
        }
        this.dateIndexedVar8 = dateIndexedVar8;
        return (this);
    }
    /**
* ```
     * Date variable name
* ```
     */
    setDateVar1(dateVar1) {
        if (!this._getIsMutateSuppressed() && this.dateVar1 !== dateVar1) {
            this._setIsDateVar1Mutated(true);
        }
        this.dateVar1 = dateVar1;
        return (this);
    }
    /**
* ```
     * Date variable name
* ```
     */
    setDateVar2(dateVar2) {
        if (!this._getIsMutateSuppressed() && this.dateVar2 !== dateVar2) {
            this._setIsDateVar2Mutated(true);
        }
        this.dateVar2 = dateVar2;
        return (this);
    }
    /**
* ```
     * Date variable name
* ```
     */
    setDateVar3(dateVar3) {
        if (!this._getIsMutateSuppressed() && this.dateVar3 !== dateVar3) {
            this._setIsDateVar3Mutated(true);
        }
        this.dateVar3 = dateVar3;
        return (this);
    }
    /**
* ```
     * Date variable name
* ```
     */
    setDateVar4(dateVar4) {
        if (!this._getIsMutateSuppressed() && this.dateVar4 !== dateVar4) {
            this._setIsDateVar4Mutated(true);
        }
        this.dateVar4 = dateVar4;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     * ImportDataFileUpload OID.
* ```
     */
    setImportDataFileUpload(importDataFileUpload) {
        if (!this._getIsMutateSuppressed() && this.importDataFileUpload !== importDataFileUpload) {
            this._setIsImportDataFileUploadMutated(true);
        }
        this.importDataFileUpload = importDataFileUpload;
        return (this);
    }
    /**
* ```
     * ImportDataMap OID.
* ```
     */
    setImportDataMap(importDataMap) {
        if (!this._getIsMutateSuppressed() && this.importDataMap !== importDataMap) {
            this._setIsImportDataMapMutated(true);
        }
        this.importDataMap = importDataMap;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar1(numberIndexedVar1) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar1 !== numberIndexedVar1) {
            this._setIsNumberIndexedVar1Mutated(true);
        }
        this.numberIndexedVar1 = numberIndexedVar1;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar2(numberIndexedVar2) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar2 !== numberIndexedVar2) {
            this._setIsNumberIndexedVar2Mutated(true);
        }
        this.numberIndexedVar2 = numberIndexedVar2;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar3(numberIndexedVar3) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar3 !== numberIndexedVar3) {
            this._setIsNumberIndexedVar3Mutated(true);
        }
        this.numberIndexedVar3 = numberIndexedVar3;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar4(numberIndexedVar4) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar4 !== numberIndexedVar4) {
            this._setIsNumberIndexedVar4Mutated(true);
        }
        this.numberIndexedVar4 = numberIndexedVar4;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar5(numberIndexedVar5) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar5 !== numberIndexedVar5) {
            this._setIsNumberIndexedVar5Mutated(true);
        }
        this.numberIndexedVar5 = numberIndexedVar5;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar6(numberIndexedVar6) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar6 !== numberIndexedVar6) {
            this._setIsNumberIndexedVar6Mutated(true);
        }
        this.numberIndexedVar6 = numberIndexedVar6;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar7(numberIndexedVar7) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar7 !== numberIndexedVar7) {
            this._setIsNumberIndexedVar7Mutated(true);
        }
        this.numberIndexedVar7 = numberIndexedVar7;
        return (this);
    }
    /**
* ```
     * Indexed number variable name
* ```
     */
    setNumberIndexedVar8(numberIndexedVar8) {
        if (!this._getIsMutateSuppressed() && this.numberIndexedVar8 !== numberIndexedVar8) {
            this._setIsNumberIndexedVar8Mutated(true);
        }
        this.numberIndexedVar8 = numberIndexedVar8;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar1(numberVar1) {
        if (!this._getIsMutateSuppressed() && this.numberVar1 !== numberVar1) {
            this._setIsNumberVar1Mutated(true);
        }
        this.numberVar1 = numberVar1;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar10(numberVar10) {
        if (!this._getIsMutateSuppressed() && this.numberVar10 !== numberVar10) {
            this._setIsNumberVar10Mutated(true);
        }
        this.numberVar10 = numberVar10;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar11(numberVar11) {
        if (!this._getIsMutateSuppressed() && this.numberVar11 !== numberVar11) {
            this._setIsNumberVar11Mutated(true);
        }
        this.numberVar11 = numberVar11;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar12(numberVar12) {
        if (!this._getIsMutateSuppressed() && this.numberVar12 !== numberVar12) {
            this._setIsNumberVar12Mutated(true);
        }
        this.numberVar12 = numberVar12;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar13(numberVar13) {
        if (!this._getIsMutateSuppressed() && this.numberVar13 !== numberVar13) {
            this._setIsNumberVar13Mutated(true);
        }
        this.numberVar13 = numberVar13;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar14(numberVar14) {
        if (!this._getIsMutateSuppressed() && this.numberVar14 !== numberVar14) {
            this._setIsNumberVar14Mutated(true);
        }
        this.numberVar14 = numberVar14;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar15(numberVar15) {
        if (!this._getIsMutateSuppressed() && this.numberVar15 !== numberVar15) {
            this._setIsNumberVar15Mutated(true);
        }
        this.numberVar15 = numberVar15;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar16(numberVar16) {
        if (!this._getIsMutateSuppressed() && this.numberVar16 !== numberVar16) {
            this._setIsNumberVar16Mutated(true);
        }
        this.numberVar16 = numberVar16;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar2(numberVar2) {
        if (!this._getIsMutateSuppressed() && this.numberVar2 !== numberVar2) {
            this._setIsNumberVar2Mutated(true);
        }
        this.numberVar2 = numberVar2;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar3(numberVar3) {
        if (!this._getIsMutateSuppressed() && this.numberVar3 !== numberVar3) {
            this._setIsNumberVar3Mutated(true);
        }
        this.numberVar3 = numberVar3;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar4(numberVar4) {
        if (!this._getIsMutateSuppressed() && this.numberVar4 !== numberVar4) {
            this._setIsNumberVar4Mutated(true);
        }
        this.numberVar4 = numberVar4;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar5(numberVar5) {
        if (!this._getIsMutateSuppressed() && this.numberVar5 !== numberVar5) {
            this._setIsNumberVar5Mutated(true);
        }
        this.numberVar5 = numberVar5;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar6(numberVar6) {
        if (!this._getIsMutateSuppressed() && this.numberVar6 !== numberVar6) {
            this._setIsNumberVar6Mutated(true);
        }
        this.numberVar6 = numberVar6;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar7(numberVar7) {
        if (!this._getIsMutateSuppressed() && this.numberVar7 !== numberVar7) {
            this._setIsNumberVar7Mutated(true);
        }
        this.numberVar7 = numberVar7;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar8(numberVar8) {
        if (!this._getIsMutateSuppressed() && this.numberVar8 !== numberVar8) {
            this._setIsNumberVar8Mutated(true);
        }
        this.numberVar8 = numberVar8;
        return (this);
    }
    /**
* ```
     * Number variable name
* ```
     */
    setNumberVar9(numberVar9) {
        if (!this._getIsMutateSuppressed() && this.numberVar9 !== numberVar9) {
            this._setIsNumberVar9Mutated(true);
        }
        this.numberVar9 = numberVar9;
        return (this);
    }
    /**
* ```
     * The organization.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setOrganization(organization, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.organization !== organization) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(organization, persistanceType);
            this._setIsOrganizationMutated(true);
        }
        this.organization = organization;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar1(stringIndexedVar1) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar1 !== stringIndexedVar1) {
            this._setIsStringIndexedVar1Mutated(true);
        }
        this.stringIndexedVar1 = stringIndexedVar1;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar10(stringIndexedVar10) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar10 !== stringIndexedVar10) {
            this._setIsStringIndexedVar10Mutated(true);
        }
        this.stringIndexedVar10 = stringIndexedVar10;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar11(stringIndexedVar11) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar11 !== stringIndexedVar11) {
            this._setIsStringIndexedVar11Mutated(true);
        }
        this.stringIndexedVar11 = stringIndexedVar11;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar12(stringIndexedVar12) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar12 !== stringIndexedVar12) {
            this._setIsStringIndexedVar12Mutated(true);
        }
        this.stringIndexedVar12 = stringIndexedVar12;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar13(stringIndexedVar13) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar13 !== stringIndexedVar13) {
            this._setIsStringIndexedVar13Mutated(true);
        }
        this.stringIndexedVar13 = stringIndexedVar13;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar14(stringIndexedVar14) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar14 !== stringIndexedVar14) {
            this._setIsStringIndexedVar14Mutated(true);
        }
        this.stringIndexedVar14 = stringIndexedVar14;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar15(stringIndexedVar15) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar15 !== stringIndexedVar15) {
            this._setIsStringIndexedVar15Mutated(true);
        }
        this.stringIndexedVar15 = stringIndexedVar15;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar16(stringIndexedVar16) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar16 !== stringIndexedVar16) {
            this._setIsStringIndexedVar16Mutated(true);
        }
        this.stringIndexedVar16 = stringIndexedVar16;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar2(stringIndexedVar2) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar2 !== stringIndexedVar2) {
            this._setIsStringIndexedVar2Mutated(true);
        }
        this.stringIndexedVar2 = stringIndexedVar2;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar3(stringIndexedVar3) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar3 !== stringIndexedVar3) {
            this._setIsStringIndexedVar3Mutated(true);
        }
        this.stringIndexedVar3 = stringIndexedVar3;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar4(stringIndexedVar4) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar4 !== stringIndexedVar4) {
            this._setIsStringIndexedVar4Mutated(true);
        }
        this.stringIndexedVar4 = stringIndexedVar4;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar5(stringIndexedVar5) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar5 !== stringIndexedVar5) {
            this._setIsStringIndexedVar5Mutated(true);
        }
        this.stringIndexedVar5 = stringIndexedVar5;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar6(stringIndexedVar6) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar6 !== stringIndexedVar6) {
            this._setIsStringIndexedVar6Mutated(true);
        }
        this.stringIndexedVar6 = stringIndexedVar6;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar7(stringIndexedVar7) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar7 !== stringIndexedVar7) {
            this._setIsStringIndexedVar7Mutated(true);
        }
        this.stringIndexedVar7 = stringIndexedVar7;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar8(stringIndexedVar8) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar8 !== stringIndexedVar8) {
            this._setIsStringIndexedVar8Mutated(true);
        }
        this.stringIndexedVar8 = stringIndexedVar8;
        return (this);
    }
    /**
* ```
     * Indexed string variable name.
* ```
     */
    setStringIndexedVar9(stringIndexedVar9) {
        if (!this._getIsMutateSuppressed() && this.stringIndexedVar9 !== stringIndexedVar9) {
            this._setIsStringIndexedVar9Mutated(true);
        }
        this.stringIndexedVar9 = stringIndexedVar9;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar1(stringVar1) {
        if (!this._getIsMutateSuppressed() && this.stringVar1 !== stringVar1) {
            this._setIsStringVar1Mutated(true);
        }
        this.stringVar1 = stringVar1;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar10(stringVar10) {
        if (!this._getIsMutateSuppressed() && this.stringVar10 !== stringVar10) {
            this._setIsStringVar10Mutated(true);
        }
        this.stringVar10 = stringVar10;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar11(stringVar11) {
        if (!this._getIsMutateSuppressed() && this.stringVar11 !== stringVar11) {
            this._setIsStringVar11Mutated(true);
        }
        this.stringVar11 = stringVar11;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar12(stringVar12) {
        if (!this._getIsMutateSuppressed() && this.stringVar12 !== stringVar12) {
            this._setIsStringVar12Mutated(true);
        }
        this.stringVar12 = stringVar12;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar13(stringVar13) {
        if (!this._getIsMutateSuppressed() && this.stringVar13 !== stringVar13) {
            this._setIsStringVar13Mutated(true);
        }
        this.stringVar13 = stringVar13;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar14(stringVar14) {
        if (!this._getIsMutateSuppressed() && this.stringVar14 !== stringVar14) {
            this._setIsStringVar14Mutated(true);
        }
        this.stringVar14 = stringVar14;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar15(stringVar15) {
        if (!this._getIsMutateSuppressed() && this.stringVar15 !== stringVar15) {
            this._setIsStringVar15Mutated(true);
        }
        this.stringVar15 = stringVar15;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar16(stringVar16) {
        if (!this._getIsMutateSuppressed() && this.stringVar16 !== stringVar16) {
            this._setIsStringVar16Mutated(true);
        }
        this.stringVar16 = stringVar16;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar17(stringVar17) {
        if (!this._getIsMutateSuppressed() && this.stringVar17 !== stringVar17) {
            this._setIsStringVar17Mutated(true);
        }
        this.stringVar17 = stringVar17;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar18(stringVar18) {
        if (!this._getIsMutateSuppressed() && this.stringVar18 !== stringVar18) {
            this._setIsStringVar18Mutated(true);
        }
        this.stringVar18 = stringVar18;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar19(stringVar19) {
        if (!this._getIsMutateSuppressed() && this.stringVar19 !== stringVar19) {
            this._setIsStringVar19Mutated(true);
        }
        this.stringVar19 = stringVar19;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar2(stringVar2) {
        if (!this._getIsMutateSuppressed() && this.stringVar2 !== stringVar2) {
            this._setIsStringVar2Mutated(true);
        }
        this.stringVar2 = stringVar2;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar20(stringVar20) {
        if (!this._getIsMutateSuppressed() && this.stringVar20 !== stringVar20) {
            this._setIsStringVar20Mutated(true);
        }
        this.stringVar20 = stringVar20;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar21(stringVar21) {
        if (!this._getIsMutateSuppressed() && this.stringVar21 !== stringVar21) {
            this._setIsStringVar21Mutated(true);
        }
        this.stringVar21 = stringVar21;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar22(stringVar22) {
        if (!this._getIsMutateSuppressed() && this.stringVar22 !== stringVar22) {
            this._setIsStringVar22Mutated(true);
        }
        this.stringVar22 = stringVar22;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar23(stringVar23) {
        if (!this._getIsMutateSuppressed() && this.stringVar23 !== stringVar23) {
            this._setIsStringVar23Mutated(true);
        }
        this.stringVar23 = stringVar23;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar24(stringVar24) {
        if (!this._getIsMutateSuppressed() && this.stringVar24 !== stringVar24) {
            this._setIsStringVar24Mutated(true);
        }
        this.stringVar24 = stringVar24;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar3(stringVar3) {
        if (!this._getIsMutateSuppressed() && this.stringVar3 !== stringVar3) {
            this._setIsStringVar3Mutated(true);
        }
        this.stringVar3 = stringVar3;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar4(stringVar4) {
        if (!this._getIsMutateSuppressed() && this.stringVar4 !== stringVar4) {
            this._setIsStringVar4Mutated(true);
        }
        this.stringVar4 = stringVar4;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar5(stringVar5) {
        if (!this._getIsMutateSuppressed() && this.stringVar5 !== stringVar5) {
            this._setIsStringVar5Mutated(true);
        }
        this.stringVar5 = stringVar5;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar6(stringVar6) {
        if (!this._getIsMutateSuppressed() && this.stringVar6 !== stringVar6) {
            this._setIsStringVar6Mutated(true);
        }
        this.stringVar6 = stringVar6;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar7(stringVar7) {
        if (!this._getIsMutateSuppressed() && this.stringVar7 !== stringVar7) {
            this._setIsStringVar7Mutated(true);
        }
        this.stringVar7 = stringVar7;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar8(stringVar8) {
        if (!this._getIsMutateSuppressed() && this.stringVar8 !== stringVar8) {
            this._setIsStringVar8Mutated(true);
        }
        this.stringVar8 = stringVar8;
        return (this);
    }
    /**
* ```
     * String variable name.
* ```
     */
    setStringVar9(stringVar9) {
        if (!this._getIsMutateSuppressed() && this.stringVar9 !== stringVar9) {
            this._setIsStringVar9Mutated(true);
        }
        this.stringVar9 = stringVar9;
        return (this);
    }
    /**
* ```
     * The workspace.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setWorkspace(workspace, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.workspace !== workspace) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(workspace, persistanceType);
            this._setIsWorkspaceMutated(true);
        }
        this.workspace = workspace;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar1',
            value: this.booleanIndexedVar1
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar2',
            value: this.booleanIndexedVar2
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar3',
            value: this.booleanIndexedVar3
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar4',
            value: this.booleanIndexedVar4
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar5',
            value: this.booleanIndexedVar5
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar6',
            value: this.booleanIndexedVar6
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar7',
            value: this.booleanIndexedVar7
        });
        this._setCurrentState2({
            attributeName: 'booleanIndexedVar8',
            value: this.booleanIndexedVar8
        });
        this._setCurrentState2({
            attributeName: 'booleanVar1',
            value: this.booleanVar1
        });
        this._setCurrentState2({
            attributeName: 'booleanVar10',
            value: this.booleanVar10
        });
        this._setCurrentState2({
            attributeName: 'booleanVar11',
            value: this.booleanVar11
        });
        this._setCurrentState2({
            attributeName: 'booleanVar12',
            value: this.booleanVar12
        });
        this._setCurrentState2({
            attributeName: 'booleanVar13',
            value: this.booleanVar13
        });
        this._setCurrentState2({
            attributeName: 'booleanVar14',
            value: this.booleanVar14
        });
        this._setCurrentState2({
            attributeName: 'booleanVar15',
            value: this.booleanVar15
        });
        this._setCurrentState2({
            attributeName: 'booleanVar16',
            value: this.booleanVar16
        });
        this._setCurrentState2({
            attributeName: 'booleanVar2',
            value: this.booleanVar2
        });
        this._setCurrentState2({
            attributeName: 'booleanVar3',
            value: this.booleanVar3
        });
        this._setCurrentState2({
            attributeName: 'booleanVar4',
            value: this.booleanVar4
        });
        this._setCurrentState2({
            attributeName: 'booleanVar5',
            value: this.booleanVar5
        });
        this._setCurrentState2({
            attributeName: 'booleanVar6',
            value: this.booleanVar6
        });
        this._setCurrentState2({
            attributeName: 'booleanVar7',
            value: this.booleanVar7
        });
        this._setCurrentState2({
            attributeName: 'booleanVar8',
            value: this.booleanVar8
        });
        this._setCurrentState2({
            attributeName: 'booleanVar9',
            value: this.booleanVar9
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar1',
            value: this.dateIndexedVar1
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar2',
            value: this.dateIndexedVar2
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar3',
            value: this.dateIndexedVar3
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar4',
            value: this.dateIndexedVar4
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar5',
            value: this.dateIndexedVar5
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar6',
            value: this.dateIndexedVar6
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar7',
            value: this.dateIndexedVar7
        });
        this._setCurrentState2({
            attributeName: 'dateIndexedVar8',
            value: this.dateIndexedVar8
        });
        this._setCurrentState2({
            attributeName: 'dateVar1',
            value: this.dateVar1
        });
        this._setCurrentState2({
            attributeName: 'dateVar2',
            value: this.dateVar2
        });
        this._setCurrentState2({
            attributeName: 'dateVar3',
            value: this.dateVar3
        });
        this._setCurrentState2({
            attributeName: 'dateVar4',
            value: this.dateVar4
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'importDataFileUpload',
            value: this.importDataFileUpload
        });
        this._setCurrentState2({
            attributeName: 'importDataMap',
            value: this.importDataMap
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar1',
            value: this.numberIndexedVar1
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar2',
            value: this.numberIndexedVar2
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar3',
            value: this.numberIndexedVar3
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar4',
            value: this.numberIndexedVar4
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar5',
            value: this.numberIndexedVar5
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar6',
            value: this.numberIndexedVar6
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar7',
            value: this.numberIndexedVar7
        });
        this._setCurrentState2({
            attributeName: 'numberIndexedVar8',
            value: this.numberIndexedVar8
        });
        this._setCurrentState2({
            attributeName: 'numberVar1',
            value: this.numberVar1
        });
        this._setCurrentState2({
            attributeName: 'numberVar10',
            value: this.numberVar10
        });
        this._setCurrentState2({
            attributeName: 'numberVar11',
            value: this.numberVar11
        });
        this._setCurrentState2({
            attributeName: 'numberVar12',
            value: this.numberVar12
        });
        this._setCurrentState2({
            attributeName: 'numberVar13',
            value: this.numberVar13
        });
        this._setCurrentState2({
            attributeName: 'numberVar14',
            value: this.numberVar14
        });
        this._setCurrentState2({
            attributeName: 'numberVar15',
            value: this.numberVar15
        });
        this._setCurrentState2({
            attributeName: 'numberVar16',
            value: this.numberVar16
        });
        this._setCurrentState2({
            attributeName: 'numberVar2',
            value: this.numberVar2
        });
        this._setCurrentState2({
            attributeName: 'numberVar3',
            value: this.numberVar3
        });
        this._setCurrentState2({
            attributeName: 'numberVar4',
            value: this.numberVar4
        });
        this._setCurrentState2({
            attributeName: 'numberVar5',
            value: this.numberVar5
        });
        this._setCurrentState2({
            attributeName: 'numberVar6',
            value: this.numberVar6
        });
        this._setCurrentState2({
            attributeName: 'numberVar7',
            value: this.numberVar7
        });
        this._setCurrentState2({
            attributeName: 'numberVar8',
            value: this.numberVar8
        });
        this._setCurrentState2({
            attributeName: 'numberVar9',
            value: this.numberVar9
        });
        this._setCurrentState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar1',
            value: this.stringIndexedVar1
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar10',
            value: this.stringIndexedVar10
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar11',
            value: this.stringIndexedVar11
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar12',
            value: this.stringIndexedVar12
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar13',
            value: this.stringIndexedVar13
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar14',
            value: this.stringIndexedVar14
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar15',
            value: this.stringIndexedVar15
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar16',
            value: this.stringIndexedVar16
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar2',
            value: this.stringIndexedVar2
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar3',
            value: this.stringIndexedVar3
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar4',
            value: this.stringIndexedVar4
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar5',
            value: this.stringIndexedVar5
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar6',
            value: this.stringIndexedVar6
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar7',
            value: this.stringIndexedVar7
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar8',
            value: this.stringIndexedVar8
        });
        this._setCurrentState2({
            attributeName: 'stringIndexedVar9',
            value: this.stringIndexedVar9
        });
        this._setCurrentState2({
            attributeName: 'stringVar1',
            value: this.stringVar1
        });
        this._setCurrentState2({
            attributeName: 'stringVar10',
            value: this.stringVar10
        });
        this._setCurrentState2({
            attributeName: 'stringVar11',
            value: this.stringVar11
        });
        this._setCurrentState2({
            attributeName: 'stringVar12',
            value: this.stringVar12
        });
        this._setCurrentState2({
            attributeName: 'stringVar13',
            value: this.stringVar13
        });
        this._setCurrentState2({
            attributeName: 'stringVar14',
            value: this.stringVar14
        });
        this._setCurrentState2({
            attributeName: 'stringVar15',
            value: this.stringVar15
        });
        this._setCurrentState2({
            attributeName: 'stringVar16',
            value: this.stringVar16
        });
        this._setCurrentState2({
            attributeName: 'stringVar17',
            value: this.stringVar17
        });
        this._setCurrentState2({
            attributeName: 'stringVar18',
            value: this.stringVar18
        });
        this._setCurrentState2({
            attributeName: 'stringVar19',
            value: this.stringVar19
        });
        this._setCurrentState2({
            attributeName: 'stringVar2',
            value: this.stringVar2
        });
        this._setCurrentState2({
            attributeName: 'stringVar20',
            value: this.stringVar20
        });
        this._setCurrentState2({
            attributeName: 'stringVar21',
            value: this.stringVar21
        });
        this._setCurrentState2({
            attributeName: 'stringVar22',
            value: this.stringVar22
        });
        this._setCurrentState2({
            attributeName: 'stringVar23',
            value: this.stringVar23
        });
        this._setCurrentState2({
            attributeName: 'stringVar24',
            value: this.stringVar24
        });
        this._setCurrentState2({
            attributeName: 'stringVar3',
            value: this.stringVar3
        });
        this._setCurrentState2({
            attributeName: 'stringVar4',
            value: this.stringVar4
        });
        this._setCurrentState2({
            attributeName: 'stringVar5',
            value: this.stringVar5
        });
        this._setCurrentState2({
            attributeName: 'stringVar6',
            value: this.stringVar6
        });
        this._setCurrentState2({
            attributeName: 'stringVar7',
            value: this.stringVar7
        });
        this._setCurrentState2({
            attributeName: 'stringVar8',
            value: this.stringVar8
        });
        this._setCurrentState2({
            attributeName: 'stringVar9',
            value: this.stringVar9
        });
        this._setCurrentState2({
            attributeName: 'workspace',
            value: this.workspace
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar1',
            value: this.booleanIndexedVar1
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar2',
            value: this.booleanIndexedVar2
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar3',
            value: this.booleanIndexedVar3
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar4',
            value: this.booleanIndexedVar4
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar5',
            value: this.booleanIndexedVar5
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar6',
            value: this.booleanIndexedVar6
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar7',
            value: this.booleanIndexedVar7
        });
        this._setInitializedState2({
            attributeName: 'booleanIndexedVar8',
            value: this.booleanIndexedVar8
        });
        this._setInitializedState2({
            attributeName: 'booleanVar1',
            value: this.booleanVar1
        });
        this._setInitializedState2({
            attributeName: 'booleanVar10',
            value: this.booleanVar10
        });
        this._setInitializedState2({
            attributeName: 'booleanVar11',
            value: this.booleanVar11
        });
        this._setInitializedState2({
            attributeName: 'booleanVar12',
            value: this.booleanVar12
        });
        this._setInitializedState2({
            attributeName: 'booleanVar13',
            value: this.booleanVar13
        });
        this._setInitializedState2({
            attributeName: 'booleanVar14',
            value: this.booleanVar14
        });
        this._setInitializedState2({
            attributeName: 'booleanVar15',
            value: this.booleanVar15
        });
        this._setInitializedState2({
            attributeName: 'booleanVar16',
            value: this.booleanVar16
        });
        this._setInitializedState2({
            attributeName: 'booleanVar2',
            value: this.booleanVar2
        });
        this._setInitializedState2({
            attributeName: 'booleanVar3',
            value: this.booleanVar3
        });
        this._setInitializedState2({
            attributeName: 'booleanVar4',
            value: this.booleanVar4
        });
        this._setInitializedState2({
            attributeName: 'booleanVar5',
            value: this.booleanVar5
        });
        this._setInitializedState2({
            attributeName: 'booleanVar6',
            value: this.booleanVar6
        });
        this._setInitializedState2({
            attributeName: 'booleanVar7',
            value: this.booleanVar7
        });
        this._setInitializedState2({
            attributeName: 'booleanVar8',
            value: this.booleanVar8
        });
        this._setInitializedState2({
            attributeName: 'booleanVar9',
            value: this.booleanVar9
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar1',
            value: this.dateIndexedVar1
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar2',
            value: this.dateIndexedVar2
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar3',
            value: this.dateIndexedVar3
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar4',
            value: this.dateIndexedVar4
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar5',
            value: this.dateIndexedVar5
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar6',
            value: this.dateIndexedVar6
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar7',
            value: this.dateIndexedVar7
        });
        this._setInitializedState2({
            attributeName: 'dateIndexedVar8',
            value: this.dateIndexedVar8
        });
        this._setInitializedState2({
            attributeName: 'dateVar1',
            value: this.dateVar1
        });
        this._setInitializedState2({
            attributeName: 'dateVar2',
            value: this.dateVar2
        });
        this._setInitializedState2({
            attributeName: 'dateVar3',
            value: this.dateVar3
        });
        this._setInitializedState2({
            attributeName: 'dateVar4',
            value: this.dateVar4
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'importDataFileUpload',
            value: this.importDataFileUpload
        });
        this._setInitializedState2({
            attributeName: 'importDataMap',
            value: this.importDataMap
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar1',
            value: this.numberIndexedVar1
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar2',
            value: this.numberIndexedVar2
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar3',
            value: this.numberIndexedVar3
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar4',
            value: this.numberIndexedVar4
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar5',
            value: this.numberIndexedVar5
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar6',
            value: this.numberIndexedVar6
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar7',
            value: this.numberIndexedVar7
        });
        this._setInitializedState2({
            attributeName: 'numberIndexedVar8',
            value: this.numberIndexedVar8
        });
        this._setInitializedState2({
            attributeName: 'numberVar1',
            value: this.numberVar1
        });
        this._setInitializedState2({
            attributeName: 'numberVar10',
            value: this.numberVar10
        });
        this._setInitializedState2({
            attributeName: 'numberVar11',
            value: this.numberVar11
        });
        this._setInitializedState2({
            attributeName: 'numberVar12',
            value: this.numberVar12
        });
        this._setInitializedState2({
            attributeName: 'numberVar13',
            value: this.numberVar13
        });
        this._setInitializedState2({
            attributeName: 'numberVar14',
            value: this.numberVar14
        });
        this._setInitializedState2({
            attributeName: 'numberVar15',
            value: this.numberVar15
        });
        this._setInitializedState2({
            attributeName: 'numberVar16',
            value: this.numberVar16
        });
        this._setInitializedState2({
            attributeName: 'numberVar2',
            value: this.numberVar2
        });
        this._setInitializedState2({
            attributeName: 'numberVar3',
            value: this.numberVar3
        });
        this._setInitializedState2({
            attributeName: 'numberVar4',
            value: this.numberVar4
        });
        this._setInitializedState2({
            attributeName: 'numberVar5',
            value: this.numberVar5
        });
        this._setInitializedState2({
            attributeName: 'numberVar6',
            value: this.numberVar6
        });
        this._setInitializedState2({
            attributeName: 'numberVar7',
            value: this.numberVar7
        });
        this._setInitializedState2({
            attributeName: 'numberVar8',
            value: this.numberVar8
        });
        this._setInitializedState2({
            attributeName: 'numberVar9',
            value: this.numberVar9
        });
        this._setInitializedState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar1',
            value: this.stringIndexedVar1
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar10',
            value: this.stringIndexedVar10
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar11',
            value: this.stringIndexedVar11
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar12',
            value: this.stringIndexedVar12
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar13',
            value: this.stringIndexedVar13
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar14',
            value: this.stringIndexedVar14
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar15',
            value: this.stringIndexedVar15
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar16',
            value: this.stringIndexedVar16
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar2',
            value: this.stringIndexedVar2
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar3',
            value: this.stringIndexedVar3
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar4',
            value: this.stringIndexedVar4
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar5',
            value: this.stringIndexedVar5
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar6',
            value: this.stringIndexedVar6
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar7',
            value: this.stringIndexedVar7
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar8',
            value: this.stringIndexedVar8
        });
        this._setInitializedState2({
            attributeName: 'stringIndexedVar9',
            value: this.stringIndexedVar9
        });
        this._setInitializedState2({
            attributeName: 'stringVar1',
            value: this.stringVar1
        });
        this._setInitializedState2({
            attributeName: 'stringVar10',
            value: this.stringVar10
        });
        this._setInitializedState2({
            attributeName: 'stringVar11',
            value: this.stringVar11
        });
        this._setInitializedState2({
            attributeName: 'stringVar12',
            value: this.stringVar12
        });
        this._setInitializedState2({
            attributeName: 'stringVar13',
            value: this.stringVar13
        });
        this._setInitializedState2({
            attributeName: 'stringVar14',
            value: this.stringVar14
        });
        this._setInitializedState2({
            attributeName: 'stringVar15',
            value: this.stringVar15
        });
        this._setInitializedState2({
            attributeName: 'stringVar16',
            value: this.stringVar16
        });
        this._setInitializedState2({
            attributeName: 'stringVar17',
            value: this.stringVar17
        });
        this._setInitializedState2({
            attributeName: 'stringVar18',
            value: this.stringVar18
        });
        this._setInitializedState2({
            attributeName: 'stringVar19',
            value: this.stringVar19
        });
        this._setInitializedState2({
            attributeName: 'stringVar2',
            value: this.stringVar2
        });
        this._setInitializedState2({
            attributeName: 'stringVar20',
            value: this.stringVar20
        });
        this._setInitializedState2({
            attributeName: 'stringVar21',
            value: this.stringVar21
        });
        this._setInitializedState2({
            attributeName: 'stringVar22',
            value: this.stringVar22
        });
        this._setInitializedState2({
            attributeName: 'stringVar23',
            value: this.stringVar23
        });
        this._setInitializedState2({
            attributeName: 'stringVar24',
            value: this.stringVar24
        });
        this._setInitializedState2({
            attributeName: 'stringVar3',
            value: this.stringVar3
        });
        this._setInitializedState2({
            attributeName: 'stringVar4',
            value: this.stringVar4
        });
        this._setInitializedState2({
            attributeName: 'stringVar5',
            value: this.stringVar5
        });
        this._setInitializedState2({
            attributeName: 'stringVar6',
            value: this.stringVar6
        });
        this._setInitializedState2({
            attributeName: 'stringVar7',
            value: this.stringVar7
        });
        this._setInitializedState2({
            attributeName: 'stringVar8',
            value: this.stringVar8
        });
        this._setInitializedState2({
            attributeName: 'stringVar9',
            value: this.stringVar9
        });
        this._setInitializedState2({
            attributeName: 'workspace',
            value: this.workspace
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar5);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar6);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar7);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanIndexedVar8);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar10);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar11);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar12);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar13);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar14);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar15);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar16);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar5);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar6);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar7);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar8);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.booleanVar9);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar5);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar6);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar7);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateIndexedVar8);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dateVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importDataFileUpload);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importDataMap);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar5);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar6);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar7);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberIndexedVar8);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar10);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar11);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar12);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar13);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar14);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar15);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar16);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar5);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar6);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar7);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar8);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.numberVar9);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.organization);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar10);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar11);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar12);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar13);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar14);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar15);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar16);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar5);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar6);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar7);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar8);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringIndexedVar9);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar1);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar10);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar11);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar12);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar13);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar14);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar15);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar16);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar17);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar18);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar19);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar2);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar20);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar21);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar22);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar23);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar24);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar3);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar4);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar5);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar6);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar7);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar8);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.stringVar9);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.workspace);
        return (hashCode_n);
    }
}
exports.ImportData = ImportData;
ImportData.metaDataForImportData = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.ImportData", ImportData);
