// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/ImportDataMapAttribute.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataMapAttribute = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataFileAttributeNameIsMutated = "_importDataFileAttributeNameIsMutated";
        this._importDataMapTypeAttributeTypeIsMutated = "_importDataMapTypeAttributeTypeIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this.createdDate = "createdDate";
        this.id = "id";
        this.importDataFileAttributeName = "importDataFileAttributeName";
        this.importDataMapTypeAttributeType = "importDataMapTypeAttributeType";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataFileAttributeNameIsMutated = "_importDataFileAttributeNameIsMutated";
        this._importDataMapTypeAttributeTypeIsMutated = "_importDataMapTypeAttributeTypeIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this.createdDate = "createdDate";
        this._id = "_id";
        this.importDataFileAttributeName = "importDataFileAttributeName";
        this.importDataMapTypeAttributeType = "importDataMapTypeAttributeType";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.ImportDataMapAttribute";
        this.mongDataModeName = "ImportDataMapAttribute";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class ImportDataMapAttribute extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._idSetBySystem = false;
        this._importDataFileAttributeNameIsMutated = false;
        this._importDataMapTypeAttributeTypeIsMutated = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this.id = null;
        this.importDataFileAttributeName = null;
        this.importDataMapTypeAttributeType = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.ImportDataMapAttribute";
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._importDataFileAttributeNameIsMutated = false;
        this._importDataMapTypeAttributeTypeIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.ImportDataMapAttribute");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportDataFileAttributeNameMutated() {
        return (this._importDataFileAttributeNameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportDataMapTypeAttributeTypeMutated() {
        return (this._importDataMapTypeAttributeTypeIsMutated);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * Given a data file such as a .csv file, this is the name of a comma separated value.
     * @returns string | null
     */
    getImportDataFileAttributeName() {
        return (this.importDataFileAttributeName);
    }
    /**
     * The import data type attriute type.
     * @returns ImportDataMapTypeAttributeType | null
     */
    getImportDataMapTypeAttributeType() {
        return (this.importDataMapTypeAttributeType);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportDataFileAttributeNameMutated(_importDataFileAttributeNameIsMutated) {
        this._importDataFileAttributeNameIsMutated = _importDataFileAttributeNameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportDataMapTypeAttributeTypeMutated(_importDataMapTypeAttributeTypeIsMutated) {
        this._importDataMapTypeAttributeTypeIsMutated = _importDataMapTypeAttributeTypeIsMutated;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     * Given a data file such as a .csv file, this is the name of a comma separated value.
* ```
     */
    setImportDataFileAttributeName(importDataFileAttributeName) {
        if (!this._getIsMutateSuppressed() && this.importDataFileAttributeName !== importDataFileAttributeName) {
            this._setIsImportDataFileAttributeNameMutated(true);
        }
        this.importDataFileAttributeName = importDataFileAttributeName;
        return (this);
    }
    /**
* ```
     * The import data type attriute type.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setImportDataMapTypeAttributeType(importDataMapTypeAttributeType, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.importDataMapTypeAttributeType !== importDataMapTypeAttributeType) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(importDataMapTypeAttributeType, persistanceType);
            this._setIsImportDataMapTypeAttributeTypeMutated(true);
        }
        this.importDataMapTypeAttributeType = importDataMapTypeAttributeType;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'importDataFileAttributeName',
            value: this.importDataFileAttributeName
        });
        this._setCurrentState2({
            attributeName: 'importDataMapTypeAttributeType',
            value: this.importDataMapTypeAttributeType
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'importDataFileAttributeName',
            value: this.importDataFileAttributeName
        });
        this._setInitializedState2({
            attributeName: 'importDataMapTypeAttributeType',
            value: this.importDataMapTypeAttributeType
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importDataFileAttributeName);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importDataMapTypeAttributeType);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        return (hashCode_n);
    }
}
exports.ImportDataMapAttribute = ImportDataMapAttribute;
ImportDataMapAttribute.metaDataForImportDataMapAttribute = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.ImportDataMapAttribute", ImportDataMapAttribute);
