// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/ImportDataAttribute.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataAttribute = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._dataIntentIsMutated = "_dataIntentIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataAttributeTypeIsMutated = "_importDataAttributeTypeIsMutated";
        this._importFileAttributeNameIsMutated = "_importFileAttributeNameIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this.createdDate = "createdDate";
        this.dataIntent = "dataIntent";
        this.id = "id";
        this.importDataAttributeType = "importDataAttributeType";
        this.importFileAttributeName = "importFileAttributeName";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._dataIntentIsMutated = "_dataIntentIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._importDataAttributeTypeIsMutated = "_importDataAttributeTypeIsMutated";
        this._importFileAttributeNameIsMutated = "_importFileAttributeNameIsMutated";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this.createdDate = "createdDate";
        this.dataIntent = "dataIntent";
        this._id = "_id";
        this.importDataAttributeType = "importDataAttributeType";
        this.importFileAttributeName = "importFileAttributeName";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.ImportDataAttribute";
        this.mongDataModeName = "ImportDataAttribute";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class ImportDataAttribute extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._dataIntentIsMutated = false;
        this._idSetBySystem = false;
        this._importDataAttributeTypeIsMutated = false;
        this._importFileAttributeNameIsMutated = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this.dataIntent = null;
        this.id = null;
        this.importDataAttributeType = null;
        this.importFileAttributeName = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.ImportDataAttribute";
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._dataIntentIsMutated = false;
        this._importDataAttributeTypeIsMutated = false;
        this._importFileAttributeNameIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.ImportDataAttribute");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsDataIntentMutated() {
        return (this._dataIntentIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportDataAttributeTypeMutated() {
        return (this._importDataAttributeTypeIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsImportFileAttributeNameMutated() {
        return (this._importFileAttributeNameIsMutated);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     * Intent for attribute name.
     * @returns string | null
     */
    getDataIntent() {
        return (this.dataIntent);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * Import file attribute type.
     * @returns string | null
     */
    getImportDataAttributeType() {
        return (this.importDataAttributeType);
    }
    /**
     * Import file attribute name.
     * @returns string | null
     */
    getImportFileAttributeName() {
        return (this.importFileAttributeName);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDataIntentMutated(_dataIntentIsMutated) {
        this._dataIntentIsMutated = _dataIntentIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportDataAttributeTypeMutated(_importDataAttributeTypeIsMutated) {
        this._importDataAttributeTypeIsMutated = _importDataAttributeTypeIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsImportFileAttributeNameMutated(_importFileAttributeNameIsMutated) {
        this._importFileAttributeNameIsMutated = _importFileAttributeNameIsMutated;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     * Intent for attribute name.
* ```
     */
    setDataIntent(dataIntent) {
        if (!this._getIsMutateSuppressed() && this.dataIntent !== dataIntent) {
            this._setIsDataIntentMutated(true);
        }
        this.dataIntent = dataIntent;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     * Import file attribute type.
* ```
     */
    setImportDataAttributeType(importDataAttributeType) {
        if (!this._getIsMutateSuppressed() && this.importDataAttributeType !== importDataAttributeType) {
            this._setIsImportDataAttributeTypeMutated(true);
        }
        this.importDataAttributeType = importDataAttributeType;
        return (this);
    }
    /**
* ```
     * Import file attribute name.
* ```
     */
    setImportFileAttributeName(importFileAttributeName) {
        if (!this._getIsMutateSuppressed() && this.importFileAttributeName !== importFileAttributeName) {
            this._setIsImportFileAttributeNameMutated(true);
        }
        this.importFileAttributeName = importFileAttributeName;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'dataIntent',
            value: this.dataIntent
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'importDataAttributeType',
            value: this.importDataAttributeType
        });
        this._setCurrentState2({
            attributeName: 'importFileAttributeName',
            value: this.importFileAttributeName
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'dataIntent',
            value: this.dataIntent
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'importDataAttributeType',
            value: this.importDataAttributeType
        });
        this._setInitializedState2({
            attributeName: 'importFileAttributeName',
            value: this.importFileAttributeName
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.dataIntent);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importDataAttributeType);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.importFileAttributeName);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        return (hashCode_n);
    }
}
exports.ImportDataAttribute = ImportDataAttribute;
ImportDataAttribute.metaDataForImportDataAttribute = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.ImportDataAttribute", ImportDataAttribute);
