// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/KmlLayer.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KmlLayer = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistenceType_1 = require("../../../data/PersistenceType");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._activeEndDateIsMutated = "_activeEndDateIsMutated";
        this._activeStartDateIsMutated = "_activeStartDateIsMutated";
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isActiveIsMutated = "_isActiveIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._kmlGeometriesIsMutated = "_kmlGeometriesIsMutated";
        this._kmlIsMutated = "_kmlIsMutated";
        this._kmlTypeIsMutated = "_kmlTypeIsMutated";
        this._nameIsMutated = "_nameIsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._productCategoryKmlLocationsIsMutated = "_productCategoryKmlLocationsIsMutated";
        this._productColorKmlLocationsIsMutated = "_productColorKmlLocationsIsMutated";
        this._productKmlLocationsIsMutated = "_productKmlLocationsIsMutated";
        this._productPricingKmlLocationsIsMutated = "_productPricingKmlLocationsIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._taxKmlLocationsIsMutated = "_taxKmlLocationsIsMutated";
        this._workspaceIsMutated = "_workspaceIsMutated";
        this.activeEndDate = "activeEndDate";
        this.activeStartDate = "activeStartDate";
        this.createdDate = "createdDate";
        this.description = "description";
        this.id = "id";
        this.isActive = "isActive";
        this.jsonType = "jsonType";
        this.kml = "kml";
        this.kmlGeometries = "kmlGeometries";
        this.kmlType = "kmlType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.organization = "organization";
        this.productCategoryKmlLocations = "productCategoryKmlLocations";
        this.productColorKmlLocations = "productColorKmlLocations";
        this.productKmlLocations = "productKmlLocations";
        this.productPricingKmlLocations = "productPricingKmlLocations";
        this.taxKmlLocations = "taxKmlLocations";
        this.workspace = "workspace";
    }
}
class MongoVirtual {
    constructor() {
        this.kmlGeometries = "kmlGeometries";
        this.productCategoryKmlLocations = "productCategoryKmlLocations";
        this.productColorKmlLocations = "productColorKmlLocations";
        this.productKmlLocations = "productKmlLocations";
        this.productPricingKmlLocations = "productPricingKmlLocations";
        this.taxKmlLocations = "taxKmlLocations";
    }
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._activeEndDateIsMutated = "_activeEndDateIsMutated";
        this._activeStartDateIsMutated = "_activeStartDateIsMutated";
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isActiveIsMutated = "_isActiveIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._kmlGeometriesIsMutated = "_kmlGeometriesIsMutated";
        this._kmlIsMutated = "_kmlIsMutated";
        this._kmlTypeIsMutated = "_kmlTypeIsMutated";
        this._nameIsMutated = "_nameIsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._productCategoryKmlLocationsIsMutated = "_productCategoryKmlLocationsIsMutated";
        this._productColorKmlLocationsIsMutated = "_productColorKmlLocationsIsMutated";
        this._productKmlLocationsIsMutated = "_productKmlLocationsIsMutated";
        this._productPricingKmlLocationsIsMutated = "_productPricingKmlLocationsIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._taxKmlLocationsIsMutated = "_taxKmlLocationsIsMutated";
        this._workspaceIsMutated = "_workspaceIsMutated";
        this.activeEndDate = "activeEndDate";
        this.activeStartDate = "activeStartDate";
        this.createdDate = "createdDate";
        this.description = "description";
        this._id = "_id";
        this.isActive = "isActive";
        this.jsonType = "jsonType";
        this.kml = "kml";
        this.kmlGeometries = "kmlGeometries";
        this.kmlType = "kmlType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.organization = "organization";
        this.productCategoryKmlLocations = "productCategoryKmlLocations";
        this.productColorKmlLocations = "productColorKmlLocations";
        this.productKmlLocations = "productKmlLocations";
        this.productPricingKmlLocations = "productPricingKmlLocations";
        this.taxKmlLocations = "taxKmlLocations";
        this.workspace = "workspace";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.KmlLayer";
        this.mongDataModeName = "KmlLayer";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class KmlLayer extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._activeEndDateIsMutated = false;
        this._activeStartDateIsMutated = false;
        this._descriptionIsMutated = false;
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isActiveIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._kmlGeometriesIsMutated = false;
        this._kmlIsMutated = false;
        this._kmlTypeIsMutated = false;
        this._nameIsMutated = false;
        this._organizationIsMutated = false;
        this._productCategoryKmlLocationsIsMutated = false;
        this._productColorKmlLocationsIsMutated = false;
        this._productKmlLocationsIsMutated = false;
        this._productPricingKmlLocationsIsMutated = false;
        this._taxKmlLocationsIsMutated = false;
        this._workspaceIsMutated = false;
        this.activeEndDate = null;
        this.activeStartDate = null;
        this.description = null;
        this.id = null;
        this.isActive = undefined;
        this.jsonType = "com.salesstryke.entity.classes.persistable.KmlLayer";
        this.kml = null;
        this.kmlGeometries = new Array();
        this.kmlType = null;
        this.name = null;
        this.organization = null;
        this.productCategoryKmlLocations = new Array();
        this.productColorKmlLocations = new Array();
        this.productKmlLocations = new Array();
        this.productPricingKmlLocations = new Array();
        this.taxKmlLocations = new Array();
        this.workspace = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._activeEndDateIsMutated = false;
        this._activeStartDateIsMutated = false;
        this._descriptionIsMutated = false;
        this._isActiveIsMutated = false;
        this._kmlIsMutated = false;
        this._kmlGeometriesIsMutated = false;
        this._kmlTypeIsMutated = false;
        this._nameIsMutated = false;
        this._organizationIsMutated = false;
        this._productCategoryKmlLocationsIsMutated = false;
        this._productColorKmlLocationsIsMutated = false;
        this._productKmlLocationsIsMutated = false;
        this._productPricingKmlLocationsIsMutated = false;
        this._taxKmlLocationsIsMutated = false;
        this._workspaceIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.KmlLayer");
    }
    /**
     * The kml layer geometry
     *
     * Default PersistenceType : PersistenceType.ATTACH
     */
    addKmlGeometry(kmlGeometry, persistanceType = PersistenceType_1.PersistenceType.ATTACH) {
        if (this._getIsTrackingMutations()) {
            kmlGeometry._trackMutations();
        }
        kmlGeometry._setRelationshipPersistableType(persistanceType.getId());
        this.kmlGeometries.push(kmlGeometry);
        return (this);
    }
    /**
     * The kml layer geometry
     * If the Entity ID is null, will not be added.
     * Default PersistenceType : PersistenceType.DETACH
     */
    removeKmlGeometry(kmlGeometry, persistanceType = PersistenceType_1.PersistenceType.DETACH) {
        if (this._getIsTrackingMutations()) {
            kmlGeometry._trackMutations();
        }
        if (persistanceType === PersistenceType_1.PersistenceType.NONE) {
            let index = this.kmlGeometries.findIndex(obj => obj === kmlGeometry);
            if (index !== -1) {
                this.kmlGeometries.splice(index, 1);
            }
            kmlGeometry._setRelationshipPersistableType(PersistenceType_1.PersistenceType.NONE.getId());
            return (this);
        }
        kmlGeometry._setRelationshipPersistableType(persistanceType.getId());
        this.kmlGeometries.push(kmlGeometry);
        return (this);
    }
    /**
     * The product category kml location
     *
     * Default PersistenceType : PersistenceType.ATTACH
     */
    addProductCategoryKmlLocation(productCategoryKmlLocation, persistanceType = PersistenceType_1.PersistenceType.ATTACH) {
        if (this._getIsTrackingMutations()) {
            productCategoryKmlLocation._trackMutations();
        }
        productCategoryKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productCategoryKmlLocations.push(productCategoryKmlLocation);
        return (this);
    }
    /**
     * The product category kml location
     * If the Entity ID is null, will not be added.
     * Default PersistenceType : PersistenceType.DETACH
     */
    removeProductCategoryKmlLocation(productCategoryKmlLocation, persistanceType = PersistenceType_1.PersistenceType.DETACH) {
        if (this._getIsTrackingMutations()) {
            productCategoryKmlLocation._trackMutations();
        }
        if (persistanceType === PersistenceType_1.PersistenceType.NONE) {
            let index = this.productCategoryKmlLocations.findIndex(obj => obj === productCategoryKmlLocation);
            if (index !== -1) {
                this.productCategoryKmlLocations.splice(index, 1);
            }
            productCategoryKmlLocation._setRelationshipPersistableType(PersistenceType_1.PersistenceType.NONE.getId());
            return (this);
        }
        productCategoryKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productCategoryKmlLocations.push(productCategoryKmlLocation);
        return (this);
    }
    /**
     * The product color kml location
     *
     * Default PersistenceType : PersistenceType.ATTACH
     */
    addProductColorKmlLocation(productColorKmlLocation, persistanceType = PersistenceType_1.PersistenceType.ATTACH) {
        if (this._getIsTrackingMutations()) {
            productColorKmlLocation._trackMutations();
        }
        productColorKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productColorKmlLocations.push(productColorKmlLocation);
        return (this);
    }
    /**
     * The product color kml location
     * If the Entity ID is null, will not be added.
     * Default PersistenceType : PersistenceType.DETACH
     */
    removeProductColorKmlLocation(productColorKmlLocation, persistanceType = PersistenceType_1.PersistenceType.DETACH) {
        if (this._getIsTrackingMutations()) {
            productColorKmlLocation._trackMutations();
        }
        if (persistanceType === PersistenceType_1.PersistenceType.NONE) {
            let index = this.productColorKmlLocations.findIndex(obj => obj === productColorKmlLocation);
            if (index !== -1) {
                this.productColorKmlLocations.splice(index, 1);
            }
            productColorKmlLocation._setRelationshipPersistableType(PersistenceType_1.PersistenceType.NONE.getId());
            return (this);
        }
        productColorKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productColorKmlLocations.push(productColorKmlLocation);
        return (this);
    }
    /**
     * The product kml location
     *
     * Default PersistenceType : PersistenceType.ATTACH
     */
    addProductKmlLocation(productKmlLocation, persistanceType = PersistenceType_1.PersistenceType.ATTACH) {
        if (this._getIsTrackingMutations()) {
            productKmlLocation._trackMutations();
        }
        productKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productKmlLocations.push(productKmlLocation);
        return (this);
    }
    /**
     * The product kml location
     * If the Entity ID is null, will not be added.
     * Default PersistenceType : PersistenceType.DETACH
     */
    removeProductKmlLocation(productKmlLocation, persistanceType = PersistenceType_1.PersistenceType.DETACH) {
        if (this._getIsTrackingMutations()) {
            productKmlLocation._trackMutations();
        }
        if (persistanceType === PersistenceType_1.PersistenceType.NONE) {
            let index = this.productKmlLocations.findIndex(obj => obj === productKmlLocation);
            if (index !== -1) {
                this.productKmlLocations.splice(index, 1);
            }
            productKmlLocation._setRelationshipPersistableType(PersistenceType_1.PersistenceType.NONE.getId());
            return (this);
        }
        productKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productKmlLocations.push(productKmlLocation);
        return (this);
    }
    /**
     * The product pricing kml location
     *
     * Default PersistenceType : PersistenceType.ATTACH
     */
    addProductPricingKmlLocation(productPricingKmlLocation, persistanceType = PersistenceType_1.PersistenceType.ATTACH) {
        if (this._getIsTrackingMutations()) {
            productPricingKmlLocation._trackMutations();
        }
        productPricingKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productPricingKmlLocations.push(productPricingKmlLocation);
        return (this);
    }
    /**
     * The product pricing kml location
     * If the Entity ID is null, will not be added.
     * Default PersistenceType : PersistenceType.DETACH
     */
    removeProductPricingKmlLocation(productPricingKmlLocation, persistanceType = PersistenceType_1.PersistenceType.DETACH) {
        if (this._getIsTrackingMutations()) {
            productPricingKmlLocation._trackMutations();
        }
        if (persistanceType === PersistenceType_1.PersistenceType.NONE) {
            let index = this.productPricingKmlLocations.findIndex(obj => obj === productPricingKmlLocation);
            if (index !== -1) {
                this.productPricingKmlLocations.splice(index, 1);
            }
            productPricingKmlLocation._setRelationshipPersistableType(PersistenceType_1.PersistenceType.NONE.getId());
            return (this);
        }
        productPricingKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.productPricingKmlLocations.push(productPricingKmlLocation);
        return (this);
    }
    /**
     * The tax kml location
     *
     * Default PersistenceType : PersistenceType.ATTACH
     */
    addTaxKmlLocation(taxKmlLocation, persistanceType = PersistenceType_1.PersistenceType.ATTACH) {
        if (this._getIsTrackingMutations()) {
            taxKmlLocation._trackMutations();
        }
        taxKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.taxKmlLocations.push(taxKmlLocation);
        return (this);
    }
    /**
     * The tax kml location
     * If the Entity ID is null, will not be added.
     * Default PersistenceType : PersistenceType.DETACH
     */
    removeTaxKmlLocation(taxKmlLocation, persistanceType = PersistenceType_1.PersistenceType.DETACH) {
        if (this._getIsTrackingMutations()) {
            taxKmlLocation._trackMutations();
        }
        if (persistanceType === PersistenceType_1.PersistenceType.NONE) {
            let index = this.taxKmlLocations.findIndex(obj => obj === taxKmlLocation);
            if (index !== -1) {
                this.taxKmlLocations.splice(index, 1);
            }
            taxKmlLocation._setRelationshipPersistableType(PersistenceType_1.PersistenceType.NONE.getId());
            return (this);
        }
        taxKmlLocation._setRelationshipPersistableType(persistanceType.getId());
        this.taxKmlLocations.push(taxKmlLocation);
        return (this);
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveEndDateMutated() {
        return (this._activeEndDateIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveStartDateMutated() {
        return (this._activeStartDateIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDescriptionMutated() {
        return (this._descriptionIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveMutated() {
        return (this._isActiveIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsKmlGeometriesMutated() {
        return (this._kmlGeometriesIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsKmlMutated() {
        return (this._kmlIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsKmlTypeMutated() {
        return (this._kmlTypeIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsOrganizationMutated() {
        return (this._organizationIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductCategoryKmlLocationsMutated() {
        return (this._productCategoryKmlLocationsIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductColorKmlLocationsMutated() {
        return (this._productColorKmlLocationsIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductKmlLocationsMutated() {
        return (this._productKmlLocationsIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductPricingKmlLocationsMutated() {
        return (this._productPricingKmlLocationsIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsTaxKmlLocationsMutated() {
        return (this._taxKmlLocationsIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsWorkspaceMutated() {
        return (this._workspaceIsMutated);
    }
    /**
     * Active end date.
     * @returns Date | null
     */
    getActiveEndDate() {
        return (this.activeEndDate);
    }
    /**
     * Active start date.
     * @returns Date | null
     */
    getActiveStartDate() {
        return (this.activeStartDate);
    }
    /**
     * The description of Kml layer.
     * @returns string | null
     */
    getDescription() {
        return (this.description);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * Kml layer is active or not.
     * @returns boolean
     */
    getIsActive() {
        return ((this.isActive === undefined) ? false : this.isActive);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The Kml for this layer.
     * @returns Kml | null
     */
    getKml() {
        return (this.kml);
    }
    /**
     * The kml layer geometry
     * @returns Array<KmlGeometry>
     */
    getKmlGeometries() {
        return (this.kmlGeometries);
    }
    /**
     * The Kml layer type Id for this layer.
     * @returns KmlType | null
     */
    getKmlType() {
        return (this.kmlType);
    }
    /**
     * The name of Kml layer.
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
     * The organization.
     * @returns Organization | null
     */
    getOrganization() {
        return (this.organization);
    }
    /**
     * The product category kml location
     * @returns Array<ProductCategoryKmlLocation>
     */
    getProductCategoryKmlLocations() {
        return (this.productCategoryKmlLocations);
    }
    /**
     * The product color kml location
     * @returns Array<ProductColorKmlLocation>
     */
    getProductColorKmlLocations() {
        return (this.productColorKmlLocations);
    }
    /**
     * The product kml location
     * @returns Array<ProductKmlLocation>
     */
    getProductKmlLocations() {
        return (this.productKmlLocations);
    }
    /**
     * The product pricing kml location
     * @returns Array<ProductPricingKmlLocation>
     */
    getProductPricingKmlLocations() {
        return (this.productPricingKmlLocations);
    }
    /**
     * The tax kml location
     * @returns Array<TaxKmlLocation>
     */
    getTaxKmlLocations() {
        return (this.taxKmlLocations);
    }
    /**
     * The workspace this kml belong to.
     * @returns Workspace | null
     */
    getWorkspace() {
        return (this.workspace);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveEndDateMutated(_activeEndDateIsMutated) {
        this._activeEndDateIsMutated = _activeEndDateIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveStartDateMutated(_activeStartDateIsMutated) {
        this._activeStartDateIsMutated = _activeStartDateIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDescriptionMutated(_descriptionIsMutated) {
        this._descriptionIsMutated = _descriptionIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveMutated(_isActiveIsMutated) {
        this._isActiveIsMutated = _isActiveIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsKmlGeometriesMutated(_kmlGeometriesIsMutated) {
        this._kmlGeometriesIsMutated = _kmlGeometriesIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsKmlMutated(_kmlIsMutated) {
        this._kmlIsMutated = _kmlIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsKmlTypeMutated(_kmlTypeIsMutated) {
        this._kmlTypeIsMutated = _kmlTypeIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsOrganizationMutated(_organizationIsMutated) {
        this._organizationIsMutated = _organizationIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductCategoryKmlLocationsMutated(_productCategoryKmlLocationsIsMutated) {
        this._productCategoryKmlLocationsIsMutated = _productCategoryKmlLocationsIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductColorKmlLocationsMutated(_productColorKmlLocationsIsMutated) {
        this._productColorKmlLocationsIsMutated = _productColorKmlLocationsIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductKmlLocationsMutated(_productKmlLocationsIsMutated) {
        this._productKmlLocationsIsMutated = _productKmlLocationsIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductPricingKmlLocationsMutated(_productPricingKmlLocationsIsMutated) {
        this._productPricingKmlLocationsIsMutated = _productPricingKmlLocationsIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsTaxKmlLocationsMutated(_taxKmlLocationsIsMutated) {
        this._taxKmlLocationsIsMutated = _taxKmlLocationsIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsWorkspaceMutated(_workspaceIsMutated) {
        this._workspaceIsMutated = _workspaceIsMutated;
        return (this);
    }
    /**
* ```
     * Active end date.
* ```
     */
    setActiveEndDate(activeEndDate) {
        if (!this._getIsMutateSuppressed() && this.activeEndDate !== activeEndDate) {
            this._setIsActiveEndDateMutated(true);
        }
        this.activeEndDate = activeEndDate;
        return (this);
    }
    /**
* ```
     * Active start date.
* ```
     */
    setActiveStartDate(activeStartDate) {
        if (!this._getIsMutateSuppressed() && this.activeStartDate !== activeStartDate) {
            this._setIsActiveStartDateMutated(true);
        }
        this.activeStartDate = activeStartDate;
        return (this);
    }
    /**
* ```
     * The description of Kml layer.
* ```
     */
    setDescription(description) {
        if (!this._getIsMutateSuppressed() && this.description !== description) {
            this._setIsDescriptionMutated(true);
        }
        this.description = description;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     * Kml layer is active or not.
* ```
     */
    setIsActive(isActive) {
        if (!this._getIsMutateSuppressed() && this.isActive !== isActive) {
            this._setIsActiveMutated(true);
        }
        this.isActive = isActive;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The Kml for this layer.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setKml(kml, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.kml !== kml) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(kml, persistanceType);
            this._setIsKmlMutated(true);
        }
        this.kml = kml;
        return (this);
    }
    /**
* ```
     * The kml layer geometry
* ```
     */
    setKmlGeometries(kmlGeometries) {
        if (!this._getIsMutateSuppressed() && this.kmlGeometries !== kmlGeometries) {
            this._setIsKmlGeometriesMutated(true);
        }
        this.kmlGeometries = kmlGeometries;
        return (this);
    }
    /**
* ```
     * The Kml layer type Id for this layer.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setKmlType(kmlType, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.kmlType !== kmlType) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(kmlType, persistanceType);
            this._setIsKmlTypeMutated(true);
        }
        this.kmlType = kmlType;
        return (this);
    }
    /**
* ```
     * The name of Kml layer.
* ```
     */
    setName(name) {
        if (!this._getIsMutateSuppressed() && this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    /**
* ```
     * The organization.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setOrganization(organization, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.organization !== organization) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(organization, persistanceType);
            this._setIsOrganizationMutated(true);
        }
        this.organization = organization;
        return (this);
    }
    /**
* ```
     * The product category kml location
* ```
     */
    setProductCategoryKmlLocations(productCategoryKmlLocations) {
        if (!this._getIsMutateSuppressed() && this.productCategoryKmlLocations !== productCategoryKmlLocations) {
            this._setIsProductCategoryKmlLocationsMutated(true);
        }
        this.productCategoryKmlLocations = productCategoryKmlLocations;
        return (this);
    }
    /**
* ```
     * The product color kml location
* ```
     */
    setProductColorKmlLocations(productColorKmlLocations) {
        if (!this._getIsMutateSuppressed() && this.productColorKmlLocations !== productColorKmlLocations) {
            this._setIsProductColorKmlLocationsMutated(true);
        }
        this.productColorKmlLocations = productColorKmlLocations;
        return (this);
    }
    /**
* ```
     * The product kml location
* ```
     */
    setProductKmlLocations(productKmlLocations) {
        if (!this._getIsMutateSuppressed() && this.productKmlLocations !== productKmlLocations) {
            this._setIsProductKmlLocationsMutated(true);
        }
        this.productKmlLocations = productKmlLocations;
        return (this);
    }
    /**
* ```
     * The product pricing kml location
* ```
     */
    setProductPricingKmlLocations(productPricingKmlLocations) {
        if (!this._getIsMutateSuppressed() && this.productPricingKmlLocations !== productPricingKmlLocations) {
            this._setIsProductPricingKmlLocationsMutated(true);
        }
        this.productPricingKmlLocations = productPricingKmlLocations;
        return (this);
    }
    /**
* ```
     * The tax kml location
* ```
     */
    setTaxKmlLocations(taxKmlLocations) {
        if (!this._getIsMutateSuppressed() && this.taxKmlLocations !== taxKmlLocations) {
            this._setIsTaxKmlLocationsMutated(true);
        }
        this.taxKmlLocations = taxKmlLocations;
        return (this);
    }
    /**
* ```
     * The workspace this kml belong to.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setWorkspace(workspace, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.workspace !== workspace) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(workspace, persistanceType);
            this._setIsWorkspaceMutated(true);
        }
        this.workspace = workspace;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'activeEndDate',
            value: this.activeEndDate
        });
        this._setCurrentState2({
            attributeName: 'activeStartDate',
            value: this.activeStartDate
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'description',
            value: this.description
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'isActive',
            value: this.isActive
        });
        this._setCurrentState2({
            attributeName: 'kml',
            value: this.kml
        });
        this._setCurrentState2({
            attributeName: 'kmlGeometries',
            value: this.kmlGeometries
        });
        this._setCurrentState2({
            attributeName: 'kmlType',
            value: this.kmlType
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._setCurrentState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setCurrentState2({
            attributeName: 'productCategoryKmlLocations',
            value: this.productCategoryKmlLocations
        });
        this._setCurrentState2({
            attributeName: 'productColorKmlLocations',
            value: this.productColorKmlLocations
        });
        this._setCurrentState2({
            attributeName: 'productKmlLocations',
            value: this.productKmlLocations
        });
        this._setCurrentState2({
            attributeName: 'productPricingKmlLocations',
            value: this.productPricingKmlLocations
        });
        this._setCurrentState2({
            attributeName: 'taxKmlLocations',
            value: this.taxKmlLocations
        });
        this._setCurrentState2({
            attributeName: 'workspace',
            value: this.workspace
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'activeEndDate',
            value: this.activeEndDate
        });
        this._setInitializedState2({
            attributeName: 'activeStartDate',
            value: this.activeStartDate
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'description',
            value: this.description
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'isActive',
            value: this.isActive
        });
        this._setInitializedState2({
            attributeName: 'kml',
            value: this.kml
        });
        this._setInitializedState2({
            attributeName: 'kmlGeometries',
            value: this.kmlGeometries
        });
        this._setInitializedState2({
            attributeName: 'kmlType',
            value: this.kmlType
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        this._setInitializedState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setInitializedState2({
            attributeName: 'productCategoryKmlLocations',
            value: this.productCategoryKmlLocations
        });
        this._setInitializedState2({
            attributeName: 'productColorKmlLocations',
            value: this.productColorKmlLocations
        });
        this._setInitializedState2({
            attributeName: 'productKmlLocations',
            value: this.productKmlLocations
        });
        this._setInitializedState2({
            attributeName: 'productPricingKmlLocations',
            value: this.productPricingKmlLocations
        });
        this._setInitializedState2({
            attributeName: 'taxKmlLocations',
            value: this.taxKmlLocations
        });
        this._setInitializedState2({
            attributeName: 'workspace',
            value: this.workspace
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.activeEndDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.activeStartDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.description);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.isActive);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.kml);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.kmlGeometries);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.kmlType);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.organization);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.productCategoryKmlLocations);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.productColorKmlLocations);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.productKmlLocations);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.productPricingKmlLocations);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.taxKmlLocations);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.workspace);
        return (hashCode_n);
    }
}
exports.KmlLayer = KmlLayer;
KmlLayer.metaDataForKmlLayer = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.KmlLayer", KmlLayer);
