// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/ProductServiceDayOfWeek.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductServiceDayOfWeek = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._activeEndDateIsMutated = "_activeEndDateIsMutated";
        this._activeStartDateIsMutated = "_activeStartDateIsMutated";
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._fridayIsMutated = "_fridayIsMutated";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._mondayIsMutated = "_mondayIsMutated";
        this._nameIsMutated = "_nameIsMutated";
        this._productIsMutated = "_productIsMutated";
        this._productPricingIsMutated = "_productPricingIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._saturdayIsMutated = "_saturdayIsMutated";
        this._sundayIsMutated = "_sundayIsMutated";
        this._thursdayIsMutated = "_thursdayIsMutated";
        this._tuesdayIsMutated = "_tuesdayIsMutated";
        this._wednesdayIsMutated = "_wednesdayIsMutated";
        this.activeEndDate = "activeEndDate";
        this.activeStartDate = "activeStartDate";
        this.createdDate = "createdDate";
        this.description = "description";
        this.friday = "friday";
        this.id = "id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.monday = "monday";
        this.name = "name";
        this.product = "product";
        this.productPricing = "productPricing";
        this.saturday = "saturday";
        this.sunday = "sunday";
        this.thursday = "thursday";
        this.tuesday = "tuesday";
        this.wednesday = "wednesday";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._activeEndDateIsMutated = "_activeEndDateIsMutated";
        this._activeStartDateIsMutated = "_activeStartDateIsMutated";
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._fridayIsMutated = "_fridayIsMutated";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._mondayIsMutated = "_mondayIsMutated";
        this._nameIsMutated = "_nameIsMutated";
        this._productIsMutated = "_productIsMutated";
        this._productPricingIsMutated = "_productPricingIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._saturdayIsMutated = "_saturdayIsMutated";
        this._sundayIsMutated = "_sundayIsMutated";
        this._thursdayIsMutated = "_thursdayIsMutated";
        this._tuesdayIsMutated = "_tuesdayIsMutated";
        this._wednesdayIsMutated = "_wednesdayIsMutated";
        this.activeEndDate = "activeEndDate";
        this.activeStartDate = "activeStartDate";
        this.createdDate = "createdDate";
        this.description = "description";
        this.friday = "friday";
        this._id = "_id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.monday = "monday";
        this.name = "name";
        this.product = "product";
        this.productPricing = "productPricing";
        this.saturday = "saturday";
        this.sunday = "sunday";
        this.thursday = "thursday";
        this.tuesday = "tuesday";
        this.wednesday = "wednesday";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.ProductServiceDayOfWeek";
        this.mongDataModeName = "ProductServiceDayOfWeek";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class ProductServiceDayOfWeek extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._activeEndDateIsMutated = false;
        this._activeStartDateIsMutated = false;
        this._descriptionIsMutated = false;
        this._fridayIsMutated = false;
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._mondayIsMutated = false;
        this._nameIsMutated = false;
        this._productIsMutated = false;
        this._productPricingIsMutated = false;
        this._saturdayIsMutated = false;
        this._sundayIsMutated = false;
        this._thursdayIsMutated = false;
        this._tuesdayIsMutated = false;
        this._wednesdayIsMutated = false;
        this.activeEndDate = null;
        this.activeStartDate = null;
        this.description = null;
        this.friday = undefined;
        this.id = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.ProductServiceDayOfWeek";
        this.monday = undefined;
        this.name = null;
        this.product = null;
        this.productPricing = null;
        this.saturday = undefined;
        this.sunday = undefined;
        this.thursday = undefined;
        this.tuesday = undefined;
        this.wednesday = undefined;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._activeEndDateIsMutated = false;
        this._activeStartDateIsMutated = false;
        this._descriptionIsMutated = false;
        this._fridayIsMutated = false;
        this._mondayIsMutated = false;
        this._nameIsMutated = false;
        this._productIsMutated = false;
        this._productPricingIsMutated = false;
        this._saturdayIsMutated = false;
        this._sundayIsMutated = false;
        this._thursdayIsMutated = false;
        this._tuesdayIsMutated = false;
        this._wednesdayIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.ProductServiceDayOfWeek");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveEndDateMutated() {
        return (this._activeEndDateIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveStartDateMutated() {
        return (this._activeStartDateIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDescriptionMutated() {
        return (this._descriptionIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsFridayMutated() {
        return (this._fridayIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsMondayMutated() {
        return (this._mondayIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductMutated() {
        return (this._productIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductPricingMutated() {
        return (this._productPricingIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsSaturdayMutated() {
        return (this._saturdayIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsSundayMutated() {
        return (this._sundayIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsThursdayMutated() {
        return (this._thursdayIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsTuesdayMutated() {
        return (this._tuesdayIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsWednesdayMutated() {
        return (this._wednesdayIsMutated);
    }
    /**
     * Active end date.
     * @returns Date | null
     */
    getActiveEndDate() {
        return (this.activeEndDate);
    }
    /**
     * Active start date.
     * @returns Date | null
     */
    getActiveStartDate() {
        return (this.activeStartDate);
    }
    /**
     * The description.
     * @returns string | null
     */
    getDescription() {
        return (this.description);
    }
    /**
     * Boolean value for the day Friday.
     * @returns boolean
     */
    getFriday() {
        return ((this.friday === undefined) ? false : this.friday);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * Boolean value for the day Monday.
     * @returns boolean
     */
    getMonday() {
        return ((this.monday === undefined) ? false : this.monday);
    }
    /**
     * The name.
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
     * The product id.
     * @returns Product | null
     */
    getProduct() {
        return (this.product);
    }
    /**
     * The product pricing id.
     * @returns ProductPricing | null
     */
    getProductPricing() {
        return (this.productPricing);
    }
    /**
     * Boolean value for the day Saturday.
     * @returns boolean
     */
    getSaturday() {
        return ((this.saturday === undefined) ? false : this.saturday);
    }
    /**
     * Boolean value for the day Sunday.
     * @returns boolean
     */
    getSunday() {
        return ((this.sunday === undefined) ? false : this.sunday);
    }
    /**
     * Boolean value for the day Thursday.
     * @returns boolean
     */
    getThursday() {
        return ((this.thursday === undefined) ? false : this.thursday);
    }
    /**
     * Boolean value for the day Tuesday.
     * @returns boolean
     */
    getTuesday() {
        return ((this.tuesday === undefined) ? false : this.tuesday);
    }
    /**
     * Boolean value for the day wWdnesday.
     * @returns boolean
     */
    getWednesday() {
        return ((this.wednesday === undefined) ? false : this.wednesday);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveEndDateMutated(_activeEndDateIsMutated) {
        this._activeEndDateIsMutated = _activeEndDateIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveStartDateMutated(_activeStartDateIsMutated) {
        this._activeStartDateIsMutated = _activeStartDateIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDescriptionMutated(_descriptionIsMutated) {
        this._descriptionIsMutated = _descriptionIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsFridayMutated(_fridayIsMutated) {
        this._fridayIsMutated = _fridayIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsMondayMutated(_mondayIsMutated) {
        this._mondayIsMutated = _mondayIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductMutated(_productIsMutated) {
        this._productIsMutated = _productIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductPricingMutated(_productPricingIsMutated) {
        this._productPricingIsMutated = _productPricingIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsSaturdayMutated(_saturdayIsMutated) {
        this._saturdayIsMutated = _saturdayIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsSundayMutated(_sundayIsMutated) {
        this._sundayIsMutated = _sundayIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsThursdayMutated(_thursdayIsMutated) {
        this._thursdayIsMutated = _thursdayIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsTuesdayMutated(_tuesdayIsMutated) {
        this._tuesdayIsMutated = _tuesdayIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsWednesdayMutated(_wednesdayIsMutated) {
        this._wednesdayIsMutated = _wednesdayIsMutated;
        return (this);
    }
    /**
* ```
     * Active end date.
* ```
     */
    setActiveEndDate(activeEndDate) {
        if (!this._getIsMutateSuppressed() && this.activeEndDate !== activeEndDate) {
            this._setIsActiveEndDateMutated(true);
        }
        this.activeEndDate = activeEndDate;
        return (this);
    }
    /**
* ```
     * Active start date.
* ```
     */
    setActiveStartDate(activeStartDate) {
        if (!this._getIsMutateSuppressed() && this.activeStartDate !== activeStartDate) {
            this._setIsActiveStartDateMutated(true);
        }
        this.activeStartDate = activeStartDate;
        return (this);
    }
    /**
* ```
     * The description.
* ```
     */
    setDescription(description) {
        if (!this._getIsMutateSuppressed() && this.description !== description) {
            this._setIsDescriptionMutated(true);
        }
        this.description = description;
        return (this);
    }
    /**
* ```
     * Boolean value for the day Friday.
* ```
     */
    setFriday(friday) {
        if (!this._getIsMutateSuppressed() && this.friday !== friday) {
            this._setIsFridayMutated(true);
        }
        this.friday = friday;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * Boolean value for the day Monday.
* ```
     */
    setMonday(monday) {
        if (!this._getIsMutateSuppressed() && this.monday !== monday) {
            this._setIsMondayMutated(true);
        }
        this.monday = monday;
        return (this);
    }
    /**
* ```
     * The name.
* ```
     */
    setName(name) {
        if (!this._getIsMutateSuppressed() && this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    /**
* ```
     * The product id.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setProduct(product, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.product !== product) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(product, persistanceType);
            this._setIsProductMutated(true);
        }
        this.product = product;
        return (this);
    }
    /**
* ```
     * The product pricing id.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setProductPricing(productPricing, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.productPricing !== productPricing) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(productPricing, persistanceType);
            this._setIsProductPricingMutated(true);
        }
        this.productPricing = productPricing;
        return (this);
    }
    /**
* ```
     * Boolean value for the day Saturday.
* ```
     */
    setSaturday(saturday) {
        if (!this._getIsMutateSuppressed() && this.saturday !== saturday) {
            this._setIsSaturdayMutated(true);
        }
        this.saturday = saturday;
        return (this);
    }
    /**
* ```
     * Boolean value for the day Sunday.
* ```
     */
    setSunday(sunday) {
        if (!this._getIsMutateSuppressed() && this.sunday !== sunday) {
            this._setIsSundayMutated(true);
        }
        this.sunday = sunday;
        return (this);
    }
    /**
* ```
     * Boolean value for the day Thursday.
* ```
     */
    setThursday(thursday) {
        if (!this._getIsMutateSuppressed() && this.thursday !== thursday) {
            this._setIsThursdayMutated(true);
        }
        this.thursday = thursday;
        return (this);
    }
    /**
* ```
     * Boolean value for the day Tuesday.
* ```
     */
    setTuesday(tuesday) {
        if (!this._getIsMutateSuppressed() && this.tuesday !== tuesday) {
            this._setIsTuesdayMutated(true);
        }
        this.tuesday = tuesday;
        return (this);
    }
    /**
* ```
     * Boolean value for the day wWdnesday.
* ```
     */
    setWednesday(wednesday) {
        if (!this._getIsMutateSuppressed() && this.wednesday !== wednesday) {
            this._setIsWednesdayMutated(true);
        }
        this.wednesday = wednesday;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'activeEndDate',
            value: this.activeEndDate
        });
        this._setCurrentState2({
            attributeName: 'activeStartDate',
            value: this.activeStartDate
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'description',
            value: this.description
        });
        this._setCurrentState2({
            attributeName: 'friday',
            value: this.friday
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'monday',
            value: this.monday
        });
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._setCurrentState2({
            attributeName: 'product',
            value: this.product
        });
        this._setCurrentState2({
            attributeName: 'productPricing',
            value: this.productPricing
        });
        this._setCurrentState2({
            attributeName: 'saturday',
            value: this.saturday
        });
        this._setCurrentState2({
            attributeName: 'sunday',
            value: this.sunday
        });
        this._setCurrentState2({
            attributeName: 'thursday',
            value: this.thursday
        });
        this._setCurrentState2({
            attributeName: 'tuesday',
            value: this.tuesday
        });
        this._setCurrentState2({
            attributeName: 'wednesday',
            value: this.wednesday
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'activeEndDate',
            value: this.activeEndDate
        });
        this._setInitializedState2({
            attributeName: 'activeStartDate',
            value: this.activeStartDate
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'description',
            value: this.description
        });
        this._setInitializedState2({
            attributeName: 'friday',
            value: this.friday
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'monday',
            value: this.monday
        });
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        this._setInitializedState2({
            attributeName: 'product',
            value: this.product
        });
        this._setInitializedState2({
            attributeName: 'productPricing',
            value: this.productPricing
        });
        this._setInitializedState2({
            attributeName: 'saturday',
            value: this.saturday
        });
        this._setInitializedState2({
            attributeName: 'sunday',
            value: this.sunday
        });
        this._setInitializedState2({
            attributeName: 'thursday',
            value: this.thursday
        });
        this._setInitializedState2({
            attributeName: 'tuesday',
            value: this.tuesday
        });
        this._setInitializedState2({
            attributeName: 'wednesday',
            value: this.wednesday
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.activeEndDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.activeStartDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.description);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.friday);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.monday);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.product);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.productPricing);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.saturday);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.sunday);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.thursday);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.tuesday);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.wednesday);
        return (hashCode_n);
    }
}
exports.ProductServiceDayOfWeek = ProductServiceDayOfWeek;
ProductServiceDayOfWeek.metaDataForProductServiceDayOfWeek = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.ProductServiceDayOfWeek", ProductServiceDayOfWeek);
