// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/ProductPricingRuleKmlLocation.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPricingRuleKmlLocation = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
const PersistableUtils_1 = require("../../../data/PersistableUtils");
class EntityAttributes {
    constructor() {
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._kmlGeometryIsMutated = "_kmlGeometryIsMutated";
        this._kmlIsMutated = "_kmlIsMutated";
        this._kmlLayerIsMutated = "_kmlLayerIsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._productPricingRuleIdIsMutated = "_productPricingRuleIdIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._workspaceIsMutated = "_workspaceIsMutated";
        this.createdDate = "createdDate";
        this.id = "id";
        this.jsonType = "jsonType";
        this.kml = "kml";
        this.kmlGeometry = "kmlGeometry";
        this.kmlLayer = "kmlLayer";
        this.modifiedDate = "modifiedDate";
        this.organization = "organization";
        this.productPricingRuleId = "productPricingRuleId";
        this.workspace = "workspace";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._kmlGeometryIsMutated = "_kmlGeometryIsMutated";
        this._kmlIsMutated = "_kmlIsMutated";
        this._kmlLayerIsMutated = "_kmlLayerIsMutated";
        this._organizationIsMutated = "_organizationIsMutated";
        this._productPricingRuleIdIsMutated = "_productPricingRuleIdIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._workspaceIsMutated = "_workspaceIsMutated";
        this.createdDate = "createdDate";
        this._id = "_id";
        this.jsonType = "jsonType";
        this.kml = "kml";
        this.kmlGeometry = "kmlGeometry";
        this.kmlLayer = "kmlLayer";
        this.modifiedDate = "modifiedDate";
        this.organization = "organization";
        this.productPricingRuleId = "productPricingRuleId";
        this.workspace = "workspace";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.ProductPricingRuleKmlLocation";
        this.mongDataModeName = "ProductPricingRuleKmlLocation";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class ProductPricingRuleKmlLocation extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._kmlGeometryIsMutated = false;
        this._kmlIsMutated = false;
        this._kmlLayerIsMutated = false;
        this._organizationIsMutated = false;
        this._productPricingRuleIdIsMutated = false;
        this._workspaceIsMutated = false;
        this.id = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.ProductPricingRuleKmlLocation";
        this.kml = null;
        this.kmlGeometry = null;
        this.kmlLayer = null;
        this.organization = null;
        this.productPricingRuleId = null;
        this.workspace = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._kmlIsMutated = false;
        this._kmlGeometryIsMutated = false;
        this._kmlLayerIsMutated = false;
        this._organizationIsMutated = false;
        this._productPricingRuleIdIsMutated = false;
        this._workspaceIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.ProductPricingRuleKmlLocation");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsKmlGeometryMutated() {
        return (this._kmlGeometryIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsKmlMutated() {
        return (this._kmlIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsKmlLayerMutated() {
        return (this._kmlLayerIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsOrganizationMutated() {
        return (this._organizationIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsProductPricingRuleIdMutated() {
        return (this._productPricingRuleIdIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsWorkspaceMutated() {
        return (this._workspaceIsMutated);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The KML.
     * @returns Kml | null
     */
    getKml() {
        return (this.kml);
    }
    /**
     * The kml polygon id.
     * @returns KmlGeometry | null
     */
    getKmlGeometry() {
        return (this.kmlGeometry);
    }
    /**
     * The kml layer.
     * @returns KmlLayer | null
     */
    getKmlLayer() {
        return (this.kmlLayer);
    }
    /**
     * The organization.
     * @returns Organization | null
     */
    getOrganization() {
        return (this.organization);
    }
    /**
     * The product pricing rule.
     * @returns ProductPricingRule | null
     */
    getProductPricingRuleId() {
        return (this.productPricingRuleId);
    }
    /**
     * The workspace this kml location belong to.
     * @returns Workspace | null
     */
    getWorkspace() {
        return (this.workspace);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsKmlGeometryMutated(_kmlGeometryIsMutated) {
        this._kmlGeometryIsMutated = _kmlGeometryIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsKmlMutated(_kmlIsMutated) {
        this._kmlIsMutated = _kmlIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsKmlLayerMutated(_kmlLayerIsMutated) {
        this._kmlLayerIsMutated = _kmlLayerIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsOrganizationMutated(_organizationIsMutated) {
        this._organizationIsMutated = _organizationIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsProductPricingRuleIdMutated(_productPricingRuleIdIsMutated) {
        this._productPricingRuleIdIsMutated = _productPricingRuleIdIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsWorkspaceMutated(_workspaceIsMutated) {
        this._workspaceIsMutated = _workspaceIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The KML.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setKml(kml, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.kml !== kml) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(kml, persistanceType);
            this._setIsKmlMutated(true);
        }
        this.kml = kml;
        return (this);
    }
    /**
* ```
     * The kml polygon id.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setKmlGeometry(kmlGeometry, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.kmlGeometry !== kmlGeometry) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(kmlGeometry, persistanceType);
            this._setIsKmlGeometryMutated(true);
        }
        this.kmlGeometry = kmlGeometry;
        return (this);
    }
    /**
* ```
     * The kml layer.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setKmlLayer(kmlLayer, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.kmlLayer !== kmlLayer) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(kmlLayer, persistanceType);
            this._setIsKmlLayerMutated(true);
        }
        this.kmlLayer = kmlLayer;
        return (this);
    }
    /**
* ```
     * The organization.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setOrganization(organization, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.organization !== organization) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(organization, persistanceType);
            this._setIsOrganizationMutated(true);
        }
        this.organization = organization;
        return (this);
    }
    /**
* ```
     * The product pricing rule.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setProductPricingRuleId(productPricingRuleId, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.productPricingRuleId !== productPricingRuleId) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(productPricingRuleId, persistanceType);
            this._setIsProductPricingRuleIdMutated(true);
        }
        this.productPricingRuleId = productPricingRuleId;
        return (this);
    }
    /**
* ```
     * The workspace this kml location belong to.
     * persistanceType?: PersistenceType   If the persistanceType is not set and the Entity.id is null or undefined, will be set to PersistenceType.CREATE.
* ```
     */
    setWorkspace(workspace, persistanceType) {
        if (!this._getIsMutateSuppressed() && this.workspace !== workspace) {
            PersistableUtils_1.PersistableUtils.mongoRefSetter(workspace, persistanceType);
            this._setIsWorkspaceMutated(true);
        }
        this.workspace = workspace;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'kml',
            value: this.kml
        });
        this._setCurrentState2({
            attributeName: 'kmlGeometry',
            value: this.kmlGeometry
        });
        this._setCurrentState2({
            attributeName: 'kmlLayer',
            value: this.kmlLayer
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setCurrentState2({
            attributeName: 'productPricingRuleId',
            value: this.productPricingRuleId
        });
        this._setCurrentState2({
            attributeName: 'workspace',
            value: this.workspace
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'kml',
            value: this.kml
        });
        this._setInitializedState2({
            attributeName: 'kmlGeometry',
            value: this.kmlGeometry
        });
        this._setInitializedState2({
            attributeName: 'kmlLayer',
            value: this.kmlLayer
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'organization',
            value: this.organization
        });
        this._setInitializedState2({
            attributeName: 'productPricingRuleId',
            value: this.productPricingRuleId
        });
        this._setInitializedState2({
            attributeName: 'workspace',
            value: this.workspace
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.kml);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.kmlGeometry);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.kmlLayer);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.organization);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.productPricingRuleId);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.workspace);
        return (hashCode_n);
    }
}
exports.ProductPricingRuleKmlLocation = ProductPricingRuleKmlLocation;
ProductPricingRuleKmlLocation.metaDataForProductPricingRuleKmlLocation = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.ProductPricingRuleKmlLocation", ProductPricingRuleKmlLocation);
