// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/SecurityPermission.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityPermission = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isActiveIsMutated = "_isActiveIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._isDeletableIsMutated = "_isDeletableIsMutated";
        this._isExcutableIsMutated = "_isExcutableIsMutated";
        this._isReadableIsMutated = "_isReadableIsMutated";
        this._isRemoveFromRelationship = "_isRemoveFromRelationship";
        this._isRemoveFromRelationshipIsMutated = "_isRemoveFromRelationshipIsMutated";
        this._isWritableIsMutated = "_isWritableIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._restrictionLevelIsMutated = "_restrictionLevelIsMutated";
        this._ssOnlyIsMutated = "_ssOnlyIsMutated";
        this.createdDate = "createdDate";
        this.description = "description";
        this.id = "id";
        this.isActive = "isActive";
        this.isDeletable = "isDeletable";
        this.isExcutable = "isExcutable";
        this.isReadable = "isReadable";
        this.isWritable = "isWritable";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.restrictionLevel = "restrictionLevel";
        this.ssOnly = "ssOnly";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isActiveIsMutated = "_isActiveIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._isDeletableIsMutated = "_isDeletableIsMutated";
        this._isExcutableIsMutated = "_isExcutableIsMutated";
        this._isReadableIsMutated = "_isReadableIsMutated";
        this._isRemoveFromRelationship = "_isRemoveFromRelationship";
        this._isRemoveFromRelationshipIsMutated = "_isRemoveFromRelationshipIsMutated";
        this._isWritableIsMutated = "_isWritableIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._relationshipPersistableType = "_relationshipPersistableType";
        this._restrictionLevelIsMutated = "_restrictionLevelIsMutated";
        this._ssOnlyIsMutated = "_ssOnlyIsMutated";
        this.createdDate = "createdDate";
        this.description = "description";
        this._id = "_id";
        this.isActive = "isActive";
        this.isDeletable = "isDeletable";
        this.isExcutable = "isExcutable";
        this.isReadable = "isReadable";
        this.isWritable = "isWritable";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.restrictionLevel = "restrictionLevel";
        this.ssOnly = "ssOnly";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.SecurityPermission";
        this.mongDataModeName = "SecurityPermission";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class SecurityPermission extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._descriptionIsMutated = false;
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isActiveIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._isDeletableIsMutated = false;
        this._isExcutableIsMutated = false;
        this._isReadableIsMutated = false;
        this._isRemoveFromRelationship = false;
        this._isRemoveFromRelationshipIsMutated = false;
        this._isWritableIsMutated = false;
        this._nameIsMutated = false;
        this._restrictionLevelIsMutated = false;
        this._ssOnlyIsMutated = false;
        this.description = null;
        this.id = null;
        this.isActive = undefined;
        this.isDeletable = undefined;
        this.isExcutable = undefined;
        this.isReadable = undefined;
        this.isWritable = undefined;
        this.jsonType = "com.salesstryke.entity.classes.persistable.SecurityPermission";
        this.name = null;
        this.restrictionLevel = 0;
        this.ssOnly = undefined;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._descriptionIsMutated = false;
        this._isActiveIsMutated = false;
        this._isDeletableIsMutated = false;
        this._isExcutableIsMutated = false;
        this._isReadableIsMutated = false;
        this._isWritableIsMutated = false;
        this._nameIsMutated = false;
        this._restrictionLevelIsMutated = false;
        this._ssOnlyIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.SecurityPermission");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     * A system attribute to identify the relationship persisable type.
     * @returns number
     */
    _getRelationshipPersistableType() {
        //return(super._getEntityState());
        return (super._getRelationshipPersistableType());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsDescriptionMutated() {
        return (this._descriptionIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsActiveMutated() {
        return (this._isActiveIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsDeletableMutated() {
        return (this._isDeletableIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsExcutableMutated() {
        return (this._isExcutableIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsReadableMutated() {
        return (this._isReadableIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRemoveFromRelationship() {
        return (this._isRemoveFromRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRemoveFromRelationshipMutated() {
        return (this._isRemoveFromRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsWritableMutated() {
        return (this._isWritableIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRestrictionLevelMutated() {
        return (this._restrictionLevelIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsSsOnlyMutated() {
        return (this._ssOnlyIsMutated);
    }
    /**
     * The description of the permission.
     * @returns string | null
     */
    getDescription() {
        return (this.description);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * Is the permission actived.
     * @returns boolean
     */
    getIsActive() {
        return ((this.isActive === undefined) ? false : this.isActive);
    }
    /**
     * Can Delete.
     * @returns boolean
     */
    getIsDeletable() {
        return ((this.isDeletable === undefined) ? false : this.isDeletable);
    }
    /**
     * Can Excute.
     * @returns boolean
     */
    getIsExcutable() {
        return ((this.isExcutable === undefined) ? false : this.isExcutable);
    }
    /**
     * Can Read.
     * @returns boolean
     */
    getIsReadable() {
        return ((this.isReadable === undefined) ? false : this.isReadable);
    }
    /**
     * Can Write.
     * @returns boolean
     */
    getIsWritable() {
        return ((this.isWritable === undefined) ? false : this.isWritable);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The name of the permission.
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
     * The restriction level of the permission.
     * @returns number
     */
    getRestrictionLevel() {
        return (this.restrictionLevel);
    }
    /**
     * salesstryke user only.
     * @returns boolean
     */
    getSsOnly() {
        return ((this.ssOnly === undefined) ? false : this.ssOnly);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     * A system attribute to identify the relationship persisable type.
     */
    _setRelationshipPersistableType(_relationshipPersistableType) {
        super._setRelationshipPersistableType(_relationshipPersistableType);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDescriptionMutated(_descriptionIsMutated) {
        this._descriptionIsMutated = _descriptionIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsActiveMutated(_isActiveIsMutated) {
        this._isActiveIsMutated = _isActiveIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDeletableMutated(_isDeletableIsMutated) {
        this._isDeletableIsMutated = _isDeletableIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsExcutableMutated(_isExcutableIsMutated) {
        this._isExcutableIsMutated = _isExcutableIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsReadableMutated(_isReadableIsMutated) {
        this._isReadableIsMutated = _isReadableIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRemoveFromRelationship(_isRemoveFromRelationship) {
        if (this._isRemoveFromRelationship !== _isRemoveFromRelationship) {
            this._setIsRemoveFromRelationshipMutated(true);
        }
        this._isRemoveFromRelationship = _isRemoveFromRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRemoveFromRelationshipMutated(_isRemoveFromRelationshipIsMutated) {
        this._isRemoveFromRelationshipIsMutated = _isRemoveFromRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsWritableMutated(_isWritableIsMutated) {
        this._isWritableIsMutated = _isWritableIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRestrictionLevelMutated(_restrictionLevelIsMutated) {
        this._restrictionLevelIsMutated = _restrictionLevelIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsSsOnlyMutated(_ssOnlyIsMutated) {
        this._ssOnlyIsMutated = _ssOnlyIsMutated;
        return (this);
    }
    /**
* ```
     * The description of the permission.
* ```
     */
    setDescription(description) {
        if (!this._getIsMutateSuppressed() && this.description !== description) {
            this._setIsDescriptionMutated(true);
        }
        this.description = description;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     * Is the permission actived.
* ```
     */
    setIsActive(isActive) {
        if (!this._getIsMutateSuppressed() && this.isActive !== isActive) {
            this._setIsActiveMutated(true);
        }
        this.isActive = isActive;
        return (this);
    }
    /**
* ```
     * Can Delete.
* ```
     */
    setIsDeletable(isDeletable) {
        if (!this._getIsMutateSuppressed() && this.isDeletable !== isDeletable) {
            this._setIsDeletableMutated(true);
        }
        this.isDeletable = isDeletable;
        return (this);
    }
    /**
* ```
     * Can Excute.
* ```
     */
    setIsExcutable(isExcutable) {
        if (!this._getIsMutateSuppressed() && this.isExcutable !== isExcutable) {
            this._setIsExcutableMutated(true);
        }
        this.isExcutable = isExcutable;
        return (this);
    }
    /**
* ```
     * Can Read.
* ```
     */
    setIsReadable(isReadable) {
        if (!this._getIsMutateSuppressed() && this.isReadable !== isReadable) {
            this._setIsReadableMutated(true);
        }
        this.isReadable = isReadable;
        return (this);
    }
    /**
* ```
     * Can Write.
* ```
     */
    setIsWritable(isWritable) {
        if (!this._getIsMutateSuppressed() && this.isWritable !== isWritable) {
            this._setIsWritableMutated(true);
        }
        this.isWritable = isWritable;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The name of the permission.
* ```
     */
    setName(name) {
        if (!this._getIsMutateSuppressed() && this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    /**
* ```
     * The restriction level of the permission.
* ```
     */
    setRestrictionLevel(restrictionLevel) {
        if (!this._getIsMutateSuppressed() && this.restrictionLevel !== restrictionLevel) {
            this._setIsRestrictionLevelMutated(true);
        }
        this.restrictionLevel = restrictionLevel;
        return (this);
    }
    /**
* ```
     * salesstryke user only.
* ```
     */
    setSsOnly(ssOnly) {
        if (!this._getIsMutateSuppressed() && this.ssOnly !== ssOnly) {
            this._setIsSsOnlyMutated(true);
        }
        this.ssOnly = ssOnly;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_isRemoveFromRelationship',
            value: this._isRemoveFromRelationship
        });
        this._setCurrentState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'description',
            value: this.description
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'isActive',
            value: this.isActive
        });
        this._setCurrentState2({
            attributeName: 'isDeletable',
            value: this.isDeletable
        });
        this._setCurrentState2({
            attributeName: 'isExcutable',
            value: this.isExcutable
        });
        this._setCurrentState2({
            attributeName: 'isReadable',
            value: this.isReadable
        });
        this._setCurrentState2({
            attributeName: 'isWritable',
            value: this.isWritable
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._setCurrentState2({
            attributeName: 'restrictionLevel',
            value: this.restrictionLevel
        });
        this._setCurrentState2({
            attributeName: 'ssOnly',
            value: this.ssOnly
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_isRemoveFromRelationship',
            value: this._isRemoveFromRelationship
        });
        this._setInitializedState2({
            attributeName: '_relationshipPersistableType',
            value: this._relationshipPersistableType
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'description',
            value: this.description
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'isActive',
            value: this.isActive
        });
        this._setInitializedState2({
            attributeName: 'isDeletable',
            value: this.isDeletable
        });
        this._setInitializedState2({
            attributeName: 'isExcutable',
            value: this.isExcutable
        });
        this._setInitializedState2({
            attributeName: 'isReadable',
            value: this.isReadable
        });
        this._setInitializedState2({
            attributeName: 'isWritable',
            value: this.isWritable
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        this._setInitializedState2({
            attributeName: 'restrictionLevel',
            value: this.restrictionLevel
        });
        this._setInitializedState2({
            attributeName: 'ssOnly',
            value: this.ssOnly
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this._isRemoveFromRelationship);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.description);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.isActive);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.isDeletable);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.isExcutable);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.isReadable);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.isWritable);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.restrictionLevel);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.ssOnly);
        return (hashCode_n);
    }
}
exports.SecurityPermission = SecurityPermission;
SecurityPermission.metaDataForSecurityPermission = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.SecurityPermission", SecurityPermission);
