// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/persistable/SystemVersion.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemVersion = void 0;
const ClassFor_1 = require("../../../data/ClassFor");
const JsonApiPersistableBase_1 = require("../../../jsonApi/JsonApiPersistableBase");
const HashCodeUtil_1 = require("../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._isRemoveFromRelationship = "_isRemoveFromRelationship";
        this._isRemoveFromRelationshipIsMutated = "_isRemoveFromRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._updateCompletedIsMutated = "_updateCompletedIsMutated";
        this._updateStartedIsMutated = "_updateStartedIsMutated";
        this._versionIsMutated = "_versionIsMutated";
        this._versionUUIDIsMutated = "_versionUUIDIsMutated";
        this.createdDate = "createdDate";
        this.description = "description";
        this.id = "id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.updateCompleted = "updateCompleted";
        this.updateStarted = "updateStarted";
        this.version = "version";
        this.versionUUID = "versionUUID";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._descriptionIsMutated = "_descriptionIsMutated";
        this._entityState = "_entityState";
        this._idSetBySystem = "_idSetBySystem";
        this._initializedDataKey = "_initializedDataKey";
        this._initializedDataKeyIsMutated = "_initializedDataKeyIsMutated";
        this._isAddToRelationship = "_isAddToRelationship";
        this._isAddToRelationshipIsMutated = "_isAddToRelationshipIsMutated";
        this._isRemoveFromRelationship = "_isRemoveFromRelationship";
        this._isRemoveFromRelationshipIsMutated = "_isRemoveFromRelationshipIsMutated";
        this._jsonStateStr = "_jsonStateStr";
        this._nameIsMutated = "_nameIsMutated";
        this._updateCompletedIsMutated = "_updateCompletedIsMutated";
        this._updateStartedIsMutated = "_updateStartedIsMutated";
        this._versionIsMutated = "_versionIsMutated";
        this._versionUUIDIsMutated = "_versionUUIDIsMutated";
        this.createdDate = "createdDate";
        this.description = "description";
        this._id = "_id";
        this.jsonType = "jsonType";
        this.modifiedDate = "modifiedDate";
        this.name = "name";
        this.updateCompleted = "updateCompleted";
        this.updateStarted = "updateStarted";
        this.version = "version";
        this.versionUUID = "versionUUID";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.persistable.SystemVersion";
        this.mongDataModeName = "SystemVersion";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class SystemVersion extends JsonApiPersistableBase_1.JsonApiPersistableBase {
    constructor(p) {
        super(p);
        this._descriptionIsMutated = false;
        this._idSetBySystem = false;
        this._initializedDataKey = null;
        this._initializedDataKeyIsMutated = false;
        this._isAddToRelationship = false;
        this._isAddToRelationshipIsMutated = false;
        this._isRemoveFromRelationship = false;
        this._isRemoveFromRelationshipIsMutated = false;
        this._nameIsMutated = false;
        this._updateCompletedIsMutated = false;
        this._updateStartedIsMutated = false;
        this._versionIsMutated = false;
        this._versionUUIDIsMutated = false;
        this.description = null;
        this.id = null;
        this.jsonType = "com.salesstryke.entity.classes.persistable.SystemVersion";
        this.name = null;
        this.updateCompleted = undefined;
        this.updateStarted = undefined;
        this.version = null;
        this.versionUUID = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        this._initializedDataKeyIsMutated = false;
        this._descriptionIsMutated = false;
        this._nameIsMutated = false;
        this._updateCompletedIsMutated = false;
        this._updateStartedIsMutated = false;
        this._versionIsMutated = false;
        this._versionUUIDIsMutated = false;
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.persistable.SystemVersion");
    }
    /**
     * CREATE, DELETE, UPDATE
     * @returns string | null
     */
    _getEntityState() {
        //return(super._getEntityState());
        return (super._getEntityState());
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     * @returns string | null
     */
    _getJsonStateStr() {
        //return(super._getEntityState());
        return (super._getJsonStateStr());
    }
    /**
     *
     * @returns Date | null
     */
    getCreatedDate() {
        //return(super._getEntityState());
        return (super.getCreatedDate());
    }
    /**
     * The modified date of the document
     * @returns Date | null
     */
    getModifiedDate() {
        //return(super._getEntityState());
        return (super.getModifiedDate());
    }
    /**
     *
     * @returns boolean
     */
    _getIsDescriptionMutated() {
        return (this._descriptionIsMutated);
    }
    /**
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
     * @returns boolean
     */
    _getIsIdSetBySystem() {
        return (this._idSetBySystem);
    }
    /**
     * This is a key that is used to initialize data one time. Example: setting the USState information.
     * @returns string | null
     */
    _getInitializedDataKey() {
        return (this._initializedDataKey);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInitializedDataKeyMutated() {
        return (this._initializedDataKeyIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationship() {
        return (this._isAddToRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddToRelationshipMutated() {
        return (this._isAddToRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRemoveFromRelationship() {
        return (this._isRemoveFromRelationship);
    }
    /**
     *
     * @returns boolean
     */
    _getIsRemoveFromRelationshipMutated() {
        return (this._isRemoveFromRelationshipIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsUpdateCompletedMutated() {
        return (this._updateCompletedIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsUpdateStartedMutated() {
        return (this._updateStartedIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsVersionMutated() {
        return (this._versionIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsVersionUUIDMutated() {
        return (this._versionUUIDIsMutated);
    }
    /**
     * The description of updated system item.
     * @returns string | null
     */
    getDescription() {
        return (this.description);
    }
    /**
     *
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * The name of the updated system item.
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
     * Was the updated completed.
     * @returns boolean
     */
    getUpdateCompleted() {
        return ((this.updateCompleted === undefined) ? false : this.updateCompleted);
    }
    /**
     * Was the updated started.
     * @returns boolean
     */
    getUpdateStarted() {
        return ((this.updateStarted === undefined) ? false : this.updateStarted);
    }
    /**
     * The version of the update system item.
     * @returns string | null
     */
    getVersion() {
        return (this.version);
    }
    /**
     * The version UUID of the updated system item.
     * @returns string | null
     */
    getVersionUUID() {
        return (this.versionUUID);
    }
    /**
     * CREATE, DELETE, UPDATE
     */
    _setEntityState(_entityState) {
        super._setEntityState(_entityState);
        return (this);
    }
    /**
     * Is a JSON string that passes the state of attributes from client to server, example: '{'attributes':[{'name':'userName','isMutated':true},{'name':'roles','isMutated':true}]}'
     */
    _setJsonStateStr(_jsonStateStr) {
        super._setJsonStateStr(_jsonStateStr);
        return (this);
    }
    /**
     *
     */
    setCreatedDate(createdDate) {
        super.setCreatedDate(createdDate);
        return (this);
    }
    /**
     * The modified date of the document
     */
    setModifiedDate(modifiedDate) {
        super.setModifiedDate(modifiedDate);
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsDescriptionMutated(_descriptionIsMutated) {
        this._descriptionIsMutated = _descriptionIsMutated;
        return (this);
    }
    /**
* ```
     * Indicates if the ID was set by the system and that the ID is used to create the documet.
* ```
     */
    _setIsIdSetBySystem(_idSetBySystem) {
        this._idSetBySystem = _idSetBySystem;
        return (this);
    }
    /**
* ```
     * This is a key that is used to initialize data one time. Example: setting the USState information.
* ```
     */
    _setInitializedDataKey(_initializedDataKey) {
        if (!this._getIsMutateSuppressed() && this._initializedDataKey !== _initializedDataKey) {
            this._setIsInitializedDataKeyMutated(true);
        }
        this._initializedDataKey = _initializedDataKey;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInitializedDataKeyMutated(_initializedDataKeyIsMutated) {
        this._initializedDataKeyIsMutated = _initializedDataKeyIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationship(_isAddToRelationship) {
        if (this._isAddToRelationship !== _isAddToRelationship) {
            this._setIsAddToRelationshipMutated(true);
        }
        this._isAddToRelationship = _isAddToRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddToRelationshipMutated(_isAddToRelationshipIsMutated) {
        this._isAddToRelationshipIsMutated = _isAddToRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRemoveFromRelationship(_isRemoveFromRelationship) {
        if (this._isRemoveFromRelationship !== _isRemoveFromRelationship) {
            this._setIsRemoveFromRelationshipMutated(true);
        }
        this._isRemoveFromRelationship = _isRemoveFromRelationship;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsRemoveFromRelationshipMutated(_isRemoveFromRelationshipIsMutated) {
        this._isRemoveFromRelationshipIsMutated = _isRemoveFromRelationshipIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsUpdateCompletedMutated(_updateCompletedIsMutated) {
        this._updateCompletedIsMutated = _updateCompletedIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsUpdateStartedMutated(_updateStartedIsMutated) {
        this._updateStartedIsMutated = _updateStartedIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsVersionMutated(_versionIsMutated) {
        this._versionIsMutated = _versionIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsVersionUUIDMutated(_versionUUIDIsMutated) {
        this._versionUUIDIsMutated = _versionUUIDIsMutated;
        return (this);
    }
    /**
* ```
     * The description of updated system item.
* ```
     */
    setDescription(description) {
        if (!this._getIsMutateSuppressed() && this.description !== description) {
            this._setIsDescriptionMutated(true);
        }
        this.description = description;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setId(id) {
        this.id = id;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * The name of the updated system item.
* ```
     */
    setName(name) {
        if (!this._getIsMutateSuppressed() && this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    /**
* ```
     * Was the updated completed.
* ```
     */
    setUpdateCompleted(updateCompleted) {
        if (!this._getIsMutateSuppressed() && this.updateCompleted !== updateCompleted) {
            this._setIsUpdateCompletedMutated(true);
        }
        this.updateCompleted = updateCompleted;
        return (this);
    }
    /**
* ```
     * Was the updated started.
* ```
     */
    setUpdateStarted(updateStarted) {
        if (!this._getIsMutateSuppressed() && this.updateStarted !== updateStarted) {
            this._setIsUpdateStartedMutated(true);
        }
        this.updateStarted = updateStarted;
        return (this);
    }
    /**
* ```
     * The version of the update system item.
* ```
     */
    setVersion(version) {
        if (!this._getIsMutateSuppressed() && this.version !== version) {
            this._setIsVersionMutated(true);
        }
        this.version = version;
        return (this);
    }
    /**
* ```
     * The version UUID of the updated system item.
* ```
     */
    setVersionUUID(versionUUID) {
        if (!this._getIsMutateSuppressed() && this.versionUUID !== versionUUID) {
            this._setIsVersionUUIDMutated(true);
        }
        this.versionUUID = versionUUID;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: '_isRemoveFromRelationship',
            value: this._isRemoveFromRelationship
        });
        this._setCurrentState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setCurrentState2({
            attributeName: 'description',
            value: this.description
        });
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._setCurrentState2({
            attributeName: 'updateCompleted',
            value: this.updateCompleted
        });
        this._setCurrentState2({
            attributeName: 'updateStarted',
            value: this.updateStarted
        });
        this._setCurrentState2({
            attributeName: 'version',
            value: this.version
        });
        this._setCurrentState2({
            attributeName: 'versionUUID',
            value: this.versionUUID
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: '_isRemoveFromRelationship',
            value: this._isRemoveFromRelationship
        });
        this._setInitializedState2({
            attributeName: 'createdDate',
            value: this.createdDate
        });
        this._setInitializedState2({
            attributeName: 'description',
            value: this.description
        });
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'modifiedDate',
            value: this.modifiedDate
        });
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        this._setInitializedState2({
            attributeName: 'updateCompleted',
            value: this.updateCompleted
        });
        this._setInitializedState2({
            attributeName: 'updateStarted',
            value: this.updateStarted
        });
        this._setInitializedState2({
            attributeName: 'version',
            value: this.version
        });
        this._setInitializedState2({
            attributeName: 'versionUUID',
            value: this.versionUUID
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this._isRemoveFromRelationship);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.createdDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.description);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.modifiedDate);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.updateCompleted);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.updateStarted);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.version);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.versionUUID);
        return (hashCode_n);
    }
}
exports.SystemVersion = SystemVersion;
SystemVersion.metaDataForSystemVersion = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.persistable.SystemVersion", SystemVersion);
