// @salesstryke/mobile-api/src/com/salesstryke/utils/AnyUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnyUtils = void 0;
const DateUtils_1 = require("./DateUtils");
class AnyUtils {
    static exists(value_any) {
        let exists = false;
        if (exists !== undefined && exists !== null) {
            exists = true;
        }
        return (exists);
    }
    static toAny(value_any) {
        let toAny = {};
        if (this.exists(value_any)) {
            toAny = value_any;
        }
        return (toAny);
    }
    static toArray(value_any) {
        let array_any = new Array();
        if (this.exists(value_any)) {
            array_any = value_any;
        }
        return (array_any);
    }
    static toBoolean(value_any) {
        let boolean_b = false;
        if (value_any !== undefined && value_any !== null) {
            if (value_any) {
                boolean_b = true;
            }
        }
        return (boolean_b);
    }
    static toDate(value_any) {
        let date = null;
        if (this.exists(value_any)) {
            date = DateUtils_1.DateUtils.stringToDate({ dateAsString: value_any });
        }
        return (date);
    }
    static toNumber(value_any) {
        let value_n = 0;
        if (this.exists(value_any)) {
            value_n = value_any;
        }
        return (value_n);
    }
    static toString(value_any) {
        let value_s = '';
        if (this.exists(value_any)) {
            value_s = value_any;
        }
        return (value_s);
    }
}
exports.AnyUtils = AnyUtils;
