// @salesstryke/mobile-api/src/com/salesstryke/utils/RandomData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RandomData = void 0;
class RandomData {
    /**
     * Returns a random email address.
     * @returns
     */
    static getRandomEmailAddress() {
        let randomData_s = "";
        let name_s = RandomData.getRandomString(8);
        let domain_s = RandomData.getRandomString(10);
        randomData_s = `${name_s}@${domain_s}.com`;
        return (randomData_s);
    }
    static getRandomHash(length_i) {
        let randomData_s = "";
        // TODO: test this eslint
        // eslint-disable-next-line
        let characters_s = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*(){}[]<>/?;:+=\|';
        let size_i = characters_s.length;
        for (let index_i = 0; index_i < length_i; index_i++) {
            randomData_s += characters_s.charAt(Math.floor(Math.random() * size_i));
        }
        return (randomData_s);
    }
    static getRandomIntegerInRange(min_i, max_i) {
        let min = Math.ceil(min_i);
        let max = Math.floor(max_i);
        return Math.floor(Math.random() * (max - min + 1)) + min;
    }
    /**
     * Returns a random number between min and max.
     * @param min_i
     * @param max_i
     * @returns
     */
    static getRandomNumberInRange(min_i, max_i) {
        let randomNumber_i = Math.random() * (max_i - min_i) + min_i;
        return (randomNumber_i);
    }
    /**
     * Returns a random string of the given lenth.
     * @param length_i
     * @returns
     */
    static getRandomString(length_i) {
        let randomData_s = "";
        let characters_s = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
        let size_i = characters_s.length;
        for (let index_i = 0; index_i < length_i; index_i++) {
            randomData_s += characters_s.charAt(Math.floor(Math.random() * size_i));
        }
        return (randomData_s);
    }
}
exports.RandomData = RandomData;
