// @salesstryke/mobile-api/src/com/salesstryke/ruleEngine/BooleanOperatorToken.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.BooleanOperatorToken = void 0;
const Token_1 = require("./Token");
class BooleanOperatorToken extends Token_1.Token {
    constructor(p) {
        super({
            defaultLabel: p.defaultLabel,
            labelI18nKey: p.labelI18nKey
        });
        this.tokens = new Array();
    }
    addToken(p) {
        this.tokens.push(p.token);
    }
    static getBooleanOperatorTypeTokens() {
        return (_a.booleanOperatorTypeTokens);
    }
    getBooleanOperatorTypeTokens() {
        return (_a.booleanOperatorTypeTokens);
    }
    getLastToken() {
        if (this.tokens.length === 0) {
            return (undefined);
        }
        else {
            return (this.tokens[this.tokens.length - 1]);
        }
    }
    getTokens() {
        return (this.tokens);
    }
}
exports.BooleanOperatorToken = BooleanOperatorToken;
_a = BooleanOperatorToken;
BooleanOperatorToken.booleanOperatorTypeTokens = new Array();
(() => {
    _a.booleanOperatorTypeTokens.push(new Token_1.Token({ defaultLabel: 'AND', labelI18nKey: '' }));
    _a.booleanOperatorTypeTokens.push(new Token_1.Token({ defaultLabel: 'OR', labelI18nKey: '' }));
})();
