// @salesstryke/mobile-api/src/com/salesstryke/ruleEngine/RuleEngine.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleEngine = void 0;
const Token_1 = require("./Token");
class RuleEngine {
    constructor(p) {
        this.attributes = new Array();
        this.tokens = new Array();
        this.attributes = p.attributes;
    }
    addToken(p) {
        this.tokens.push(p.token);
        return (this);
    }
    getAttributes() {
        return (this.attributes);
    }
    static getComparators() {
        return (_a.comparators);
    }
    getComparators() {
        return (_a.comparators);
    }
    getTokens() {
        return (this.tokens);
    }
}
exports.RuleEngine = RuleEngine;
_a = RuleEngine;
RuleEngine.comparators = new Array();
(() => {
    _a.comparators.push(new Token_1.Token({ defaultLabel: 'Equal', labelI18nKey: 'f5e2b3bf-016e-44fa-8b3e-7cf73e5f8fd7' }));
    _a.comparators.push(new Token_1.Token({ defaultLabel: 'Greater Than', labelI18nKey: '' }));
    _a.comparators.push(new Token_1.Token({ defaultLabel: 'Less Than', labelI18nKey: '' }));
    _a.comparators.push(new Token_1.Token({ defaultLabel: 'Not Equal', labelI18nKey: '' }));
    _a.comparators.push(new Token_1.Token({ defaultLabel: 'Not Greater Than', labelI18nKey: '' }));
    _a.comparators.push(new Token_1.Token({ defaultLabel: 'Not Less Than', labelI18nKey: '' }));
})();
