// @salesstryke/mobile-api/src/com/salesstryke/dailysalesreportconfig/springCreek.salesReportConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.springCreekResidentialSalesReportConfig = void 0;
const salesReportUtiil_1 = require("../utils/salesreport/salesReportUtiil");
// TODO: Delete after database is setup
// need to create something that could set and update header key and name easy in the front end.
const residentialHeaders = {
    "Signup Date": "0",
    "Start Date": "1",
    "Service First Name": "2",
    "Service Last Name": "3",
    "Service Address": "4",
    "Service City": "5",
    "Service State": "6",
    "Service Zip": "7",
    "Service Phone": "8",
    "Service Email": "9",
    // "Service Lat": "10",
    // "Service Lng": "11",
    "Bill Name": "10",
    "Bill Address": "11",
    "Bill City": "12",
    "Bill State": "13",
    "Bill Zip": "14",
    "Bill Phone": "15",
    "Bill Email": "16",
    "Billing Frequency": "17",
    "Service Day": "18",
    "Trash Service": "20",
    "Trash Service Rate": "21",
    "Discount": "22",
    "Total Paid": "23",
    "E-Billing": "24",
    "Autopay": "25",
    "Autopay Payment Method": "26",
    "Document Status": "27",
    "Routing Software Sync Status": "28",
    // "Modify_Date",
    // "Panda_ID",
};
const residentialQueries = [
    // FIELD
    // {
    //     type: 'lat',
    //     searchParams: [
    //         {
    //             conditional: true,
    //             section: 'field',
    //             params: [
    //                 {
    //                     statement: 'none',
    //                     key: 'lat',
    //                     value: ''
    //                 }
    //             ]
    //         }
    //     ],
    //     columnId: residentialHeaders["Service Lat"],
    // },
    // {
    //     type: 'lng',
    //     searchParams: [
    //         {
    //             conditional: true,
    //             section: 'field',
    //             params: [
    //                 {
    //                     statement: 'none',
    //                     key: 'lng',
    //                     value: ''
    //                 }
    //             ]
    //         }
    //     ],
    //     columnId: residentialHeaders["Service Lng"],
    // },
    {
        type: 'date',
        // dateFormat: 'YYYY-MM-DD',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'signatureDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Signup Date"],
    },
    {
        type: 'date',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceStartDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Start Date"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceFirstName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service First Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Last Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'servicePhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Email"],
    },
    {
        type: 'concat',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingFirstName',
                        value: ''
                    }
                ]
            },
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingPhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Email"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceDay',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Day"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'RadioButtons',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Autopay Payment Method"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'RadioButtons1',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Autopay Payment Method"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'paperless',
                        value: ''
                    }
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "",
                newValue: "N"
            },
            {
                defaultValue: "false",
                newValue: "N"
            },
            {
                defaultValue: "null",
                newValue: "N"
            },
            {
                defaultValue: "true",
                newValue: "Y"
            }
        ],
        columnId: residentialHeaders["Autopay"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'paperless',
                        value: ''
                    }
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "",
                newValue: "N"
            },
            {
                defaultValue: "false",
                newValue: "N"
            },
            {
                defaultValue: "null",
                newValue: "N"
            },
            {
                defaultValue: "true",
                newValue: "Y"
            }
        ],
        columnId: residentialHeaders["E-Billing"],
    },
    // END OF FIELD
    ///////////////////////////////
    // ITEM
    // Service Item
    {
        type: 'name',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Trash Service'
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Trash Service"],
        value: ''
    },
    // Service Rate
    {
        type: 'compound_price',
        searchParams: [
            {
                section: 'price_item',
                conditional: false,
                optional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Trash Service']
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Trash Service Rate"],
        value: 0,
    },
    // Billing Frequency
    {
        type: 'default',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Monthly Billing'
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Billing Frequency"],
        value: 'Monthly'
    },
    {
        type: 'default',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Quarterly Billing'
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Billing Frequency"],
        value: 'Quarterly'
    },
    {
        type: 'default',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Semi-Annual Billing'
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Billing Frequency"],
        value: 'Semi-Annual'
    },
    {
        type: 'default',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Yearly Billing'
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Billing Frequency"],
        value: 'Yearly'
    },
    // Discount
    {
        type: 'price',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Discount']
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Discount"],
        value: 0,
    },
];
const residentialSheetConfig = {
    sheetName: 'Residentail',
    showRootField: {
        grandTotalAmount: residentialHeaders["Total Paid"],
    },
    workspaceDocumentVariables: [
        {
            columnId: residentialHeaders["Document Status"],
            variableName: 'pandaDocStatus',
        },
        {
            columnId: residentialHeaders["Routing Software Sync Status"],
            variableName: 'routingSoftwareSyncStatus',
        },
    ],
    queries: residentialQueries,
    headers: (0, salesReportUtiil_1.createSalesReportHeaderFromObject)(residentialHeaders),
};
exports.springCreekResidentialSalesReportConfig = {
    reportName: 'Spring Creek',
    sheetConfigs: [residentialSheetConfig]
};
