// @salesstryke/mobile-api/src/com/salesstryke/dailysalesreportconfig/trigg.salesReportConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.triggResidentialSalesReportConfig = void 0;
const salesReportUtiil_1 = require("../utils/salesreport/salesReportUtiil");
// TODO: Delete after database is setup
// need to create something that could set and update header key and name easy in the front end.
const residentialHeaders = {
    "Signup Date": "0",
    "Start Date": "1",
    "Srv First Name": "2",
    "Srv Last Name": "3",
    "Srv Address": "4",
    "Srv City": "5",
    "Srv State": "6",
    "Srv Zip": "7",
    "Srv Phone": "8",
    "Srv Email": "9",
    // "Srv Lat": "10",
    // "Srv Lng": "11",
    "Bill Name": "10",
    "Bill Address": "11",
    "Bill City": "12",
    "Bill State": "13",
    "Bill Zip": "14",
    "Bill Phone": "15",
    "Bill Email": "16",
    "Service Day": "17",
    "# of Carts": "18",
    "Billing Frequency": "19",
    "Billing Amount": "20",
    "Fuel Surcharge": "21",
    "Setup Fee": "22",
    "Pink Cart": "23",
    "Total Paid": "24",
    "E-Billing": "25",
    "Autopay": "26",
    "Document Status": "28",
    "Routing Software Sync Status": "29",
    // "Modify_Date",
    // "Panda_ID",
};
const residentialQueries = [
    // FIELD
    // {
    //     type: 'lat',
    //     searchParams: [
    //         {
    //             conditional: true,
    //             section: 'field',
    //             params: [
    //                 {
    //                     statement: 'none',
    //                     key: 'lat',
    //                     value: ''
    //                 }
    //             ]
    //         }
    //     ],
    //     columnId: residentialHeaders["Srv Lat"],
    // },
    // {
    //     type: 'lng',
    //     searchParams: [
    //         {
    //             conditional: true,
    //             section: 'field',
    //             params: [
    //                 {
    //                     statement: 'none',
    //                     key: 'lng',
    //                     value: ''
    //                 }
    //             ]
    //         }
    //     ],
    //     columnId: residentialHeaders["Srv Lng"],
    // },
    {
        type: 'date',
        // dateFormat: 'YYYY-MM-DD',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'signatureDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Signup Date"],
    },
    {
        type: 'date',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceStartDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Start Date"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceFirstName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv First Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv Last Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'servicePhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Srv Email"],
    },
    {
        type: 'concat',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingFirstName',
                        value: ''
                    }
                ]
            },
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingPhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Email"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceDay',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Day"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'autopay',
                        value: ''
                    }
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "",
                newValue: "N"
            },
            {
                defaultValue: "false",
                newValue: "N"
            },
            {
                defaultValue: "null",
                newValue: "N"
            },
            {
                defaultValue: "true",
                newValue: "Y"
            }
        ],
        columnId: residentialHeaders["Autopay"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'paperless',
                        value: ''
                    }
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "",
                newValue: "N"
            },
            {
                defaultValue: "false",
                newValue: "N"
            },
            {
                defaultValue: "null",
                newValue: "N"
            },
            {
                defaultValue: "true",
                newValue: "Y"
            }
        ],
        columnId: residentialHeaders["E-Billing"],
    },
    // END OF FIELD
    ///////////////////////////////
    // ITEM
    // trash cart number
    {
        type: 'name',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Residential Trash Service -'
                    },
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "Residential Trash Service - ",
                newValue: ""
            },
            {
                defaultValue: " Carts",
                newValue: ""
            },
            {
                defaultValue: " Cart",
                newValue: ""
            }
        ],
        columnId: residentialHeaders['# of Carts'],
        value: ''
    },
    // Billing amount
    {
        type: 'compound_price',
        searchParams: [
            {
                section: 'price_item',
                conditional: false,
                optional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Residential Trash Service']
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Billing Amount"],
        value: 0,
    },
    // Billing Frequency
    {
        type: 'name',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Billing Frequency'
                    },
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "Billing Frequency: ",
                newValue: ""
            },
            {
                defaultValue: " Billing",
                newValue: ""
            },
            {
                defaultValue: " ",
                newValue: ""
            }
        ],
        columnId: residentialHeaders["Billing Frequency"],
        value: ''
    },
    // Fuel Surcharge
    {
        type: 'price',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Fuel Surcharge']
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Fuel Surcharge"],
        value: 0,
    },
    // Setup Fee
    {
        type: 'price',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Setup Fee']
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Setup Fee"],
        value: 0,
    },
    // Pink Can
    {
        type: 'price',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Pink Can']
                    },
                ]
            }
        ],
        columnId: residentialHeaders["Pink Cart"],
        value: 0,
    },
];
let residentialSheetConfig = {
    sheetName: 'Residentail',
    showRootField: {
        grandTotalAmount: residentialHeaders["Total Paid"],
    },
    workspaceDocumentVariables: [
        {
            columnId: residentialHeaders["Document Status"],
            variableName: 'pandaDocStatus',
        },
        {
            columnId: residentialHeaders["Routing Software Sync Status"],
            variableName: 'routingSoftwareSyncStatus',
        },
    ],
    queries: residentialQueries,
    headers: (0, salesReportUtiil_1.createSalesReportHeaderFromObject)(residentialHeaders),
};
exports.triggResidentialSalesReportConfig = {
    reportName: 'Trigg',
    sheetConfigs: [residentialSheetConfig]
};
