// @salesstryke/mobile-api/src/com/salesstryke/dailysalesreportconfig/whiteTail.salesReportConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.whiteTailRollOffSalesReportConfig = exports.whiteTailResidentialSalesReportConfig = void 0;
const salesReportUtiil_1 = require("../utils/salesreport/salesReportUtiil");
// TODO: Delete after database is setup
// need to create something that could set and update header key and name easy in the front end.
const residentialHeaders = {
    "Signup Date": "0",
    "Start Date": "1",
    "Service First Name": "2",
    "Service Last Name": "3",
    "Service Address": "4",
    "Service City": "5",
    "Service State": "6",
    "Service Zip": "7",
    "Service Phone": "8",
    "Service Email": "9",
    "Service Lat": "10",
    "Service Lng": "11",
    "Bill First Name": "12",
    "Bill Last Name": "13",
    "Bill Address": "14",
    "Bill City": "15",
    "Bill State": "16",
    "Bill Zip": "17",
    "Bill Phone": "18",
    "Bill Email": "19",
    "Service Day": "20",
    "Bill Method": "21",
    "Autopay Info": "22",
    "Pickup Per Week": "23",
    "Trash Cart Size": "24",
    "Trash Cart #": "25",
    "Trash Rate": "26",
    "Recycling Rate": "27",
    "Total Paid": "28",
    "Promo Code": "29",
    "Document Status": "30",
    "Softpak Status": "31",
    "Lockbox added Date": "32",
    // "Modify_Date",
    // "Panda_ID",
};
const rollOffHeaders = {
    "Signup Date": '0',
    "Delivery Date": '1',
    "Service First Name": '2',
    "Service Last Name": '3',
    "Service Address": '4',
    "Service City": '5',
    "Service State": '6',
    "Service Zip": '7',
    "Service Phone": '8',
    "Service Email": '9',
    "Service Lat": '10',
    "Service Lng": '11',
    "Bill First Name": '12',
    "Bill Last Name": '13',
    "Bill Address": '14',
    "Bill City": '15',
    "Bill State": '16',
    "Bill Zip": '17',
    "Bill Phone": '18',
    "Bill Email": '19',
    "Business Type": '20',
    "Container Size": '21',
    "Material Type": '22',
    "Container Location": '23',
    "Dumpster Rate": '24',
    "Fuel Surcharge": '25',
    "Total Paid": '26',
    "Promo Code": '27',
    "Document Status": '28',
    "Softpak Status": '29',
    "Lockbox added Date": '30',
};
const residentialQueries = [
    // FIELD
    {
        type: 'lat',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'lat',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Lat"],
    },
    {
        type: 'lng',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'lng',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Lng"],
    },
    {
        type: 'date',
        // dateFormat: 'YYYY-MM-DD',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'signatureDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Signup Date"],
    },
    {
        type: 'date',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceStartDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Start Date"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceFirstName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service First Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Last Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'servicePhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Email"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingFirstName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill First Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Last Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingPhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Bill Email"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceDay',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Day"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'autopay',
                        value: ''
                    }
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "",
                newValue: "N"
            },
            {
                defaultValue: "false",
                newValue: "N"
            },
            {
                defaultValue: "null",
                newValue: "N"
            },
            {
                defaultValue: "true",
                newValue: "Y"
            }
        ],
        columnId: residentialHeaders["Autopay Info"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'paperless',
                        value: ''
                    }
                ]
            }
        ],
        replaceValues: [
            {
                defaultValue: "",
                newValue: "N"
            },
            {
                defaultValue: "false",
                newValue: "N"
            },
            {
                defaultValue: "null",
                newValue: "N"
            },
            {
                defaultValue: "true",
                newValue: "Y"
            }
        ],
        columnId: residentialHeaders["Bill Method"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'promoCode',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Promo Code"],
    },
    // END OF FIELD
    ///////////////////////////////
    // ITEM
    // pickup per week
    {
        type: 'default',
        columnId: residentialHeaders["Pickup Per Week"],
        value: '1x',
    },
    {
        type: "default",
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: "contain",
                        key: "name",
                        value: "Twice Per Week",
                    },
                ]
            },
        ],
        columnId: residentialHeaders['Pickup Per Week'],
        value: "2x"
    },
    // trash cart size
    {
        type: 'default',
        columnId: residentialHeaders['Trash Cart Size'],
        value: '65'
    },
    {
        type: 'default',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: '96 Gallon'
                    },
                ]
            }
        ],
        columnId: residentialHeaders['Trash Cart Size'],
        value: '96'
    },
    // trash cart number
    {
        type: 'default',
        columnId: residentialHeaders["Trash Cart #"],
        value: '1'
    },
    {
        type: 'default',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: '2 Carts'
                    },
                ]
            }
        ],
        columnId: residentialHeaders['Trash Cart #'],
        value: '2'
    },
    // trash Rate
    {
        type: 'compound_price',
        searchParams: [
            {
                section: 'price_item',
                conditional: false,
                optional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Gallon Trash']
                    },
                ]
            }
        ],
        columnId: residentialHeaders['Trash Rate'],
        value: 0,
    },
    // Recycling
    {
        type: 'compound_price',
        searchParams: [
            {
                section: 'price_item',
                conditional: false,
                optional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: ['Weekly Recycling', 'Every Other Week']
                    },
                ]
            }
        ],
        columnId: residentialHeaders['Recycling Rate'],
        value: 0,
    },
];
const rollOffQueries = [
    ///////////////////////////////
    // FIELD
    ///////////////////////////////
    {
        type: 'lat',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'lat',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Lat"],
    },
    {
        type: 'lng',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'lng',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Service Lng"],
    },
    {
        type: 'date',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'signatureDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Signup Date"],
    },
    {
        type: 'date',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'deliveryDate',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Delivery Date"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceFirstName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service First Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service Last Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'servicePhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'serviceEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Service Email"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingFirstName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill First Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingLastName',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill Last Name"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingAddress',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill Address"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingCity',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill City"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingState',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill State"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingZip',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill Zip"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingPhone',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill Phone"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'billingEmail',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Bill Email"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'location',
                        value: ''
                    }
                ]
            }
        ],
        columnId: rollOffHeaders["Container Location"],
    },
    {
        type: 'result',
        searchParams: [
            {
                conditional: true,
                section: 'field',
                params: [
                    {
                        statement: 'none',
                        key: 'promoCode',
                        value: ''
                    }
                ]
            }
        ],
        columnId: residentialHeaders["Promo Code"],
    },
    // {
    //     type: 'result',
    //     searchParams: [
    //         {
    //             conditional: true,
    //             section: 'field',
    //             params: [
    //                 {
    //                     statement: 'none',
    //                     key: 'autopay',
    //                     value: ''
    //                 }
    //             ]
    //         }
    //     ],
    //     replaceValues: [
    //         {
    //             defaultValue: "",
    //             newValue: "N"
    //         },
    //         {
    //             defaultValue: "false",
    //             newValue: "N"
    //         },
    //         {
    //             defaultValue: "null",
    //             newValue: "N"
    //         },
    //         {
    //             defaultValue: "true",
    //             newValue: "Y"
    //         }
    //     ],
    //     columnId: rollOffHeaders["Autopay Info"],
    // },
    // {
    //     type: 'result',
    //     searchParams: [
    //         {
    //             conditional: true,
    //             section: 'field',
    //             params: [
    //                 {
    //                     statement: 'none',
    //                     key: 'paperless',
    //                     value: ''
    //                 }
    //             ]
    //         }
    //     ],
    //     replaceValues: [
    //         {
    //             defaultValue: "",
    //             newValue: "N"
    //         },
    //         {
    //             defaultValue: "false",
    //             newValue: "N"
    //         },
    //         {
    //             defaultValue: "null",
    //             newValue: "N"
    //         },
    //         {
    //             defaultValue: "true",
    //             newValue: "Y"
    //         }
    //     ],
    //     columnId: rollOffHeaders["Bill Method"],
    // },
    ///////////////////////////////
    // END OF FIELD
    ///////////////////////////////
    ///////////////////////////////
    // ITEM
    ///////////////////////////////
    {
        type: "name",
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: "contain",
                        key: "name",
                        value: "Business Type",
                    },
                ]
            },
        ],
        replaceValues: [
            {
                defaultValue: 'Business Type: ',
                newValue: '',
            }
        ],
        columnId: rollOffHeaders["Business Type"]
    },
    // Material Type
    {
        type: "name",
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: "contain",
                        key: "name",
                        value: "Material Type",
                    },
                ]
            },
        ],
        replaceValues: [
            {
                defaultValue: 'Material Type: ',
                newValue: '',
            }
        ],
        columnId: rollOffHeaders['Material Type'],
        value: ""
    },
    // Container Size
    {
        type: "name",
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                params: [
                    {
                        statement: "contain",
                        key: "name",
                        value: "yd Dumpster",
                    },
                ]
            },
        ],
        replaceValues: [
            {
                defaultValue: ' Dumpster',
                newValue: '',
            }
        ],
        columnId: rollOffHeaders['Container Size'],
        value: ""
    },
    // dumpster Rate
    {
        type: 'compound_price',
        searchParams: [
            {
                section: 'price_item',
                conditional: true,
                optional: true,
                params: [
                    {
                        statement: 'contain',
                        key: 'name',
                        value: 'Dumpster'
                    },
                ]
            }
        ],
        columnId: rollOffHeaders['Dumpster Rate'],
        value: 0,
    },
    // Fuel Surcharge
    {
        type: 'price',
        columnId: rollOffHeaders["Fuel Surcharge"],
        value: 0
    }
    ///////////////////////////////
    // END OF ITEM
    ///////////////////////////////
];
let residentialSheetConfig = {
    sheetName: 'Residentail',
    templateFilters: ['Whitetail Residential Service Template'],
    showRootField: {
        grandTotalAmount: residentialHeaders["Total Paid"],
    },
    workspaceDocumentVariables: [
        {
            columnId: residentialHeaders["Document Status"],
            variableName: 'pandaDocStatus',
        },
        {
            columnId: residentialHeaders["Softpak Status"],
            variableName: 'routingSoftwareSyncStatus',
        },
        {
            columnId: residentialHeaders["Lockbox added Date"],
            variableName: 'dsrSyncDate',
            // defaultValue: 'Synced to Softpak. Added to lockboxfile.'
        }
    ],
    queries: residentialQueries,
    headers: (0, salesReportUtiil_1.createSalesReportHeaderFromObject)(residentialHeaders),
};
let rollOffSheetConfig = {
    sheetName: 'Roll Off',
    templateFilters: ['Whitetail Commercial Service Template', 'Whitetail Roll Off Service Template'],
    showRootField: {
        grandTotalAmount: rollOffHeaders["Total Paid"],
    },
    workspaceDocumentVariables: [
        {
            columnId: rollOffHeaders["Document Status"],
            variableName: 'pandaDocStatus',
        },
        {
            columnId: rollOffHeaders["Softpak Status"],
            variableName: 'routingSoftwareSyncStatus',
        },
        {
            columnId: rollOffHeaders["Lockbox added Date"],
            variableName: 'dsrSyncDate',
            // defaultValue: 'Synced to Softpak. Added to lockboxfile.'
        }
    ],
    queries: rollOffQueries,
    headers: (0, salesReportUtiil_1.createSalesReportHeaderFromObject)(rollOffHeaders),
};
exports.whiteTailResidentialSalesReportConfig = {
    reportName: 'Whitetail-residentail',
    sheetConfigs: [residentialSheetConfig]
};
exports.whiteTailRollOffSalesReportConfig = {
    reportName: 'Whitetail-non-residentail',
    sheetConfigs: [rollOffSheetConfig]
};
