// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/AgreementDocumentVersionEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgreementDocumentVersionEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.AgreementDocumentVersion';
class AgreementDocumentVersionEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('42f56114-a7fe-4d49-9383-66f1380b443f')
            .setDataModelName('AgreementDocumentVersion')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('82da493a-5489-407f-b7bb-a7f68bdc2ea2');
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('bc957a62-7424-4032-8178-300089a8278d'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('35654287-fcec-49e2-9c5d-8e2a2c1041c6'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('bc010e8d-2f67-4bba-917e-db101952db0e'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('d6b47be0-b13f-42c8-ae7d-01a57919da01'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('f0743ec6-16da-4b6a-9996-db8a9a407e33'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '4f8af500-68b4-4ffd-93e3-63bb9e01187a'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('21ad99b4-7139-498b-8555-ef97af91eb7f'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('abc18cbb-7b07-4979-b1b5-394ed8124d59'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '33abf646-82d4-443d-bfab-cc2c156e615b',
            comment: 'Indicates if the agreement document is active.',
            isMongoIndexed: false,
            isMongoRequired: true
        }));
        // version
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'version',
            attributeUUID: 'f7eac62a-0cdd-47b6-a88b-de0844831ce4',
            comment: 'The version of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: 'b688c636-2180-4e4f-acce-2d5e4119a7e6',
            comment: 'The description of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '081e8e2e-fced-4c38-87c5-5c7f8d6affca',
            comment: 'The name of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: true
        }));
        // text
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'text',
            attributeUUID: '51943bd7-03f4-4bc7-899d-e0e86d5c2338',
            comment: 'The text of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // html
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'html',
            attributeUUID: '42ffde3b-e399-4caf-b263-90d42eb4179a',
            comment: 'The HTML content of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: '3e3120a9-18e3-4017-802b-b9b0bd339de8',
            comment: 'Active start date.',
            isMongoRequired: false,
            isMongoIndexed: false
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: 'e22a79c4-911d-48ea-8749-df6c723b0faf',
            comment: 'Active end date.',
            isMongoRequired: false,
            isMongoIndexed: false
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '1d56c141-5236-4995-b9a0-cc35f77184e5',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '15c6e357-2cb2-49a1-9308-e6bf38d38017',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * AgreementDocument
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'agreementDocument',
            attributeUUID: '53b26d3d-f77a-4b46-b1fc-431a1d7011eb',
            comment: 'The agreement document.',
            foreignRefEntityNamespace: persistable_1.AgreementDocument.metaDataForAgreementDocument.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // agreementType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'agreementType',
            attributeUUID: '47613b05-cb42-459b-8ab9-f099256b9a3d',
            comment: 'The agreement type.',
            foreignRefEntityNamespace: persistable_1.AgreementType.metaDataForAgreementType.namespace,
            isMongoIndexed: false,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : 'a1dc3cab-4397-4729-b334-dd3de78edc24',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            isMongoIndexed            : true,
            isMongoRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'ede35a8d-ee01-4f6c-8539-ffbb4d5560e4',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (AgreementDocumentVersionEntityModel.INSTANCE);
    }
}
exports.AgreementDocumentVersionEntityModel = AgreementDocumentVersionEntityModel;
AgreementDocumentVersionEntityModel.INSTANCE = new AgreementDocumentVersionEntityModel();
