// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/AgreementDocumentEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgreementDocumentEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const Organization_1 = require("../../classes/persistable/Organization");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.AgreementDocument';
class AgreementDocumentEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('6567f427-c944-4b2d-b35f-4119917dd195')
            .setDataModelName('AgreementDocument')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('1ae7cddb-4ed9-4897-8227-e50accc5fd7e');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('2f4472aa-42b9-481f-adfa-e2be92731926'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('421b7851-a9f3-4135-8ac4-1168257cb329'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('2b3607e5-2c88-4e5e-ad37-af24e663e90c'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('c378c553-bab2-40c3-bd1f-071339e1eb3b'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('d9ded085-4a6a-4d12-8eb7-20fbd68827ee'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('bdd5be0d-b173-478c-9239-ade197b047d2'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '9d5d06e5-1e98-47ba-b5fa-dc6269dc724f'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('c281698c-32c3-4a66-9e08-b7ccfb0a71fc'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('c509f1f2-ffcf-4a21-9c00-898480e1e285'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: 'c7cd280b-6f44-48c7-acd6-dc1059050380',
            comment: 'Indicates if the agreement document is active.',
            isMongoIndexed: false,
            isMongoRequired: true
        }));
        // version
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'version',
            attributeUUID: 'ce1227d6-d706-46a6-b920-244cc1175c46',
            comment: 'The version of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '060480df-7edf-45bd-b42b-87251c214c45',
            comment: 'The description of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '54054ff8-d3fb-4fc3-9368-109b0830e7ea',
            comment: 'The name of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: true
        }));
        // text
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'text',
            attributeUUID: '314f9eea-67b1-44f9-8f17-1a818c6f5378',
            comment: 'The text of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // html
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'html',
            attributeUUID: '7f19a1ca-5fe4-4a57-9276-dc306f8f3efa',
            comment: 'The HTML content of the agreement document.',
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: '032a06c9-b44e-4123-a450-65ed97664d66',
            comment: 'Active start date.',
            isMongoRequired: false,
            isMongoIndexed: false
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: '120ff56c-1ae6-4f42-824a-91c903a7fdb5',
            comment: 'Active end date.',
            isMongoRequired: false,
            isMongoIndexed: false
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'f1d0c855-b849-412b-aa0c-e32b2ac3d0ed',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '53b26d3d-f77a-4b46-b1fc-431a1d7011eb',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // agreementType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'agreementType',
            attributeUUID: '812e49c3-6c45-4a99-a707-4a70f2b4a7e3',
            comment: 'The agreement type.',
            foreignRefEntityNamespace: persistable_1.AgreementType.metaDataForAgreementType.namespace,
            isMongoIndexed: false,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'a2eb7bfe-ce5c-4c88-97c8-ba638bbfe6dc',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (AgreementDocumentEntityModel.INSTANCE);
    }
}
exports.AgreementDocumentEntityModel = AgreementDocumentEntityModel;
AgreementDocumentEntityModel.INSTANCE = new AgreementDocumentEntityModel();
