// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ApplicationEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.Application';
class ApplicationEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('4d56f842-fbec-4532-b7d4-9789f12e216c')
            .setDataModelName('Application')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('3b5e6114-7414-4d03-afa9-6628881011ff');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('3c3a208a-6a27-43ce-b53d-dc5c4831b7bf'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('af78a183-1dae-4c15-a00e-547b11865116'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('4aad66a6-e5ee-4714-9c60-6de4abc3e53a'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('f70eba0a-454c-4ce0-aac6-0c2dd19de82f'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('5a35baed-a84f-4559-8ee1-e71ba6fe0ec1'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('e3fe6847-7471-41c7-960e-91c5397c2099'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '362edc49-26eb-4a20-bc0f-f97fda53a57d'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('d73d1b45-091c-43aa-b7c8-0353032d9a43'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('f2618d02-9952-4265-a6b0-d337a2e8ea1e'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '99547188-bf1a-41da-b7cc-cc014f125113',
            comment: 'The name.'
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '59f2c809-ddc8-4c4a-9542-8eb55a437148',
            comment: 'The description.'
        }));
        /**
         * SercurityPermission.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: 'securityPermissions',
            attributeNameSingular: 'securityPermission',
            attributeUUID: 'a679f811-84cf-4aeb-af49-1b6f5f16c55a',
            comment: "The security permission required for this application.",
            foreignRefEntityNamespace: persistable_1.SecurityPermission.metaDataForSecurityPermission.namespace
        }));
        // Organization
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
        //     attributeName             : 'organization',
        //     attributeUUID             : 'f061c977-aee3-49cd-b846-b6d0ceab40e8',
        //     comment                   : 'The organization.',
        //     foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
        //     isMongoIndexed            : true,
        //     isMongoRequired           : true
        // }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '73a004af-f6dc-40ec-91e6-cdf4ecc6101e',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ApplicationEntityModel.INSTANCE);
    }
}
exports.ApplicationEntityModel = ApplicationEntityModel;
ApplicationEntityModel.INSTANCE = new ApplicationEntityModel();
