// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/CnsCampaignEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsCampaignEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const RegularExpressionRegexConstUtil_1 = require("../../../utils/validation/RegularExpressionRegexConstUtil");
const CnsCampaignList_1 = require("../../classes/persistable/CnsCampaignList");
const CnsMessageTemplate_1 = require("../../classes/persistable/CnsMessageTemplate");
const CnsSegment_1 = require("../../classes/persistable/CnsSegment");
const TypeReference_1 = require("../../../data/entity/TypeReference");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.CnsCampaign';
class CnsCampaignEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('b65ef4b4-7aeb-4d80-9cb1-8ce6b2faeb1c')
            .setDataModelName('CnsCampaign')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('956f82e4-0e09-4dc4-a231-4e989dde3478');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('97f7ff23-3481-46d8-8fd2-7dea4227cc55'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('ef4cdf98-a333-4eac-b59f-8dc4c7b88d5c'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('a4725567-10dc-47d9-bb48-b261c421ff94'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('d4f59782-8e28-46ec-b7e8-313964aa0710'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('c5c2a45b-e4a4-46af-9756-24ba394f3208'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('25dfbc1a-553b-4b1d-aed7-2e721d516482'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '21612e38-5b06-4c04-b785-e4516edc6105'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('ace9b961-0dce-408b-9a75-a173e0851488'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('12bd3d06-9bf5-49ff-8b1e-b274405ef067'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'b8a3b3e5-0199-4d8c-8090-0d144c9acd8d',
            comment: 'The name of the campaign.',
            validator: zod_1.z.string().min(1, "Name cannot be empty")
                .max(255, "Name is too long")
                .regex(RegularExpressionRegexConstUtil_1.Regex.alphanumericWithSpace, "Name cannot contain special characters"),
        }));
        // awsData
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableJsonString2({
            attributeName: 'awsData',
            attributeUUID: '9e14b01e-27bf-4e19-9e60-c0bbc86102bc',
            comment: 'AWS Date for this cns campaign, such as the id.',
            variableType: new TypeReference_1.TypeReference({
                typeName: 'Object'
            }),
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '9e14b01e-27bf-4e19-9e60-c0bbc86102bc',
            comment: 'Campaign description.',
            isMongoRequired: false,
            validator: zod_1.z.string().max(5000, "Description is too long"),
        }));
        // cnsCampaignType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'cnsCampaignType',
            attributeUUID: '9e14b01e-27bf-4e19-9e60-c0bbc86102bc',
            comment: 'Campaign type.',
            validator: zod_1.z.string(),
        }));
        // endpoint - uuid
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'endpoint',
            attributeUUID: '9e14b01e-27bf-4e19-9e60-c0bbc86102bc',
            comment: 'Data for endpoint, should the cnsCampaignType be endpoint.',
            isMongoRequired: false,
            validator: zod_1.z.string(),
        }));
        // cnsMessageTemplate []
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: 'cnsMessageTemplate',
            attributeNameSingular: 'cnsMessageTemplate',
            attributeUUID: '636c5dd1-69cf-487f-8d39-4ec2b728430e',
            comment: 'Template that are use in this campaign.',
            foreignRefEntityNamespace: CnsMessageTemplate_1.CnsMessageTemplate.metaDataForCnsMessageTemplate.namespace,
            mongoIsRequired: false,
        }));
        // cnsCampaignList mongoRef
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'cnsCampaignList',
            attributeUUID: '9e14b01e-27bf-4e19-9e60-c0bbc86102bc',
            comment: 'Campaign cnsCampaignList.',
            foreignRefEntityNamespace: CnsCampaignList_1.CnsCampaignList.metaDataForCnsCampaignList.namespace,
            isMongoRequired: false,
            validator: zod_1.z.string(),
        }));
        // cnsSegment AKA RuleBuilder MongoRef
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'cnsSegment',
            attributeUUID: '9e14b01e-27bf-4e19-9e60-c0bbc86102bc',
            comment: 'Campaign segment.',
            foreignRefEntityNamespace: CnsSegment_1.CnsSegment.metaDataForCnsSegment.namespace,
            isMongoRequired: false,
            validator: zod_1.z.string(),
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : 'fd9d6d6f-d9f1-4d3b-b679-7a07029bfd0f',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            isMongoIndexed            : true,
            isMongoRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '4ff878f7-9309-4959-ab43-85db1db8f0fe',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (CnsCampaignEntityModel.INSTANCE);
    }
}
exports.CnsCampaignEntityModel = CnsCampaignEntityModel;
CnsCampaignEntityModel.INSTANCE = new CnsCampaignEntityModel();
