// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/CnsCampaignListItemEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsCampaignListItemEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const CnsCampaignList_1 = require("../../classes/persistable/CnsCampaignList");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.CnsCampaignListItem';
class CnsCampaignListItemEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('af963c90-7f18-43db-b906-535253ef1718')
            .setDataModelName('CnsCampaignListItem')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('5440eb92-12bc-4a79-90f4-d611089ccbcc');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('9e17ebc3-261e-48bf-bbe3-387367824159'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('262943a4-1a0d-41a4-a30a-e8d642de9b5c'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('1012f12f-81aa-41e7-94ed-15fd5aa9c8e9'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('f16bebe2-1284-49d8-8224-400ee0488b91'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('ef5f782c-d79f-4aed-86ec-3f0fa4bf8528'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('35454271-4b3b-432b-a6d6-6e73416aa47a'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '75fd5acf-ab0e-4c9d-9c19-863ada51a698'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('df7fab87-31e2-40cf-a58c-71121ea15ea6'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('349d62fc-5e05-418e-84ce-4904c4401c39'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // customerId
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'customerId',
            attributeUUID: '245aa4c2-bc2e-4bd1-8d41-65b7b241f001',
            comment: 'The CustomerId.',
            validator: zod_1.z.string(),
        }));
        // awsData
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'awsData',
            attributeUUID: '245aa4c2-bc2e-4bd1-8d41-65b7b241f001',
            comment: 'AwsData for this item, such as recorded id in aws for the customerId.',
            validator: zod_1.z.string(),
        }));
        // cnsCampaignList
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'cnsCampaignList',
            attributeUUID: '245aa4c2-bc2e-4bd1-8d41-65b7b241f001',
            comment: 'The campaign list this item belong to.',
            foreignRefEntityNamespace: CnsCampaignList_1.CnsCampaignList.metaDataForCnsCampaignList.namespace,
            isMongoRequired: false,
            isMongoIndexed: false,
            validator: zod_1.z.string(),
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : '37dbe5df-9ba5-46cf-b1db-91bc2373c9b0',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            isMongoIndexed            : true,
            isMongoRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'a964d6ad-4f68-459d-85d2-4b66901f47d6',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (CnsCampaignListItemEntityModel.INSTANCE);
    }
}
exports.CnsCampaignListItemEntityModel = CnsCampaignListItemEntityModel;
CnsCampaignListItemEntityModel.INSTANCE = new CnsCampaignListItemEntityModel();
