// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/CnsMessageTemplateEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsMessageTemplateEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const RegularExpressionRegexConstUtil_1 = require("../../../utils/validation/RegularExpressionRegexConstUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.CnsMessageTemplate';
class CnsMessageTemplateEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('4c2444d6-9ba1-4b4f-9935-00542089d31b')
            .setDataModelName('CnsMessageTemplate')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('f0bb5b91-d622-483d-983a-dccfbe79f1ff');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('90bf33bc-0ad3-4a3c-871f-fb53689f9274'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('fa888a76-e110-4832-8dcb-bb9abca60e3c'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('1267a00c-5bc1-4b1e-8241-3741040cb953'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('b97876cb-9767-4c33-bdd0-8fcc9e6f2976'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('63884ad9-9d9a-4551-9a7f-5ce8bbc16f86'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('c31a8062-fcf2-4c64-95d7-2a9d1a5ca33c'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'b36ba973-4890-4a1d-85eb-2b712acc5e63'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('6efd5c58-b20f-4481-81cc-2b0c3d5356f4'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('489aa921-1ca3-486d-8ae3-5660e3d45609'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'c6a48dc1-d5d7-4c5d-a6ba-4cb06415f498',
            comment: 'The name of the template.',
            validator: zod_1.z.string().min(1, "Name cannot be empty")
                .max(255, "Name is too long")
                .regex(RegularExpressionRegexConstUtil_1.Regex.alphanumericWithSpace, "Name cannot contain special characters")
        }));
        // channelType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'channelType',
            attributeUUID: '1435e49c-b5c7-4485-a248-5c770a4dd636',
            comment: 'What channel type of message this template is for.'
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '08844d6e-1ea1-48eb-a3fc-1dbaddfe804a',
            comment: 'Template description.',
            isMongoRequired: false,
            validator: zod_1.z.string().max(5000, "Description is too long")
        }));
        // awsData - id / name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'awsData',
            attributeUUID: 'cf230adf-1b5d-4f04-868b-c7d0df58bceb',
            comment: 'AWS data for this template.',
            validator: zod_1.z.string(),
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '1592318e-2145-49d5-b6e7-4e7559362f8f',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            validator: zod_1.z.string(),
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '9c3ce47a-e030-440a-9386-14255c934664',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (CnsMessageTemplateEntityModel.INSTANCE);
    }
}
exports.CnsMessageTemplateEntityModel = CnsMessageTemplateEntityModel;
CnsMessageTemplateEntityModel.INSTANCE = new CnsMessageTemplateEntityModel();
