// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/CnsSyncEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsSyncEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const CnsSyncStatusType_1 = require("../../classes/persistable/CnsSyncStatusType");
const Organization_1 = require("../../classes/persistable/Organization");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.CnsSync';
class CnsSyncEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('e0be8a14-5727-4da1-9560-b14bbbdd630d')
            .setDataModelName('CnsSync')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('63c4953c-9580-4f01-9759-56916635ed8d');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('a3118a33-6afa-4c8c-b7c1-620fdacbb204'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('3ae91f62-e9d7-4c88-bd83-46f7c2775742'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('8b2df002-b97e-4791-b85f-15b13a65a76c'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('38d45733-0979-4e90-9425-a4166f5633d6'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('1e82945f-30f4-4d4c-8a2a-92d1afabb500'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('9edbfbee-24ac-4e37-9758-ef9f2d28733c'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'ff52cbf4-7c5c-4b65-9600-eb8865a9f05e'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('49baed5e-c71b-4b4f-a1ca-784ef34743f6'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('8a1fb9a6-bcb5-4fa5-aee7-0ee172b1061f'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        /**
         * appCnsSyncStatusType
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'appCnsSyncStatusType',
            attributeUUID: 'a868ed79-f895-453d-9122-23590462162d',
            comment: 'The APP CnsSyncStatusType.',
            foreignRefEntityNamespace: CnsSyncStatusType_1.CnsSyncStatusType.metaDataForCnsSyncStatusType.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * emailCnsSyncStatusType
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'emailCnsSyncStatusType',
            attributeUUID: 'fec10c7a-7cf7-4a6e-aa3e-5d16ec89ca6d',
            comment: 'The EMAIL CnsSyncStatusType.',
            foreignRefEntityNamespace: CnsSyncStatusType_1.CnsSyncStatusType.metaDataForCnsSyncStatusType.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * ImportDataFileUploadSync
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'importDataFileUploadSync',
            attributeUUID: '11e19fb6-eeaf-4aca-83e7-e407397aff83',
            comment: 'The ImportDataFileUploadSync.',
            foreignRefEntityNamespace: persistable_1.ImportDataFileUpload.metaDataForImportDataFileUpload.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // // boolean
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableBoolean2({
        //     attributeName   : 'aBoolean',
        //     attributeUUID   : '616fd349-ada5-4ff1-835b-764be53e9df0',
        //     comment         : 'The name of a boolean value.'
        // }));
        // // number
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableNumber2({
        //     attributeName   : 'aNumber',
        //     attributeUUID   : 'c1f93b1f-cfc8-43cb-99f9-508e04aacc25',
        //     comment         : 'The name of a number value.'
        // }));
        // // string
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableString2({
        //     attributeName   : 'aString',
        //     attributeUUID   : 'f3c1a4a6-e275-4327-8609-4a252e426dd0',
        //     comment         : 'The name of a string value.'
        // }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '4024031b-cf16-4490-a512-3c35ba5bc18b',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * textCnsSyncStatusType
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'textCnsSyncStatusType',
            attributeUUID: 'c8a26bc8-938b-4193-94a7-65dc568f0a6a',
            comment: 'The EMAIL CnsSyncStatusType.',
            foreignRefEntityNamespace: CnsSyncStatusType_1.CnsSyncStatusType.metaDataForCnsSyncStatusType.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'e1f7b49e-ec14-4ce5-8b9c-48a4cbc88adb',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (CnsSyncEntityModel.INSTANCE);
    }
}
exports.CnsSyncEntityModel = CnsSyncEntityModel;
CnsSyncEntityModel.INSTANCE = new CnsSyncEntityModel();
