// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/EmailAddressEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailAddressEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.EmailAddress';
class EmailAddressEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('2ac5d7d7-5fce-4700-a800-f9dc48ee5da8')
            .setDataModelName('EmailAddress')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('e8a9be23-0fe4-4f1e-9545-5764eaf92458');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('23c37a0f-7393-43fa-b866-d81ec003faca'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('4d862de6-b8d3-451e-8441-0b5e817fd85a'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('18f6a96f-b4a7-4fb7-b47e-2798ca54b9a4'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('4a0659e8-7dad-4da9-807a-22b250b88814'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('04685be2-80d0-4a7e-9720-50ec1692d7ec'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('afaf0efa-c2c1-42fa-a680-5215cd141828'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'f52e09c5-1c02-4097-9fc8-cae135a0d6e8'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('c1801ba3-7f23-4099-86cd-dfc500532a3f'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('56e583a3-1954-4bf1-88e8-91ce06b19537'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'emailAddress',
            attributeUUID: 'deb84c1b-940f-4cfc-87e0-f53440e3a861',
            comment: 'The email address.',
            isMongoIndexed: false,
            isMongoRequired: true,
        }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '226fa96b-1e59-4952-aada-0cf751e4b3a7',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
    }
    static getInstance() {
        return (EmailAddressEntityModel.INSTANCE);
    }
}
exports.EmailAddressEntityModel = EmailAddressEntityModel;
EmailAddressEntityModel.INSTANCE = new EmailAddressEntityModel();
