// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/HistoryEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoryEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.History'; // *** EDIT THIS ***
class HistoryEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true); // most Entities need this.
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('618b68e5-b02f-4afb-a56c-94b9025a43a4')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('436a02f7-3825-4f42-8bde-6e18c25e7639');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('dbe27842-cc28-4c56-8ae2-8da5f39bb8dc'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('a9df2d40-ff42-4360-806d-ab83edf14022'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('21787952-5b25-4ed2-905a-7965648bf79e'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('a9dece5f-d9c7-4210-aaeb-383a060e441b'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('3eaf5fbf-bf49-47f4-8bc3-08130dd1d26a'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('e48bd5d0-fe30-4ca9-b362-961e89ad5a06'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('9098dac0-8cc6-47f3-afae-26803c780207'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '3c6c03df-8c8c-483e-ac46-2826d15d691e'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('c1ce27fa-476c-4e40-9665-9a368d3e9c7d'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType('8dd0b930-3004-450f-aa58-3473372c1bb1'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('e576c4a7-0995-43ac-a9a2-c3ec8b77a267'));
        // /**
        //  * booleanValue
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getRecordBoolean(
        //     'booleanValue',             // *** EDIT THIS ***
        //     '7f651a6d-d3b9-44de-8a96-0186eaa0bca2',
        //     'A boolean.'                // // *** EDIT THIS ***
        // )); 
        // /**
        //  * currencyValue
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableCurrency(
        //     'currencyValue',                        // *** EDIT THIS ***
        //     'd298362c-9a3c-417e-9d8c-0ce216af7cc8',
        //     'A currency value'                      // *** EDIT THIS ***
        // ));  
        // /**
        //  * dateValue
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableDate(
        //     'dateValue',                            // *** EDIT THIS ***
        //     '94ad9175-909a-4c4e-9cf6-4ff115773f42',
        //     'A date value.'                         // *** EDIT THIS ***
        //     )
        //     //.setMongoIsRequired(false)
        // );
        /**
         * mongoRef
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef(
           'mongoRef',                         // *** EDIT THIS ***
           User.metaDataForUser.namespace,   // *** EDIT THIS ***
           '71ceaf47-5ec0-4887-b8a4-4b1ffbea71e8',
           'A mongo ref'                       // *** EDIT THIS ***
       ));
       */
        /**
         * mongoRefArray
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName             : 'mongoRefArrays',                                   // *** EDIT THIS ***
                attributeNameSingular     : 'mongoRefArray',                                    // *** EDIT THIS ***
                comment                   : 'A mongo ref array.',                               // *** EDIT THIS ***
                foreignRefEntityNamespace : SecurityRole.metaDataForSecurityRole.namespace,   // *** EDIT THIS, replace with correct foreign ref ***
                uuid                      : '35d81ce3-4db7-463e-bfce-ea9c9735a447'
            })
            .setMongoRefArrayAllowDuplicate(false)
        );
        */
        // /**
        //  * numberValue
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableNumber(
        //     'numberValue',          // *** EDIT THIS ***
        //     '167d89c9-c449-481b-aee4-59620cc8842c',
        //     'A number value.'       // *** EDIT THIS ***
        // ));      
        /**
         * attributeUUID
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString('attributeUUID', // *** EDIT THIS ***
        '1528be38-d454-46d0-a19b-332de7d60bef', 'The Entity Attribute UUID value.' // *** EDIT THIS ***
        ));
        /**
         * attributeValue
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString('attributeValue', // *** EDIT THIS ***
        '0ed7f9cb-b2d2-4975-b00c-5a1614ddbed0', 'The Entity Attribute UUID value.' // *** EDIT THIS ***
        ));
        /**
         * attributeDataType
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableInteger('attributeDataType', // *** EDIT THIS ***
        '74f56bbb-84a0-4c90-b0a8-6c98888c61d0', 'The the attribute DataType.' // *** EDIT THIS ***
        ));
        /**
         * entityId
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString('entityId', // *** EDIT THIS ***
        '3d89d58c-6607-4455-b3ea-401d366e7d26', 'The Entity ID value for the specific document.' // *** EDIT THIS ***
        ));
        /**
         * entityUUID
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString('entityUUID', // *** EDIT THIS ***
        '005a1a13-0d17-4d99-a9e3-37d9b07568c8', 'The Entity UUID value, for the Entity type. This is NOT a document ID.' // *** EDIT THIS ***
        ));
        /**
         * entityNamespace.
         *
         * DO NOT USE THIS TO Query against. If the Entity Name is changed, this will not be accurate.
         * ONLY query by entityUUID.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString('entityNamespace', // *** EDIT THIS ***
        '1f147ff1-ae1d-420c-8698-7fea252aa855', 'The Entity namespace value. DO NOT USE THIS TO Query against. If the Entity Name is changed, this will not be accurate. ONLY query by entityUUID.' // *** EDIT THIS ***
        ));
        /**
         * userUUID.
         *
         * DO NOT USE THIS TO Query against. If the Entity Name is changed, this will not be accurate.
         * ONLY query by entityUUID.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString('userUUID', // *** EDIT THIS ***
        '2fb3727f-25a2-4ac3-b31b-f76cc1f01260', 'The userUUID that made the change.' // *** EDIT THIS ***
        ));
    }
    static getInstance() {
        return (HistoryEntityModel.INSTANCE);
    }
}
exports.HistoryEntityModel = HistoryEntityModel;
HistoryEntityModel.INSTANCE = new HistoryEntityModel();
