// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/HttpApiUsageEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpApiUsageEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const Import_1 = require("../../../data/entity/Import");
const TypeReference_1 = require("../../../data/entity/TypeReference");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.HttpApiUsage';
class HttpApiUsageEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('d35d0930-89c0-44fb-bd8a-551e7a69e6e1')
            .setDataModelName('HttpApiUsage')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('8ce29e45-5a62-48a6-a62e-0c984bdd0662');
        entityModel
            .addImport({
            import: new Import_1.Import({
                moduleSpecifier: 'react-querybuilder',
                namedBindings: [
                    new Import_1.NamedBinding({ propertyName: 'RuleGroupType' })
                ]
            })
        });
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('0ce9f443-fcd1-4b57-8615-54dc2b8f242a'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('8b32a868-a77c-4194-bfcd-ef807a531211'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('cb733cb9-5294-4c1d-87df-2f817bcce576'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('9c432ab3-4bc8-4c69-8d29-2bb844a0c6f0'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('64c8c005-a652-401c-9abc-b84a2a4473dd'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('ab2fe02d-ca30-4796-b2f1-f3190c535c01'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '198b3c7d-8a72-4b9a-8e6c-c1f2c4ca04d6'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('2d7a56fc-8182-4acb-8f3f-7395afbdb355'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('0539c32b-28f2-458f-a808-ca6cc37f4bd4'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // // boolean
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableBoolean2({
        //     attributeName   : 'aBoolean',
        //     attributeUUID   : '0802a866-c090-4ebd-8ef7-3404fcf65cc8',
        //     comment         : 'The name of a boolean value.'
        // }));
        // // number
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableNumber2({
        //     attributeName   : 'aNumber',
        //     attributeUUID   : '75e2cf71-e4ea-4ada-ba5d-01fb4492ad6b',
        //     comment         : 'The name of a number value.'
        // }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableJsonString2({
            attributeName: 'jsonString',
            attributeUUID: 'f41d4aa7-2ffd-4e9d-8a2f-1e8a4f875089',
            comment: 'a test jsonString',
            variableType: new TypeReference_1.TypeReference({
                typeName: 'RuleGroupType' //,
                // typeArguments : [
                //     DataType.STRING,
                //     DataType.STRING
                // ]
            })
        }));
        // .setVariableType({ 
        //     variableType : new TypeReference({ 
        //         typeName : 'Array', 
        //         typeArguments : [ DataType.STRING ] 
        //     })
        // });
        // url
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'url',
            attributeUUID: '0d831872-7d82-4ba3-9603-84acc9a2961f',
            comment: 'The url of the API usage.'
        }));
        /**
         * Organization
         */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
        //     attributeName             : 'organization',
        //     attributeUUID             : 'ca383170-0072-4dfd-805a-0117e1f5455d',
        //     comment                   : 'The organization.',
        //     foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
        //     mongoIsIndexed            : true,
        //     mongoIsRequired           : true
        // }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '5a6cee53-522a-4b35-af10-e70a89c08f48',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (HttpApiUsageEntityModel.INSTANCE);
    }
}
exports.HttpApiUsageEntityModel = HttpApiUsageEntityModel;
HttpApiUsageEntityModel.INSTANCE = new HttpApiUsageEntityModel();
