// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/I18nLocaleTypeEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.I18nLocaleTypeEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.I18nLocaleType';
class I18nLocaleTypeEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('7e589fca-13aa-4dce-9ae3-0e3045eaaf62')
            .setDataModelName('I18nLocaleType')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('74af659c-fc3c-48f8-8318-2d46535268fd');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('3033053d-b300-453a-83af-97a4ad371ee5'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('c20cba8f-b136-4f47-a3c5-4df568e05f05'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('d8c570b8-1ac3-4595-963e-ea4aa20531ce'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('0de81a7a-8810-4d51-b3c5-a33f454d45f1'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('baaa6c5e-07ff-4cdd-99c0-85ab08a6ba5d'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('5d91300a-4cd8-424b-ab8f-c47af94bd69e'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '9ba5a70e-02df-42e1-9138-344b01b9c97b'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('dd427dc3-ab5d-4a52-bb65-056c84ccd618'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('5b429cce-0161-4659-9d49-d0f4920de5c6'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: 'e5a842dc-41f6-4639-b163-4fc14521e415',
            comment: 'The description of the category. Example: Basic UI controls.',
            isMongoRequired: true
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'd8e35e5a-efc2-4f8c-b147-33f5a942fe1d',
            comment: 'The name for the category.',
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : '242b8290-a967-48e1-aaaa-0e88edaff9aa',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            mongoIsIndexed            : true,
            mongoIsRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'f0cfcb7b-45dc-45a2-bc2e-b349a41d782e',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (I18nLocaleTypeEntityModel.INSTANCE);
    }
}
exports.I18nLocaleTypeEntityModel = I18nLocaleTypeEntityModel;
I18nLocaleTypeEntityModel.INSTANCE = new I18nLocaleTypeEntityModel();
