// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/I18nTextEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.I18nTextEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const I18nKeyType_1 = require("../../classes/persistable/I18nKeyType");
const I18nLocaleType_1 = require("../../classes/persistable/I18nLocaleType");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.I18nText';
class I18nTextEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('dd6e99f8-6321-4b42-9d8f-2356247cc9f2')
            .setDataModelName('I18nText')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('80b7c93e-f21d-4b97-a93b-9dfd3a5a73dc');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('f38c7849-ce4f-499b-a95c-a5504d732761'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('8ef2b284-a4c1-4647-8291-6871a1465193'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('195bae7a-7c8b-4bfe-bd9b-8c4fc64a3623'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('73b33600-8201-42d5-b35b-14de3e719209'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('614857b0-0dbe-4d35-96d2-cddae02602c2'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('87d90104-a41c-4317-881e-14feb10c8d9c'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '7a35fb50-0c6a-4364-9974-7b6df84b3751'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('20a9af7a-dfe6-4864-979b-246a94d46fdb'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('386c7f6a-34df-4bbc-9cc8-20113811a2bc'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // template
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'template',
            attributeUUID: '3d0a70d0-539a-4d2d-8805-d40fd2116eaa',
            comment: 'The template for the localized string.'
        }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'i18nKeyType',
            attributeUUID: 'fc3a36b4-f1d9-4016-9773-2b129c0d62f9',
            comment: 'The key.',
            foreignRefEntityNamespace: I18nKeyType_1.I18nKeyType.metaDataForI18nKeyType.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'i18nLocaleType',
            attributeUUID: 'd8e35e5a-efc2-4f8c-b147-33f5a942fe1d',
            comment: 'The Locale.',
            foreignRefEntityNamespace: I18nLocaleType_1.I18nLocaleType.metaDataForI18nLocaleType.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : 'fc3a36b4-f1d9-4016-9773-2b129c0d62f9',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            mongoIsIndexed            : true,
            mongoIsRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '527a8221-0a00-4d75-a7cc-79c3cb1dca8e',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (I18nTextEntityModel.INSTANCE);
    }
}
exports.I18nTextEntityModel = I18nTextEntityModel;
I18nTextEntityModel.INSTANCE = new I18nTextEntityModel();
