// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/I18nCategoryEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.I18nCategoryEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.I18nCategory';
class I18nCategoryEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('cf76e614-6f6f-4e10-a73e-312375bceb0b')
            .setDataModelName('I18nCategory')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('b5b09674-492e-4c32-a63a-a246da428892');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('5ee02cf0-c33d-4d6b-b5da-86ea85ed5fb2'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('665f1b77-d659-4230-a593-e77ec27e2409'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('51f07543-0aae-4250-b1d8-07068a919637'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('a17989bf-fe00-4c8f-817c-ecdd56071ed0'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('7b5d54e1-962d-4ba0-a37d-7fec37567eaf'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('c9a3bcad-9973-4346-9cde-c17a37b95951'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '5d0131df-4ac4-47a0-91d2-ad4e776f4928'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('429a4f9c-abed-4bbd-90ff-0b17f4c4f5b1'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('1dfcd70c-32eb-4676-ae51-9bcb45e24e49'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '2091ea2c-4aff-4870-89d9-31585b3265f0',
            comment: 'The description of the category. Example: Basic UI controls.'
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '964ea13a-a849-4d0a-8a7b-f6fef30ca2d9',
            comment: 'The name for the category.'
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : 'c6319fe2-b961-4ee8-a500-d4b691c16933',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            mongoIsIndexed            : true,
            mongoIsRequired           : true
        }));
        */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: "i18nTexts",
            attributeNameSingular: "i18nText",
            attributeUUID: '3148af70-2ca8-45df-9ee4-1be496fd5112',
            comment: "The i18nText items that are in the given category.",
            foreignRefEntityNamespace: persistable_1.I18nText.metaDataForI18nText.namespace
        }));
    }
    static getInstance() {
        return (I18nCategoryEntityModel.INSTANCE);
    }
}
exports.I18nCategoryEntityModel = I18nCategoryEntityModel;
I18nCategoryEntityModel.INSTANCE = new I18nCategoryEntityModel();
