// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ImportDataAttributeEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataAttributeEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ImportDataAttribute';
class ImportDataAttributeEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('4c1449ad-9975-4174-b8ff-1c72d65c41cc')
            .setDataModelName('ImportDataAttribute')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('c27b89a1-fa9c-47e8-88fe-02a94876937b');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('91b97317-5ffc-4e39-ab3c-4a78efe31cd8'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('324429e9-dfbf-4d12-96ab-87000e995322'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('ece4726b-56ed-44f1-9acb-f9fe95a12025'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('a62d0245-920d-448f-983c-0ed777359819'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('093652d5-3c63-4b95-9859-25de29987610'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('448a8cc8-23ad-497c-840e-7b82eecff8ca'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '9a368389-ef5f-4dbe-82cb-4baa5579b982'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('fba76a8b-62b4-4543-be7d-cf1ffabddf13'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('c39784d4-93f0-4477-ac1d-66b891925ccb'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: '9dbfc127-c869-4d09-83c6-719339728c5d' }));
        // string
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'importFileAttributeName',
            attributeUUID: 'ac538fb2-cbcd-471c-8471-a19151693d12',
            comment: 'Import file attribute name.',
            validator: zod_1.z.string()
        }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'importDataAttributeType',
            attributeUUID: '8cb6e9dd-af53-4361-b96d-caa09379f5b2',
            comment: 'Import file attribute type.'
        }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'dataIntent',
            attributeUUID: 'eff98a70-f273-4e2e-b22b-e15600ea5039',
            comment: 'Intent for attribute name.'
        }));
    }
    static getInstance() {
        return (ImportDataAttributeEntityModel.INSTANCE);
    }
}
exports.ImportDataAttributeEntityModel = ImportDataAttributeEntityModel;
ImportDataAttributeEntityModel.INSTANCE = new ImportDataAttributeEntityModel();
