// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ImportDataEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const Organization_1 = require("../../classes/persistable/Organization");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ImportData';
class ImportDataEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('e47f24fe-5811-4273-bd0e-80cf3dd17be6')
            .setDataModelName('ImportData')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('18430eb4-d8d2-48dc-8e1f-410518c66b46');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('8c902301-322a-4ed9-98b8-ce1bb2aa94da'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('45b2595a-3568-4625-800d-7c71a4b9c945'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('e1f742f2-dd72-4bd2-a43d-9a36dae683e1'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('64f046bd-3b8b-40c3-ad6b-3cf117f28764'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('f093be87-18bf-4e2c-b814-5b62585ad545'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('f87ce54d-7aea-407b-9131-2e4dabf88d3c'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '8c89699d-9015-487a-a98e-bc5eb9f98864'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('775368a8-0780-4bf6-8e64-78505cd93c40'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('b6bea63e-979c-41f5-a957-0f068fa714eb'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'e354ef9f-b635-438b-ae18-79433156d94d' }));
        // TODO mongo ref objectId
        // importDataFileUpload
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'importDataFileUpload',
            attributeUUID: '12ff9f28-12cc-4385-b402-aada393fca5e',
            comment: 'ImportDataFileUpload OID.',
            isMongoRequired: false,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '895a4adc-99b1-45d7-9569-d93fcb06f4f4',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: 'b2f272f9-61ad-46e9-899f-c32951e361c1',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // BOOLEAN ------------------------------------------------------------
        // booleanIndexedVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar1',
            attributeUUID: '837aa16f-bfd4-40e0-b45b-a7f82e8fef27',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanIndexedVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar2',
            attributeUUID: 'd6859074-2fa0-42b2-9bef-226c43e9f43b',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanIndexedVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar3',
            attributeUUID: '64037e1c-975d-478f-b694-595f174bce59',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanIndexedVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar4',
            attributeUUID: '1b671044-067f-45b5-bd08-0826456a09e7',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanIndexedVar5
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar5',
            attributeUUID: '80100eca-04d6-4920-90d4-3a98bf4e798a',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanIndexedVar6
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar6',
            attributeUUID: '4bfda67b-0741-4a7b-bbe7-f0e50a1fd1ac',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanIndexedVar7
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar7',
            attributeUUID: 'd7b9eb2a-1f7a-4bc1-8ac7-71d7bfb65e70',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanIndexedVar8
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanIndexedVar8',
            attributeUUID: '79c18e5f-17c2-4d87-94a1-1db5cf24464d',
            comment: 'Indexed boolean variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // booleanVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar1',
            attributeUUID: '10fe21eb-381f-4eed-8611-d6588c5dd9ff',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar2',
            attributeUUID: 'c503f1bc-4704-4c2a-9bcf-20780787b917',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar3',
            attributeUUID: 'b0e20ac0-0602-43ee-abbe-0bbb5bf11c18',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar4',
            attributeUUID: '1f71eb7b-e058-48bc-bdf7-7658053ffb2e',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar5
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar5',
            attributeUUID: 'a0ff7b0d-6839-4f9c-a471-a22031de48a3',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar6
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar6',
            attributeUUID: '3f66187f-91f1-4a8a-822b-ae050d0a54ce',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar7
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar7',
            attributeUUID: '4c5e0514-30e1-4321-9cda-641c10ed1c64',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar8
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar8',
            attributeUUID: '1a0b4d17-5bba-4120-b3ae-a37df305c1fb',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar9
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar9',
            attributeUUID: '2176fa46-e0b5-449d-9a1e-4d916e100e7e',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar10
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar10',
            attributeUUID: '491ffc0c-aca4-41a3-aa6a-1e5fa4a5b16b',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar11
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar11',
            attributeUUID: '0bd8a610-dc4b-43d8-b483-0cd7d6da92c9',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar12
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar12',
            attributeUUID: '57b0265a-9ab7-4554-b354-de9264880fd9',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar13
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar13',
            attributeUUID: '2ab8164f-bba3-40ac-a369-8200841b9870',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar14
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar14',
            attributeUUID: 'bcb2bdff-5878-480c-883e-9919123a869a',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar15
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar15',
            attributeUUID: '0d612612-8c13-400e-804f-eb8203da4c53',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // booleanVar16
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'booleanVar16',
            attributeUUID: 'd2b95a11-c235-4504-ae5a-33faab6b57b6',
            comment: 'Boolean variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // BOOLEAN ============================================================
        // DATE ---------------------------------------------------------------
        // dateIndexedVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar1',
            attributeUUID: 'f6c0b520-d8ff-4441-b209-45c0d577615d',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateIndexedVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar2',
            attributeUUID: 'be675c5f-894e-428e-b298-580e02ef688e',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateIndexedVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar3',
            attributeUUID: 'fdaddf8f-ed6a-4d77-93aa-766663796606',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateIndexedVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar4',
            attributeUUID: '694e6f9d-0cdf-4178-baec-7a783d42d1ab',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateIndexedVar5
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar5',
            attributeUUID: '97ac9f19-d74a-4013-b71f-52315e167b20',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateIndexedVar6
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar6',
            attributeUUID: 'c40dbe02-04f7-4cc8-9d53-de46bb9a7edd',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateIndexedVar7
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar7',
            attributeUUID: '0ba2a5a6-2cef-4528-8480-294b8dea3ebd',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateIndexedVar8
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateIndexedVar8',
            attributeUUID: '9c358df5-579d-4199-a6b8-abca624690d0',
            comment: 'Indexed date variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // dateVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateVar1',
            attributeUUID: 'df2300bd-100c-4be2-b2a6-b9501b97e9e5',
            comment: 'Date variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // dateVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateVar2',
            attributeUUID: '56e4f013-3c1e-433c-b370-482aa36eefbd',
            comment: 'Date variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // dateVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateVar3',
            attributeUUID: '266917ef-03c5-4e5b-bdc3-91a091477212',
            comment: 'Date variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // dateVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'dateVar4',
            attributeUUID: 'ef73bab4-56fc-46ca-a920-5e9bfd58b5b8',
            comment: 'Date variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // DATE ===============================================================
        // importDataMap
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'importDataMap',
            attributeUUID: 'ad10f7cd-94fa-41f9-9b4a-5da5c5a0846c',
            comment: 'ImportDataMap OID.',
            isMongoRequired: false,
        }));
        // number -------------------------------------------------------------
        // numberIndexedVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar1',
            attributeUUID: '222f47c5-2cf4-4793-9c41-103c51be3b10',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberIndexedVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar2',
            attributeUUID: 'a7e7f2e3-649c-417e-af1c-bfcc3de55893',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberIndexedVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar3',
            attributeUUID: '9b26df02-6d93-42ef-bd3a-6bf8a0131ab1',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberIndexedVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar4',
            attributeUUID: 'c7a26b38-f36f-47a9-8f10-437742789263',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberIndexedVar5
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar5',
            attributeUUID: '0d845637-9ed7-472c-8132-2e674fc21ab0',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberIndexedVar6
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar6',
            attributeUUID: '73bc32b4-199e-4ce7-a56f-fa1c4ed8a2a6',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberIndexedVar7
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar7',
            attributeUUID: '7981b57a-0237-4bf6-9589-9cb7ff3dd131',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberIndexedVar8
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberIndexedVar8',
            attributeUUID: '46c359fc-bd35-44ec-aa61-a6edd8109038',
            comment: 'Indexed number variable name',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // numberVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar1',
            attributeUUID: '2a547a96-4990-45d6-a8f1-6172c756b0a6',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar2',
            attributeUUID: '66920f77-9e81-4f32-9c24-c7964796a4d8',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar3',
            attributeUUID: '455834ef-111d-46ca-bf90-f4a61c0fc9ea',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar4',
            attributeUUID: 'ff553c1c-186b-41cf-8b8a-971d6c8ca184',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar5
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar5',
            attributeUUID: 'e2006159-0c2d-47d6-9909-c17be644ed04',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar6
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar6',
            attributeUUID: 'd1fa2718-a8f7-4eef-a01b-268209978e94',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar7
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar7',
            attributeUUID: '02f1e7c9-d240-462f-84f1-150d9c6a6521',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar8
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar8',
            attributeUUID: '9ff048d5-5944-4ffd-bd27-5e687d2b624b',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar9
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar9',
            attributeUUID: '464e26ca-25cc-44c1-b40d-06343635c9a8',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar10
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar10',
            attributeUUID: '5d172ee0-9a78-48c2-a560-cc1e5ac4a6ed',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar11
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar11',
            attributeUUID: 'fe07a38b-ec60-43a8-8f6b-8529ec08c271',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar12
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar12',
            attributeUUID: '085bdbd7-d2af-4768-8659-63b443059cff',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar13
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar13',
            attributeUUID: '63475398-918d-4096-86c7-e92021e50ca7',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar14
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar14',
            attributeUUID: '975c8aa5-283e-4684-800d-aa935c68a96c',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar15
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar15',
            attributeUUID: '280e290c-f0a8-4aaa-ae09-85b625a82c14',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // numberVar16
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'numberVar16',
            attributeUUID: 'cbedbfe9-ff9e-48f8-b648-b8d04112b398',
            comment: 'Number variable name',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // number ============================================================= 
        // string -------------------------------------------------------------
        // stringIndexedVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar1',
            attributeUUID: 'a1f20257-c323-4756-b0e5-13d9531ede32',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar2',
            attributeUUID: '8d0b455a-c37e-4ccc-8ee1-542ccaadf4f3',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar3',
            attributeUUID: '52a523e0-ad3c-462e-ae46-459d717a7b34',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar4',
            attributeUUID: 'b7c9a599-f27c-4d12-96cb-37ec85564a23',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar5
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar5',
            attributeUUID: '143f2f46-d1cd-4ad6-8f7e-bb7cd1f12788',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar6
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar6',
            attributeUUID: '7146a8d5-9996-4c85-864c-efa5f9538669',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar7
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar7',
            attributeUUID: '84bee8ba-decc-49f6-b106-802d6489c0d0',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar8
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar8',
            attributeUUID: '3565ead6-a197-4f98-a2c6-dac163a7ff32',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar9
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar9',
            attributeUUID: '2f33ce75-e49b-4521-9913-da536a7a8f55',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar10
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar10',
            attributeUUID: '72c3d613-9e49-4a92-b218-9e3b565ce179',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar11
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar11',
            attributeUUID: '9dce65cf-3592-4f71-9677-f5a79074d613',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar12
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar12',
            attributeUUID: '3d7b6234-daf2-412e-8bb6-edc6230fe4da',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar13
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar13',
            attributeUUID: '51072887-4f99-4076-b6f9-504427f2206a',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar14
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar14',
            attributeUUID: 'bbb7d80b-1d62-407f-9c4a-be396b7b4f16',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar15
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar15',
            attributeUUID: '5d24c744-ff89-498f-b4cb-c96488e1e286',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringIndexedVar16
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringIndexedVar16',
            attributeUUID: '0bd0f29f-d0d7-4957-b6de-4b45ab0df6b0',
            comment: 'Indexed string variable name.',
            isMongoIndexed: true,
            isMongoRequired: false,
        }));
        // stringVar1 
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar1',
            attributeUUID: 'c8af35e0-0dfb-4f43-86bf-60a146874c01',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar2',
            attributeUUID: '63ba9c46-162b-44f0-888f-34bcf59c5078',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar3',
            attributeUUID: '1b6f42ff-3b21-4e9f-8e67-0b2409be897e',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar4
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar4',
            attributeUUID: 'b6a9ea0a-06a1-445d-b5d4-75d03917df04',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar5
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar5',
            attributeUUID: 'e5c44300-4b07-4b23-8e48-d8cdab049d65',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar6
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar6',
            attributeUUID: 'a7387087-105e-49ab-b4dc-551ca6765d2a',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar7
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar7',
            attributeUUID: 'ce76b407-267b-4172-aa50-0681eb25d49d',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar8
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar8',
            attributeUUID: 'fe389d95-4da2-452d-ab5f-dadaf26be9c8',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar9
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar9',
            attributeUUID: 'a51ee228-9e0c-4d90-bd78-ce38eafa1d5b',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar10
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar10',
            attributeUUID: '6e2384e4-8b47-4660-959b-852ec62e6e36',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar11
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar11',
            attributeUUID: '2563fa5b-364c-4ac6-bd07-1c97f7df2f23',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar12
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar12',
            attributeUUID: 'd1676f4f-daca-44aa-8a07-3b619bf1737e',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar13
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar13',
            attributeUUID: 'dd538fc3-0222-4550-8c3f-0f8a9381cae2',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar14
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar14',
            attributeUUID: '99975981-5ddc-4b17-8dbe-610d1f494d8b',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar15
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar15',
            attributeUUID: 'f7c533ac-f24f-4b13-98fb-56f4bfa17ab8',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar16
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar16',
            attributeUUID: '4de76589-4b00-455d-a243-9636ee3dcee4',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar17
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar17',
            attributeUUID: '1ba2a31d-9a45-4a75-91b8-b8b257830d99',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar18
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar18',
            attributeUUID: '33590440-2dd2-4778-83af-4ee53e04f4f2',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar19
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar19',
            attributeUUID: '176cf5db-114a-4ee9-83aa-ed5f98495a70',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar20
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar20',
            attributeUUID: 'e8e32390-257a-4160-99d7-4361ddc164ae',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar21
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar21',
            attributeUUID: '0014d61a-efb8-45d4-b7e5-7bc1138fce23',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar22
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar22',
            attributeUUID: '39110b8c-e394-49f6-a6bd-e7f29dd3d071',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar23
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar23',
            attributeUUID: '297ccdc9-a984-4ddd-91ab-241b6945d1a4',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // stringVar24
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar24',
            attributeUUID: 'aba884ba-4b3a-432b-a7ec-8eca4fdd56d2',
            comment: 'String variable name.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // string =============================================================
    }
    static getInstance() {
        return (ImportDataEntityModel.INSTANCE);
    }
}
exports.ImportDataEntityModel = ImportDataEntityModel;
ImportDataEntityModel.INSTANCE = new ImportDataEntityModel();
