// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ImportDataMapTypeEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataMapTypeEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const RegularExpressionRegexConstUtil_1 = require("../../../utils/validation/RegularExpressionRegexConstUtil");
const ImportDataMapTypeAttributeType_1 = require("../../classes/persistable/ImportDataMapTypeAttributeType");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ImportDataMapType';
class ImportDataMapTypeEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('cbc5c10e-1d78-4cf2-8f2f-514b50d3d408')
            .setDataModelName('ImportDataMapType')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('9d3564a4-36e2-4d3e-888f-d98695e42fa8');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('a9ca9822-fe14-4deb-8d4b-85eff2c3dd29'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('e8178099-4f98-4d9d-bfe5-dab167966489'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('1a749d66-4dc3-4f6b-8a5b-29a9e304a9a1'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('3377743b-6634-451f-89e6-663c29c9eb25'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('78ab189d-aa38-4bd4-9d1f-0e19c9ea688e'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('b437139a-8dcd-408a-916d-49e27ef524d1'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'd5596166-dc42-4ef2-b543-b31c17967db4'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('afc41f87-af5c-4758-8e9d-98d80d7dc584'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('3581d433-5fe7-4fb6-8763-a424fb8a5ed0'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: '83ebe714-53f4-4511-be72-ba01a6d3f147' }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "description",
            attributeUUID: "6b55111e-1a9b-4904-bf45-bdee704553b7",
            comment: "The description of the ImportDataMapType.",
            validator: zod_1.z.string().max(5000, 'Description is too long')
        }));
        // 
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: "importDataMapTypeAttributeTypes",
            attributeNameSingular: "importDataMapTypeAttributeType",
            attributeUUID: 'c068afe0-01e5-4784-b076-cf384adf7714',
            comment: "The ImportData",
            foreignRefEntityNamespace: ImportDataMapTypeAttributeType_1.ImportDataMapTypeAttributeType.metaDataForImportDataMapTypeAttributeType.namespace //SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: 'd8de9df0-2936-4db9-9c11-15745899d33d',
            comment: "Is the type active."
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'ab2a3b15-85a2-4204-a3bf-bdcd3eebe2a5',
            comment: 'The name of the type.',
            validator: zod_1.z.string().min(1, "Name cannot be empty")
                .max(255, "Name is too long")
                .regex(RegularExpressionRegexConstUtil_1.Regex.alphanumericWithSpace, "Name cannot contain special characters")
        }));
    }
    static getInstance() {
        return (ImportDataMapTypeEntityModel.INSTANCE);
    }
}
exports.ImportDataMapTypeEntityModel = ImportDataMapTypeEntityModel;
ImportDataMapTypeEntityModel.INSTANCE = new ImportDataMapTypeEntityModel();
