// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ImportDataMapAttributeEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataMapAttributeEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const ImportDataMapTypeAttributeType_1 = require("../../classes/persistable/ImportDataMapTypeAttributeType");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ImportDataMapAttribute';
class ImportDataMapAttributeEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('1dc7051a-c5b2-479c-b1d1-c903111217e8')
            .setDataModelName('ImportDataMapAttribute')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('7de08986-8211-4df7-b224-d7acaf9b9a66');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('a425418e-7ebd-401e-bebf-b4a988c279b4'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('7f854a6c-400a-447b-9070-1128f2b44ae2'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('56e45dc8-225e-47b6-8d23-42d05d105377'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('afee436a-c090-4954-9a7b-ed110e0ecda3'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('36e269eb-bfb9-451f-9de3-0b9bb190bf9b'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('07fcf3ae-d573-48ef-8d5e-1338c0632643'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'c4e3bd41-1841-460b-ba65-f47c4d3f0aab'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('0276b1c1-a72c-46a3-8c5b-472ea1e77f40'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('f97f2987-f961-4147-bc21-c8d9cd623ffe'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: '55f611ec-5b65-4bcb-918c-56413413d05b' }));
        // // boolean
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableBoolean2({
        //     attributeName   : 'aBoolean',
        //     attributeUUID   : 'fc7eb3a0-e7c1-483b-a6a7-5d6651fb141b',
        //     comment         : 'The name of a boolean value.'
        // }));
        // // number
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableNumber2({
        //     attributeName   : 'aNumber',
        //     attributeUUID   : 'ad357b6a-12ef-471a-a8aa-ef5040139b42',
        //     comment         : 'The name of a number value.'
        // }));
        // importDataFileAttributeName
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'importDataFileAttributeName',
            attributeUUID: 'dba9c7d1-fab0-414e-8504-7ba145107b42',
            comment: 'Given a data file such as a .csv file, this is the name of a comma separated value.'
        }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'importDataMapTypeAttributeType',
            attributeUUID: '470a4302-7b81-4a2d-b003-8102e6220704',
            comment: 'The import data type attriute type.',
            foreignRefEntityNamespace: ImportDataMapTypeAttributeType_1.ImportDataMapTypeAttributeType.metaDataForImportDataMapTypeAttributeType.namespace,
            isMongoIndexed: false,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : 'cf3e4fe3-8672-40fd-8486-e7c03b0e542d',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            mongoIsIndexed            : true,
            mongoIsRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName: "salesReportConfigParamValues",
                attributeNameSingular: "salesReportConfigParamValue",
                comment: "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace: SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace,
                attributeUUID: 'e708a1a6-273c-4e3f-a59e-29933e921cc1', // uuid
            }
       ));
       */
    }
    static getInstance() {
        return (ImportDataMapAttributeEntityModel.INSTANCE);
    }
}
exports.ImportDataMapAttributeEntityModel = ImportDataMapAttributeEntityModel;
ImportDataMapAttributeEntityModel.INSTANCE = new ImportDataMapAttributeEntityModel();
