// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/KmlEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KmlEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const RegularExpressionRegexConstUtil_1 = require("../../../utils/validation/RegularExpressionRegexConstUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const zod_1 = require("zod");
const persistable_1 = require("../../classes/persistable");
const ProductKmlLocation_1 = require("../../classes/persistable/ProductKmlLocation");
const Application_1 = require("../../classes/persistable/Application");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.Kml";
class KmlEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('dc5a1ba6-479b-4fae-b112-288db7887c7a')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('babb5f51-a79c-4b03-976e-af5e47a6baba');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('865fe1a4-d65b-48e9-92b1-322d88ffebec'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('23af1cce-f665-4448-aa0f-b39ded2918a5'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('410e7b51-a099-497a-91c6-f197101a6952'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('d7380994-8fbe-450e-bf4f-5d026c2b81e7'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('ff9d448b-52e1-460e-9748-a98ecf507119'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('ef13d49f-de94-4c1d-9248-5711d870deae'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('8abb964d-1a00-442a-9cfc-0c9003c20ae2'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '91600f31-3bb2-4421-81e3-2f82089f7933'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('19909581-aeb5-4f85-aef7-bce7b4978aa3'));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '7e15d0f0-d62f-4140-82d2-2490b46a2ffa',
            comment: 'The name of the KML file.',
            validator: zod_1.z.string().min(1, "Name cannot be empty")
                .max(255, "Name is too long")
                .regex(RegularExpressionRegexConstUtil_1.Regex.alphanumericWithSpace, "Name cannot contain special characters")
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '6765e83f-7509-4861-a5f4-3e28def5860d',
            comment: 'The description.',
            validator: zod_1.z.string().max(1000, 'Description is to long'),
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: 'fcf4f72a-61dd-4728-bf96-9f4e241d0540',
            comment: 'Entity is active or not.',
            isMongoIndexed: false,
            isMongoRequired: false,
        }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: 'bcbbc88c-898e-4357-9680-13d278013f16',
            comment: 'Active start date.',
            isMongoRequired: false,
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: '971b4cc9-f8c0-4a1b-98f3-ee39e7d2d7ad',
            comment: 'Active end date.',
            isMongoRequired: false,
        }));
        // kmlType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kmlType',
            attributeUUID: 'a0dcd68b-d2d6-4c4d-85c1-ff8a497c5c97',
            comment: 'The Kml type Id for this layer.',
            foreignRefEntityNamespace: persistable_1.KmlType.metaDataForKmlType.namespace,
            // TODO ASSOCIATE WITH KMLLayerType, change both to true
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // kmlLayers
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('kmlLayers', 'kmlLayer', persistable_1.KmlLayer.metaDataForKmlLayer.namespace, persistable_1.KmlLayer.metaDataForKmlLayer.mongoAttributes.kml, 'defad548-5f85-46de-a6d2-356a5f4d21a5', 'The KmlLayer for the kml.', 1));
        // productPricingKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productPricingKmlLocations', 'productPricingKmlLocation', persistable_1.ProductPricingKmlLocation.metaDataForProductPricingKmlLocation.namespace, persistable_1.ProductPricingKmlLocation.metaDataForProductPricingKmlLocation.mongoAttributes.kml, 'a860bc09-f718-498c-afb5-1da263418f88', // uuid
        'The product pricing kml location', 1));
        // productColorKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productColorKmlLocations', 'productColorKmlLocation', persistable_1.ProductColorKmlLocation.metaDataForProductColorKmlLocation.namespace, persistable_1.ProductColorKmlLocation.metaDataForProductColorKmlLocation.mongoAttributes.kml, 'ce732dd2-3aa9-4fb4-9628-937042699b82', 'The product color kml location', 1));
        // productCategoryKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productCategoryKmlLocations', 'productCategoryKmlLocation', persistable_1.ProductCategoryKmlLocation.metaDataForProductCategoryKmlLocation.namespace, persistable_1.ProductCategoryKmlLocation.metaDataForProductCategoryKmlLocation.mongoAttributes.kml, '1dc360bf-336b-485a-b575-48b58e70718b', 'The product category kml location', 1));
        // productKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productKmlLocations', 'productKmlLocation', ProductKmlLocation_1.ProductKmlLocation.metaDataForProductKmlLocation.namespace, ProductKmlLocation_1.ProductKmlLocation.metaDataForProductKmlLocation.mongoAttributes.kml, '1bde90ec-94b0-4dcd-abf4-e1d44a4e4b4f', 'The product kml location', 1));
        // taxKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('taxKmlLocations', 'taxKmlLocation', persistable_1.TaxKmlLocation.metaDataForTaxKmlLocation.namespace, persistable_1.TaxKmlLocation.metaDataForTaxKmlLocation.mongoAttributes.kml, '8b076d3b-a8f9-4b19-8912-0b677d2cc65e', 'The tax kml location', 1));
        // // kmlGeometry
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoVirtualArray(
        //     'kmlGeometries',
        //     'kmlGeometry',
        //     KmlGeometry.metaDataForKmlGeometry.namespace,
        //     KmlGeometry.metaDataForKmlGeometry.mongoAttributes.kml,
        //     '0ae284b3-bbf1-417d-97ff-4159e17f78ed',
        //     'The kml geometry',
        //     1
        //     )
        // );
        /**
         * organization oid
         * The Tenant Organization this workspace belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "organization",
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: 'ba11ba65-1bac-454f-9286-a0edf0348b6f',
            comment: "The tenant Organization.",
            isMongoRequired: true,
        }));
        // application
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "application",
            foreignRefEntityNamespace: Application_1.Application.metaDataForApplication.namespace,
            attributeUUID: '11599057-8143-4735-8b63-96e94065f809',
            comment: "The application kml belong to.",
            isMongoRequired: false,
        }));
        // workspace
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "workspace",
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            attributeUUID: '5ead8288-dd69-420f-a9ea-69d743ede708',
            comment: "The workspace this kml belong to.",
            isMongoRequired: false,
        }));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('ad213eaa-637e-4ed1-8698-118a9b0b8375'));
        // s3Url
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 's3Url',
            attributeUUID: '06156c6c-81f7-411a-93ea-f3d4c3ad05d9',
            comment: 'The name of the s3Url path.',
            validator: zod_1.z.string(),
            isMongoRequired: false,
        }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({
            uuid: 'f39b5626-276d-41bd-b8cd-d7dc25450b0c'
        }));
    }
    static getInstance() {
        return (KmlEntityModel.INSTANCE);
    }
}
exports.KmlEntityModel = KmlEntityModel;
KmlEntityModel.INSTANCE = new KmlEntityModel();
