// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/KmlLayerEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KmlLayerEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ProductKmlLocation_1 = require("../../classes/persistable/ProductKmlLocation");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.KmlLayer';
class KmlLayerEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('791fab77-d2e4-44e9-82fb-e1f18736e92b')
            .setDataModelName('KmlLayer')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('70d8e5a3-84bb-4063-b1f5-9287c318507c');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('c5885b24-5607-4472-bc00-88affa067039'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('f1324c29-06ef-41cc-a4d5-6f36baab0b2e'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('427bf688-1b16-435b-83dd-d4023972cdf3'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('fd3949d7-2242-4d5e-ae23-80be5cdad4aa'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('e428fdf2-e9e1-4603-ba84-a4a9ab6b6587'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('06346a9b-9995-4237-8bb4-4fafdb6ee95c'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'af675a5d-0c85-46d2-9c6c-1978d4d534f4'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('0b16b598-d923-4e3d-a786-92f3c82cf51e'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('44ba9e6d-3138-411a-9eb9-f0d6af1dfd04'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '55496d3f-dc3b-4e11-8020-a87ae5384b0d',
            comment: 'Kml layer is active or not.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '8c101ba1-50bd-4d23-b28e-263699b0bf38',
            comment: 'The name of Kml layer.'
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '8c101ba1-50bd-4d23-b28e-263699b0bf38',
            comment: 'The description of Kml layer.',
            validator: zod_1.z.string().max(5000, 'Description is too long')
        }));
        // kml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kml',
            attributeUUID: 'eefc5f76-3168-4754-9c76-05fb7819dced',
            comment: 'The Kml for this layer.',
            foreignRefEntityNamespace: persistable_1.Kml.metaDataForKml.namespace,
            // TODO ASSOCIATE WITH KML, change both to true
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // kmlType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kmlType',
            attributeUUID: 'eefc5f76-3168-4754-9c76-05fb7819dced',
            comment: 'The Kml layer type Id for this layer.',
            foreignRefEntityNamespace: persistable_1.KmlType.metaDataForKmlType.namespace,
            // TODO ASSOCIATE WITH KMLLayerType, change both to true
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        // kmlGeometry
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('kmlGeometries', 'kmlGeometry', persistable_1.KmlGeometry.metaDataForKmlGeometry.namespace, persistable_1.KmlGeometry.metaDataForKmlGeometry.mongoAttributes.kmlLayer, 'bc81ccdf-e62c-46f7-ac19-38b954d805ab', 'The kml layer geometry', 1));
        // productPricingKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productPricingKmlLocations', 'productPricingKmlLocation', persistable_1.ProductPricingKmlLocation.metaDataForProductPricingKmlLocation.namespace, persistable_1.ProductPricingKmlLocation.metaDataForProductPricingKmlLocation.mongoAttributes.kmlLayer, 'a07036e9-626e-440f-8167-f0d1df17c0c4', // uuid
        'The product pricing kml location', 1));
        // productColorKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productColorKmlLocations', 'productColorKmlLocation', persistable_1.ProductColorKmlLocation.metaDataForProductColorKmlLocation.namespace, persistable_1.ProductColorKmlLocation.metaDataForProductColorKmlLocation.mongoAttributes.kmlLayer, '6f649796-1cd5-498e-bea3-1a54a7e40267', 'The product color kml location', 1));
        // productCategoryKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productCategoryKmlLocations', 'productCategoryKmlLocation', persistable_1.ProductCategoryKmlLocation.metaDataForProductCategoryKmlLocation.namespace, persistable_1.ProductCategoryKmlLocation.metaDataForProductCategoryKmlLocation.mongoAttributes.kmlLayer, '00bddbe7-9ac0-40df-b17e-3a88b35e0301', 'The product category kml location', 1));
        // productKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productKmlLocations', 'productKmlLocation', ProductKmlLocation_1.ProductKmlLocation.metaDataForProductKmlLocation.namespace, ProductKmlLocation_1.ProductKmlLocation.metaDataForProductKmlLocation.mongoAttributes.kmlLayer, 'cabbd11b-af03-4923-8975-b1ab1685662f', 'The product kml location', 1));
        // taxKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('taxKmlLocations', 'taxKmlLocation', persistable_1.TaxKmlLocation.metaDataForTaxKmlLocation.namespace, persistable_1.TaxKmlLocation.metaDataForTaxKmlLocation.mongoAttributes.kml, '058695e0-4946-480b-805a-4e0063394181', 'The tax kml location', 1));
        // workspace
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "workspace",
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            attributeUUID: '50ae0aac-63ac-4661-8de2-e6ad263a90d1',
            comment: "The workspace this kml belong to.",
            isMongoRequired: false,
        }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: '5d94dcc3-5c85-477e-bbf2-caaa0e8ccbb7',
            comment: 'Active start date.',
            isMongoRequired: false,
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: 'db2adbd8-a7db-49ae-ac70-60585325fb23',
            comment: 'Active end date.',
            isMongoRequired: false,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'eefc5f76-3168-4754-9c76-05fb7819dced',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoRequired: false
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '0ce6fe74-9fb2-4c8e-ba8c-5ba7b7a034ff',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (KmlLayerEntityModel.INSTANCE);
    }
}
exports.KmlLayerEntityModel = KmlLayerEntityModel;
KmlLayerEntityModel.INSTANCE = new KmlLayerEntityModel();
