// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/OrganizationEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const OrganizationType_1 = require("../../classes/persistable/OrganizationType");
const Organization_1 = require("../../classes/persistable/Organization");
const Application_1 = require("../../classes/persistable/Application");
const TypeReference_1 = require("../../../data/entity/TypeReference");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.Organization';
class OrganizationEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        //this.setAddPersistableAttributes(true);
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('d970d8e0-907a-429b-930d-239f2e516872')
            .setDataModelName('Organization')
            .setCrudOverWriteSourceCode(false) // not use in new entity model build
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('63160314-e1e9-4a0b-bc62-ee320ce4e36c');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('5fbef652-83a2-492c-8805-9f876f8e5888'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('2e3457a3-1490-470f-bcad-d16f0a0e3a0b'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('5d91ccb3-2aa3-4dbb-a6c5-94001ba6bdf8'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('423a993f-4244-4cea-bb4f-d769eff84987'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('987c060f-3720-4b91-b469-32f5231180ac'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('30b5eca7-643c-4e7a-b12e-d06246dbe6ec'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('91f6eea6-9db5-4813-867a-d5d5e2f8249a'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'ce883aa3-59ea-47c4-b602-2699f8397629'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('686a6524-b700-4ea6-9e5a-1bc99bb941e4'));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '1f87690e-a2d4-408f-bb38-6d409bab5eef',
            comment: 'The description for the Organization',
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '58745954-a64e-447f-bf49-ee24288f06af',
            comment: 'Identifies if the Organization is active.',
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'fcca3eda-8ec3-457c-a252-3be94e28d118',
            comment: 'The name of the Organization',
        }));
        // setting
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableJsonString2({
            attributeName: 'setting',
            attributeUUID: '8b7b236a-13ba-4b3a-a7d0-7bc3c77a6ef5',
            comment: 'The organization app setting',
            variableType: new TypeReference_1.TypeReference({
                typeName: 'Object'
            }),
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        /**
         * organizationType
         * Many Organization to One OrganizationType
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organizationType',
            foreignRefEntityNamespace: OrganizationType_1.OrganizationType.metaDataForOrganizationType.namespace,
            attributeUUID: '703dc016-5853-45fc-b57e-2d5181606ccb',
            comment: 'The Organization Type.'
        }));
        /**
         * Application.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: 'applications',
            attributeNameSingular: 'application',
            attributeUUID: '1bde90ec-94b0-4dcd-abf4-e1d44a4e4b4f',
            comment: "The organization application.",
            foreignRefEntityNamespace: Application_1.Application.metaDataForApplication.namespace
        }));
        /**
         * parentOrganization
         * Many child Organization to One parent Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'parentOrganization',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: 'd95da75e-d797-4cf8-b1b3-3d95eb599cbd',
            comment: 'The Parent Organization.'
        }));
    }
    static getInstance() {
        return (OrganizationEntityModel.INSTANCE);
    }
}
exports.OrganizationEntityModel = OrganizationEntityModel;
OrganizationEntityModel.INSTANCE = new OrganizationEntityModel();
