// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/PandaDocTemplateEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PandaDocTemplateEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.PandaDocTemplate";
class PandaDocTemplateEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID("b3633035-8d94-410d-bff2-7cd1546ad59a")
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID("b294f50a-604a-4146-9f91-93dd52d5e885");
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('97474b4b-7613-486c-be4c-46f024a875f2'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState("7c620605-9b77-417a-a30d-3c883904f7bd"));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem("f8c777e2-6983-4cc2-863f-1134fee1d4f2"));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey("9b79a458-aab9-49f9-bcdf-c3753b73526d"));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate("d77f4a2a-cba2-406a-882d-74440d70ad57"));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId("ca40d194-9f80-4b38-b9c2-9666003da079"));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, "94f6e060-ac77-4116-8f6f-f1582f7b504a"));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate("49b1a9c7-b3a3-48ee-a7cf-869656842a88"));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship(""));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "typeScriptDataType",
            attributeUUID: "73d2061b-5734-4022-be08-83f62a2e3faa", // uuid
            comment: "The workspace description.",
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "name",
            attributeUUID: "68fdfae4-478b-4da2-956b-e86c0ffc669e", // uuid
            comment: "The data type",
        }));
        /**
         * organization oid
         * The Tenant Organization this workspace belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "organization",
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: "090033c1-2c28-4aac-bbc8-5fc24c77581e",
            comment: "The tenant Organization."
        }));
    }
    static getInstance() {
        return (PandaDocTemplateEntityModel.INSTANCE);
    }
}
exports.PandaDocTemplateEntityModel = PandaDocTemplateEntityModel;
PandaDocTemplateEntityModel.INSTANCE = new PandaDocTemplateEntityModel();
