// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/PandaDocWorkspaceEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PandaDocWorkspaceEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.PandaDocWorkspace";
// TODO: DELETE_FILE and clean up related files
class PandaDocWorkspaceEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID("52ee3857-f113-405a-b54c-3452882914b0")
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID("5ab506f3-547b-4fc2-a21d-121f70a166b1");
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('e2dc4155-0d5d-4976-b3b7-bf5730c29a96'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState("9f1bb526-39f5-4c1f-9bef-160a82566420"));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem("ef721bb2-f38c-41f5-a4c9-cfc1b4ca35cf"));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey("1678508b-4d7d-4561-8119-fce270b176ed"));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate("23203fff-7134-4a79-9a6a-20d020d141ac"));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId("b5415015-6707-4161-9f7b-75abff5d5819"));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, "79ccf84b-48bd-4ea1-afb8-aeaed2ded0ef"));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate("37bdd486-165d-443b-9927-557b0c9d9d3e"));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship("9c806307-362a-45ff-b95c-363d4d5f8b0b"));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "decription",
            attributeUUID: "0eee26b4-9d1a-423d-a3e0-fdb006cfd541", // uuid
            comment: "The workspace description."
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: "isActive",
            attributeUUID: "db70829e-dabf-4b77-a018-2ccb067aaf49",
            comment: "Identifies if the WorkSpace is active."
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "name",
            attributeUUID: "dca92073-0819-466a-8afd-befc447926b2", // uuid
            comment: "The data type"
        }));
        // securityCode
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "securityCode",
            attributeUUID: "412261d1-29d1-45be-affc-16385b58dab3", // uuid
            comment: "The security code for the workspace."
        }));
        // securityCodeExpirationDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: "securityCodeExpirationDate",
            attributeUUID: "462b9b94-3fe4-4951-a545-1cf9afd6bcd9", // uuid
            comment: "The security code expiration date."
        }));
        /**
         * organization oid
         * The Tenant Organization this workspace belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "organization",
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: "29643f11-a4f1-4864-9782-ed839bbee058",
            comment: "The tenant Organization."
        }));
    }
    static getInstance() {
        return (PandaDocWorkspaceEntityModel.INSTANCE);
    }
}
exports.PandaDocWorkspaceEntityModel = PandaDocWorkspaceEntityModel;
PandaDocWorkspaceEntityModel.INSTANCE = new PandaDocWorkspaceEntityModel();
