// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductPricingRuleEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPricingRuleEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductPricingRule';
class ProductPricingRuleEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('c00b7416-57ea-4096-b052-7a6a2ebadbed')
            .setDataModelName('ProductPricingRule')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('2ee1bfe8-1bc8-490d-9ac6-6d6f4ef35fa5');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('a6e65d65-17e0-43ed-b445-0b0ea02d5174'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('44cf366d-2f64-4066-a344-5f7fb215b049'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('13166cf2-7e1d-419b-97a5-919b6e5c7e8f'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('cc1ff800-19c7-4a07-aa54-1828a923473f'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('6da558e1-cc29-40a4-8ebe-c929fcd77f09'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('8df7cc02-c6da-4021-afe7-3b43b917e767'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '7746f472-2922-4115-b9a5-c0c754fcd32d'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('1e19cd47-28fd-439d-8415-9c7459f0fa44'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('c85decab-5ae2-4fb8-b9b8-3540557c1a91'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'd24d5eef-e2a4-4d64-aaa6-470517772860',
            comment: 'Name of product rule.',
            validator: zod_1.z.string()
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '208eabad-9086-46b4-af6e-fd4f8d323b6b',
            comment: 'Description of the product rule.',
            validator: zod_1.z.string(),
            isMongoRequired: false
        }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: '830ef0f0-d2fc-4596-b827-3ac57eb867ca',
            comment: 'Active start date.',
            validator: zod_1.z.date(),
            isMongoRequired: false
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: '36fc284e-12a7-4d1a-80ec-4ac83c3fccac',
            comment: 'Active end date.',
            validator: zod_1.z.date(),
            isMongoRequired: false
        }));
        // rules
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'rules',
            attributeUUID: 'be1c0789-c5c7-4cac-968b-f9641fc40f3e',
            comment: 'Rules.',
            validator: zod_1.z.string(),
            isMongoRequired: false
        }));
        // product
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'product',
            attributeUUID: '6a83dfaf-b7b6-4a40-ad2a-411574f73e13',
            comment: 'The Product.',
            foreignRefEntityNamespace: persistable_1.Product.metaDataForProduct.namespace,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'f9ca1b48-3b4b-4c3f-8788-8a6af6346a15',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            validator: zod_1.z.string(),
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '4db71a2f-9354-42e1-a8b6-f3c7e8d19e02',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '6ba4e726-7c03-4a2f-b691-edbf85a24cec',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductPricingRuleEntityModel.INSTANCE);
    }
}
exports.ProductPricingRuleEntityModel = ProductPricingRuleEntityModel;
ProductPricingRuleEntityModel.INSTANCE = new ProductPricingRuleEntityModel();
