// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductServiceDayOfWeekRouteEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductServiceDayOfWeekRouteEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductServiceDayOfWeekRoute';
class ProductServiceDayOfWeekRouteEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('570ddc99-b514-49e5-bf50-4d30ed2c91d5')
            .setDataModelName('ProductServiceDayOfWeekRoute')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('0e6cb6a3-0f5a-4705-b6da-bb1f639f72c6');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('d397078c-83d4-4484-988a-37d2570ca4e8'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('d09f7ed3-4dac-4480-9b6a-20947894cb2b'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('a18dbd6e-42ef-4668-9a60-15eb45cbe53c'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('38498784-b878-4ef1-90eb-f9adb34ea0e5'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('f00438f9-7aa0-423a-916c-145b6a967913'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('a91b6e76-27b4-496a-b064-e19e2c03992b'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '89812e40-827c-4369-8bc9-fb824cf82f59'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('dbd0bc6b-eec2-49ab-9ad4-785996859006'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('38ba6ac0-4ddf-4cc5-979f-ebd6dd9ddd48'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        /**
         * ProductServiceDayOfWeek
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'product',
            attributeUUID: '1aae1088-d8b1-4539-ba9d-f1f4e0cdc3d4',
            comment: 'The product.',
            foreignRefEntityNamespace: persistable_1.ProductServiceDayOfWeek.metaDataForProductServiceDayOfWeek.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Route
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'route',
            attributeUUID: '9f1e931d-d7cf-4812-9b20-7ab60901a01a',
            comment: 'The route.',
            foreignRefEntityNamespace: persistable_1.Route.metaDataForRoute.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '23f9dd5f-4bff-4472-8424-13ff7c2d12d9',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '593b1d2e-0c95-493c-b603-e53b00f50d84',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '43e2cc71-8589-401f-bb42-02be5270bfde',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductServiceDayOfWeekRouteEntityModel.INSTANCE);
    }
}
exports.ProductServiceDayOfWeekRouteEntityModel = ProductServiceDayOfWeekRouteEntityModel;
ProductServiceDayOfWeekRouteEntityModel.INSTANCE = new ProductServiceDayOfWeekRouteEntityModel();
