// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductServiceDayOfWeekKmlLocationEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductServiceDayOfWeekKmlLocationEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductServiceDayOfWeekKmlLocation';
class ProductServiceDayOfWeekKmlLocationEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('42920117-5375-4cbd-a72a-e530ec9d9cac')
            .setDataModelName('ProductServiceDayOfWeekKmlLocation')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('61f3de88-26de-468f-a8d2-65bcaa42994b');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('6a0971a9-da92-4d75-b270-c4a5017c2e4f'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('083f50c2-5620-4e22-9f37-1f36abdb937f'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('ba615491-a6cb-4c85-adcc-cc51997dcd62'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('f9c4ef88-d699-4780-bd5a-d2a67f5f2231'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('9afec211-c7d0-4f60-934b-df542aa42477'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('be144cda-dd61-408c-af6f-188b4371c86b'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'cfa574e3-2fb9-4696-b438-393214a175a4'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('373a9084-b58f-47fc-88d1-4531d287a309'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('7113ef28-5fea-4d84-b926-73705f065dfe'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // kml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kml',
            attributeUUID: 'ca26b436-28e2-433a-94fc-e6fe0431a925',
            comment: 'The KML.',
            foreignRefEntityNamespace: persistable_1.Kml.metaDataForKml.namespace,
        }));
        // kmlLayer
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kmlLayer',
            attributeUUID: '49f34169-373f-4416-a135-ef2e7dd86301',
            comment: 'The KML layer id.',
            foreignRefEntityNamespace: persistable_1.KmlLayer.metaDataForKmlLayer.namespace,
            isMongoRequired: true,
            isMongoIndexed: true,
        }));
        // kmlLayerPolygon // TODO
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
        //     attributeName               : 'kmlLayerPolygon',
        //     attributeUUID               : 'de31f67c-045f-4091-9816-46fdf6a78d08',
        //     comment                     : 'The KML layer polygon.',
        //     foreignRefEntityNamespace   : KmlLayerPolygon....
        // }));
        // productServiceDayOfWeek
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'productServiceDayOfWeek',
            attributeUUID: 'a18e1656-e0ba-4357-bb96-dee48c3d72c5',
            comment: 'Product Service Day of Week mongo ref.',
            foreignRefEntityNamespace: persistable_1.ProductServiceDayOfWeek.metaDataForProductServiceDayOfWeek.namespace,
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '9ab906d1-397a-4ad8-9ee9-7249063bf037',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoRequired: false
        }));
        /**
         * workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '176577ac-6ef0-4d4d-bb74-8de37ba9d2ba',
            comment: 'The workspace this kml location belong to.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: false
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '79b2babf-05cf-4bb3-9233-c2f17b930217',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductServiceDayOfWeekKmlLocationEntityModel.INSTANCE);
    }
}
exports.ProductServiceDayOfWeekKmlLocationEntityModel = ProductServiceDayOfWeekKmlLocationEntityModel;
ProductServiceDayOfWeekKmlLocationEntityModel.INSTANCE = new ProductServiceDayOfWeekKmlLocationEntityModel();
