// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductServiceDayOfWeekEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductServiceDayOfWeekEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const RegularExpressionRegexConstUtil_1 = require("../../../utils/validation/RegularExpressionRegexConstUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductServiceDayOfWeek';
class ProductServiceDayOfWeekEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('e479dcb9-3ee8-4c95-a2ff-4ec0798ef380')
            .setDataModelName('ProductServiceDayOfWeek')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('79c4d7fb-b822-4bcb-8cea-c1262718d0ac');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('fbf77f5c-7a93-4b29-bf8f-b9188f220b46'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('a33d6a95-91da-4b6a-9f9d-753599e45159'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('131e577c-2fc5-4f5b-bb5b-7dfccf7b76bd'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('de63a57a-60c2-4a88-ab3b-f3189ab49644'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('c39ec6e5-eda2-42cb-bec2-115b06912d93'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('20eb0dc3-2813-4cfe-9b98-d79894470ea9'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '04d7fab5-829b-4ae9-b446-a2519f4339b6'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('afd240f3-1234-4f11-8f46-6a22735737cc'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('7b9b99d1-39ca-4fcd-a547-abe3f0a7b03c'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'de0306ef-27a6-41e6-8fb3-fda9900188c7',
            comment: 'The name.',
            isMongoRequired: false,
            validator: zod_1.z.string().min(1, "Name cannot be empty")
                .max(255, "Name is too long")
                .regex(RegularExpressionRegexConstUtil_1.Regex.alphanumericWithSpace, "Name cannot contain special characters"),
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '81469010-6842-4f79-affb-a9d7abb99bb8',
            comment: 'The description.',
            isMongoRequired: false,
            validator: zod_1.z.string().max(1000, 'Description is to long'),
        }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: '974470e2-3c8e-4fd4-be69-4e34da2d9e58',
            comment: 'Active start date.'
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: 'd84e3272-47e6-41ec-a2ff-53e0c553ffdb',
            comment: 'Active end date.'
        }));
        // monday
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'monday',
            attributeUUID: '115f1511-016f-4e73-9db6-6f1b3461d276',
            comment: 'Boolean value for the day Monday.'
        }));
        // tuesday
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'tuesday',
            attributeUUID: '8a6da53e-529f-4c1c-b343-10468a93c8ac',
            comment: 'Boolean value for the day Tuesday.'
        }));
        // wednesday
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'wednesday',
            attributeUUID: '23abf585-94b8-4604-a655-152e7d939e9e',
            comment: 'Boolean value for the day wWdnesday.'
        }));
        // thursday
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'thursday',
            attributeUUID: 'c1ac4274-ac2c-4438-8c4c-26763c5e7c0e',
            comment: 'Boolean value for the day Thursday.'
        }));
        // friday
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'friday',
            attributeUUID: '35d2194c-f9ea-448b-941e-06a2c8773e50',
            comment: 'Boolean value for the day Friday.'
        }));
        // saturday
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'saturday',
            attributeUUID: '436878c3-b0e7-4bad-8e1f-e1ee72d79218',
            comment: 'Boolean value for the day Saturday.'
        }));
        // sunday
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'sunday',
            attributeUUID: 'ed3cb334-7b1a-4457-99a5-cbad8c5d980e',
            comment: 'Boolean value for the day Sunday.'
        }));
        // product
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'product',
            attributeUUID: '01c31b40-6c02-4c74-acea-09d103db46a7',
            comment: 'The product id.',
            foreignRefEntityNamespace: persistable_1.Product.metaDataForProduct.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // productPricing
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'productPricing',
            attributeUUID: '01c31b40-6c02-4c74-acea-09d103db46a7',
            comment: 'The product pricing id.',
            foreignRefEntityNamespace: persistable_1.ProductPricing.metaDataForProductPricing.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : '01c31b40-6c02-4c74-acea-09d103db46a7',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            isMongoIndexed            : true,
            isMongoRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'b05f4d71-b792-4e23-880c-0407b5766cfc',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductServiceDayOfWeekEntityModel.INSTANCE);
    }
}
exports.ProductServiceDayOfWeekEntityModel = ProductServiceDayOfWeekEntityModel;
ProductServiceDayOfWeekEntityModel.INSTANCE = new ProductServiceDayOfWeekEntityModel();
