// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductPricingKmlLocationEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPricingKmlLocationEntityModel = void 0;
const zod_1 = require("zod");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductPricingKmlLocation';
class ProductPricingKmlLocationEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('e0b3fa91-785c-4d28-b294-dc7758a72aa9')
            .setDataModelName('ProductPricingKmlLocation')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('52450102-8bc0-4019-a7dd-34f27f3d0ffc');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('504372d1-f0eb-4bc2-9f8a-6c460dc9d57a'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('509d09d0-de4d-4425-a541-cbed1c9be42a'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('d29fe51e-a6f1-4f2f-8981-57dcfc60d25c'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('4adf8673-cfff-4b85-8cc1-b254ca11d843'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('8f8118bf-4e66-46ee-b194-79ca76e3b096'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('69b046cb-d6b1-4500-9721-7ab960003e2d'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'd29776ff-dfde-401d-9c06-044fa433679d'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('185a5991-9b46-439f-8ace-49c8c2474c4c'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('efeb485a-21e5-4402-9198-af6cd7ada633'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // kmlLayer
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kmlLayer',
            attributeUUID: 'c1112e01-9b08-445c-9560-d1131a0db8f2',
            comment: 'The kml layer id.',
            foreignRefEntityNamespace: persistable_1.KmlLayer.metaDataForKmlLayer.namespace,
            isMongoRequired: false,
        }));
        // kml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kml',
            attributeUUID: 'b5cba705-e494-4ec5-b70c-12736ff47912',
            comment: 'The kml id.',
            foreignRefEntityNamespace: persistable_1.Kml.metaDataForKml.namespace,
            isMongoRequired: false,
        }));
        // kmlGeometry
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kmlGeometry',
            attributeUUID: '411d8395-ed96-4d3b-a047-dc8fb9bbcdd3',
            comment: 'The kml polygon id.',
            foreignRefEntityNamespace: persistable_1.KmlGeometry.metaDataForKmlGeometry.namespace,
            isMongoRequired: false,
        }));
        // productPricing
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'productPricing',
            attributeUUID: '51b4b1c9-a99a-4138-96d4-5f3fc1c62fae',
            comment: 'The product pricing id.',
            foreignRefEntityNamespace: persistable_1.ProductPricing.metaDataForProductPricing.namespace,
            isMongoRequired: false,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'd727a15a-5a1c-43f1-b416-0c0c841b4f86',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            validator: zod_1.z.string(),
            isMongoIndexed: true,
            isMongoRequired: false
        }));
        /**
         * workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '8666fac4-c594-4886-9335-bcbeea9fe78b',
            comment: 'The workspace this kml location belong to.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: false
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '12e93074-28c1-40a2-83bf-fe6e44fe83d2',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductPricingKmlLocationEntityModel.INSTANCE);
    }
}
exports.ProductPricingKmlLocationEntityModel = ProductPricingKmlLocationEntityModel;
ProductPricingKmlLocationEntityModel.INSTANCE = new ProductPricingKmlLocationEntityModel();
