// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductImageRouteEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductImageRouteEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const Route_1 = require("../../classes/persistable/Route");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductImageRoute';
class ProductImageRouteEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('50ad70ab-ba20-4ca6-a12a-ec5e4583a9a0')
            .setDataModelName('ProductImageRoute')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('924b5c97-5f29-48ac-912d-c976f7f91f3c');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('530dbc6e-1188-4c21-b65d-09ab273440d4'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('0c0d09bd-f726-4521-9013-d923a121be32'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('03d785a1-3dda-4528-9f28-764a481d49c0'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('7b1c5922-ad0e-4ec6-9b03-195bd603f3b4'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('1a654945-ccb3-428c-8b99-e8b1ea3337f3'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('391076c9-c1b9-4f4b-b158-1dd68cfea963'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'fe1e2d19-3e63-498f-983f-195c064ed015'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('c9c45171-9663-4dcc-bf42-846e5d86d2a4'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('a6060a2c-5a8d-45c6-93b6-36ac7bfa86ac'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        /**
         * ProductImage
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'productImage',
            attributeUUID: '4f8e9bc8-d9c6-405d-a166-7a88fb2437e1',
            comment: 'The product image.',
            foreignRefEntityNamespace: persistable_1.ProductImage.metaDataForProductImage.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Route
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'route',
            attributeUUID: 'bffc20c0-7242-43ac-9f5d-20c7eca0662e',
            comment: 'The route.',
            foreignRefEntityNamespace: Route_1.Route.metaDataForRoute.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'f28ab40c-3c53-4df1-923b-5d904e44fd0a',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '598d5025-794b-4eee-b5ff-255d2468d484',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'f3a17a72-c83b-4c61-b979-621b18a1fb87',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductImageRouteEntityModel.INSTANCE);
    }
}
exports.ProductImageRouteEntityModel = ProductImageRouteEntityModel;
ProductImageRouteEntityModel.INSTANCE = new ProductImageRouteEntityModel();
