// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductTaxesEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductTaxesEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const ProductPricing_1 = require("../../classes/persistable/ProductPricing");
const Tax_1 = require("../../classes/persistable/Tax");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductTaxes';
class ProductTaxesEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('264ffad5-b2ea-4d14-a085-10950ff6aa1e')
            .setDataModelName('ProductTaxes')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('a81c7509-affc-4fbc-95e3-db371b3f0635');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('2e41e268-0b52-4145-9c10-5a2bf3c1185c'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('9fbef5d4-15b4-444e-b3f6-50f61f047fac'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('d6d3e4b4-598d-42fb-afcb-59e24c1be351'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('fd00babd-992c-42af-bace-21244667c8cd'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('e5fbfe2b-722d-4437-895b-c1620eee6dbb'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('40a918ee-0c46-4e59-bdc0-ba50881f4202'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '88735981-c72d-4a86-849a-d65d8e0aa843'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('44489720-2afc-4b33-b288-308845ee8273'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('fe0bbe2a-96d8-45dc-82fd-e73ac100e9f2'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: '39dd3688-841d-46a5-8aaf-16cbbc87bbb4',
            comment: 'Active start date.'
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: 'fa3de96a-dec9-4a4b-b58a-940ea6d6f880',
            comment: 'Active end date.'
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '3d07f10f-25ae-4335-a089-20474ce0cbdb',
            comment: 'Identifies if the product tax is active.'
        }));
        // taxId
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'taxId',
            attributeUUID: '22c98607-3770-4349-8299-6629fe4f0269',
            comment: 'The Tax Type id.',
            foreignRefEntityNamespace: Tax_1.Tax.metaDataForTax.namespace,
            isMongoRequired: false
        }));
        // productPricing
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'productPricing',
            attributeUUID: '038838eb-5666-40f2-8fd6-61b4ac3f0435',
            comment: 'The Product Pricing id.',
            foreignRefEntityNamespace: ProductPricing_1.ProductPricing.metaDataForProductPricing.namespace,
            isMongoRequired: false
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'ae20611b-1407-4402-9d0a-28a06e7a86a5',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '889704d4-f75e-4538-afd1-c1ee4ec93068',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductTaxesEntityModel.INSTANCE);
    }
}
exports.ProductTaxesEntityModel = ProductTaxesEntityModel;
ProductTaxesEntityModel.INSTANCE = new ProductTaxesEntityModel();
