// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/ProductPricingRouteEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPricingRouteEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const Route_1 = require("../../classes/persistable/Route");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.ProductPricingRoute';
class ProductPricingRouteEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('9b4dabee-fbd2-47bb-b9d3-a44f9c7ef820')
            .setDataModelName('ProductPricingRoute')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('08d3120f-54de-4d22-a846-acd068bb65ad');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('30c3792a-bf39-4039-b8a9-17dc5762104c'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('644838fb-df79-4d93-9159-596d51f3054b'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('3b7df51a-45b7-410e-b752-450652f072b6'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('63ff50cc-a366-4355-9f16-a5bbf103f58c'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('a683e500-f269-4f3d-9338-02762e7d2a90'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('0ba23164-74a6-4d25-bffd-379b4b0abfde'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'e32c2d2c-0330-4570-9e1d-5340e1e95d91'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('e33a7b0c-b5d5-4892-b494-2bb909ff4342'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('9215cffe-2f01-4e6c-ab3c-d66ae328b841'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        /**
         * ProductPricing
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'productPricing',
            attributeUUID: 'c2e27e90-8c05-424e-bd9a-b7f310588487',
            comment: 'The product pricing.',
            foreignRefEntityNamespace: persistable_1.ProductPricing.metaDataForProductPricing.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Route
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'route',
            attributeUUID: '25eb57b2-559e-440f-835b-0718e2e45655',
            comment: 'The route.',
            foreignRefEntityNamespace: Route_1.Route.metaDataForRoute.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'ec6cd33e-8829-4568-8baa-0d63736bdacc',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: 'a7a22da1-b281-4e8f-97df-27420378d742',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'df7182a8-4a14-40b1-b29a-6455bf130b11',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (ProductPricingRouteEntityModel.INSTANCE);
    }
}
exports.ProductPricingRouteEntityModel = ProductPricingRouteEntityModel;
ProductPricingRouteEntityModel.INSTANCE = new ProductPricingRouteEntityModel();
