// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/RouteEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.Route';
class RouteEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('8d399cac-3748-4e08-aaa6-e25c3a579ef8')
            .setDataModelName('Route')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('f2a1eb60-fb3d-4b4f-8138-9d39262e54f4');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('b7334ec8-2f92-4dea-8537-4d13d42fdb61'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('8846c909-2d01-4327-82f4-3a0cc40e5225'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('0371a7ed-a919-4d9d-9ca7-40eea09a3dc2'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('23d59c45-bddf-459e-b198-b170a7011a97'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('6df810ba-5002-40b9-8cce-62d3989e765e'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('02ffdf15-fe7f-416c-a822-c74501e5a598'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'ddf77e3d-9d32-4880-b33e-14b282f18945'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('2d106aa8-cd18-4c9c-b3c3-4f0a02cf3953'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('f5c07465-314c-4319-9226-b3f4397b4db6'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '58b73572-9372-480e-9239-e22a83e38671',
            comment: 'The name of the route.'
        }));
        // clientRouteId
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'clientRouteId',
            attributeUUID: 'be2f7a7b-aaa7-4326-8052-f90ed3b6c3a9',
            comment: 'The client route ID.'
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '9d0c614c-c1f3-40c9-8a6f-e01bdefddf19',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // productPricingRoute
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productPricingRoutes', 'productPricingRoute', persistable_1.ProductPricingRoute.metaDataForProductPricingRoute.namespace, persistable_1.ProductPricingRoute.metaDataForProductPricingRoute.mongoAttributes.route, '3ec05175-f14f-4bcd-ba86-f09d28036b5a', // uuid
        'The product pricing route', 1));
        // productColorRoute
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productColorRoutes', 'productColorRoute', persistable_1.ProductColorRoute.metaDataForProductColorRoute.namespace, persistable_1.ProductColorRoute.metaDataForProductColorRoute.mongoAttributes.route, 'eb3fd788-f6c2-4601-82a9-86eb3480d7d1', 'The product color route', 1));
        // productCategoryRoute
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productCategoryRoutes', 'productCategoryRoute', persistable_1.ProductCategoryRoute.metaDataForProductCategoryRoute.namespace, persistable_1.ProductCategoryRoute.metaDataForProductCategoryRoute.mongoAttributes.route, 'eeb32bea-8c4d-4199-aeac-2d33d8f65815', 'The product category route', 1));
        // productKmlLocation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('productRoutes', 'productRoute', persistable_1.ProductRoute.metaDataForProductRoute.namespace, persistable_1.ProductRoute.metaDataForProductRoute.mongoAttributes.route, 'ae829b80-dd02-4415-a2b8-ac2ff4bb05f4', 'The product route', 1));
        // // taxRoute
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoVirtualArray(
        //     'taxRoutes',
        //     'taxRoute',
        //     TaxRoute.metaDataForTaxRoute.namespace,
        //     TaxRoute.metaDataForTaxRoute.mongoAttributes.route,
        //     '967d1400-701e-4a05-bee3-e0101d972e8e',
        //     'The tax kml location',
        //     1
        //     )
        // );
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '6a313b40-8214-404d-8166-0c6bd8278fa0',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'fb58bb02-10a9-46da-9898-7576563715c4',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (RouteEntityModel.INSTANCE);
    }
}
exports.RouteEntityModel = RouteEntityModel;
RouteEntityModel.INSTANCE = new RouteEntityModel();
