// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/RoutingSoftwareAccountEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingSoftwareAccountEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const Organization_1 = require("../../classes/persistable/Organization");
const RoutingSoftwareType_1 = require("../../classes/persistable/RoutingSoftwareType");
const RoutingSoftwareVersion_1 = require("../../classes/persistable/RoutingSoftwareVersion");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.RoutingSoftwareAccount';
class RoutingSoftwareAccountEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('4978c406-5e4f-4ac5-b242-54594c4b0c56')
            .setDataModelName('RoutingSoftwareAccount')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('dd30e231-2183-411f-ac5b-9a23c2fe9c77');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('85c3853f-27b0-427e-985a-4196a8795d7e'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('26fa1a4d-90a5-42b8-b21e-2175d110ed37'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('8ae49355-8ac9-4a6b-8972-73d39d1e51d9'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('7d1ca04b-ada8-472b-87d0-f7ee56ffc611'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('7ca553f3-6d8a-418f-806a-e322c2d4a44c'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('2f89200e-349e-4c4d-8fde-fb17b1403f16'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '786e6ffc-59bf-4ada-9ed6-b07b3d7c5ba7'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('6c154799-19a5-4524-91b8-149ec50aff08'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('3a4bb578-717a-4be4-9f94-c3c81f03f0ae'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '31127a85-bced-402e-be5c-9f3ce8edb31c',
            comment: 'The name of routing software account.',
            isMongoRequired: false,
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '3ae423e6-3fa6-440c-8484-b00425241024',
            comment: 'The description of routing software account.',
            isMongoRequired: false,
        }));
        // accessToken
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'accessToken',
            attributeUUID: '07eafdbe-cbdb-42f4-b208-9f155d1d1c84',
            comment: 'The accessToken of routing software account.',
            isMongoRequired: false,
        }));
        // refreshToken
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'refreshToken',
            attributeUUID: 'd7228694-29e0-4ef2-bb43-173511a6fa08',
            comment: 'The refreshToken of routing software account.',
            isMongoRequired: false,
        }));
        // tokenExpirationDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'tokenExpirationDate',
            attributeUUID: 'ae4ce537-67de-4083-9bd5-298f10db92af',
            comment: 'The tokenExpirationDate of routing software account.',
            isMongoRequired: false,
        }));
        /**
         * routingSoftwareType
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'routingSoftwareType',
            attributeUUID: '43ba9b1e-4084-4d4b-8144-a2b2d4acd184',
            comment: 'The routingSoftwareType.',
            foreignRefEntityNamespace: RoutingSoftwareType_1.RoutingSoftwareType.metaDataForRoutingSoftwareType.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * routingSoftwareVersion
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'routingSoftwareVersion',
            attributeUUID: 'd2fd61fd-d25a-4a1d-8399-bc18892aaf7d',
            comment: 'The routingSoftwareVersion.',
            foreignRefEntityNamespace: RoutingSoftwareVersion_1.RoutingSoftwareVersion.metaDataForRoutingSoftwareVersion.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // TODO uncomment this code and build entity, after script template is fixed
        // /**
        //  * Login
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableString2({
        //     attributeName             : 'routingSoftwareVersion',
        //     attributeUUID             : 'cfc77ef3-870f-4d3c-a9c7-7ba112b0289d',
        //     comment                   : 'Login.',
        //     isMongoRequired           : false
        // }));
        // /**
        //  * Password
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableString2({
        //     attributeName             : 'routingSoftwareVersion',
        //     attributeUUID             : '979a66fa-54d7-4b6f-b8de-a5ba4dea897f',
        //     comment                   : 'Login passwword.',
        //     isMongoRequired           : false
        // }));
        // /**
        //  * Port
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableString2({
        //     attributeName             : 'routingSoftwareVersion',
        //     attributeUUID             : '2a6f6a48-3bf6-4625-be1f-b6ea2977787c',
        //     comment                   : 'Port connection.',
        //     isMongoRequired           : false
        // }));
        // /**
        //  * Host
        //  */
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableString2({
        //     attributeName             : 'routingSoftwareVersion',
        //     attributeUUID             : '20c776fc-e383-4d8f-8329-c99c02039b6a',
        //     comment                   : 'Host url.',
        //     isMongoRequired           : false
        // }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'd2fd61fd-d25a-4a1d-8399-bc18892aaf7d',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // stringVar1
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar1',
            attributeUUID: '36471484-52b2-4678-b18a-678220ab5987',
            comment: 'The stringVar1 of routing software account.',
            isMongoRequired: false,
        }));
        // stringVar2
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar2',
            attributeUUID: '4468dbdd-ac3e-44e6-934d-525d3b90fdf8',
            comment: 'The stringVar2 of routing software account.',
            isMongoRequired: false,
        }));
        // stringVar3
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stringVar3',
            attributeUUID: '75a737db-d286-4814-9211-50b941e766fc',
            comment: 'The stringVar3 of routing software account.',
            isMongoRequired: false,
        }));
    }
    static getInstance() {
        return (RoutingSoftwareAccountEntityModel.INSTANCE);
    }
}
exports.RoutingSoftwareAccountEntityModel = RoutingSoftwareAccountEntityModel;
RoutingSoftwareAccountEntityModel.INSTANCE = new RoutingSoftwareAccountEntityModel();
