// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/RoutingSoftwareVersionEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingSoftwareVersionEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const RoutingSoftwareType_1 = require("../../classes/persistable/RoutingSoftwareType");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.RoutingSoftwareVersion';
class RoutingSoftwareVersionEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('004075af-36dd-466c-889a-7e6edea152c8')
            .setDataModelName('RoutingSoftwareVersion')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('b50c609f-e376-4494-af42-350040cc8c1d');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('0e335e85-18e1-48e1-9cde-14ed038f9577'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('a80503da-311c-49ad-b954-975cb7386831'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('57d0ba0d-a22a-42a4-8f96-e652a8e7e9e4'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('1017d449-170d-419b-9a1a-259f33688b54'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('9ba35672-b18c-4f8c-949c-413088ad35c4'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('b7f783fc-77a8-4ad0-bfa8-44d4ce4f304e'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '58a601e4-b93b-4171-99d0-e20f2bc4e870'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('4c41f771-8e58-44fc-9b19-6e921d10a794'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('ebf0814b-d464-40d1-bda3-d469db2cd840'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // version
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'version',
            attributeUUID: '2a3b5d36-8164-4368-b99a-d3a64d4935e9',
            comment: 'The version of the softwareType'
        }));
        // routingSoftwareType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'routingSoftwareType',
            attributeUUID: '5abee711-1826-46cb-9d01-0eb06bc17a28',
            comment: 'The RoutingSoftwareType.',
            foreignRefEntityNamespace: RoutingSoftwareType_1.RoutingSoftwareType.metaDataForRoutingSoftwareType.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
    }
    static getInstance() {
        return (RoutingSoftwareVersionEntityModel.INSTANCE);
    }
}
exports.RoutingSoftwareVersionEntityModel = RoutingSoftwareVersionEntityModel;
RoutingSoftwareVersionEntityModel.INSTANCE = new RoutingSoftwareVersionEntityModel();
