// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingWorkspaceBPSEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceBPSEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingWorkspaceBPS';
class SettingWorkspaceBPSEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('fe283297-b796-4142-a975-ad85e5f5e801')
            .setDataModelName('SettingWorkspaceBPS')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('2502bb64-258d-4e36-a689-37c4028d0125');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('2aa7affa-2a56-41c2-82e9-638c56c1083e'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('0dc697c3-4a1e-4672-8830-f28fd213f991'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('9510a2ae-c787-4e21-a696-2aaf3925c065'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('15365c51-a200-4fe4-adc4-d7236b473771'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('3a9303ba-b28d-420b-a391-4f59081dbf0c'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('ea0a740d-a5a2-4174-b1f8-33787c021cac'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'f9855e73-e418-41eb-9ca4-dc85de2d7d90'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('f15fe7e7-cd71-40e3-a04c-e8f7db2c44ee'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('581b49da-20c4-48cb-9883-eb6891b0dcba'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: '8662283d-0f22-44cf-bd47-4f78bb5ec5e9',
            comment: 'isDefault.',
            isMongoRequired: false,
        }));
        // isKml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isKml',
            attributeUUID: '36b0dbc3-e731-400a-8f75-c2773688c4dd',
            comment: 'Is Kml if not, we will assume it is route base.',
            isMongoRequired: false,
        }));
        // settingIntegration
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: "settingIntegrations",
            attributeNameSingular: "settingIntegration",
            attributeUUID: '92f416b5-e75b-43ac-9195-f8941448a496',
            comment: "The settingIntegration.",
            foreignRefEntityNamespace: persistable_1.SettingIntegration.metaDataForSettingIntegration.namespace
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'abec3e6c-e6f0-4098-b418-5964e95f4280',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: 'baa29884-4ba3-479d-b488-a70cac1f86be',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'd34b1d79-5885-4032-9171-de77f7076efa',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (SettingWorkspaceBPSEntityModel.INSTANCE);
    }
}
exports.SettingWorkspaceBPSEntityModel = SettingWorkspaceBPSEntityModel;
SettingWorkspaceBPSEntityModel.INSTANCE = new SettingWorkspaceBPSEntityModel();
