// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingWorkspaceOSSEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceOSSEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const SettingIntegration_1 = require("../../classes/persistable/SettingIntegration");
const Organization_1 = require("../../classes/persistable/Organization");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingWorkspaceOSS';
class SettingWorkspaceOSSEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('2f9faa97-1d27-4479-9a67-4c6ce8bf8215')
            .setDataModelName('SettingWorkspaceOSS')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('ff35d917-52c8-4356-99e8-220b903596b4');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('385a371d-11fe-4d0c-9119-5acaf2fdc5ab'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('c48a0edd-1a4a-464b-865a-18e8c70ff3d3'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('ea645786-5535-4265-8b20-e753c0691af7'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('df700d7f-28d0-4177-ad43-d134f7568e9b'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('546852a8-b30c-4216-8b45-260c02dffbc4'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('5919d7f3-dc59-4889-8e98-09817a18a028'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'ae907424-4fb0-4ee1-a1e5-61659a1e7a1f'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('6d1f69e8-3c4a-4e22-8959-9399f4fb00fd'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('c5056cbd-2319-4340-8657-a8e3f27dbb12'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: '5e0bd067-611d-4d71-86f6-f8c9efd9c12d',
            comment: 'isDefault.',
            isMongoRequired: false,
        }));
        // isKml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isKml',
            attributeUUID: '4d0689a4-080b-41fe-9c87-1b9c7c368688',
            comment: 'Is Kml if not, we will assume it is route base.',
            isMongoRequired: false,
        }));
        // settingIntegration
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: "settingIntegrations",
            attributeNameSingular: "settingIntegration",
            attributeUUID: '1607b26e-3146-4804-b232-ba6df2a34a89',
            comment: "The settingIntegration.",
            foreignRefEntityNamespace: SettingIntegration_1.SettingIntegration.metaDataForSettingIntegration.namespace
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'dc07578c-05b5-4433-ab0b-c78562e9c3e8',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '090fa5ed-3ee6-40a8-85b1-3201ddb1f8a9',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
    }
    static getInstance() {
        return (SettingWorkspaceOSSEntityModel.INSTANCE);
    }
}
exports.SettingWorkspaceOSSEntityModel = SettingWorkspaceOSSEntityModel;
SettingWorkspaceOSSEntityModel.INSTANCE = new SettingWorkspaceOSSEntityModel();
