// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingOrganizationEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingOrganizationEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const Organization_1 = require("../../classes/persistable/Organization");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingOrganization';
class SettingOrganizationEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('eac67e3a-dedb-4734-9dea-5fc28b529369')
            .setDataModelName('SettingOrganization')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('aac94215-0a99-4a6c-81c0-f9f916be2d43');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('62bd1e13-55ec-4076-b693-8cfd07833f40'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('c6f51648-b870-406f-8456-1230005409cb'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('e649ac5e-eecb-487d-aa4a-67c2eee5ea68'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('275a64f6-3a05-4ae8-9125-616a0dbf5ad7'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('3d090b13-3be4-4099-8764-4effdda2d85d'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('c98a4f58-ec89-4074-8580-b6d9ecfb78f0'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '2752bad9-28bd-4303-a884-6d4226fc0ad6'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('ef0f40a2-8617-426c-9c25-dda3c00f763f'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('2b7f8dd5-af9a-414f-8bb1-21d8ef646932'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: '34fbde81-550a-4464-b90e-13dc2c196a66',
            comment: 'Is setting default.',
            isMongoRequired: false,
        }));
        // logo
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'logo',
            attributeUUID: '3fe4acf8-f4ca-4ef6-9f33-5a2025e85644',
            comment: 'Organization logo.',
            isMongoRequired: false,
        }));
        // isOSSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isOSSEnabled',
            attributeUUID: 'bf78da4c-6ed0-47c2-b90f-743688c68367',
            comment: 'is OSS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isSESEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isSESEnabled',
            attributeUUID: 'b4f4f121-9019-4b47-a156-1f36ccb9ae87',
            comment: 'is SES Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isCNSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isCNSEnabled',
            attributeUUID: 'f12ff5e8-5faf-461e-8085-0bead00650dc',
            comment: 'is CNS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isCSPEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isCSPEnabled',
            attributeUUID: 'eb24ea2c-a481-446e-b134-188b7186521f',
            comment: 'is CSP Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isBPSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isBPSEnabled',
            attributeUUID: 'a6516e5a-cfbc-40c7-91a9-6140e2eaf472',
            comment: 'is BPS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isDGSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDGSEnabled',
            attributeUUID: '0ffe5d36-ea1c-4cc4-9597-d9463a0b9770',
            comment: 'is DGS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isCALEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isCALEnabled',
            attributeUUID: '87e7ee06-56a6-4f0d-90be-46162d9e27e2',
            comment: 'is CAL Enabled for this organization.',
            isMongoRequired: false,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'e60eff7d-4db1-4424-8511-0de7c8e7f759',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: false
        }));
    }
    static getInstance() {
        return (SettingOrganizationEntityModel.INSTANCE);
    }
}
exports.SettingOrganizationEntityModel = SettingOrganizationEntityModel;
SettingOrganizationEntityModel.INSTANCE = new SettingOrganizationEntityModel();
