// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingWorkspaceEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const Organization_1 = require("../../classes/persistable/Organization");
const Workspace_1 = require("../../classes/persistable/Workspace");
const TypeReference_1 = require("../../../data/entity/TypeReference");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingWorkspace';
// update - 
//. -- prefix "system" to to any field that only we can see
class SettingWorkspaceEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('b1ae6dfc-4d9f-463d-ac01-f5437e2158c1')
            .setDataModelName('SettingWorkspace')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('5c8bdde1-fa33-474c-9425-94f39eb91e47');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('cd13a7aa-3147-43cc-a97c-ad595c1ffacf'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('fb33689e-5e48-490c-bff4-dacd6132c4d6'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('379926fd-e743-4500-9029-abdce7c2aaf6'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('70a62ecf-67e3-4827-b762-01853e162168'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('69d22518-8680-459a-9457-6b3fc611cb84'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('553ce89e-2929-4a50-9332-0d093e315292'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '5f1902e2-b406-4419-aa25-4f0a67a635b6'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('f9e44ea6-03b3-453a-9e54-bb50a1d4dd55'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('9d8e2efc-bb13-484c-8c95-577760260a31'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: '85e3a21f-9e78-4bea-8e75-efb10c085e56',
            comment: 'Is setting default.',
            isMongoRequired: false,
        }));
        // location
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'location',
            attributeUUID: 'b0145125-437f-442c-91be-d53de7e49f24',
            comment: 'The location that the workspace is associated with.',
            isMongoRequired: false,
        }));
        // isOSSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isOSSEnabled',
            attributeUUID: '6cafbde1-f161-4b20-9dc3-969fab135087',
            comment: 'is OSS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isSESEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isSESEnabled',
            attributeUUID: '02752d67-f94a-44d4-a0ee-eff5dd9bf276',
            comment: 'is SES Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isCNSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isCNSEnabled',
            attributeUUID: '623f2221-ff0d-4896-b3cb-206b9e264fb1',
            comment: 'is CNS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isCSPEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isCSPEnabled',
            attributeUUID: '01fde01b-db46-4a1d-9cf3-8a1aef7b6b6f',
            comment: 'is CSP Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isBPSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isBPSEnabled',
            attributeUUID: 'e23dca5c-ebdc-4f6a-a94a-dfba86ed8329',
            comment: 'is BPS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isDGSEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDGSEnabled',
            attributeUUID: '0b2c373f-91de-4eb5-b378-01c864348a75',
            comment: 'is DGS Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isCALEnabled
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isCALEnabled',
            attributeUUID: '47e7bafe-063c-4035-ab01-ac2e536323d7',
            comment: 'is CAL Enabled for this organization.',
            isMongoRequired: false,
        }));
        // isRouteBased
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isRouteBased',
            attributeUUID: 'e7172aad-2da0-4dad-bf7f-3eb3c9242d05',
            comment: 'is workspace route based',
            isMongoRequired: false,
        }));
        // kmlMaxCount -- move to application level settings?
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableInteger2({
        //     attributeName   : 'kmlMaxCount',
        //     attributeUUID   : '859eeb4f-5ea0-4723-aab0-53466990acaf',
        //     comment         : 'max kml polygon.',
        // }));
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableJsonString2({
            attributeName: 'settingIntegrations',
            attributeUUID: '01d87b3b-db99-4f7c-82fc-d393b4675c58',
            comment: 'A JSON text that represents the settingIntegrations data',
            isMongoRequired: false,
            variableType: new TypeReference_1.TypeReference({
                typeName: 'Object'
            })
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '187a20e0-0fcd-4f06-ad41-e71315a7dd06',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '0d896e65-4328-4d61-890e-098b21645657',
            comment: 'The workspace.',
            foreignRefEntityNamespace: Workspace_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
    }
    static getInstance() {
        return (SettingWorkspaceEntityModel.INSTANCE);
    }
}
exports.SettingWorkspaceEntityModel = SettingWorkspaceEntityModel;
SettingWorkspaceEntityModel.INSTANCE = new SettingWorkspaceEntityModel();
