// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingWorkspaceCALEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceCALEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingWorkspaceCAL';
class SettingWorkspaceCALEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('57abd836-1a41-4f8d-8559-c85058a37a9d')
            .setDataModelName('SettingWorkspaceCAL')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('aaeb7c10-415f-4ba4-8636-b31287ac5bd2');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('423933cd-7089-490a-bd4c-118ec8b66f1a'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('cd5b23b3-2c26-48c3-8b09-88717d2b6d0c'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('f1289623-f2fe-4c5b-aba4-7b1f4399a1f2'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('70836ea3-00f3-4a91-b345-fcfb527c2540'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('13a4a7d8-fa99-4251-a7d2-adf1aeb24832'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('68634b1c-01c3-4099-9561-d552482e0813'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '3c90ee8d-2a12-4ccb-9373-8d44de2c1cf3'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('2c6acdae-0d0d-42e9-925a-28c55069b216'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('538cdbbc-5a85-47f2-8496-b5abfca64299'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: '29e77d73-54dc-46b6-bb9f-7f8e2d31226d',
            comment: 'isDefault.',
            isMongoRequired: false,
        }));
        // isKml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isKml',
            attributeUUID: '3f04c259-5720-443f-8f95-4ab4d705e087',
            comment: 'Is Kml if not, we will assume it is route base.',
            isMongoRequired: false,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'c4116a37-d3f9-4d1f-a500-d387c6539e16',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // settingIntegration
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: "settingIntegrations",
            attributeNameSingular: "settingIntegration",
            attributeUUID: 'b3cc984e-48a7-41d9-8b67-9083c0ad317d',
            comment: "The settingIntegration.",
            foreignRefEntityNamespace: persistable_1.SettingIntegration.metaDataForSettingIntegration.namespace
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '697a8fcc-55be-43d0-999d-b319698d5dfd',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'ff7e8454-527c-4499-b123-ecac32beaf20',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (SettingWorkspaceCALEntityModel.INSTANCE);
    }
}
exports.SettingWorkspaceCALEntityModel = SettingWorkspaceCALEntityModel;
SettingWorkspaceCALEntityModel.INSTANCE = new SettingWorkspaceCALEntityModel();
