// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingWorkspaceCNSEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceCNSEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingWorkspaceCNS';
class SettingWorkspaceCNSEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('53a5f9e0-90bd-4eae-8464-202f98fe09fe')
            .setDataModelName('SettingWorkspaceCNS')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('8ee930b4-a2bb-496b-b4ae-f4601a67000e');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('f772a956-340f-4f96-a70e-9323190cba9f'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('ee3c7304-426a-41b3-887e-01a40e3d2b7c'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('c40eb68a-cdbc-4e62-94c6-d0fa7f471b07'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('a90192d3-9602-42d5-b24d-1f71292e05d4'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('9dff4a5f-aa25-4aac-a15e-c458e424afc1'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('c8f824f9-ae40-4da9-ba00-814239614e38'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '848fe702-9447-4b69-b1b0-85cd460a5769'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('d296c542-1666-4c16-bd04-141e6c80ec0b'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('c34444b4-9451-453c-9e3d-e882ef24b40d'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: '19978170-c4d2-47b7-9222-72505aacdf32',
            comment: 'isDefault.',
            isMongoRequired: false,
        }));
        // isKml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isKml',
            attributeUUID: '796f9a2b-671b-44e1-a98a-869423d5c475',
            comment: 'Is Kml if not, we will assume it is route base.',
            isMongoRequired: false,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'c7688107-02ef-4c26-89a4-db950f07c3d4',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        // settingIntegration
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: "settingIntegrations",
            attributeNameSingular: "settingIntegration",
            attributeUUID: 'b49c381f-8ca6-4aa8-aa0f-9d9daef94e39',
            comment: "The settingIntegration.",
            foreignRefEntityNamespace: persistable_1.SettingIntegration.metaDataForSettingIntegration.namespace
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '7ffc338b-91a2-4b85-845d-3f241051c2d1',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '0a9fbfe9-2690-4809-a1df-518893758970',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (SettingWorkspaceCNSEntityModel.INSTANCE);
    }
}
exports.SettingWorkspaceCNSEntityModel = SettingWorkspaceCNSEntityModel;
SettingWorkspaceCNSEntityModel.INSTANCE = new SettingWorkspaceCNSEntityModel();
