// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingNotificationEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingNotificationEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingNotification';
class SettingNotificationEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('38816274-64df-45e5-8fcf-44623fda0131')
            .setDataModelName('SettingNotification')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('61dbe54b-6011-4ac2-9b3d-f5ab169dbe9d');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('6bbc064a-d32c-42e9-8c8a-cbf83d402423'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('19c1108a-5a09-4240-8326-2f2307560474'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('92830e7d-ada0-4bdf-931e-dd06a73cf5f3'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('95ea1d4b-f3b1-4069-bcc1-e6d2c57fbc29'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('099b3808-67ff-4ce2-a6fb-13ff6ec6c349'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('bcce3d27-1961-4be7-92cf-09a3b5fd4da4'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '3ff037ec-b987-4795-9afb-5983e78672d9'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('2b8dbc99-8945-42be-9fcf-fea29d684696'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('25fc52ee-e297-444b-a7a8-d39967f0965e'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: 'f4e8360b-f07d-46e5-b2b4-343dac48261a',
            comment: 'is Default.',
            isMongoRequired: false,
        }));
        // isClient
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isClient',
            attributeUUID: 'd7706322-0fd6-43aa-badc-de0e5a6cb22c',
            comment: 'notification is for client.',
            isMongoRequired: false,
        }));
        // isCustomer
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isCustomer',
            attributeUUID: 'c4629ff1-4f68-4ebf-921e-ac3c61be0f52',
            comment: 'notification is for customer.',
            isMongoRequired: false,
        }));
        // user
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'user',
            attributeUUID: 'a5442245-d438-4ed1-a823-c8c033abbb70',
            comment: 'User reference.',
            foreignRefEntityNamespace: persistable_1.User.metaDataForUser.namespace,
            isMongoRequired: false
        }));
        // organization
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: '517b8f3d-3b8d-4cda-b910-18004c72854e',
            comment: 'Organization reference.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoRequired: false,
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : '2bb95262-94f7-480d-8782-fc5cbbcecf31',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            isMongoIndexed            : true,
            isMongoRequired           : true
        }));
        */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : 'd29e676e-c9ea-4d65-9394-574d391153ad',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (SettingNotificationEntityModel.INSTANCE);
    }
}
exports.SettingNotificationEntityModel = SettingNotificationEntityModel;
SettingNotificationEntityModel.INSTANCE = new SettingNotificationEntityModel();
