// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SettingWorkspaceSESEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceSESEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const SettingIntegration_1 = require("../../classes/persistable/SettingIntegration");
const Organization_1 = require("../../classes/persistable/Organization");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SettingWorkspaceSES';
class SettingWorkspaceSESEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('017aa02a-fa14-4af3-b031-3085c709c6ec')
            .setDataModelName('SettingWorkspaceSES')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('d86a031b-e2f6-4ca6-bec1-e292968b425a');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('372501c8-6a53-45df-9fd1-81406735302f'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('e2aaab4e-3d33-497b-8af3-bdc36abaaf56'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('5d6111a1-7105-4e87-9b3a-1dab2d26a027'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('1dbdd186-299b-4065-b519-81722a5b28b0'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('d34c9b47-5745-46c0-b55d-48f60b74365a'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('4970bbfd-6a77-470d-9811-cbaaab4d4772'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '7c9da30e-25b6-4ed3-b113-e86071af0126'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('72caeb3c-a64f-4541-9ca1-955108531f6c'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('1700cb30-aa5b-40e8-ae79-8848c22a4182'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // string
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'aString',
            attributeUUID: 'fee6d930-c59a-4bcd-b3f3-eaadaaf3d886',
            comment: 'The name of a string value.'
        }));
        // isDefault
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isDefault',
            attributeUUID: 'f656385e-9c34-4a80-9470-d9d30d1c71fb',
            comment: 'isDefault.',
            isMongoRequired: false,
        }));
        // isKml
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isKml',
            attributeUUID: 'ed5ef3f9-e581-4008-a83d-4c79fbf80778',
            comment: 'Is Kml if not, we will assume it is route base.',
            isMongoRequired: false,
        }));
        // settingIntegration
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: "settingIntegrations",
            attributeNameSingular: "settingIntegration",
            attributeUUID: '1fdf7f91-2cc2-4ff1-bc0f-2f3bf94dd6d4',
            comment: "The settingIntegration.",
            foreignRefEntityNamespace: SettingIntegration_1.SettingIntegration.metaDataForSettingIntegration.namespace
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'c7c2d165-fe28-4f1e-a0cb-0f092a8b3bcf',
            comment: 'The organization.',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * Workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: 'c705e773-0039-42c7-90a5-27c4b68c6d52',
            comment: 'The workspace.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
    }
    static getInstance() {
        return (SettingWorkspaceSESEntityModel.INSTANCE);
    }
}
exports.SettingWorkspaceSESEntityModel = SettingWorkspaceSESEntityModel;
SettingWorkspaceSESEntityModel.INSTANCE = new SettingWorkspaceSESEntityModel();
