// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SoftPakCycleCodeEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoftPakCycleCodeEntityModel = void 0;
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
// No implementation
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.SoftPakCycleCode';
class SoftPakCycleCodeEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('4954af83-1366-4090-a044-bd8a80597944')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('14d1ad04-f506-4b21-8201-42d16a5d2440');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('981f8b79-6b7a-4965-81ea-3c46842949e6'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('1dd7d2eb-9d90-424e-bd34-8d30b2d14f5f'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('96bd9ddd-10aa-4093-aab0-34d538565b9e'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('dd6153fa-cf60-45c5-9951-70939a7f3393'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('2221389e-ca00-4cfa-8c97-180596a51f78'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('f1d9e0e9-4a5a-4021-87f2-e1f9624af130'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('ba508a4e-8ec3-45cf-80b6-87160fe0ad3b'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '938277e3-415f-4978-a496-735bdc6618a5'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('e9813fa6-9355-4f63-9bde-394263879a7d'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType('c5feab78-92fa-499a-8751-b1f61284711f'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('9f9dc363-a7d3-46de-a49f-9ca9f75991fc'));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: '2297e409-72aa-4950-aa4b-d9e0d01ca4de',
            comment: 'The organization.'
        }));
        // cycleCode
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'cycleCode',
            attributeUUID: '30da4cf1-8dfe-421b-8de1-5522ddd9a026', // uuid
            comment: 'The cycle code.'
        }));
        // startServiceDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRecordDate('startServiceDate', '5d53da28-a2c9-41ee-a888-213dad3f5a50', // uuid
        'The start date of the service.'));
    }
    static getInstance() {
        return (SoftPakCycleCodeEntityModel.INSTANCE);
    }
}
exports.SoftPakCycleCodeEntityModel = SoftPakCycleCodeEntityModel;
SoftPakCycleCodeEntityModel.INSTANCE = new SoftPakCycleCodeEntityModel();
