// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SystemWidgetAttributeTemplateEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemWidgetAttributeTemplateEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const SystemWidgetTemplate_1 = require("../../classes/persistable/SystemWidgetTemplate");
const SystemWidgetViewTemplate_1 = require("../../classes/persistable/SystemWidgetViewTemplate");
const WidgetAttributeType_1 = require("../../classes/persistable/WidgetAttributeType");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.SystemWidgetAttributeTemplate";
class SystemWidgetAttributeTemplateEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('99916e16-c0a7-4db6-8a88-854c4952168f')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('bd5de2b9-0252-4997-812c-7ed1da276127');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('39819049-b47c-42b9-905c-f4001278768c'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('4cbfd817-997e-486d-b1f0-0208e79bb1e8'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('e0b18e6e-8106-4c5a-90e0-ca2ba8dd126f'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('52fcfd84-6d5e-4d59-8373-06bfb0a8b96e'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('81d56d0d-0327-4a0b-9e95-e90b834a4564'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('47e84785-6226-42be-9208-41babd26e612'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('86b02659-a1b2-4d08-a805-33438e60024c'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '708bf954-32aa-4746-9d2a-4c839f37352d'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('5a356563-8ec0-4988-a6e0-768b0529475d'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('9883852d-28b7-48f5-993a-2e31f9bf93ca'));
        // defaultValue
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'defaultValue',
            attributeUUID: '74331354-740d-4d49-a901-e86cb8e7ed84', // uuid
            comment: 'The default value of the attribute.',
            isMongoRequired: false,
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '9e5dcbfa-9200-4b1c-ac57-1a1b488e0e65', // uuid
            comment: 'The description of the attribute.'
        }));
        // functionValidation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'functionValidation',
            attributeUUID: '1bbbad5e-2716-4918-a514-060672c6a88f', // uuid
            comment: 'A function for validation of a value of the attribute.',
            isMongoRequired: false,
        }));
        // functionValidationErrorMessage
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'functionValidationErrorMessage',
            attributeUUID: '367c341d-9a1a-4fc0-9b4a-c4ffb406a2bf', // uuid
            comment: 'An error message for validation of a value of the attribute.',
            isMongoRequired: false,
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '90eba789-5d08-4007-b2af-f496f2710efc',
            comment: 'Identifies if the attribute is active.',
        }));
        // label
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'label',
            attributeUUID: '3d4cab3d-e5df-4387-bf75-442d36a9d2fa', // uuid
            comment: 'The label for the attribute.'
        }));
        // labelI18nKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'labelI18nKey',
            attributeUUID: 'f148a200-cd71-4fc1-9761-e55fede83974', // uuid
            comment: 'The label I18N for the attribute.',
            isMongoRequired: false,
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'ba609a65-5f1f-42a5-bbec-3598c2ae2c5d', // uuid
            comment: 'The widget name.'
        }));
        // pandaDocKeyName
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'pandaDocKeyName',
            attributeUUID: 'e51b97b0-f31b-4b71-a7c7-893ed278a215', // uuid
            comment: 'The pandaDocKeyName attribute name.'
        }));
        // regexValidation
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'regexValidation',
            attributeUUID: 'eaf7e1d5-143c-493c-b59e-d356d4f4b538', // uuid
            comment: 'An regex for validation of a value of the attribute.',
            isMongoRequired: false,
        }));
        // regexValidationErrorMessage
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'regexValidationErrorMessage',
            attributeUUID: 'f4ff54b0-6120-45f0-9167-87dc8a15bea1', // uuid
            comment: 'An error message for validation of a value of the attribute.',
            isMongoRequired: false,
        }));
        // regexValidationErrorMessageI18nKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'regexValidationErrorMessageI18nKey',
            attributeUUID: '356c2bb6-9ee6-43ed-b02f-db06bf3107b3', // uuid
            comment: 'An error message for validation of a value of the attribute.',
            isMongoRequired: false
        }));
        // salesStrykeKeyName
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'salesStrykeKeyName',
            attributeUUID: '103410a9-c94f-4676-be57-a3562bcd42f5', // uuid
            comment: 'The salesStrykeKeyName attribute name.'
        }));
        /**
         * SystemWidgetTemplate oid
         * The Widget this view belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "systemWidgetTemplate",
            foreignRefEntityNamespace: SystemWidgetTemplate_1.SystemWidgetTemplate.metaDataForSystemWidgetTemplate.namespace,
            attributeUUID: 'ea867485-d3a8-4c21-862d-491a85d7057f',
            comment: "The SystemWidgetTemplate this attribute belongs to."
        }));
        /**
         * SystemWidgetViewTemplate oid
         * The Widget this view belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "systemWidgetViewTemplate",
            foreignRefEntityNamespace: SystemWidgetViewTemplate_1.SystemWidgetViewTemplate.metaDataForSystemWidgetViewTemplate.namespace,
            attributeUUID: '8c93c05a-4223-431d-895a-03496d683202',
            comment: "The SystemWidgetViewTemplate this attribute belongs to."
        }));
        /**
         * WidgetAttributeType oid
         * The Widget this view belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "widgetAttributeType",
            foreignRefEntityNamespace: WidgetAttributeType_1.WidgetAttributeType.metaDataForWidgetAttributeType.namespace,
            attributeUUID: '1ceb9b7b-2f08-41d7-b0e0-ac8f85959276',
            comment: "The WidgetAttributeType this attribute belongs to."
        }));
    }
    static getInstance() {
        return (SystemWidgetAttributeTemplateEntityModel.INSTANCE);
    }
}
exports.SystemWidgetAttributeTemplateEntityModel = SystemWidgetAttributeTemplateEntityModel;
SystemWidgetAttributeTemplateEntityModel.INSTANCE = new SystemWidgetAttributeTemplateEntityModel();
