// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/SystemWidgetViewTemplateEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemWidgetViewTemplateEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const SystemWidgetTemplate_1 = require("../../classes/persistable/SystemWidgetTemplate");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.SystemWidgetViewTemplate";
class SystemWidgetViewTemplateEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('09856721-2736-4e4b-81ea-c9566ccce0b0')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('63f50231-b023-45c2-b336-cf8706e15f85');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('f714ef60-d9a2-43c0-a8a6-5ea6950f9f36'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('bd991004-ad00-4f9c-ab51-f54dd0ccced4'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('f3000acc-15e5-4712-bddb-7f6be701db58'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('e7de9527-95da-4a41-923b-58a606df0be9'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('046d955a-a6cf-4a35-bc00-482e013f61cb'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('ae5db902-95b0-4ab6-bfb7-b368fb7c5e34'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('92531567-edd8-4d88-a3d5-d996370c9e77'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '907b1293-3cb8-41cd-9f5c-222cd72e31a2'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('b3c63399-ed17-4672-99ef-9ac1f5c67f92'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('120e3623-4757-449d-81b9-1e1469877de8'));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '3d7726ff-fbc9-420f-b709-a95183dc0bb0', // uuid
            comment: 'The description of the widget view.'
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '558a2ac1-05be-44a6-92de-6f609d7fde16',
            comment: 'Identifies if the widget view is active.'
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '0c5982d5-c6dc-495c-b889-0443205f9765', // uuid
            comment: 'The widget view name.'
        }));
        /**
         * SystemWidgetTemplate oid
         * The Widget this view belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "systemWidgetTemplate",
            foreignRefEntityNamespace: SystemWidgetTemplate_1.SystemWidgetTemplate.metaDataForSystemWidgetTemplate.namespace,
            attributeUUID: 'aa42a07f-dad6-48ec-8786-b6a2832f5c87',
            comment: "The SystemWidgetTemplate this view belongs to."
        }));
    }
    static getInstance() {
        return (SystemWidgetViewTemplateEntityModel.INSTANCE);
    }
}
exports.SystemWidgetViewTemplateEntityModel = SystemWidgetViewTemplateEntityModel;
SystemWidgetViewTemplateEntityModel.INSTANCE = new SystemWidgetViewTemplateEntityModel();
