// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/TaxEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.Tax';
class TaxEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('6e990dee-0537-454a-a13b-3e58ea9659c2')
            .setDataModelName('Tax')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('ccbda423-b69f-4372-bc46-dd9906e7884b');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('a466d434-5a06-4cc1-8a19-757310d8806b'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('442cc3d0-7b71-4038-97e0-2ce464230413'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('a891d5c1-6ace-4954-9295-2b5369f2dc0c'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('29e0fe52-9bb0-4d7f-bfbe-c355df94dcdb'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('88a90fb5-0aaf-4732-9a0e-866933b17741'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('2dec4f5b-d46d-4a3d-ba4b-5ea81255e4c1'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '050c9157-114d-4075-b94b-f3635b850720'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('8a95b2dc-85e0-4e1e-bf81-581953c2a825'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('3b73f34e-0fbf-4d3f-a2ef-23c38fa6ecfb'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '5fe21043-24c2-4aac-a531-2e8c93a1326d',
            comment: 'is this Tax Active.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        // flatFee
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'flatFee',
            attributeUUID: '368da577-7592-4911-bab8-cfc0bf2e3ded',
            comment: 'is the Tax a flat fee.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        // percentage
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableNumber2({
            attributeName: 'percentage',
            attributeUUID: '368da577-7592-4911-bab8-cfc0bf2e3ded',
            comment: 'is the Tax a percentage.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'af6ab5e8-06ed-4df9-a2d3-f10b9d0f237c',
            comment: 'The name of the tax.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '98b94dea-47c9-422f-97fd-acb8bd36655b',
            comment: 'The name of the tax.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeStartDate',
            attributeUUID: 'e0d72b93-be6f-4b76-913c-576cd425808a',
            comment: 'Active start date.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate2({
            attributeName: 'activeEndDate',
            attributeUUID: '345a8536-381f-40d0-b1ca-ca7413ac89b7',
            comment: 'Active end date.',
            isMongoRequired: false,
            isMongoIndexed: false,
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            attributeUUID: 'aee99e2c-c1df-4ef0-877c-45ff863fe7a3',
            comment: 'The organization.',
            foreignRefEntityNamespace: persistable_1.Organization.metaDataForOrganization.namespace,
            isMongoIndexed: false,
            isMongoRequired: false
        }));
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRefArray(
            {
                attributeName              : "salesReportConfigParamValues",
                attributeNameSingular      : "salesReportConfigParamValue",
                attributeUUID              : '1385ba02-98e2-444f-9d14-f62e778887cc',
                comment                    : "The SalesReportConfigParamValue.",
                foreignRefEntityNamespace  : SalesReportConfigParamValue.metaDataForSalesReportConfigParamValue.namespace
            }
       ));
       */
    }
    static getInstance() {
        return (TaxEntityModel.INSTANCE);
    }
}
exports.TaxEntityModel = TaxEntityModel;
TaxEntityModel.INSTANCE = new TaxEntityModel();
