// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/TaxKmlLocationEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxKmlLocationEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const persistable_1 = require("../../classes/persistable");
const Tax_1 = require("../../classes/persistable/Tax");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.TaxKmlLocation';
class TaxKmlLocationEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID('b044efa1-f890-4029-b85b-af455e90f5e7')
            .setDataModelName('TaxKmlLocation')
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('a8ff5c79-dbac-42ab-a7cb-e37faba034e7');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('4bdcebe9-b89a-4375-ac42-a1f4f2b31009'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('e39ab398-6b59-4c63-9de5-56203ee3b003'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('b06628b9-b447-40b8-9a43-8a9098d86f9e'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('a6474a72-87ab-4482-9583-7d9c90e5a944'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('a2d01f96-e84a-421e-9449-8dcb6f927c1d'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('4bf0b4fe-c52d-46e7-aa5c-4c3c4fb12d17'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'ad23c665-bb0f-40a1-b436-2f834daeec97'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('e48ef787-d5a2-4bca-b8f1-098373e1000b'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('b2463f1c-9a93-4010-91d3-8d57f4c0f3b0'));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: 'relationshipPersistableTypeUUID' }));
        /**
         * tax
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'tax',
            attributeUUID: '3cd974ae-8cd0-4688-a9e5-9cef744f1acc',
            comment: 'The tax.',
            foreignRefEntityNamespace: Tax_1.Tax.metaDataForTax.namespace,
            isMongoIndexed: true,
            isMongoRequired: true
        }));
        /**
         * kml
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kml',
            attributeUUID: 'a1ee3b80-ad24-44a5-aea2-57ec6feaea3b',
            comment: 'The kml.',
            foreignRefEntityNamespace: persistable_1.Kml.metaDataForKml.namespace,
            isMongoIndexed: true,
            isMongoRequired: false
        }));
        /**
         * kmlLayer
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kmlLayer',
            attributeUUID: '5411e538-724b-45bf-9984-01fef922d2d6',
            comment: 'The kml layer.',
            foreignRefEntityNamespace: persistable_1.KmlLayer.metaDataForKmlLayer.namespace,
            isMongoIndexed: true,
            isMongoRequired: false
        }));
        /**
         * kmlGeometry
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'kmlGeometry',
            attributeUUID: '48692e42-0b8b-4ee4-ba8c-28c90351c522',
            comment: 'The kml geometry.',
            foreignRefEntityNamespace: persistable_1.KmlGeometry.metaDataForKmlGeometry.namespace,
            isMongoIndexed: true,
            isMongoRequired: false
        }));
        /**
         * workspace
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            attributeUUID: '0a7f80e2-2dc5-4d0a-8064-322c99a0b162',
            comment: 'The workspace this kml location belong to.',
            foreignRefEntityNamespace: persistable_1.Workspace.metaDataForWorkspace.namespace,
            isMongoIndexed: true,
            isMongoRequired: false
        }));
        /**
         * Organization
         */
        /*
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName             : 'organization',
            attributeUUID             : '2fef53f3-7766-4c5b-b8cf-c4f0fb0ebd06',
            comment                   : 'The organization.',
            foreignRefEntityNamespace : Organization.metaDataForOrganization.namespace,
            isMongoIndexed            : true,
            isMongoRequired           : true
        }));
        */
    }
    static getInstance() {
        return (TaxKmlLocationEntityModel.INSTANCE);
    }
}
exports.TaxKmlLocationEntityModel = TaxKmlLocationEntityModel;
TaxKmlLocationEntityModel.INSTANCE = new TaxKmlLocationEntityModel();
