// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/UserEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserEntityModel = void 0;
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const Organization_1 = require("../../classes/persistable/Organization");
const UserType_1 = require("../../classes/persistable/UserType");
const zod_1 = require("zod");
const RegularExpressionRegexConstUtil_1 = require("../../../utils/validation/RegularExpressionRegexConstUtil");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.User";
class UserEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        entityModel
            .setCrudCreateAuthorizationUUID("669161b7-8340-497a-8566-38ba39b00359")
            .setDataModelName("User")
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID("02bfbc1a-0dd2-4bfc-9b88-59f00572ba1e");
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('96b91335-c280-43c0-b34c-9c79b9e05316'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState("6ff6bb6c-88ac-42d1-a72a-404a19e196f8"));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem("116f3ebd-0bc1-444d-bc57-16cb24ae1f56"));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey("116f3ebd-0bc1-444d-bc57-16cb24ae1f56"));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate("ec62eaf3-821f-4fed-aa3e-0654557131ef"));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId("8b195cb4-e797-45dd-93c2-cc0a422fb993"));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, "c8f0e8cf-3b11-4592-b187-9f6383652a88"));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate("cf1faf2b-eb0f-4ffe-af11-bf3e9f8d4511"));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship("a47caa65-d6b6-4b42-ad54-ee3d75441c50"));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType2({ uuid: '5076e4cf-3665-4681-8540-17522e663714' }));
        // firstName
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "firstName",
            attributeUUID: "0e4d40b9-ee09-4276-90f7-677ed1afb1ca", // uuid
            comment: "The user first name.",
        }));
        // lastName
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "lastName",
            attributeUUID: "b49d1769-4aa4-4239-ac32-d516e57e1c0e", // uuid
            comment: "The user LastName"
        }));
        // email
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "email",
            attributeUUID: "61921239-366e-4279-b980-056cb66f0672", // uuid
            comment: "The user email",
            validator: zod_1.z.string().email()
        }));
        // phone
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "phone",
            attributeUUID: "61921239-366e-4279-b980-056cb66f0672", // uuid
            comment: "The user phone number",
            isMongoRequired: false,
            validator: zod_1.z.string().min(10).max(10).regex(RegularExpressionRegexConstUtil_1.Regex.numberOnly).regex(/^(?!0000000000$).*/)
        }));
        /**
         * Organization
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "organization",
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: "ee54dc9b-b837-426b-9247-b19e570b43da",
            comment: "The organization."
        }));
        // softPakCustomerId
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "softPakCustomerId",
            attributeUUID: "caa2df81-ada3-43fb-b6d1-27d9ffd3fb03", // uuid
            comment: "The Soft-Pak customerid.",
            isMongoRequired: false,
        }));
        /**
         * UserType
         * Many Users to One User Type
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "userType",
            foreignRefEntityNamespace: UserType_1.UserType.metaDataForUserType.namespace,
            attributeUUID: "02b3249a-cc8d-4bb6-b257-ac9c515b59c8",
            comment: "The User Type."
        }));
    }
    static getInstance() {
        return (UserEntityModel.INSTANCE);
    }
}
exports.UserEntityModel = UserEntityModel;
UserEntityModel.INSTANCE = new UserEntityModel();
