// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/WorkspaceWidgetEndpointEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceWidgetEndpointEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const Workspace_1 = require("../../classes/persistable/Workspace");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.WorkspaceWidgetEndpoint";
class WorkspaceWidgetEndpointEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('a591a3b9-18c8-4bec-8a58-2e57df511b73')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('3fdc1934-7281-4d6b-a52a-d093b247e4c0');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('fafda214-bee7-4d23-afa3-ea7870de25e3'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('9efb53ee-9c31-4495-a893-fed082e6f497'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('24270bcd-776e-4e61-9376-1dc314d5ed5b'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('d5449786-d77a-4a9b-964d-3018349b01a8'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('7923e04e-cc41-4fc6-a458-afd9b60476d2'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('3e84b290-1662-4fba-ba69-87cd8f49f1ac'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('1fff3f07-08c4-4a75-9398-9e6fe63f5579'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '4917d2e6-90f2-4a8b-92c8-e1591df4cb6e'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('afc44173-dc61-4275-9a30-5dd1fd48bf75'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('567598a0-bfe0-471c-ac7b-036983536dd3'));
        // // deploymentDomainName
        // this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableString(
        //     'deploymentDomainName',
        //     'df6e212e-875a-458e-bb9c-337a3d7aaa8a', // uuid
        //     'The domain where the wiget will be deployed.'
        // ));
        // deployedURL
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'deployedURL',
            attributeUUID: '6911703e-c747-48ae-a722-84f3a2c010ed', // uuid
            comment: 'The endpoint deployed URL.'
        }));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '75ae470b-b351-418f-a990-1f35c86d7e83', // uuid
            comment: 'The workspace description.'
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: '6de1135f-f31e-4783-956d-74c75f4f14e2',
            comment: 'Identifies if the WorkSpace is active.'
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '4adaaeab-4a3d-4ee6-9b6b-d110bb209636', // uuid
            comment: 'The endpoint name.'
        }));
        /**
         * organization oid
         * The Tenant Organization this workspace belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: 'fb7fd319-0e69-4cea-add8-6737e8ee12d0',
            comment: 'The tenant Organization.'
        }));
        // securityToken
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'securityToken',
            attributeUUID: '903e8b12-9ed9-45d4-878b-cdbd9d100e72', // uuid
            comment: 'The endpoint security token.'
        }));
        // securityTokenExpirationDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate('securityTokenExpirationDate', '3e548a68-fc90-49c6-b676-7350f111e838', // uuid
        'The security token expiration date.')
            .setMongoIsRequired(false));
        /**
         * Workspace oid
         * The workspace this template belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            foreignRefEntityNamespace: Workspace_1.Workspace.metaDataForWorkspace.namespace,
            attributeUUID: '3c3c3f42-fe1b-459f-b607-7b86b9bcc60f',
            comment: 'The workspace this template belongs to.'
        }));
    }
    static getInstance() {
        return (WorkspaceWidgetEndpointEntityModel.INSTANCE);
    }
}
exports.WorkspaceWidgetEndpointEntityModel = WorkspaceWidgetEndpointEntityModel;
WorkspaceWidgetEndpointEntityModel.INSTANCE = new WorkspaceWidgetEndpointEntityModel();
