// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/WorkspaceTemplateAttributeEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceTemplateAttributeEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const WorkspaceTemplate_1 = require("../../classes/persistable/WorkspaceTemplate");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.WorkspaceTemplateAttribute";
class WorkspaceTemplateAttributeEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('1e3b57d7-0d67-429c-9241-f3eb0f44fa34')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('f937b788-78d0-4373-a8aa-11b7119cbf73');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('e6b4bacc-1c45-4b1b-afae-279a7512ba5f'));
        // _entityState
        let iEntityPersistableType = {
            attributeUUID: 'eb59d7f8-beb5-4ee4-99b0-43c778c4b098',
            mutatedFlagAttributeUUID: 'bdd24db5-9165-4f2f-8436-ea2bf0ed08aa'
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityPersistableType(iEntityPersistableType));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('b261cc91-2ddc-4ab7-8aec-2d039cd6dc9e'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('3bf16476-258d-4db7-badf-f4684e5a317f'));
        // createdDate
        let iCreatedTimestamp = {
            attributeUUID: '7fbec0ef-b742-4767-b59b-432ed520ef52',
            mutatedFlagAttributeUUID: '7d53ae48-c4d2-4927-a183-d0b166df5182'
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate2(iCreatedTimestamp));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('c7484f47-bae4-4540-8d0f-da9b9ddd9fdc'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '823c4818-6ef4-4b9a-bd39-33323a8d748f'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('49ef779e-b94f-4f65-a38a-f87858ceba19'));
        // RelationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType('62ebdabe-4329-4ba4-9873-66a4ca802f75'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('07605620-199d-4d84-b0b9-bb06f02dce2f'));
        // dataType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'dataType',
            attributeUUID: '2cfb0274-ff1c-4aa1-9e83-c5704c47f2ef', // uuid
            comment: 'The attribute data type.'
        }));
        // description
        let description = {
            attributeName: 'description',
            attributeUUID: '7cd3b042-94c4-41f0-88ac-00772599876d',
            comment: 'The attribute description.'
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2(description));
        // defaultValue
        let defaultValue = {
            attributeName: 'defaultValue',
            attributeUUID: '3610adf1-5cbd-4251-88b6-cd852b059e7b',
            mongoIsRequired: false,
            comment: 'The attribute default value.'
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2(defaultValue));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean("isActive", '6ec1a578-f20d-47bd-a2b4-8736e0aadc3c', "Identifies if the attribute is active."), '71d98e5b-69a4-4730-b48f-a0a798ea9cbe' // mutatedFlagAttributeUUID_s
        );
        // isHidden
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean("isHidden", 'e3b44042-1255-4202-bc63-3b621e4c5188', "Identifies if the attribute is hidden.")
            .setMongoIsRequired(false), '108e6751-dc37-4e24-8ee5-75b4c0d11829' // mutatedFlagAttributeUUID_s
        );
        // name
        let name = {
            attributeName: '',
            attributeUUID: '',
            comment: ''
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString("name", 'c4b46b44-a82f-488b-a489-7233b67fd514', // uuid
        "The name of the attribute")
            .setMongoIsRequired(false), '86d011d7-eb35-4315-a456-3b3b17c59b22' // mutatedFlagAttributeUUID_s
        );
        /**
         * organization oid
         * The Tenant Organization this belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef("organization", Organization_1.Organization.metaDataForOrganization.namespace, 'b6b995c0-df0a-413d-a969-3802fcf085fb', "The tenant Organization."), '37351698-ece9-483c-a435-eca4a66e75a5' // mutatedFlagAttributeUUID_s
        );
        // pandaDocKeyName
        let pandaDocKeyName = {
            attributeName: 'pandaDocKeyName',
            attributeUUID: '48e34377-d1b4-4246-a521-97e4fa87c8e3',
            mongoIsRequired: false,
            comment: 'The name of the PandaDoc template attribute.'
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2(pandaDocKeyName));
        // salesStrykeKeyName
        let salesStrykeKeyName = {
            attributeName: 'salesStrykeKeyName',
            attributeUUID: 'd9611341-92e6-4d2f-9f9d-73c83fc5dffa',
            comment: 'The name of the SaleStryke template attribute. This is also used as the name of the UI input value.'
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2(salesStrykeKeyName));
        // UILabel
        let uiLabel = {
            attributeName: 'uiLabel',
            attributeUUID: '5da02938-e358-4e30-9e38-0f945c50cf17',
            mongoIsRequired: false,
            comment: 'The UI Label'
        };
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2(uiLabel));
        /**
         * WorkspaceTemplate oid
         * The workspace template this attribute belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef("workspaceTemplate", WorkspaceTemplate_1.WorkspaceTemplate.metaDataForWorkspaceTemplate.namespace, '02a1a8fe-ddd7-4892-b3db-164fb2b7381c', 'The workspace template this attribute belongs to.'), '6bdf016e-f47a-4317-99be-dce550545319' // mutatedFlagAttributeUUID_s
        );
    }
    static getInstance() {
        return (WorkspaceTemplateAttributeEntityModel.INSTANCE);
    }
}
exports.WorkspaceTemplateAttributeEntityModel = WorkspaceTemplateAttributeEntityModel;
WorkspaceTemplateAttributeEntityModel.INSTANCE = new WorkspaceTemplateAttributeEntityModel();
