// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/WorkspaceTemplateEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceTemplateEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const Workspace_1 = require("../../classes/persistable/Workspace");
const WorkspaceTemplateAttribute_1 = require("../../classes/persistable/WorkspaceTemplateAttribute");
const QuoteBlock_1 = require("../../classes/persistable/QuoteBlock");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.WorkspaceTemplate';
class WorkspaceTemplateEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('6f20a2b8-e51c-4131-9d61-5a98487fd71a')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('effa9bad-e5d6-4013-94aa-6e161f38e6fc');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('b0bb7f0c-f377-41c8-a7bd-7d493058b3ff'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('5ba6cb02-42e5-4bbb-b62a-a48fee0b49df'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('7b1d4a90-2933-4590-b1c6-2698f23fb19c'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('8f312708-2fd7-44a0-b42b-31996555ee7f'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('51112afb-b425-4e96-9ae0-8fd03e105607'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('7a81adbc-559b-4bc7-bc54-ba4a63b8fb4e'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('8e39ecaa-4363-417a-9af1-03adb96ca63f'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'e586ec9e-a1f7-4a4c-9bfc-17c9a09c1065'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('e3a332ce-5c52-40dc-ac77-5fa9ca82b217'));
        // RelationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType('eafc18a9-d549-4bad-9a96-57808767c654'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('1b392f20-29a2-4b58-93be-9ac78925e9c6'));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate('activeEndDate', 'c3e085dd-e784-4610-9c53-fd7d3f715507', // uuid
        'The template active end date.'));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate('activeStartDate', 'fe540f9e-c692-4497-addd-a3f7912fc863', // uuid
        'The template active start date.'));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'decription',
            attributeUUID: '709776ca-b7c1-472a-890e-7c23efe71ad1', // uuid
            comment: 'The workspace description.'
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'isActive',
            attributeUUID: 'c8eee364-7af0-482b-8204-c4625b606bab',
            comment: 'Identifies if the WorkSpace is active.'
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '20ccdaed-f37a-452c-b6da-5823b51739ee', // uuid
            comment: 'The data type'
        }));
        // pandaDocTemplateUUID
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'pandaDocTemplateUUID',
            attributeUUID: '1e0bc643-66d3-4984-8c88-86d799b8aca2', // uuid
            comment: 'The PandaDoc Template UUID',
            isMongoRequired: false,
        }));
        /**
         * quoteBlock oid
         * The quoteblock this workspacetemplate belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'quoteBlock',
            foreignRefEntityNamespace: QuoteBlock_1.QuoteBlock.metaDataForQuoteBlock.namespace,
            attributeUUID: '9c30250c-4445-4926-a102-242730ef403d',
            comment: 'The quoteblock.'
        }));
        /**
         * organization oid
         * The Tenant Organization this workspace belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: '5cacf552-315a-4a40-91eb-4a1ad0c3473f',
            comment: 'The tenant Organization.'
        }));
        /**
         * Workspace oid
         * The workspace this template belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspace',
            foreignRefEntityNamespace: Workspace_1.Workspace.metaDataForWorkspace.namespace,
            attributeUUID: '02a1a8fe-ddd7-4892-b3db-164fb2b7381c',
            comment: 'The workspace this template belongs to.'
        }));
        /**
         * WorkspaceTemplateAttributes
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('workspaceTemplateAttributes', 'workspaceTemplateAttribute', WorkspaceTemplateAttribute_1.WorkspaceTemplateAttribute.metaDataForWorkspaceTemplateAttribute.namespace, WorkspaceTemplateAttribute_1.WorkspaceTemplateAttribute.metaDataForWorkspaceTemplateAttribute.mongoAttributes.workspaceTemplate, 'a724a7c7-f08e-4573-ab93-98fa0624ce78', // uuid
        'The workspace template attributes.', 1));
    }
    static getInstance() {
        return (WorkspaceTemplateEntityModel.INSTANCE);
    }
}
exports.WorkspaceTemplateEntityModel = WorkspaceTemplateEntityModel;
WorkspaceTemplateEntityModel.INSTANCE = new WorkspaceTemplateEntityModel();
