// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/WorkspaceDocumentAttributeEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceDocumentAttributeEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const WorkspaceDocument_1 = require("../../classes/persistable/WorkspaceDocument");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.WorkspaceDocumentAttribute";
class WorkspaceDocumentAttributeEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('ac55ad74-454c-4ec8-8eca-75e83de5cbb4')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('a08ab6b6-ed3a-490e-8dd2-d175fcf23e9b');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('35c3ced0-c87f-40cf-b41e-ad2a4301feb0'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('bad7eb18-eae9-4c59-88ad-c0cadefdb22d'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('1bb632c9-a8e6-4eb8-80e6-ef197afa4c93'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('2754eae6-494c-4ba3-ae80-5b3d0146aa8d'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('93a0a517-315e-420c-82c2-07049ce869e5'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('6481c8fb-a016-4632-b53c-a3cafa35caca'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('2e208164-89ad-4d12-bd7e-a37abf6513d4'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '4252fabd-e059-41c0-9636-050994db3c1e'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('a5beb929-9279-412c-9d3b-bde16764adca'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('a0a25286-bf18-43b8-830d-e3b5bd60a471'));
        // value
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: "value",
            attributeUUID: '5b1846e3-2167-4a20-b14c-2b4e0625342e', // uuid
            comment: "The attribute value.",
            isMongoRequired: false,
        }));
        /**
         * organization oid
         * The Tenant Organization this belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: 'f914e6f7-4d68-41ad-852e-e23bab8560ad',
            comment: 'The tenant Organization.'
        }));
        // relationshipPersistableType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRelationshipPersistableType('6864a639-2041-48ae-86d8-0944285046ff'));
        /**
         * WorkspaceTemplate oid
         * The workspace template this attribute belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "workspaceDocument",
            foreignRefEntityNamespace: WorkspaceDocument_1.WorkspaceDocument.metaDataForWorkspaceDocument.namespace,
            attributeUUID: '94bd3c06-de9e-4deb-afcd-209d101d2bc4',
            comment: 'The workspace document this attribute belongs to.'
        }));
    }
    static getInstance() {
        return (WorkspaceDocumentAttributeEntityModel.INSTANCE);
    }
}
exports.WorkspaceDocumentAttributeEntityModel = WorkspaceDocumentAttributeEntityModel;
WorkspaceDocumentAttributeEntityModel.INSTANCE = new WorkspaceDocumentAttributeEntityModel();
