// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/WorkspaceWidgetViewEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceWidgetViewEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const WorkspaceWidget_1 = require("../../classes/persistable/WorkspaceWidget");
const SystemWidgetViewTemplate_1 = require("../../classes/persistable/SystemWidgetViewTemplate");
const ENTITY_CLASS_NAMESPACE = "com.salesstryke.entity.classes.persistable.WorkspaceWidgetView";
class WorkspaceWidgetViewEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('e7cc7dc6-472b-4ede-af52-0e9616b76685')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('0537feab-2a03-439d-9812-fbae8b8541d7');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('c02d34de-a8d5-4968-8b12-e1127eb700e7'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('7483aba8-0099-48f1-ac8e-cab95236eec2'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('3c7e5ac8-4f39-448c-9762-0b14b1db772f'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('4fba558c-196e-41f1-9716-f3182efb9d6c'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('db6b69db-3adc-441e-ba9a-7e1b3f863473'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('6195d37e-a8fa-418c-ad5b-76954dfd87af'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('b4291168-7558-492e-8fa5-06fc4e6cf8d2'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, 'cedf54da-e1cc-4c0b-b098-c7120844b907'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('4ea9e3a9-cfd6-4845-8d23-1094fd7e22c0'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('2ea9e489-a705-4a82-9c2a-a30428ad687c'));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: '58998203-e73b-4734-866b-1ffe701abb7a',
            comment: 'The description for the widget view'
        }));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: '53a83b72-ad8f-435b-adf8-868274a98816',
            comment: 'The name of the widget view'
        }));
        /**
         * organization oid
         * The Tenant Organization this workspace belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: 'ab8fd6db-cefe-4476-a6ea-de0b6a6ccbc7',
            comment: 'The tenant Organization.'
        }));
        /**
         * SystemWidgetViewTemplate oid
         * The view type this widget belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'systemWidgetViewTemplate',
            foreignRefEntityNamespace: SystemWidgetViewTemplate_1.SystemWidgetViewTemplate.metaDataForSystemWidgetViewTemplate.namespace,
            attributeUUID: '9ee17b85-dcc7-4812-8459-4f8ec1f8fb86',
            comment: 'The view type this widget belongs to.'
        }));
        /**
         * WorkspaceWidget oid
         * The workspace this template belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: "workspaceWidget",
            foreignRefEntityNamespace: WorkspaceWidget_1.WorkspaceWidget.metaDataForWorkspaceWidget.namespace,
            attributeUUID: '4f56f583-9028-4b85-bfd9-5493f536d8f3',
            comment: "The workspace widget this view belongs to."
        }));
    }
    static getInstance() {
        return (WorkspaceWidgetViewEntityModel.INSTANCE);
    }
}
exports.WorkspaceWidgetViewEntityModel = WorkspaceWidgetViewEntityModel;
WorkspaceWidgetViewEntityModel.INSTANCE = new WorkspaceWidgetViewEntityModel();
