// @salesstryke/mobile-api/src/com/salesstryke/entity/model/persistable/WorkspaceWidgetEntityModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceWidgetEntityModel = void 0;
const EntityBase_1 = require("../../../data/entity/EntityBase");
const EntityType_1 = require("../../../data/entity/EntityType");
const DataModelAttributeHelperUtil_1 = require("../../../utils/DataModelAttributeHelperUtil");
const NamespaceUtils_1 = require("../../../utils/NamespaceUtils");
const Organization_1 = require("../../classes/persistable/Organization");
const WorkspaceTemplate_1 = require("../../classes/persistable/WorkspaceTemplate");
const WorkspaceWidgetEndpoint_1 = require("../../classes/persistable/WorkspaceWidgetEndpoint");
const WorkspaceWidgetView_1 = require("../../classes/persistable/WorkspaceWidgetView");
const WorkspaceWidgetAttribute_1 = require("../../classes/persistable/WorkspaceWidgetAttribute");
const ENTITY_CLASS_NAMESPACE = 'com.salesstryke.entity.classes.persistable.WorkspaceWidget';
class WorkspaceWidgetEntityModel extends EntityBase_1.EntityBase {
    constructor() {
        super();
        this.setAddPersistableAttributes(true);
        this.setIsOrganizationRefRequired(true);
        let entityModel = this.getEntityModel();
        let simpleName_s = NamespaceUtils_1.NamespaceUtils.namespaceToSimpleName(ENTITY_CLASS_NAMESPACE);
        entityModel
            .setCrudCreateAuthorizationUUID('8326cb65-55a8-45c2-aba5-881c9ff47ac1')
            .setDataModelName(simpleName_s)
            .setEntityType(EntityType_1.EntityType.PERSISTABLE)
            .setIsGraphUpdateable(true)
            .setNamespace(ENTITY_CLASS_NAMESPACE)
            .setUUID('583d7dfb-4bc0-4123-bc40-56f3218a2204');
        // _jsonStateStr
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonStateStr('5aaff69a-ad9d-4a59-a92d-02ce92960f45'));
        // _entityState
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getEntityState('1317dc96-0eec-44f2-951f-1a9aa735fe92'));
        // _idSetBySystem
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIdSetBySystem('c80ecb4d-bf66-4cd9-bec4-dcacc9d30f48'));
        // _initializedDataKey
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getIntializedDataKey('1f7e67d5-3f86-4edc-8ef0-1129b679423a'));
        // addToRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getAddToRelationship('037c842e-ba1e-4715-b39b-fa5da92a1514'));
        // createdDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getCreatedDate('c6fa4d59-71c8-49f1-9a78-4b12888c5c34'));
        // id
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getId('92274af8-ebf2-4e0b-829f-32a03208efae'));
        // jsonType
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getJsonType(ENTITY_CLASS_NAMESPACE, '4c6420ff-91e2-451a-bd93-448332b287b0'));
        // modifiedDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getModifiedDate('e6e38428-2446-403d-93c1-7b7a7614a54c'));
        // removeFromRelationship
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getRemoveFromRelationship('207867af-d7d3-4846-8255-26f778a36018'));
        // activeEndDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate('activeEndDate', '984b8ac0-bc7e-4317-9f06-5e223d1edd58', // uuid
        'The widget active end date.'));
        // activeStartDate
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableDate('activeStartDate', '86b9b5a4-79e1-4cd2-818a-3b92fad77c9e', // uuid
        'The widget active start date.'));
        // description
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'description',
            attributeUUID: 'a1433ef4-846b-4e24-be4d-3fa2beef1097', // uuid
            comment: 'The widget description.'
        }));
        // isActive
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean('isActive', '8ed9b6ea-bf20-467d-8626-8e8a2f3a94b9', 'Identifies if the widget is active.'));
        // mapShowPolygons
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean('mapShowPolygons', 'df828ab3-98f1-4b68-a3fb-e8cb510d4aea', 'Identifies if the map polygons should be shown from the Google Map.'));
        // name
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'name',
            attributeUUID: 'b3cb9255-5ea1-49fb-b404-6d883bdb236c', // uuid
            comment: 'The widget name.'
        }));
        /**
         * organization oid
         * The Tenant Organization this workspace belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'organization',
            foreignRefEntityNamespace: Organization_1.Organization.metaDataForOrganization.namespace,
            attributeUUID: '7ca63174-d291-4a64-81ca-ce01fc36513e',
            comment: 'The tenant Organization.'
        }));
        // mapShowPolygons
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableBoolean2({
            attributeName: 'pandaDocWebHookEnabled',
            attributeUUID: '5d0a87e0-cd3a-4d0f-843d-7ed09fdfa70d',
            comment: 'Identifies if the PandaDoc webhook has been enabled.'
        }));
        // siteButtonHoverBackgroundColor
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'siteButtonHoverBackgroundColor',
            attributeUUID: '2cf6b8ae-a019-4eab-b421-ecf057eda652', // uuid
            comment: 'The widget button hover background color, should match the site default. Is required.'
        }));
        // siteCustomerSupportNumber
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'siteCustomerSupportNumber',
            attributeUUID: '06b37941-5a99-492e-ae4f-eda33c9dc84d', // uuid
            comment: 'The widget text that will show the customers support number. Is optional.',
            isMongoRequired: false,
        }));
        // siteDescription
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'siteDescription',
            attributeUUID: '1756b109-995d-4567-a22c-c8e51e816a8a', // uuid
            comment: 'The widget text that displays the site description. Is optional.',
            isMongoRequired: false,
        }));
        // siteEmail
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'siteEmail',
            attributeUUID: '7696b90f-b7e6-4790-a298-3ea8536ba6a9', // uuid
            comment: 'The widget text that displays the site contact email address. Is optional.',
            isMongoRequired: false,
        }));
        // siteLogoHeight
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableInteger('siteLogoHeight', '43518dc5-6093-4255-b654-41f8bd1f293f', // uuid
        'The site logo height. Is optional.')
            .setMongoIsRequired(false));
        // siteLogoPath
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'siteLogoPath',
            attributeUUID: 'dffa34b7-5c99-4db5-a505-d9d5bf1edd39', // uuid
            comment: 'The path to the site logo. Is optional.',
            isMongoRequired: false,
        }));
        // siteUrlPath
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'siteUrlPath',
            attributeUUID: 'bb8ccf30-46a7-453a-bf4a-19e9f9e4dcb0', // uuid
            comment: 'The path to the website. Is required.'
        }));
        // styleButtonEnabledBackgroundColor
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'styleButtonEnabledBackgroundColor',
            attributeUUID: '7995dd99-664e-4b99-990e-288462c85347', // uuid
            comment: 'The widget button enabled background color. Is required.'
        }));
        // styleButtonEnabledColor
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'styleButtonEnabledColor',
            attributeUUID: 'db38cff7-8ead-4665-9e01-ee78ca1a3e2d', // uuid
            comment: 'The widget button enabled color. Is required.'
        }));
        // styleLayoutBoarderColor
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'styleLayoutBoarderColor',
            attributeUUID: 'ebe80999-2c98-4ba3-8123-d3b32247bfee', // uuid
            comment: 'The widget layout boarder color. Is required.'
        }));
        // stylePrimaryColor
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'stylePrimaryColor',
            attributeUUID: '7b584b7b-9a5c-4eda-851f-7db83ef91886', // uuid
            comment: 'The widget primary color. Is required.'
        }));
        // styleSecondaryColor
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'styleSecondaryColor',
            attributeUUID: 'ef29d16c-9f96-475a-899e-54cb11498624', // uuid
            comment: 'The widget secondary color. Is required.'
        }));
        // styleTeriaryColor
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableString2({
            attributeName: 'styleTeriaryColor',
            attributeUUID: 'be3eb303-76bc-4928-b824-f0ffb2d82be6', // uuid
            comment: 'The widget teriary color. Is required.'
        }));
        /**
         * WorkspaceTemplateAry
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRefArray({
            attributeName: 'workspaceTemplates',
            attributeNameSingular: 'workspaceTemplate',
            comment: 'A WorkspaceTemplate for this Widget.',
            foreignRefEntityNamespace: WorkspaceTemplate_1.WorkspaceTemplate.metaDataForWorkspaceTemplate.namespace,
            attributeUUID: '3e3e69cd-7c73-4f5d-bcff-18d8730a11b0'
        }));
        // /**
        //  * WorkspaceDocuments
        //  */
        //  this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableOneToMany(
        //     'workspaceDocuments',
        //     'workspaceDocument',
        //     WorkspaceDocument.metaDataForWorkspaceDocument.namespace,
        //     WorkspaceDocument.metaDataForWorkspaceDocument.mongoAttributes.workspace,
        //     '6c3d5c78-d79a-4558-a164-e100d20c9520lk', // uuid
        //     'The documents for the workspace.',
        //     1
        // ));  
        // /**
        //  * WorkspaceTemplates
        //  */
        //  this.getEntityModel().addEntityAttribute(DataModelAttributeHelper.getPersistableOneToMany(
        //     'workspaceTemplates',
        //     'workspaceTemplate',
        //     WorkspaceTemplate.metaDataForWorkspaceTemplate.namespace,
        //     WorkspaceTemplate.metaDataForWorkspaceTemplate.mongoAttributes.workspace,
        //     'afdd500b-c65e-46a8-a985-ef1d522a122b', // uuid
        //     'The workspace templates for the workspace.',
        //     1
        // ));  
        /**
         * WorkspaceWidgetAttributes
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('workspaceWidgetAttributes', 'workspaceWidgetAttribute', WorkspaceWidgetAttribute_1.WorkspaceWidgetAttribute.metaDataForWorkspaceWidgetAttribute.namespace, WorkspaceWidgetAttribute_1.WorkspaceWidgetAttribute.metaDataForWorkspaceWidgetAttribute.mongoAttributes.workspaceWidget, '1b45d9ce-d0f1-41ae-9508-41465f49cfb3', // uuid
        'The attributes for the widget.'));
        /**
         * WorkspaceWidgetView
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoVirtualArray('workspaceWidgetViews', 'workspaceWidgetView', WorkspaceWidgetView_1.WorkspaceWidgetView.metaDataForWorkspaceWidgetView.namespace, WorkspaceWidgetView_1.WorkspaceWidgetView.metaDataForWorkspaceWidgetView.mongoAttributes.workspaceWidget, 'be03f997-74d8-4cba-a5da-eb3ab621c5c5', // uuid
        'The views for the widget.'));
        /**
         * WorkspaceWidgetEndpoint oid
         * The workspace this template belongs to.
         */
        this.getEntityModel().addEntityAttribute(DataModelAttributeHelperUtil_1.DataModelAttributeHelper.getPersistableMongoRef2({
            attributeName: 'workspaceWidgetEndpoint',
            foreignRefEntityNamespace: WorkspaceWidgetEndpoint_1.WorkspaceWidgetEndpoint.metaDataForWorkspaceWidgetEndpoint.namespace,
            attributeUUID: 'b58e6b0e-c915-4f68-b91f-c17fdae91f0f',
            comment: 'The endpoint this widget belongs to.'
        }));
    }
    static getInstance() {
        return (WorkspaceWidgetEntityModel.INSTANCE);
    }
}
exports.WorkspaceWidgetEntityModel = WorkspaceWidgetEntityModel;
WorkspaceWidgetEntityModel.INSTANCE = new WorkspaceWidgetEntityModel();
