// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/AgreementTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgreementTypeEnum = void 0;
const EnumBase_1 = require("../../lang/EnumBase");
const AgreementType_1 = require("../classes/persistable/AgreementType");
const NAMESPACE = 'com.salesstryke.entity.enums.AgreementTypeEnum';
class AgreementTypeEnum extends EnumBase_1.EnumBase {
    constructor(p) {
        super({
            idOverride: p.enumId,
            name: p.name,
            namespace: NAMESPACE,
        });
        this.agreementType = new AgreementType_1.AgreementType()
            .setId(p.databaseUUID)
            .setDescription(p.description)
            .setName(p.name);
        this.agreementType._setIsUserSetId(true);
    }
    getAgreementType() {
        return this.agreementType;
    }
}
exports.AgreementTypeEnum = AgreementTypeEnum;
AgreementTypeEnum.BULK_PICKUP = new AgreementTypeEnum({ enumId: 1, name: 'BULK_PICKUP', databaseUUID: '64f1c2e5a2b3c4d5e6f7a8b9', description: 'Bulk Pickup Agreement' });
AgreementTypeEnum.DISPOSAL_GUIDE = new AgreementTypeEnum({ enumId: 2, name: 'DISPOSAL_GUIDE', databaseUUID: '64f1c2e5a2b3c4d5e6f7a8c0', description: 'Disposal Guide Agreement' });
AgreementTypeEnum.WORKSPACE = new AgreementTypeEnum({ enumId: 3, name: 'WORKSPACE', databaseUUID: '64f1c2e5a2b3c4d5e6f7a8c1', description: 'Workspace Agreement' });
AgreementTypeEnum.ORGANIZATION = new AgreementTypeEnum({ enumId: 4, name: 'ORGANIZATION', databaseUUID: '64f1c2e5a2b3c4d5e6f7a8c2', description: 'Organization Agreement' });
AgreementTypeEnum.GENERAL = new AgreementTypeEnum({ enumId: 5, name: 'GENERAL', databaseUUID: '64f1c2e5a2b3c4d5e6f7a8c3', description: 'General Agreement' });
