// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/CnsSyncStatusTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsSyncStatusTypeEnum = void 0;
const EnumBase_1 = require("../../lang/EnumBase");
const CnsSyncStatusType_1 = require("../classes/persistable/CnsSyncStatusType");
const NAMESPACE = 'com.salesstryke.entity.enums.CnsSyncStatusTypeEnum';
class CnsSyncStatusTypeEnum extends EnumBase_1.EnumBase {
    constructor(p) {
        super({
            idOverride: p.enumId,
            name: p.name,
            namespace: NAMESPACE
        });
        this.cnsEndPointSignupStatusType = new CnsSyncStatusType_1.CnsSyncStatusType()
            .setId(p.databaseUUID)
            .setDescription(p.description)
            .setIsActive(p.isActive)
            .setName(p.name);
        this.cnsEndPointSignupStatusType._setIsUserSetId(true);
    }
    getCnsEndPointSignupStatusType() {
        return (this.cnsEndPointSignupStatusType);
    }
}
exports.CnsSyncStatusTypeEnum = CnsSyncStatusTypeEnum;
CnsSyncStatusTypeEnum.COMPLETED = new CnsSyncStatusTypeEnum({ enumId: 10, name: 'COMPLETED', databaseUUID: '664473ed14f6af2585020b59', description: 'Sync Status is completed.', isActive: true });
CnsSyncStatusTypeEnum.INPROGRESS = new CnsSyncStatusTypeEnum({ enumId: 5, name: 'INPROGRESS', databaseUUID: '664473ed14f6af2585020b5a', description: 'Sync Status is inprogress', isActive: true });
CnsSyncStatusTypeEnum.PENDING = new CnsSyncStatusTypeEnum({ enumId: 1, name: 'PENDING', databaseUUID: '664473ed14f6af2585020b5b', description: 'Sync status is pending', isActive: true });
