// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/I18nKeyTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.I18nKeyTypeEnum = void 0;
const EnumPersistableBase_1 = require("../../lang/EnumPersistableBase");
const I18nKeyType_1 = require("../classes/persistable/I18nKeyType");
const NAMESPACE = 'com.salesstryke.entity.enums.I18nKeyTypeEnum';
class I18nKeyTypeEnum extends EnumPersistableBase_1.EnumPersistableBase {
    constructor(p) {
        super({
            databaseOID: p.databaseOID,
            name: p.name,
            namespace: NAMESPACE
        });
        this.i18nKeyType = new I18nKeyType_1.I18nKeyType()
            .setId(p.databaseOID)
            .setDescription(p.description)
            .setName(p.name);
        this.i18nKeyType._setIsUserSetId(true);
    }
    getI18nKeyType() {
        return (this.i18nKeyType);
    }
}
exports.I18nKeyTypeEnum = I18nKeyTypeEnum;
I18nKeyTypeEnum.UI_COMMON_BUTTON_SAVE = new I18nKeyTypeEnum({ name: 'ui.common.button.save', databaseOID: '65f1e6a547277444c6089463', description: 'The common ui save button.' });
I18nKeyTypeEnum.UI_COMMON_BUTTON_CANCEL = new I18nKeyTypeEnum({ name: 'ui.common.button.cancel', databaseOID: '65f1e6a547277444c6089464', description: 'The common ui cancel button.' });
I18nKeyTypeEnum.UI_COMMON_BUTTON_CLOSE = new I18nKeyTypeEnum({ name: 'ui.common.button.close', databaseOID: '65f1e6a547277444c6089465', description: 'The common ui close button.' });
I18nKeyTypeEnum.UI_USER_FIRST_NAME = new I18nKeyTypeEnum({ name: 'ui.user.first.name', databaseOID: '65f1e6a547277444c6089466', description: 'The user first name.' });
I18nKeyTypeEnum.VALIDATION_IS_REQUIRED = new I18nKeyTypeEnum({ name: 'validation.isrequired', databaseOID: '65f1e6a547277444c6089468', description: 'A validation to test is required.' });
I18nKeyTypeEnum.VALIDATION_MIN = new I18nKeyTypeEnum({ name: 'validation.min', databaseOID: '65f1e6a547277444c6089469', description: 'A validation to test max length.' });
I18nKeyTypeEnum.VALIDATION_MAX = new I18nKeyTypeEnum({ name: 'validation.max', databaseOID: '65f1e6a547277444c6089467', description: 'A validation to test min length.' });
