// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/I18nLocaleTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.I18nLocaleTypeEnum = void 0;
const EnumPersistableBase_1 = require("../../lang/EnumPersistableBase");
const persistable_1 = require("../classes/persistable");
const NAMESPACE = 'com.salesstryke.entity.enums.I18nLocaleTypeEnum';
class I18nLocaleTypeEnum extends EnumPersistableBase_1.EnumPersistableBase {
    constructor(p) {
        super({
            databaseOID: p.databaseOID,
            name: p.name,
            namespace: NAMESPACE
        });
        this.i18nLocaleType = new persistable_1.I18nLocaleType()
            .setId(p.databaseOID)
            .setDescription(p.description)
            .setName(p.name);
        this.i18nLocaleType._setIsUserSetId(true);
    }
    getI18nLocaleType() {
        return (this.i18nLocaleType);
    }
}
exports.I18nLocaleTypeEnum = I18nLocaleTypeEnum;
I18nLocaleTypeEnum.EN_US = new I18nLocaleTypeEnum({ name: 'en-US', databaseOID: '65f1e6a547277444c6089460', description: 'English, US' });
I18nLocaleTypeEnum.ES_US = new I18nLocaleTypeEnum({ name: 'es-US', databaseOID: '65f1e6a547277444c6089461', description: 'Spanish, US' });
