// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/KmlTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KmlTypeEnum = void 0;
const EnumBase_1 = require("../../lang/EnumBase");
const KmlType_1 = require("../classes/persistable/KmlType");
const NAMESPACE = 'com.salesstryke.entity.enums.KmlTypeEnum';
// TODO: delete file and clean up references - not need for future cases
class KmlTypeEnum extends EnumBase_1.EnumBase {
    constructor(p) {
        super({
            idOverride: p.enumId,
            name: p.name,
            namespace: NAMESPACE
        });
        this.KmlType = new KmlType_1.KmlType()
            .setId(p.databaseUUID)
            .setDescription(p.description)
            .setIsCustomerForOrganization(p.isCustomerForOrganization)
            .setZIndex(p.zIndex)
            .setName(p.name);
        this.KmlType._setIsUserSetId(true);
    }
    getKmlType() {
        return (this.KmlType);
    }
}
exports.KmlTypeEnum = KmlTypeEnum;
KmlTypeEnum.SERVICEZONE = new KmlTypeEnum({ enumId: 1, name: 'SERVICEZONE', databaseUUID: '67910ad55ff983000c9f23c1', description: 'Service Zone Kml', isCustomerForOrganization: true, zIndex: 10000 }); //
KmlTypeEnum.DISPOSALGUIDE = new KmlTypeEnum({ enumId: 2, name: 'DISPOSALGUIDE', databaseUUID: '67910ad55ff983000c9f23c2', description: 'Disposable guide Kml', isCustomerForOrganization: true, zIndex: 10000 - 100 }); // after service area
KmlTypeEnum.BULKPICKUP = new KmlTypeEnum({ enumId: 3, name: 'BULKPICKUP', databaseUUID: '67910ad55ff983000c9f23c3', description: 'Bulk pickup Kml', isCustomerForOrganization: true, zIndex: 10000 - 200 }); // after service area
KmlTypeEnum.SERVICEAREA = new KmlTypeEnum({ enumId: 4, name: 'SERVICEAREA', databaseUUID: '67910ad55ff983000c9f23c4', description: 'Service Area Kml', isCustomerForOrganization: true, zIndex: 10000 - 300 }); // 3
KmlTypeEnum.SALESAREA = new KmlTypeEnum({ enumId: 5, name: 'SALESAREA', databaseUUID: '67910ad55ff983000c9f23c5', description: 'Sales Area Kml', isCustomerForOrganization: true, zIndex: 10000 - 300 }); // 3
KmlTypeEnum.GENERAL = new KmlTypeEnum({ enumId: 6, name: 'GENERAL', databaseUUID: '67910ad55ff983000c9f23c6', description: 'General Kml.', isCustomerForOrganization: true, zIndex: 10000 - 400 }); // top LEVEL
KmlTypeEnum.PRODUCT = new KmlTypeEnum({ enumId: 7, name: 'PRODUCT', databaseUUID: '67910ad55ff983000c9f23c7', description: 'Product Kml', isCustomerForOrganization: true, zIndex: 10000 - 500 });
KmlTypeEnum.PROMOTION = new KmlTypeEnum({ enumId: 8, name: 'PROMOTION', databaseUUID: '67910ad55ff983000c9f23c8', description: 'Promotion Kml', isCustomerForOrganization: true, zIndex: 10000 - 600 }); // - 100 between each
KmlTypeEnum.TAX = new KmlTypeEnum({ enumId: 9, name: 'TAX', databaseUUID: '67910ad55ff983000c9f23c9', description: 'Tax Kml', isCustomerForOrganization: true, zIndex: 10000 - 700 });
