// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/RoutingSoftwareTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingSoftwareTypeEnum = void 0;
const EnumPersistableBase_1 = require("../../lang/EnumPersistableBase");
const RoutingSoftwareType_1 = require("../classes/persistable/RoutingSoftwareType");
const NAMESPACE = 'com.salesstryke.entity.enums.RoutingSoftwareTypeEnum';
class RoutingSoftwareTypeEnum extends EnumPersistableBase_1.EnumPersistableBase {
    constructor(p) {
        super({
            name: p.name, namespace: NAMESPACE, databaseOID: p.databaseOID
        });
        this.routingSoftwareType = new RoutingSoftwareType_1.RoutingSoftwareType()
            .setId(p.databaseOID)
            .setName(p.name);
        this.routingSoftwareType._setIsUserSetId(true);
    }
    getRoutingSoftwareTypeID() {
        return this.routingSoftwareType.getId();
    }
    getRoutingSoftwareType() {
        return this.routingSoftwareType;
    }
    getByDatabaseOID(p) {
        let RoutingSoftwareTypeEnum = null;
        return (RoutingSoftwareTypeEnum);
    }
}
exports.RoutingSoftwareTypeEnum = RoutingSoftwareTypeEnum;
RoutingSoftwareTypeEnum.SOFTPAK = new RoutingSoftwareTypeEnum({ name: 'Softpak', databaseOID: '666c986c4ca3c93a76109da4' });
RoutingSoftwareTypeEnum.HAULER_HERO = new RoutingSoftwareTypeEnum({ name: 'Hauler Hero', databaseOID: '666c986c4ca3c93a76109da5' });
