// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/SecurityRoleEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityRoleEnum = void 0;
const data_1 = require("../../data");
const EnumBase_1 = require("../../lang/EnumBase");
const SecurityPermissionEnum_1 = require("./SecurityPermissionEnum");
const SecurityRole_1 = require("../classes/persistable/SecurityRole");
const NAMESPACE = 'com.salesstryke.entity.enums.SecurityRoleEnum';
// TODO: need to add permission to the role... Otherwise delete the role since it's just a place holder for nothing.
class SecurityRoleEnum extends EnumBase_1.EnumBase {
    constructor(p) {
        super({
            name: p.name,
            namespace: NAMESPACE
        });
        this.securityRole = new SecurityRole_1.SecurityRole()
            .setId(p.databaseUUID)
            .setDescription(p.description)
            .setName(p.name);
        // if (p.organization) {
        //     this.securityRole.setOrganization(p.organization);
        // }
        let index_i = 0;
        let size_i = p.permissions.length;
        for (index_i = 0; index_i < size_i; index_i++) {
            let permission = p.permissions[index_i];
            this.securityRole.addSecurityPermission(permission, data_1.PersistenceType.ATTACH);
        }
        this.securityRole._setIsUserSetId(true);
        this.securityRole._clearIsMutated();
    }
    getSecurityRole() {
        return (this.securityRole);
    }
}
exports.SecurityRoleEnum = SecurityRoleEnum;
/**
 *
 * ROLE
 * Administration Full Access
 * Administration View
 * Administration Update
 * Administration Create
 * Administration Delete
 *
 * Directory
 *
 * Service
 *
 * Workspace
 *
 * Settings
 *
 *
 */
SecurityRoleEnum.SalesStryke_Root = new SecurityRoleEnum({
    name: 'SalesStryke Root',
    databaseUUID: '65e67553883af4b7d00817e2',
    description: 'The Sales Stryke Root Role.',
    permissions: [
        SecurityPermissionEnum_1.SecurityPermissionEnum.ROOT_USER.getPermission(),
    ]
});
SecurityRoleEnum.SalesStryke_Admin = new SecurityRoleEnum({
    name: 'SalesStryke Admin',
    databaseUUID: '65e67553883af4b7d00817e3',
    description: 'The Sales Stryke Admin Role.',
    permissions: []
});
SecurityRoleEnum.SalesStryke_Editor = new SecurityRoleEnum({
    name: 'SalesStryke Editor',
    databaseUUID: '65e75553883af4b7d00817f1',
    description: 'The Sales Stryke Editor Role.',
    permissions: []
});
SecurityRoleEnum.SalesStryke_Viewer = new SecurityRoleEnum({
    name: 'SalesStryke Viewer',
    databaseUUID: '65e75553883af4b7d00817f2',
    description: 'The Sales Stryke Viewer Role.',
    permissions: []
});
SecurityRoleEnum.SalesStryke_Support = new SecurityRoleEnum({
    name: 'SalesStryke Support',
    databaseUUID: '65e75553883af4b7d00817f3',
    description: 'The Sales Stryke Support Role.',
    permissions: []
});
SecurityRoleEnum.Client_Admin = new SecurityRoleEnum({
    name: 'Client Admin',
    databaseUUID: '65e75553883af4b7d00817f4',
    description: 'The Client Admin Role',
    permissions: [
    // SecurityPermissionEnum.ADMIN.getPermission(),
    // SecurityPermissionEnum.ADMIN_CREATE.getPermission(),
    // SecurityPermissionEnum.ADMIN_READ.getPermission(),
    // SecurityPermissionEnum.ADMIN_UPDATE.getPermission(),
    // SecurityPermissionEnum.ADMIN_DELETE.getPermission(),
    // SecurityPermissionEnum.CLIENT.getPermission(),
    // SecurityPermissionEnum.ORGANIZATION_FULL_ACCESS.getPermission(),
    // SecurityPermissionEnum.UI_SERVICE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_ADMINISTRATION_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_SETTINGS_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_DIRECTORY_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_WORKSPACE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.USER_READ.getPermission(),
    // SecurityPermissionEnum.USER_CREATE.getPermission(),
    // SecurityPermissionEnum.USER_UPDATE.getPermission(),
    // SecurityPermissionEnum.USER_DELETE.getPermission(),
    // SecurityPermissionEnum.ROLE_READ.getPermission(),
    // SecurityPermissionEnum.ROLE_CREATE.getPermission(),
    // SecurityPermissionEnum.ROLE_UPDATE.getPermission(),
    // SecurityPermissionEnum.ROLE_DELETE.getPermission(),
    // SecurityPermissionEnum.OSS_CREATE.getPermission(),
    // SecurityPermissionEnum.OSS_READ.getPermission(),
    // SecurityPermissionEnum.OSS_UPDATE.getPermission(),
    // SecurityPermissionEnum.OSS_DELETE.getPermission(),
    // SecurityPermissionEnum.SES_CREATE.getPermission(),
    // SecurityPermissionEnum.SES_READ.getPermission(),
    // SecurityPermissionEnum.SES_UPDATE.getPermission(),
    // SecurityPermissionEnum.SES_DELETE.getPermission(),
    // SecurityPermissionEnum.CNS_CREATE.getPermission(),
    // SecurityPermissionEnum.CNS_READ.getPermission(),
    // SecurityPermissionEnum.CNS_UPDATE.getPermission(),
    // SecurityPermissionEnum.CNS_DELETE.getPermission(),
    // SecurityPermissionEnum.CSP_CREATE.getPermission(),
    // SecurityPermissionEnum.CSP_READ.getPermission(),
    // SecurityPermissionEnum.CSP_UPDATE.getPermission(),
    // SecurityPermissionEnum.CSP_DELETE.getPermission()
    ],
});
SecurityRoleEnum.Client_Editor = new SecurityRoleEnum({
    name: 'Client Editor',
    databaseUUID: '65e75553883af4b7d00817f5',
    description: 'The Client Editor Role',
    permissions: [
    // SecurityPermissionEnum.CLIENT.getPermission(),
    // SecurityPermissionEnum.ORGANIZATION_READ.getPermission(),
    // SecurityPermissionEnum.ORGANIZATION_UPDATE.getPermission(),
    // SecurityPermissionEnum.UI_SERVICE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_DIRECTORY_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_WORKSPACE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.USER_READ.getPermission(),
    // SecurityPermissionEnum.ROLE_READ.getPermission(),
    // SecurityPermissionEnum.OSS_CREATE.getPermission(),
    // SecurityPermissionEnum.OSS_READ.getPermission(),
    // SecurityPermissionEnum.OSS_UPDATE.getPermission(),
    // SecurityPermissionEnum.SES_CREATE.getPermission(),
    // SecurityPermissionEnum.SES_READ.getPermission(),
    // SecurityPermissionEnum.SES_UPDATE.getPermission(),
    // SecurityPermissionEnum.CNS_CREATE.getPermission(),
    // SecurityPermissionEnum.CNS_READ.getPermission(),
    // SecurityPermissionEnum.CNS_UPDATE.getPermission(),
    // SecurityPermissionEnum.CSP_CREATE.getPermission(),
    // SecurityPermissionEnum.CSP_READ.getPermission(),
    // SecurityPermissionEnum.CSP_UPDATE.getPermission()
    ]
});
SecurityRoleEnum.Client_Read_Only = new SecurityRoleEnum({
    name: 'Client Read Only',
    databaseUUID: '65e75553883af4b7d00817f6',
    description: 'The Client Read Only Role',
    permissions: [
    // SecurityPermissionEnum.CLIENT.getPermission(),
    // SecurityPermissionEnum.ORGANIZATION_READ.getPermission(),
    // SecurityPermissionEnum.UI_SERVICE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_DIRECTORY_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_WORKSPACE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.USER_READ.getPermission(),
    // SecurityPermissionEnum.ROLE_READ.getPermission(),
    // SecurityPermissionEnum.OSS_READ.getPermission(),
    // SecurityPermissionEnum.SES_READ.getPermission(),
    // SecurityPermissionEnum.CNS_READ.getPermission(),
    // SecurityPermissionEnum.CSP_READ.getPermission()
    ]
});
SecurityRoleEnum.Customer = new SecurityRoleEnum({
    name: 'Customer',
    databaseUUID: '65e75553883af4b7d00817f7',
    description: 'The Customer Role',
    permissions: [
    // SecurityPermissionEnum.CUSTOMER.getPermission(),
    // SecurityPermissionEnum.UI_SERVICE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.OSS_READ.getPermission(),
    // SecurityPermissionEnum.CNS_READ.getPermission(),
    // SecurityPermissionEnum.CSP_READ.getPermission(),
    // SecurityPermissionEnum.CSP_CREATE.getPermission(),
    // SecurityPermissionEnum.CSP_UPDATE.getPermission()
    ]
});
SecurityRoleEnum.System_Admin_Read_Only = new SecurityRoleEnum({
    name: 'System Admin Read Only',
    databaseUUID: '66c45f3ed47e2a19b5c23dd9',
    description: 'System Admin with Read-Only Access',
    permissions: [
    // SecurityPermissionEnum.SYSTEM_ADMIN_READ_ONLY.getPermission(),
    // SecurityPermissionEnum.ADMIN_READ.getPermission(),
    // SecurityPermissionEnum.CLIENT.getPermission(),
    // SecurityPermissionEnum.ORGANIZATION_READ.getPermission(),
    // SecurityPermissionEnum.UI_SERVICE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_ADMINISTRATION_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_SETTINGS_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_DIRECTORY_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.UI_WORKSPACE_NAVIGATION.getPermission(),
    // SecurityPermissionEnum.USER_READ.getPermission(),
    // SecurityPermissionEnum.ROLE_READ.getPermission(),
    // SecurityPermissionEnum.OSS_READ.getPermission(),
    // SecurityPermissionEnum.SES_READ.getPermission(),
    // SecurityPermissionEnum.CNS_READ.getPermission(),
    // SecurityPermissionEnum.CSP_READ.getPermission()
    ]
});
