// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/SettingDefaultEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingDefaultEnum = void 0;
const EnumBase_1 = require("../../lang/EnumBase");
const Setting_1 = require("../classes/persistable/Setting");
const SettingNotification_1 = require("../classes/persistable/SettingNotification");
const SettingOrganization_1 = require("../classes/persistable/SettingOrganization");
const SettingWorkspace_1 = require("../classes/persistable/SettingWorkspace");
const NAMESPACE = 'com.salesstryke.entity.enums.SettingDefaultEnum';
class SettingDefaultEnum extends EnumBase_1.EnumBase {
    constructor(p) {
        super({
            idOverride: p.enumId,
            name: p.name,
            namespace: NAMESPACE
        });
        switch (p.class) {
            case Setting_1.Setting:
                // create default setting
                this.Setting = new Setting_1.Setting()
                    .setId(p.databaseUUID)
                    .setIsDefault(true)
                    .setSettingOrganization(new SettingOrganization_1.SettingOrganization().setId('67910ad55ff983000c9f23c2'))
                    .setSettingWorkspace(new SettingWorkspace_1.SettingWorkspace().setId('67910ad55ff983000c9f23c4'))
                    .setSettingNotification(new SettingNotification_1.SettingNotification().setId('67910ad55ff983000c9f23c3'))
                    .setLogo('https://ss-image-temp.s3.us-east-2.amazonaws.com/preview.png')
                    // .setTypography({
                    //     fontFamily: '"Comic Sans MS","Comic Sans","Chalkboard","cursive"',
                    //     fontSize: 14,
                    //     fontWeightLight: 300,
                    //     fontWeightRegular: 400,
                    //     fontWeightMedium: 500,
                    //     h1: {
                    //         fontSize: '2.5rem',
                    //         fontWeight: 500,
                    //     },
                    //     h2: {
                    //         fontSize: '2rem',
                    //         fontWeight: 500,
                    //     },
                    //     body1: {
                    //         fontSize: '1rem',
                    //     },
                    //     button: {
                    //         textTransform: 'none',
                    //     }
                    // })
                    .setDisplayedName('SalesStryke APP')
                    .setLogoWidth(180)
                    .setLogoHeight(60);
                break;
            case SettingOrganization_1.SettingOrganization:
                this.SettingOrganization = new SettingOrganization_1.SettingOrganization()
                    .setId(p.databaseUUID)
                    .setIsDefault(true)
                    .setLogo('https://example.com/logo.png')
                    .setIsBPSEnabled(false)
                    .setIsCALEnabled(false)
                    .setIsCNSEnabled(false)
                    .setIsCSPEnabled(false)
                    .setIsDGSEnabled(false)
                    .setIsOSSEnabled(false)
                    .setIsSESEnabled(false);
                break;
            case SettingNotification_1.SettingNotification:
                this.SettingNotification = new SettingNotification_1.SettingNotification()
                    .setId(p.databaseUUID)
                    .setIsDefault(true)
                    .setIsClient(false)
                    .setIsCustomer(false);
                // .setUser(null)
                break;
            case SettingWorkspace_1.SettingWorkspace:
                this.SettingWorkspace = new SettingWorkspace_1.SettingWorkspace()
                    .setId(p.databaseUUID)
                    // .setKmlMaxCount(1000)
                    .setIsDefault(true);
                break;
        }
    }
    getSetting() {
        return (this.Setting);
    }
    getSettingOrganization() {
        return (this.SettingOrganization);
    }
    getSettingNotification() {
        return (this.SettingNotification);
    }
    getSettingWorkspace() {
        return (this.SettingWorkspace);
    }
}
exports.SettingDefaultEnum = SettingDefaultEnum;
SettingDefaultEnum.SETTING = new SettingDefaultEnum({ enumId: 0, name: 'SETTING', databaseUUID: '67910ad55ff983000c9f23c0', description: 'Setting', class: Setting_1.Setting });
SettingDefaultEnum.ORGANIZATION = new SettingDefaultEnum({ enumId: 2, name: 'ORGANIZATION', databaseUUID: '67910ad55ff983000c9f23c2', description: 'Organization Settings', class: SettingOrganization_1.SettingOrganization });
SettingDefaultEnum.NOTIFICATION = new SettingDefaultEnum({ enumId: 3, name: 'NOTIFICATION', databaseUUID: '67910ad55ff983000c9f23c3', description: 'Notification Settings', class: SettingNotification_1.SettingNotification });
SettingDefaultEnum.WORKSPACE = new SettingDefaultEnum({ enumId: 4, name: 'WORKSPACE', databaseUUID: '67910ad55ff983000c9f23c4', description: 'Workspace Settings', class: SettingWorkspace_1.SettingWorkspace });
