// @salesstryke/mobile-api/src/com/salesstryke/entity/enums/SettingIntegrationTypeEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingIntegrationTypeEnum = void 0;
const EnumBase_1 = require("../../lang/EnumBase");
const persistable_1 = require("../classes/persistable");
const NAMESPACE = 'com.salesstryke.entity.enums.SettingIntegrationTypeEnum';
class SettingIntegrationTypeEnum extends EnumBase_1.EnumBase {
    constructor(p) {
        super({
            idOverride: p.enumId,
            name: p.name,
            namespace: NAMESPACE
        });
        this.settingIntegrationType = new persistable_1.SettingIntegrationType()
            .setId(p.databaseUUID)
            .setName(p.name);
        this.settingIntegrationType._setIsIdSetBySystem(true);
    }
    getSettingIntegration() {
        return (this.settingIntegrationType);
    }
}
exports.SettingIntegrationTypeEnum = SettingIntegrationTypeEnum;
SettingIntegrationTypeEnum.PANDADOC = new SettingIntegrationTypeEnum({ enumId: 1, name: 'PANDADOC', databaseUUID: '67a10ad55ff983000c9f24c1', description: 'PandaDoc' });
SettingIntegrationTypeEnum.SOFTPAK = new SettingIntegrationTypeEnum({ enumId: 2, name: 'SOFTPAK', databaseUUID: '67a10ad55ff983000c9f24c2', description: 'Softpak' });
SettingIntegrationTypeEnum.HAULERHERO = new SettingIntegrationTypeEnum({ enumId: 3, name: 'HAULERHERO', databaseUUID: '67a10ad55ff983000c9f24c3', description: 'Hauler Hero' });
SettingIntegrationTypeEnum.NAVUSOFT = new SettingIntegrationTypeEnum({ enumId: 4, name: 'NAVUSOFT', databaseUUID: '67a10ad55ff983000c9f24c4', description: 'Navu Soft' });
SettingIntegrationTypeEnum.TRUX = new SettingIntegrationTypeEnum({ enumId: 5, name: 'TRUX', databaseUUID: '67a10ad55ff983000c9f24c5', description: 'Trux' });
