// @salesstryke/mobile-api/src/com/salesstryke/validation/ValidationBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationBase = void 0;
class ValidationBase {
    constructor() {
        this.attributeName = null;
        this.attributeUUID = '';
        this.error = null;
        this.i18nKey = null;
        this.isValid_b = false;
        this.localizedTemplate = '';
        this.localizedError = null;
        this.value = null;
        this.validationName = '';
    }
    getAttributeName() {
        return (this.attributeName);
    }
    getAttributeUUID() {
        return (this.attributeUUID);
    }
    getClone() {
        return (new ValidationBase());
    }
    getError() {
        return (this.error);
    }
    getI18nKey() {
        return (this.i18nKey);
    }
    /**
     * Default is English
     * @returns
     */
    getLocalizedTemplate() {
        return (this.localizedTemplate);
    }
    getLocalizedError() {
        return (this.localizedError);
    }
    getValue() {
        return (this.value);
    }
    isValid() {
        return (this.isValid_b);
    }
    setAttributeName(p) {
        this.attributeName = p.attributeName;
        return (this);
    }
    setAttibuteUUID(p) {
        this.attributeUUID = p.attributeUUID;
        return (this);
    }
    setError(p) {
        this.error = p.error;
        return (this);
    }
    setI18nKey(p) {
        this.i18nKey = p.key;
        return (this);
    }
    setLocalizedTemplate(p) {
        this.localizedTemplate = p.localizedTemplate;
        return (this);
    }
    setLocalizedError(p) {
        this.localizedError = p.localizedError;
        return (this);
    }
    setValue(p) {
        this.value = p.value;
        return (this);
    }
}
exports.ValidationBase = ValidationBase;
