// @salesstryke/mobile-api/src/com/salesstryke/jsonApi/util/JsonStringBuilder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JsonStringBuilder = void 0;
class JsonStringBuilder {
    constructor(currentPadLength_i_p, padLength_i_p, addNewLine_b_p) {
        this.addNewLine_b_c = false;
        this.currentPadLength_i_c = 0;
        this.json_s_c = "";
        this.newLine_s_c = "";
        this.padLength_i_c = 0;
        this.currentPadLength_i_c = currentPadLength_i_p;
        this.addNewLine_b_c = addNewLine_b_p;
        this.padLength_i_c = padLength_i_p;
        if (addNewLine_b_p) {
            this.newLine_s_c = "\n";
        }
    }
    addLineAtCurrentPadLength(line_s_p) {
        let line_s_m = this.pad(line_s_p, this.currentPadLength_i_c) + this.newLine_s_c;
        this.json_s_c += line_s_m;
        return (this);
    }
    addLineAtCurrentFullLength(line_s_p) {
        let line_s_m = this.pad(line_s_p, this.currentPadLength_i_c + this.padLength_i_c) + this.newLine_s_c;
        this.json_s_c += line_s_m;
        return (this);
    }
    addLineNoPad(line_s_p) {
        this.json_s_c += line_s_p;
        return (this);
    }
    addLine(line_s_p) {
        let line_s_m = this.pad(line_s_p, this.padLength_i_c) + this.newLine_s_c;
        this.json_s_c += line_s_m;
        return (this);
    }
    pad(value_s_p, padLength_i_m) {
        let index_i_m = 0;
        let pad_s_m = "";
        for (index_i_m = 0; index_i_m < padLength_i_m; index_i_m++) {
            pad_s_m += " ";
        }
        return (pad_s_m + value_s_p);
    }
    toString() {
        return (this.json_s_c);
    }
}
exports.JsonStringBuilder = JsonStringBuilder;
