// @salesstryke/mobile-api/src/com/salesstryke/jsonApi/util/QueryParam.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryParam = void 0;
const DataType_1 = require("../../data/entity/DataType");
const QueryParamBase_1 = require("../QueryParamBase");
class QueryParam extends QueryParamBase_1.QueryParamBase {
    /**
     *
     * @param name_s
     * @param value
     */
    constructor(name_s, dataType, value) {
        super(name_s, dataType);
        this.value = '';
        this.setEntityType(QueryParam.entityType);
        this.value = value;
    }
    getValue() {
        return (this.value);
    }
    /**
     * will return 0 if its NAN
     * @returns number
     */
    getValueAsNumber() {
        if (this.getDataType() === DataType_1.DataType.FLOAT.getId() && typeof this.value === 'number') {
            return this.value;
        }
        return 0;
    }
    /**
     * will return '' if its not a string
     * @returns string
     */
    getValueAsString() {
        if (this.getDataType() === DataType_1.DataType.STRING.getId() && typeof this.value === 'string') {
            return this.value;
        }
        return '';
    }
    /**
     * will return false if its not a boolean
     * @returns boolean
     */
    getValueAsBoolean() {
        if (this.getDataType() === DataType_1.DataType.STRING.getId() && typeof this.value === 'boolean') {
            return this.value;
        }
        return false;
    }
    /**
     * will return a new Date if its not a Date
     * @returns boolean
     */
    getValueAsDate() {
        if (this.getDataType() === DataType_1.DataType.DATE.getId()) {
            if (this.value instanceof Date) {
                return this.value;
            }
            // Date is converted to string when transforming form TypscriptToJsonApi
            else if (typeof this.value === 'string') {
                return new Date(this.value);
            }
        }
        return new Date();
    }
    setValue(value) {
        this.value = value;
    }
}
exports.QueryParam = QueryParam;
QueryParam.entityType = 'com.salesstryke.jsonapi.QueryParam';
