// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/record/documentQuery/DocumentQueryInteger.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentQueryInteger = void 0;
const ClassFor_1 = require("../../../../data/ClassFor");
const JsonApiRecordBase_1 = require("../../../../jsonApi/JsonApiRecordBase");
const HashCodeUtil_1 = require("../../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._valueIsMutated = "_valueIsMutated";
        this.value = "value";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._valueIsMutated = "_valueIsMutated";
        this.value = "value";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.record.documentQuery.DocumentQueryInteger";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class DocumentQueryInteger extends JsonApiRecordBase_1.JsonApiRecordBase {
    constructor(p) {
        super(p);
        this._valueIsMutated = false;
        this.value = 0;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.record.documentQuery.DocumentQueryInteger");
    }
    /**
     *
     * @returns boolean
     */
    _getIsValueMutated() {
        return (this._valueIsMutated);
    }
    /**
     * the status value of the Integer
     * @returns number
     */
    getValue() {
        return (this.value);
    }
    /**
* ```
     *
* ```
     */
    _setIsValueMutated(_valueIsMutated) {
        this._valueIsMutated = _valueIsMutated;
        return (this);
    }
    /**
* ```
     * the status value of the Integer
* ```
     */
    setValue(value) {
        if (this.value !== value) {
            this._setIsValueMutated(true);
        }
        this.value = value;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: 'value',
            value: this.value
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: 'value',
            value: this.value
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.value);
        return (hashCode_n);
    }
}
exports.DocumentQueryInteger = DocumentQueryInteger;
DocumentQueryInteger.metaDataForDocumentQueryInteger = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.record.documentQuery.DocumentQueryInteger", DocumentQueryInteger);
