// @salesstryke/mobile-api/src/com/salesstryke/client/api/AgreementDocumentClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgreementDocumentClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.AgreementDocumentClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public agreementDocument : AgreementDocumentClientApi;


        this.agreementDocument = new AgreementDocumentClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static AGREEMENT_DOCUMENT__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document/delete");
        public static AGREEMENT_DOCUMENT__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document/getall");
        public static AGREEMENT_DOCUMENT__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document/getbyid");
        public static AGREEMENT_DOCUMENT__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document/save");

*/
class AgreementDocumentClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: AgreementDocument
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '86bf77ea-c8f7-4c74-b85f-613adfc7e537', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'a16d7250-b0ca-4f0a-9e89-d507fee3b547', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: AgreementDocument by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'a36dd737-517e-4dee-9f26-e9db484f2ea1', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '60944882-28b5-49ca-9aab-4a7b4a0a69f1', error: error });
                throw error;
            }
        });
    }
}
exports.AgreementDocumentClientApi = AgreementDocumentClientApi;
