// @salesstryke/mobile-api/src/com/salesstryke/client/api/AgreementDocumentVersionClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgreementDocumentVersionClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.AgreementDocumentVersionClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public agreementDocumentVersion : AgreementDocumentVersionClientApi;


        this.agreementDocumentVersion = new AgreementDocumentVersionClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static AGREEMENT_DOCUMENT_VERSION__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document_version/delete");
        public static AGREEMENT_DOCUMENT_VERSION__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document_version/getall");
        public static AGREEMENT_DOCUMENT_VERSION__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document_version/getbyid");
        public static AGREEMENT_DOCUMENT_VERSION__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_document_version/save");

*/
class AgreementDocumentVersionClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: AgreementDocumentVersion
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT_VERSION__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '862f817d-320f-4fe9-b0d1-02713ff64c1a', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT_VERSION__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'c2ed3131-6f05-4b37-86f1-dd5e53389114', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: AgreementDocumentVersion by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT_VERSION__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '0d23bc18-c9f8-44a1-8671-9fdd4d71adb1', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_DOCUMENT_VERSION__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '600959fd-bb1c-49b2-8321-f53df25f91cc', error: error });
                throw error;
            }
        });
    }
}
exports.AgreementDocumentVersionClientApi = AgreementDocumentVersionClientApi;
