// @salesstryke/mobile-api/src/com/salesstryke/client/api/AgreementTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgreementTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.AgreementTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public agreementType : AgreementTypeClientApi;


        this.agreementType = new AgreementTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static AGREEMENT_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_type/delete");
        public static AGREEMENT_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_type/getall");
        public static AGREEMENT_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_type/getbyid");
        public static AGREEMENT_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/agreement_type/save");

*/
class AgreementTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: AgreementType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_TYPE__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '45fc5f92-6e68-4b42-9dc6-c4283e3794a9', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.AGREEMENT_TYPE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'ca4cf251-7107-495c-b304-9dfb401db014', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: AgreementType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'f7b69013-04c3-4f79-b0e6-7d627f2705a8', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.AGREEMENT_TYPE__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'e863fd7a-8844-4d00-92c9-c67bf4b926a0', error: error });
                throw error;
            }
        });
    }
}
exports.AgreementTypeClientApi = AgreementTypeClientApi;
