// @salesstryke/mobile-api/src/com/salesstryke/client/api/ApiCallBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiCallBase = void 0;
class ApiCallBase {
    constructor(httpClient, httpMethodType, path) {
        this.jsonApiDocument = null;
        this.jsonApiQuery = null;
        this.path_s = "";
        this.httpClient = httpClient;
        this.httpMethodType = httpMethodType;
        this.path_s = path;
    }
    getHttpClient() {
        return (this.httpClient);
    }
    getHttpMethodType() {
        return (this.httpMethodType);
    }
    getJsonApiDocument() {
        return (this.jsonApiDocument);
    }
    getJsonApiQuery() {
        return (this.jsonApiQuery);
    }
    getPath() {
        return (this.path_s);
    }
    /**
     * resolves the async call back if there is an error, returns true. Else returns false if no error.
     * @param reject
     * @param response_jad
     * @returns
     */
    handleResponseErrors(reject, response_jad) {
        let hadError_b = false;
        if (response_jad !== undefined && response_jad !== null) {
            if (response_jad.getErrors().length > 0) {
                let errors = response_jad.getErrors();
                hadError_b = true;
                reject(errors);
            }
        }
        return (hadError_b);
    }
    setJsonApiDocument(jsonApiDocument) {
        this.jsonApiDocument = jsonApiDocument;
        return (this);
    }
    setJsonApiPagination(jsonApiPagination) {
        this.jsonApiQuery = jsonApiPagination;
        return (this);
    }
}
exports.ApiCallBase = ApiCallBase;
