// @salesstryke/mobile-api/src/com/salesstryke/client/api/ApplicationClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ApplicationClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public application : ApplicationClientApi;


        this.application = new ApplicationClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static APPLICATION__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/application/delete");
        public static APPLICATION__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/application/getall");
        public static APPLICATION__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/application/getbyid");
        public static APPLICATION__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/application/save");

*/
class ApplicationClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: Application
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.APPLICATION__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '29aa234f-89a1-4616-ba29-dec062f6d1fe', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.APPLICATION__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'ddeaae3b-276f-467a-83c0-7c83d98eea4c', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: Application by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.APPLICATION__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '3fb2093b-12ed-458f-8474-73a276975777', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.APPLICATION__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '30d13e0a-046b-4802-a426-4c04ec064e64', error: error });
                throw error;
            }
        });
    }
}
exports.ApplicationClientApi = ApplicationClientApi;
