// @salesstryke/mobile-api/src/com/salesstryke/client/api/CalendarItemClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalendarItemClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.CalendarItemClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public calendarItem : CalendarItemClientApi;


        this.calendarItem = new CalendarItemClientApi(this.httpClient);


    Copy to EndPoint.ts

        public static CALENDAR_ITEM__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/calendar_item/delete");
        public static CALENDAR_ITEM__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/calendar_item/getall");
        public static CALENDAR_ITEM__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/calendar_item/getbyid");
        public static CALENDAR_ITEM__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/calendar_item/save");

*/
class CalendarItemClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CalendarItem
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CALENDAR_ITEM__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'c7535b8d-7ea6-482b-8f71-3a258b781ce4', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.CALENDAR_ITEM__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'a9e3d8b0-4376-4652-8fb6-0f721671d096', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CalendarItem by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CALENDAR_ITEM__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'd311f3a3-934d-4e0b-88d0-78ac49aee7da', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CALENDAR_ITEM__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '86cfd308-3e6c-452b-a9ed-1c2902162d05', error: error });
                throw error;
            }
        });
    }
}
exports.CalendarItemClientApi = CalendarItemClientApi;
