// @salesstryke/mobile-api/src/com/salesstryke/client/api/CartItemClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CartItemClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CartItemClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cartItem : CartItemClientApi;


        this.cartItem = new CartItemClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CART_ITEM__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_item/delete");
        public static CART_ITEM__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_item/getall");
        public static CART_ITEM__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_item/getbyid");
        public static CART_ITEM__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_item/save");

*/
class CartItemClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CartItem
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_ITEM__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '294a5e60-3661-418c-b298-de12989a91fc', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CART_ITEM__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'c3139bed-4475-4428-bdde-a518698ff357', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CartItem by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_ITEM__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '43b9aacc-3b1a-49fd-a938-10e164305d25', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_ITEM__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'afb9901d-74b4-4a6f-aa80-b11c95b2b16e', error: error });
                throw error;
            }
        });
    }
}
exports.CartItemClientApi = CartItemClientApi;
