// @salesstryke/mobile-api/src/com/salesstryke/client/api/CartProductRuleClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CartProductRuleClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CartProductRuleClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cartProductRule : CartProductRuleClientApi;


        this.cartProductRule = new CartProductRuleClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CART_PRODUCT_RULE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule/delete");
        public static CART_PRODUCT_RULE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule/getall");
        public static CART_PRODUCT_RULE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule/getbyid");
        public static CART_PRODUCT_RULE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule/save");

*/
class CartProductRuleClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CartProductRule
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '3d90a27a-320a-4083-9a62-36399efa1e6d', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '8e51f213-1708-4d69-8292-30b667e37fcb', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CartProductRule by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '12479b66-ad11-4a03-9dd9-c007bff5a10d', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'b3ac5f8b-fb0e-42c8-8376-7a2058f6ec96', error: error });
                throw error;
            }
        });
    }
}
exports.CartProductRuleClientApi = CartProductRuleClientApi;
