// @salesstryke/mobile-api/src/com/salesstryke/client/api/CartProductRuleTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CartProductRuleTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CartProductRuleTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cartProductRuleType : CartProductRuleTypeClientApi;


        this.cartProductRuleType = new CartProductRuleTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CART_PRODUCT_RULE_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule_type/delete");
        public static CART_PRODUCT_RULE_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule_type/getall");
        public static CART_PRODUCT_RULE_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule_type/getbyid");
        public static CART_PRODUCT_RULE_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_product_rule_type/save");

*/
class CartProductRuleTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CartProductRuleType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '6b9df99d-f8b8-46f5-878e-ee9ce2f5096f', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE_TYPE__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'cf57bf79-34d1-42bd-9c7a-a652f69b9da4', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CartProductRuleType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'ecaf412a-ea7e-4443-b4c1-5a10ec099b81', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PRODUCT_RULE_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'da538dde-9e0f-4f0a-8b11-f6acb5c9333c', error: error });
                throw error;
            }
        });
    }
}
exports.CartProductRuleTypeClientApi = CartProductRuleTypeClientApi;
