// @salesstryke/mobile-api/src/com/salesstryke/client/api/CartSessionClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CartSessionClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CartSessionClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cartSession : CartSessionClientApi;


        this.cartSession = new CartSessionClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CART_SESSION__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_session/delete");
        public static CART_SESSION__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_session/getall");
        public static CART_SESSION__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_session/getbyid");
        public static CART_SESSION__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_session/save");

*/
class CartSessionClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CartSession
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_SESSION__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '03538b18-af5d-49c2-b3ef-ccb06a2c9bbb', error: error });
                throw error;
            }
        });
    }
    getAll() {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CART_SESSION__GET_ALL }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '11b23d6a-dd70-4e64-a689-1e746328f623', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CartSession by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_SESSION__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '7d870b40-5e96-47e6-9fc4-e832683943a7', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_SESSION__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '8d2390bc-1cf3-49df-bc0d-e11a4857072d', error: error });
                throw error;
            }
        });
    }
}
exports.CartSessionClientApi = CartSessionClientApi;
