// @salesstryke/mobile-api/src/com/salesstryke/client/api/CartServiceDayOfWeekClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CartServiceDayOfWeekClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CartServiceDayOfWeekClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cartServiceDayOfWeek : CartServiceDayOfWeekClientApi;


        this.cartServiceDayOfWeek = new CartServiceDayOfWeekClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CART_SERVICE_DAY_OF_WEEK__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_service_day_of_week/delete");
        public static CART_SERVICE_DAY_OF_WEEK__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_service_day_of_week/getall");
        public static CART_SERVICE_DAY_OF_WEEK__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_service_day_of_week/getbyid");
        public static CART_SERVICE_DAY_OF_WEEK__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_service_day_of_week/save");

*/
class CartServiceDayOfWeekClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CartServiceDayOfWeek
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_SERVICE_DAY_OF_WEEK__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'cc06d89a-1b8d-44a1-bbf8-c3b09e96c03e', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.CART_SERVICE_DAY_OF_WEEK__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'e686dc91-c32d-4cad-94bc-4eb472fc3c92', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CartServiceDayOfWeek by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_SERVICE_DAY_OF_WEEK__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'e31fabda-76d6-4e67-b21f-30da8f05dbbe', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_SERVICE_DAY_OF_WEEK__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'ba4f82d4-1a16-4686-b97f-7c5e8cda04f8', error: error });
                throw error;
            }
        });
    }
}
exports.CartServiceDayOfWeekClientApi = CartServiceDayOfWeekClientApi;
