// @salesstryke/mobile-api/src/com/salesstryke/client/api/CartPromotionMaxClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CartPromotionMaxClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CartPromotionMaxClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cartPromotionMax : CartPromotionMaxClientApi;


        this.cartPromotionMax = new CartPromotionMaxClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CART_PROMOTION_MAX__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_promotion_max/delete");
        public static CART_PROMOTION_MAX__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_promotion_max/getall");
        public static CART_PROMOTION_MAX__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_promotion_max/getbyid");
        public static CART_PROMOTION_MAX__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cart_promotion_max/save");

*/
class CartPromotionMaxClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CartPromotionMax
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PROMOTION_MAX__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '573e7620-ac9f-4208-855c-7ee026559e68', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CART_PROMOTION_MAX__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'cc7eaba6-4163-4523-a52f-06dab3ef9d15', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CartPromotionMax by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PROMOTION_MAX__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'b89bd065-6381-4da4-ba2f-97c39c74b51a', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CART_PROMOTION_MAX__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '9190887a-fea2-4270-a3dd-651f1a9ed3bc', error: error });
                throw error;
            }
        });
    }
}
exports.CartPromotionMaxClientApi = CartPromotionMaxClientApi;
