// @salesstryke/mobile-api/src/com/salesstryke/client/api/CnsEndPointClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsEndPointClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CnsEndPointClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cnsEndPoint : CnsEndPointClientApi;


        this.cnsEndPoint = new CnsEndPointClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CNS_END_POINT__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point/delete");
        public static CNS_END_POINT__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point/getall");
        public static CNS_END_POINT__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point/getbyid");
        public static CNS_END_POINT__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point/save");

*/
class CnsEndPointClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CnsEndPoint
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_END_POINT__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '134c514e-247a-4c2e-9f81-1161bf7afd2d', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CNS_END_POINT__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'e9d957de-20c8-4782-af28-32384a5a8504', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CnsEndPoint by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_END_POINT__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'de40884b-9a22-4895-b4b3-efc3e92670cf', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_END_POINT__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '7cb02a3b-e67c-41a4-8281-5c1dc0521f5d', error: error });
                throw error;
            }
        });
    }
}
exports.CnsEndPointClientApi = CnsEndPointClientApi;
