// @salesstryke/mobile-api/src/com/salesstryke/client/api/CnsEndPointChannelTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsEndPointChannelTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CnsEndPointChannelTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cnsEndPointChannelType : CnsEndPointChannelTypeClientApi;


        this.cnsEndPointChannelType = new CnsEndPointChannelTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CNS_END_POINT_CHANNEL_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point_channel_type/delete");
        public static CNS_END_POINT_CHANNEL_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point_channel_type/getall");
        public static CNS_END_POINT_CHANNEL_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point_channel_type/getbyid");
        public static CNS_END_POINT_CHANNEL_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_end_point_channel_type/save");

*/
class CnsEndPointChannelTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CnsEndPointChannelType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_END_POINT_CHANNEL_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '1bdf6b76-7ac7-4d9a-aa8e-ab5e326cf363', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CNS_END_POINT_CHANNEL_TYPE__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'de7b5f35-2a5c-45b1-a203-cf6e4edeea8b', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CnsEndPointChannelType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_END_POINT_CHANNEL_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '7e1a5e37-07b4-43a1-bc3c-f4481172f1a6', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_END_POINT_CHANNEL_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '6350fd00-4437-43fb-b50d-ede8a5af8518', error: error });
                throw error;
            }
        });
    }
}
exports.CnsEndPointChannelTypeClientApi = CnsEndPointChannelTypeClientApi;
