// @salesstryke/mobile-api/src/com/salesstryke/client/api/CnsSyncStatusTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsSyncStatusTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CnsSyncStatusTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cnsSyncStatusType : CnsSyncStatusTypeClientApi;


        this.cnsSyncStatusType = new CnsSyncStatusTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CNS_SYNC_STATUS_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_sync_status_type/delete");
        public static CNS_SYNC_STATUS_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_sync_status_type/getall");
        public static CNS_SYNC_STATUS_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_sync_status_type/getbyid");
        public static CNS_SYNC_STATUS_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_sync_status_type/save");

*/
class CnsSyncStatusTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CnsSyncStatusType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_SYNC_STATUS_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '06b30671-6858-415a-bfa0-0a4ce85a335e', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CNS_SYNC_STATUS_TYPE__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'f0fe2153-fe2f-42c2-b697-3205d620e53b', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CnsSyncStatusType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_SYNC_STATUS_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'd869633a-a732-4994-a23e-4f978ef6619d', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_SYNC_STATUS_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '72bae679-f3bc-4d47-bb94-9bcc5ed37ed3', error: error });
                throw error;
            }
        });
    }
}
exports.CnsSyncStatusTypeClientApi = CnsSyncStatusTypeClientApi;
