// @salesstryke/mobile-api/src/com/salesstryke/client/api/CnsSegmentClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CnsSegmentClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.CnsSegmentClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public cnsSegment : CnsSegmentClientApi;


        this.cnsSegment = new CnsSegmentClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static CNS_SEGMENT__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_segment/delete");
        public static CNS_SEGMENT__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_segment/getall");
        public static CNS_SEGMENT__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_segment/getbyid");
        public static CNS_SEGMENT__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/cns_segment/save");

*/
class CnsSegmentClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: CnsSegment
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_SEGMENT__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '7b1cedf7-1c06-4003-a534-36433a2a9e89', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.CNS_SEGMENT__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'c5bb7893-479a-42ad-a7b7-83da50aa5f0c', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: CnsSegment by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_SEGMENT__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '2eb38351-4e31-4353-96f8-f04d394851c2', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.CNS_SEGMENT__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '5bf36bfc-0253-4287-8248-b46589648939', error: error });
                throw error;
            }
        });
    }
}
exports.CnsSegmentClientApi = CnsSegmentClientApi;
