// @salesstryke/mobile-api/src/com/salesstryke/client/api/DataArrayClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataArrayClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.DataArrayClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public dataArray : DataArrayClientApi;


        this.dataArray = new DataArrayClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static DATA_ARRAY__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/data_array/delete");
        public static DATA_ARRAY__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/data_array/getall");
        public static DATA_ARRAY__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/data_array/getbyid");
        public static DATA_ARRAY__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/data_array/save");

*/
class DataArrayClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: DataArray
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.DATA_ARRAY__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'f11f0622-3f09-48aa-89e2-87916fcc0d0b', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.DATA_ARRAY__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'df7baf31-48bd-4a85-bb1a-fa8812aabbcb', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: DataArray by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.DATA_ARRAY__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '1cfcc752-c30d-4104-a9f8-f91d9eed494c', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.DATA_ARRAY__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '0263ed56-9499-4995-9bfb-0c9e49ea5f55', error: error });
                throw error;
            }
        });
    }
}
exports.DataArrayClientApi = DataArrayClientApi;
