// @salesstryke/mobile-api/src/com/salesstryke/client/api/DurationTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DurationTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.DurationTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public durationType : DurationTypeClientApi;


        this.durationType = new DurationTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static DURATION_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/duration_type/delete");
        public static DURATION_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/duration_type/getall");
        public static DURATION_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/duration_type/getbyid");
        public static DURATION_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/duration_type/save");

*/
class DurationTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: DurationType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.DURATION_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'b5e372eb-7b6b-4148-8fb2-998e9e19f788', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.DURATION_TYPE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '93a33460-62eb-43d9-beaa-3d0b69903ba5', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: DurationType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.DURATION_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '3376c73b-dc7e-47b8-93d1-c86359717203', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.DURATION_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '072334b3-fc3d-4386-9bbe-d42b6558f52a', error: error });
                throw error;
            }
        });
    }
}
exports.DurationTypeClientApi = DurationTypeClientApi;
