// @salesstryke/mobile-api/src/com/salesstryke/client/api/EmailAddressClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailAddressClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.EmailAddressClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public emailAddress : EmailAddressClientApi;


        this.emailAddress = new EmailAddressClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static EMAIL_ADDRESS__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/email_address/delete");
        public static EMAIL_ADDRESS__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/email_address/getall");
        public static EMAIL_ADDRESS__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/email_address/getbyid");
        public static EMAIL_ADDRESS__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/email_address/save");

*/
class EmailAddressClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: EmailAddress
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.EMAIL_ADDRESS__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'eb1a9ff4-b8f4-41db-a480-9531caa7f387', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.EMAIL_ADDRESS__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '249c403b-6b2b-4e69-a1df-578394c12ff4', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: EmailAddress by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.EMAIL_ADDRESS__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '14ee4ed1-7150-413c-b975-284ecd0ad447', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.EMAIL_ADDRESS__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'c0b33cfc-0223-43e8-9080-108e2f3b04ff', error: error });
                throw error;
            }
        });
    }
}
exports.EmailAddressClientApi = EmailAddressClientApi;
