// @salesstryke/mobile-api/src/com/salesstryke/client/api/I18nCategoryClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.I18nCategoryClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const lang_1 = require("../../lang");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.I18nCategoryClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public I18nCategory : I18nCategoryClientApi;


        this.I18nCategory = new I18nCategoryClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static I18N_CATEGORY__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_category/delete");
        public static I18N_CATEGORY__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_category/getall");
        public static I18N_CATEGORY__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_category/getbyid");
        public static I18N_CATEGORY__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_category/save");

*/
class I18nCategoryClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: I18nCategory
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.I18N_CATEGORY__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.delete.name, 'f0d26883-a573-4e7e-9616-157db670c010', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.I18N_CATEGORY__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getAll.name, '213bf213-285b-4639-99b6-d3a4362031c3', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Get Entity: I18nCategory by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.I18N_CATEGORY__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getById.name, 'c0a02417-eb6d-425c-82bb-f3b694cb4d56', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.I18N_CATEGORY__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.save.name, 'd4d7e1c9-05dd-4dd3-936a-0f51ef556413', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
}
exports.I18nCategoryClientApi = I18nCategoryClientApi;
