// @salesstryke/mobile-api/src/com/salesstryke/client/api/I18nLocaleTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.I18nLocaleTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const lang_1 = require("../../lang");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.I18nLocaleTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public I18nLocaleType : I18nLocaleTypeClientApi;


        this.I18nLocaleType = new I18nLocaleTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static I18N_LOCALE_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_locale_type/delete");
        public static I18N_LOCALE_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_locale_type/getall");
        public static I18N_LOCALE_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_locale_type/getbyid");
        public static I18N_LOCALE_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/i18n_locale_type/save");

*/
class I18nLocaleTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: I18nLocaleType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.I18N_LOCALE_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.delete.name, '012c917c-6d3a-4335-b861-57e4a47dbbba', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.I18N_LOCALE_TYPE__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getAll.name, 'a3887cd0-50bf-437e-bbc1-4474f2c11552', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Get Entity: I18nLocaleType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.I18N_LOCALE_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getById.name, 'fa31e93a-d796-4e72-a617-57f848dbf88d', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.I18N_LOCALE_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.save.name, '38f9773a-df17-4636-8a48-21e2a5493449', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
}
exports.I18nLocaleTypeClientApi = I18nLocaleTypeClientApi;
