// @salesstryke/mobile-api/src/com/salesstryke/client/api/ImportNotificationUserSyncClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportNotificationUserSyncClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const lang_1 = require("../../lang");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ImportNotificationUserSyncClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public importNotificationUserSync : ImportNotificationUserSyncClientApi;


        this.importNotificationUserSync = new ImportNotificationUserSyncClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static IMPORT_NOTIFICATION_USER_SYNC__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_notification_user_sync/delete");
        public static IMPORT_NOTIFICATION_USER_SYNC__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_notification_user_sync/getall");
        public static IMPORT_NOTIFICATION_USER_SYNC__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_notification_user_sync/getbyid");
        public static IMPORT_NOTIFICATION_USER_SYNC__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_notification_user_sync/save");

*/
class ImportNotificationUserSyncClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ImportNotificationUserSync
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_NOTIFICATION_USER_SYNC__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.delete.name, 'ceee272f-d6da-4db5-88c4-8263af55649a', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.IMPORT_NOTIFICATION_USER_SYNC__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getAll.name, '00b8d156-51dd-4e71-b23e-c4969ef9d19c', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Get Entity: ImportNotificationUserSync by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_NOTIFICATION_USER_SYNC__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getById.name, 'ed785b76-e6f1-45c1-a8c6-8b8abf8a4deb', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_NOTIFICATION_USER_SYNC__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.save.name, 'a33b90ad-4f23-42a6-85e2-ba2bc50e6b15', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
}
exports.ImportNotificationUserSyncClientApi = ImportNotificationUserSyncClientApi;
