// @salesstryke/mobile-api/src/com/salesstryke/client/api/ImportDataAttributeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataAttributeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const lang_1 = require("../../lang");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ImportDataAttributeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public importDataAttribute : ImportDataAttributeClientApi;


        this.importDataAttribute = new ImportDataAttributeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static IMPORT_DATA_ATTRIBUTE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute/delete");
        public static IMPORT_DATA_ATTRIBUTE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute/getall");
        public static IMPORT_DATA_ATTRIBUTE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute/getbyid");
        public static IMPORT_DATA_ATTRIBUTE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute/save");

*/
class ImportDataAttributeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ImportDataAttribute
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.delete.name, 'b5423f4e-8b21-4545-9846-0ccd6e6855cc', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getAll.name, 'fc8c4a09-eb94-490b-80e1-fe9de8e4676a', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Get Entity: ImportDataAttribute by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getById.name, 'cc62c325-71b5-4794-b519-a74bf8b1f836', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.save.name, 'dbc5f8ed-b346-4de9-ae3b-8f47194db2e2', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
}
exports.ImportDataAttributeClientApi = ImportDataAttributeClientApi;
