// @salesstryke/mobile-api/src/com/salesstryke/client/api/ImportDataAttributeTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportDataAttributeTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const lang_1 = require("../../lang");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ImportDataAttributeTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public importDataAttributeType : ImportDataAttributeTypeClientApi;


        this.importDataAttributeType = new ImportDataAttributeTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static IMPORT_DATA_ATTRIBUTE_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute_type/delete");
        public static IMPORT_DATA_ATTRIBUTE_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute_type/getall");
        public static IMPORT_DATA_ATTRIBUTE_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute_type/getbyid");
        public static IMPORT_DATA_ATTRIBUTE_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/import_data_attribute_type/save");

*/
class ImportDataAttributeTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ImportDataAttributeType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.delete.name, '8f715cc9-fdc8-4f9f-8f24-692c69fa83ae', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE_TYPE__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getAll.name, '98a5595e-db87-4266-829f-c338d276697f', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Get Entity: ImportDataAttributeType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getById.name, '70b2112a-00ca-4dd5-92b4-8573f4a3da57', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.IMPORT_DATA_ATTRIBUTE_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.save.name, '808857bf-334d-424d-ba36-619eddc77efc', error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
}
exports.ImportDataAttributeTypeClientApi = ImportDataAttributeTypeClientApi;
