// @salesstryke/mobile-api/src/com/salesstryke/client/api/KmlLayerClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KmlLayerClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.KmlLayerClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public kmlLayer : KmlLayerClientApi;


        this.kmlLayer = new KmlLayerClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static KML_LAYER__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/kml_layer/delete");
        public static KML_LAYER__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/kml_layer/getall");
        public static KML_LAYER__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/kml_layer/getbyid");
        public static KML_LAYER__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/kml_layer/save");

*/
class KmlLayerClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: KmlLayer
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.KML_LAYER__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'd63d590d-c43e-4902-b95e-4531347abde4', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.KML_LAYER__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'a90e006e-e1ee-4bd9-b724-d85e05a27582', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: KmlLayer by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.KML_LAYER__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '6060541a-3c54-460d-9485-84da24f6349d', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.KML_LAYER__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'a3e9170d-d806-4310-9070-a83b60085cbf', error: error });
                throw error;
            }
        });
    }
}
exports.KmlLayerClientApi = KmlLayerClientApi;
