// @salesstryke/mobile-api/src/com/salesstryke/client/api/PandaDocWorkspaceClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PandaDocWorkspaceClientApi = void 0;
const tslib_1 = require("tslib");
const endpoint_1 = require("../../endpoint");
const lang_1 = require("../../lang");
const log_1 = require("../../utils/log");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const ApiBase_1 = require("./ApiBase");
const NAMESPACE = "com.salesstryke.client.api.PandaDocWorkspaceClientApi";
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
// NOTE: Report api is in workspace and should not be here.
// TODO: Review Do we need this and if so then need to complete server and monogo api.
class PandaDocWorkspaceClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
        this.delete = this.delete.bind(this);
    }
    /**
     * Return PandaDocTemplate
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PANDADOC_WORKSPACE_GET_BY_ID
                })
                    .addEntity(request)
                    .request();
                return (_c = (_b = (_a = response_api.getJsonApiDocument().getDataAs()[0]) !== null && _a !== void 0 ? _a : null) !== null && _b !== void 0 ? _b : null) !== null && _c !== void 0 ? _c : null;
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getById.name, "d5fe24e9-15b2-4d63-a099-7d4b93fb0641", error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Return Array<PandaDocWorkspace>
     * @returns
     */
    getAll(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PANDADOC_WORKSPACE_GET_ALL
                })
                    .addEntity(request)
                    .request();
                return response_api.getJsonApiDocument().getDataAs();
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.getAll.name, "eb99fb76-4f95-4403-ae45-30d109db14d2", error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Return PandaDocWorkspace
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PANDADOC_WORKSPACE_SAVE
                })
                    .addEntity(request)
                    .request();
                return (_c = (_b = (_a = response_api.getJsonApiDocument().getDataAs()[0]) !== null && _a !== void 0 ? _a : null) !== null && _b !== void 0 ? _b : null) !== null && _c !== void 0 ? _c : null;
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.save.name, "c6f29894-b989-41bf-bbdb-7683aeff914c", error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
    /**
     * Return PandaDocWorkspace
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a, _b, _c;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PANDADOC_WORKSPACE_DELETE
                })
                    .addEntity(request)
                    .request();
                return (_c = (_b = (_a = response_api.getJsonApiDocument().getDataAs()[0]) !== null && _a !== void 0 ? _a : null) !== null && _b !== void 0 ? _b : null) !== null && _c !== void 0 ? _c : null;
            }
            catch (error_any) {
                let error = error_any;
                let errorBase = new lang_1.ErrorBase(error.message, NAMESPACE, this.delete.name, "403074a2-b290-4942-9d4d-b65d4f251ae5", error);
                if (LOGGER.isErrorEnabled()) {
                    LOGGER.logErrorBase(errorBase);
                }
                throw (error_any);
            }
        });
    }
}
exports.PandaDocWorkspaceClientApi = PandaDocWorkspaceClientApi;
