// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductServiceDayOfWeekRouteClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductServiceDayOfWeekRouteClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ProductServiceDayOfWeekRouteClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productServiceDayOfWeekRoute : ProductServiceDayOfWeekRouteClientApi;


        this.productServiceDayOfWeekRoute = new ProductServiceDayOfWeekRouteClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_day_of_week_route/delete");
        public static PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_day_of_week_route/getall");
        public static PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_day_of_week_route/getbyid");
        public static PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_day_of_week_route/save");

*/
class ProductServiceDayOfWeekRouteClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductServiceDayOfWeekRoute
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '00bfc8de-b81a-4ee4-8ed6-4cb75892c9a2', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'a4eb3867-bea8-4229-b5f8-2a02da4619f7', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductServiceDayOfWeekRoute by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '81c9c512-b100-4e03-9db7-5a4d8e292ef6', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_DAY_OF_WEEK_ROUTE__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'd2e4553d-fb4b-42d3-826a-a4e79e7ada6c', error: error });
                throw error;
            }
        });
    }
}
exports.ProductServiceDayOfWeekRouteClientApi = ProductServiceDayOfWeekRouteClientApi;
