// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductStartServiceMinimumDaysFromPurchaseClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductStartServiceMinimumDaysFromPurchaseClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.ProductStartServiceMinimumDaysFromPurchaseClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productStartServiceMinimumDaysFromPurchase : ProductStartServiceMinimumDaysFromPurchaseClientApi;


        this.productStartServiceMinimumDaysFromPurchase = new ProductStartServiceMinimumDaysFromPurchaseClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase/delete");
        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase/getall");
        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase/getbyid");
        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase/save");

*/
class ProductStartServiceMinimumDaysFromPurchaseClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductStartServiceMinimumDaysFromPurchase
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '84eba083-e0d6-4070-8b34-fe5f9d15b16a', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'c0ff003a-e6b4-44d2-bb04-cfac2da76ec9', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductStartServiceMinimumDaysFromPurchase by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '41365874-5636-410a-981f-6ceb298d1995', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '58638320-5cca-463a-8dbc-912730918519', error: error });
                throw error;
            }
        });
    }
}
exports.ProductStartServiceMinimumDaysFromPurchaseClientApi = ProductStartServiceMinimumDaysFromPurchaseClientApi;
