// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductTaxesClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductTaxesClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.ProductTaxesClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productTaxes : ProductTaxesClientApi;


        this.productTaxes = new ProductTaxesClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_TAXES__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_taxes/delete");
        public static PRODUCT_TAXES__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_taxes/getall");
        public static PRODUCT_TAXES__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_taxes/getbyid");
        public static PRODUCT_TAXES__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_taxes/save");

*/
class ProductTaxesClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductTaxes
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_TAXES__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '81c2c5bb-d4db-4974-8e80-9eb6a45a429f', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_TAXES__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'ab7e3d58-1544-4f5c-bb5d-a0b5211b9187', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductTaxes by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_TAXES__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'edb6a7ed-cc0c-457f-9eff-8ed0337dda96', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_TAXES__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '726803cf-1889-470a-9206-337011b24421', error: error });
                throw error;
            }
        });
    }
}
exports.ProductTaxesClientApi = ProductTaxesClientApi;
