// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductServiceViewStatusAuditClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductServiceViewStatusAuditClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ProductServiceViewStatusAuditClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productServiceViewStatusAudit : ProductServiceViewStatusAuditClientApi;


        this.productServiceViewStatusAudit = new ProductServiceViewStatusAuditClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_SERVICE_VIEW_STATUS_AUDIT__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_view_status_audit/delete");
        public static PRODUCT_SERVICE_VIEW_STATUS_AUDIT__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_view_status_audit/getall");
        public static PRODUCT_SERVICE_VIEW_STATUS_AUDIT__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_view_status_audit/getbyid");
        public static PRODUCT_SERVICE_VIEW_STATUS_AUDIT__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_service_view_status_audit/save");

*/
class ProductServiceViewStatusAuditClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductServiceViewStatusAudit
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_VIEW_STATUS_AUDIT__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '1ef49ad5-0af2-4c21-950d-85214f375f65', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_VIEW_STATUS_AUDIT__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'dff122cf-384b-4f32-8c88-6b45442197c6', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductServiceViewStatusAudit by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_VIEW_STATUS_AUDIT__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '352bd15c-374a-4106-b8c7-6a4e294e251d', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_SERVICE_VIEW_STATUS_AUDIT__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '2c3cce7a-7b77-4775-8c3b-c609804eb4a9', error: error });
                throw error;
            }
        });
    }
}
exports.ProductServiceViewStatusAuditClientApi = ProductServiceViewStatusAuditClientApi;
