// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductPricingRuleClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPricingRuleClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ProductPricingRuleClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productPricingRule : ProductPricingRuleClientApi;


        this.productPricingRule = new ProductPricingRuleClientApi(this.httpClient);


    Copy to EndPoint.ts

        public static PRODUCT_PRICING_RULE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_rule/delete");
        public static PRODUCT_PRICING_RULE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_rule/getall");
        public static PRODUCT_PRICING_RULE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_rule/getbyid");
        public static PRODUCT_PRICING_RULE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_rule/save");

*/
class ProductPricingRuleClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductPricingRule
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RULE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '3a547e56-06b3-4cb7-b8a3-29ec45c372e9', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RULE__GET_ALL }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '90accd96-95de-4c90-8197-43e6bcc465ba', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductPricingRule by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RULE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '6935048f-0c02-4047-bc07-b21e7259b1cc', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RULE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'ef601ed9-f35b-475f-9733-40ba549d7c61', error: error });
                throw error;
            }
        });
    }
}
exports.ProductPricingRuleClientApi = ProductPricingRuleClientApi;
