// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductColorRouteClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductColorRouteClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ProductColorRouteClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productColorRoute : ProductColorRouteClientApi;


        this.productColorRoute = new ProductColorRouteClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_COLOR_ROUTE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_color_route/delete");
        public static PRODUCT_COLOR_ROUTE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_color_route/getall");
        public static PRODUCT_COLOR_ROUTE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_color_route/getbyid");
        public static PRODUCT_COLOR_ROUTE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_color_route/save");

*/
class ProductColorRouteClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductColorRoute
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_COLOR_ROUTE__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '0309baad-44aa-4cb8-92e8-02bd5a14b6e4', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_COLOR_ROUTE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '7e390daf-e147-4b0f-9473-ddcfc42aba3b', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductColorRoute by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_COLOR_ROUTE__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'ec85817a-e2c4-4d9f-b0da-50d8354d7e26', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_COLOR_ROUTE__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '82ea2379-fb26-4dea-9114-b3cabbcbe590', error: error });
                throw error;
            }
        });
    }
}
exports.ProductColorRouteClientApi = ProductColorRouteClientApi;
