// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productStartServiceMinimumDaysFromPurchaseKmlLocation : ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi;


        this.productStartServiceMinimumDaysFromPurchaseKmlLocation = new ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase_kml_locations/delete");
        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase_kml_locations/getall");
        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase_kml_locations/getbyid");
        public static PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_start_service_minimum_days_from_purchase_kml_locations/save");

*/
class ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductStartServiceMinimumDaysFromPurchaseKmlLocation
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'bfe73408-0d67-4972-88c6-017a28a88e66', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '87145678-c2c6-40d5-b197-20b45ce3d9ec', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductStartServiceMinimumDaysFromPurchaseKmlLocation by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'b5b23b35-85a7-46d6-a05a-241cdfc09fe6', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_START_SERVICE_MINIMUM_DAYS_FROM_PURCHASE_KML_LOCATIONS__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'bc5e35e8-b42a-4783-9cb9-f071ad061053', error: error });
                throw error;
            }
        });
    }
}
exports.ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi = ProductStartServiceMinimumDaysFromPurchaseKmlLocationClientApi;
