// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductCategoryImageRouteClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductCategoryImageRouteClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ProductCategoryImageRouteClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productCategoryImageRoute : ProductCategoryImageRouteClientApi;


        this.productCategoryImageRoute = new ProductCategoryImageRouteClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_CATEGORY_IMAGE_ROUTE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_category_image_route/delete");
        public static PRODUCT_CATEGORY_IMAGE_ROUTE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_category_image_route/getall");
        public static PRODUCT_CATEGORY_IMAGE_ROUTE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_category_image_route/getbyid");
        public static PRODUCT_CATEGORY_IMAGE_ROUTE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_category_image_route/save");

*/
class ProductCategoryImageRouteClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductCategoryImageRoute
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_CATEGORY_IMAGE_ROUTE__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'fbbb78f0-3449-4253-af44-99d939ddaed4', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_CATEGORY_IMAGE_ROUTE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '44466104-ec58-42d3-a082-459c2954c09a', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductCategoryImageRoute by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_CATEGORY_IMAGE_ROUTE__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'aca8d7cc-1c05-4f9b-8bd9-d215e9e75c16', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_CATEGORY_IMAGE_ROUTE__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'eaf6a94a-8cbc-47ce-bdfa-bbab4067b3f9', error: error });
                throw error;
            }
        });
    }
}
exports.ProductCategoryImageRouteClientApi = ProductCategoryImageRouteClientApi;
