// @salesstryke/mobile-api/src/com/salesstryke/client/api/ProductPricingRecurringPeriodTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPricingRecurringPeriodTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.ProductPricingRecurringPeriodTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public productPricingRecurringPeriodType : ProductPricingRecurringPeriodTypeClientApi;


        this.productPricingRecurringPeriodType = new ProductPricingRecurringPeriodTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PRODUCT_PRICING_RECURRING_PERIOD_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_recurring_period_type/delete");
        public static PRODUCT_PRICING_RECURRING_PERIOD_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_recurring_period_type/getall");
        public static PRODUCT_PRICING_RECURRING_PERIOD_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_recurring_period_type/getbyid");
        public static PRODUCT_PRICING_RECURRING_PERIOD_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/product_pricing_recurring_period_type/save");

*/
class ProductPricingRecurringPeriodTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ProductPricingRecurringPeriodType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RECURRING_PERIOD_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'a7b51802-32c9-46b2-bbbd-0dec37cd2ab9', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RECURRING_PERIOD_TYPE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'eb5a9107-eda3-45e2-ae14-7fa39b28dc29', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ProductPricingRecurringPeriodType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RECURRING_PERIOD_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '366f99af-ff28-4bdd-8696-3383631650c8', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PRODUCT_PRICING_RECURRING_PERIOD_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '01798bb1-d73a-4a18-be29-797eabbb87c9', error: error });
                throw error;
            }
        });
    }
}
exports.ProductPricingRecurringPeriodTypeClientApi = ProductPricingRecurringPeriodTypeClientApi;
