// @salesstryke/mobile-api/src/com/salesstryke/client/api/PromotionClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromotionClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.PromotionClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public promotion : PromotionClientApi;


        this.promotion = new PromotionClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static PROMOTION__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/promotion/delete");
        public static PROMOTION__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/promotion/getall");
        public static PROMOTION__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/promotion/getbyid");
        public static PROMOTION__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/promotion/save");

*/
class PromotionClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: Promotion
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PROMOTION__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '42d200b7-7c4b-4f8d-a653-4e58b8a56941', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.PROMOTION__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '13c842dd-488a-44fa-b657-6ea68c497e60', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: Promotion by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PROMOTION__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '709d08cf-bcc9-4b77-91e2-96889ec74082', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.PROMOTION__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '06694126-5604-43c6-a178-f7b3f69ac0f4', error: error });
                throw error;
            }
        });
    }
}
exports.PromotionClientApi = PromotionClientApi;
