// @salesstryke/mobile-api/src/com/salesstryke/client/api/RouteClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.RouteClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public route : RouteClientApi;


        this.route = new RouteClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static ROUTE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/route/delete");
        public static ROUTE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/route/getall");
        public static ROUTE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/route/getbyid");
        public static ROUTE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/route/save");

*/
class RouteClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: Route
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.ROUTE__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '6cf9781c-a2b6-4d48-8ff3-94d731255817', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.ROUTE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '633973ad-92a9-4ce0-b1e3-f76874a802d9', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: Route by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.ROUTE__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'f0b78a1f-8ee3-4c64-9065-8d3c50d8ae5d', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.ROUTE__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '632d0509-f850-45ba-8eee-1a293f38c1e2', error: error });
                throw error;
            }
        });
    }
}
exports.RouteClientApi = RouteClientApi;
