// @salesstryke/mobile-api/src/com/salesstryke/client/api/RuleBuilderClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleBuilderClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.RuleBuilderClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public ruleBuilder : RuleBuilderClientApi;


        this.ruleBuilder = new RuleBuilderClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static RULE_BUILDER__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/rule_builder/delete");
        public static RULE_BUILDER__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/rule_builder/getall");
        public static RULE_BUILDER__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/rule_builder/getbyid");
        public static RULE_BUILDER__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/rule_builder/save");

*/
class RuleBuilderClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: RuleBuilder
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.RULE_BUILDER__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'cba8dab4-ac57-412e-91de-987fa9abf13d', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.RULE_BUILDER__GET_ALL, jsonApiQuery: jsonApiQuery }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'cc576f1a-0a68-48cc-9d63-2721df67b65c', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: RuleBuilder by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.RULE_BUILDER__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '7f7d3cf8-36bf-4ca2-98b2-e71e188ffd29', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.RULE_BUILDER__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '97ffc86d-be13-459e-9b9b-14ce70e6c193', error: error });
                throw error;
            }
        });
    }
}
exports.RuleBuilderClientApi = RuleBuilderClientApi;
