// @salesstryke/mobile-api/src/com/salesstryke/client/api/ServiceGeoClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceGeoClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.ServiceGeoClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public serviceGeo : ServiceGeoClientApi;


        this.serviceGeo = new ServiceGeoClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SERVICE_GEO__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/service_geo/delete");
        public static SERVICE_GEO__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/service_geo/getall");
        public static SERVICE_GEO__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/service_geo/getbyid");
        public static SERVICE_GEO__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/service_geo/save");

*/
class ServiceGeoClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: ServiceGeo
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SERVICE_GEO__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '80080c1a-4c7a-41c8-a2cf-e985d87a1ce5', error: error });
                throw error;
            }
        });
    }
    getAll() {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.callEndPointUrlAndReturnEntityArray({ endPointUrl: endpoint_1.EndPoint.SERVICE_GEO__GET_ALL }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '522cb118-2e30-460e-bec3-df8cec2786fd', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: ServiceGeo by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SERVICE_GEO__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '4a8a1fba-740b-4198-91c8-975ff035d939', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SERVICE_GEO__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'e0d08026-6f43-4e3b-ac52-057f0d3792d2', error: error });
                throw error;
            }
        });
    }
}
exports.ServiceGeoClientApi = ServiceGeoClientApi;
