// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingNotificationClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingNotificationClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingNotificationClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public settingNotification : SettingNotificationClientApi;


        this.settingNotification = new SettingNotificationClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING_NOTIFICATION__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_notification/delete");
        public static SETTING_NOTIFICATION__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_notification/getall");
        public static SETTING_NOTIFICATION__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_notification/getbyid");
        public static SETTING_NOTIFICATION__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_notification/save");

*/
class SettingNotificationClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: SettingNotification
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_NOTIFICATION__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '625d097c-652b-438b-a9e1-b595594f07ee', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING_NOTIFICATION__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'a2e7f167-60a2-4c42-848a-5c19f363aeb3', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: SettingNotification by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_NOTIFICATION__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '68a1c773-dc4d-4638-abdf-0dd52fbc686d', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_NOTIFICATION__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '21488d21-b286-4e51-9cc0-6c5707eaceed', error: error });
                throw error;
            }
        });
    }
}
exports.SettingNotificationClientApi = SettingNotificationClientApi;
