// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public setting : SettingClientApi;


        this.setting = new SettingClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting/delete");
        public static SETTING__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting/getall");
        public static SETTING__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting/getbyid");
        public static SETTING__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting/save");

*/
class SettingClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: Setting
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'ef94cbb9-81f5-4300-8ec0-dd30b0291475', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '7e365619-e418-4ec3-88d0-01eca69b1376', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: Setting by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '422d5aa8-d90b-4091-9d25-32968512dfe0', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '4fa3e040-8001-446e-8482-ad7575483382', error: error });
                throw error;
            }
        });
    }
}
exports.SettingClientApi = SettingClientApi;
