// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingWorkspaceBPSClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceBPSClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingWorkspaceBPSClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public settingWorkspaceBPS : SettingWorkspaceBPSClientApi;


        this.settingWorkspaceBPS = new SettingWorkspaceBPSClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING_WORKSPACE_B_P_S__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_b_p_s/delete");
        public static SETTING_WORKSPACE_B_P_S__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_b_p_s/getall");
        public static SETTING_WORKSPACE_B_P_S__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_b_p_s/getbyid");
        public static SETTING_WORKSPACE_B_P_S__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_b_p_s/save");

*/
class SettingWorkspaceBPSClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: SettingWorkspaceBPS
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_B_P_S__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '4e7a0fb8-e386-4231-949d-6d97344d7e3e', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_B_P_S__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '39437fa3-0241-4430-9874-f566b0be12af', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: SettingWorkspaceBPS by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_B_P_S__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '0828d523-5ddc-41f3-a1b9-73e32f5ec264', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_B_P_S__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '8b8ac5e7-5702-470d-914e-32fcc9550b2a', error: error });
                throw error;
            }
        });
    }
}
exports.SettingWorkspaceBPSClientApi = SettingWorkspaceBPSClientApi;
