// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingWorkspaceOSSClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceOSSClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingWorkspaceOSSClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public settingWorkspaceOSS : SettingWorkspaceOSSClientApi;


        this.settingWorkspaceOSS = new SettingWorkspaceOSSClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING_WORKSPACE_O_S_S__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_o_s_s/delete");
        public static SETTING_WORKSPACE_O_S_S__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_o_s_s/getall");
        public static SETTING_WORKSPACE_O_S_S__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_o_s_s/getbyid");
        public static SETTING_WORKSPACE_O_S_S__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_o_s_s/save");

*/
class SettingWorkspaceOSSClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: SettingWorkspaceOSS
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_O_S_S__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '0cbf8d6d-3d96-4ecb-9c15-96573d1f34f3', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_O_S_S__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'e8b8a6eb-e643-4507-a4f8-ae8e694fe7b2', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: SettingWorkspaceOSS by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_O_S_S__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'f9687223-d5d2-41c8-a0a0-9425ff49076c', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_O_S_S__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '2d3738df-3a00-4c9a-9ac1-911429535a43', error: error });
                throw error;
            }
        });
    }
}
exports.SettingWorkspaceOSSClientApi = SettingWorkspaceOSSClientApi;
