// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingIntegrationTypeClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingIntegrationTypeClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingIntegrationTypeClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public settingIntegrationType : SettingIntegrationTypeClientApi;


        this.settingIntegrationType = new SettingIntegrationTypeClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING_INTEGRATION_TYPE__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_integration_type/delete");
        public static SETTING_INTEGRATION_TYPE__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_integration_type/getall");
        public static SETTING_INTEGRATION_TYPE__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_integration_type/getbyid");
        public static SETTING_INTEGRATION_TYPE__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_integration_type/save");

*/
class SettingIntegrationTypeClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: SettingIntegrationType
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_INTEGRATION_TYPE__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'b31a3f42-ef11-4d2f-bc08-548084e8adc8', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING_INTEGRATION_TYPE__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: 'f276d78a-b622-4737-b563-547c4a2cd823', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: SettingIntegrationType by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_INTEGRATION_TYPE__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '1bd38f66-4532-4396-bddc-5ca264606973', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_INTEGRATION_TYPE__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '4e80a2f0-a3c2-4b77-acd2-7becb4ed8370', error: error });
                throw error;
            }
        });
    }
}
exports.SettingIntegrationTypeClientApi = SettingIntegrationTypeClientApi;
