// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingWorkspaceCALClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingWorkspaceCALClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingWorkspaceCALClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public settingWorkspaceCAL : SettingWorkspaceCALClientApi;


        this.settingWorkspaceCAL = new SettingWorkspaceCALClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING_WORKSPACE_C_A_L__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_c_a_l/delete");
        public static SETTING_WORKSPACE_C_A_L__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_c_a_l/getall");
        public static SETTING_WORKSPACE_C_A_L__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_c_a_l/getbyid");
        public static SETTING_WORKSPACE_C_A_L__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_workspace_c_a_l/save");

*/
class SettingWorkspaceCALClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: SettingWorkspaceCAL
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_C_A_L__DELETE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: 'f6dbf0e6-370f-4718-9e3c-ad6faf161b01', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_C_A_L__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '091c413d-a56a-4186-9481-4f5ba048a465', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: SettingWorkspaceCAL by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_C_A_L__GET_BY_ID }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: '558054fa-a972-40e8-9b6b-ff3a5b732fba', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_WORKSPACE_C_A_L__SAVE }));
            }
            catch (error_any) {
                const error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'cbe8a682-5888-406f-ae08-9946cae9d270', error: error });
                throw error;
            }
        });
    }
}
exports.SettingWorkspaceCALClientApi = SettingWorkspaceCALClientApi;
