// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingTokenClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingTokenClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingTokenClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public settingToken : SettingTokenClientApi;


        this.settingToken = new SettingTokenClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING_TOKEN__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_token/delete");
        public static SETTING_TOKEN__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_token/getall");
        public static SETTING_TOKEN__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_token/getbyid");
        public static SETTING_TOKEN__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_token/save");

*/
class SettingTokenClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: SettingToken
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_TOKEN__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '93a28824-f918-4f85-82f4-e3cdc68d6f08', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING_TOKEN__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '68d51f1f-6a30-4eb7-88e7-6525a5a7e5e0', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: SettingToken by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_TOKEN__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'a30affbf-d57b-46e5-9ff6-b29198ea5290', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_TOKEN__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'ea91cf64-fc3b-4923-acfe-d09fb465898a', error: error });
                throw error;
            }
        });
    }
}
exports.SettingTokenClientApi = SettingTokenClientApi;
