// @salesstryke/mobile-api/src/com/salesstryke/client/api/SettingOrganizationClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SettingOrganizationClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const NAMESPACE = 'com.salesstryke.client.api.SettingOrganizationClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public settingOrganization : SettingOrganizationClientApi;


        this.settingOrganization = new SettingOrganizationClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static SETTING_ORGANIZATION__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_organization/delete");
        public static SETTING_ORGANIZATION__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_organization/getall");
        public static SETTING_ORGANIZATION__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_organization/getbyid");
        public static SETTING_ORGANIZATION__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/setting_organization/save");

*/
class SettingOrganizationClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: SettingOrganization
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_ORGANIZATION__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '5936c1a3-11ed-4ec5-9ab2-ae21c617a775', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            var _a;
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.SETTING_ORGANIZATION__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return (_a = response_api.getJsonApiDocument().getDataAs()) !== null && _a !== void 0 ? _a : null;
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '3e1101d3-b232-4bd9-b8f4-9fe1aa0cb762', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: SettingOrganization by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_ORGANIZATION__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'f0ab03aa-d07c-46a6-be01-ed7d56efcd67', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.SETTING_ORGANIZATION__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: 'a1539703-d48f-41dc-8995-645dcf12540a', error: error });
                throw error;
            }
        });
    }
}
exports.SettingOrganizationClientApi = SettingOrganizationClientApi;
