// @salesstryke/mobile-api/src/com/salesstryke/client/api/TaxClientApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxClientApi = void 0;
const tslib_1 = require("tslib");
const ApiBase_1 = require("./ApiBase");
const ClientToServerApiCall_1 = require("../call/ClientToServerApiCall");
const endpoint_1 = require("../../endpoint");
const log_1 = require("../../utils/log");
const ProcessApiDataListUtil_1 = require("../../jsonApi/util/ProcessApiDataListUtil");
const NAMESPACE = 'com.salesstryke.client.api.TaxClientApi';
const LOGGER = log_1.LogManager.getLogger(NAMESPACE);
/*

    Make sure this file is added to the correct index.ts

    Add To SalesStrykeClientApi:
        public tax : TaxClientApi;


        this.tax = new TaxClientApi(this.httpClient);
    

    Copy to EndPoint.ts

        public static TAX__DELETE:     EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/tax/delete");
        public static TAX__GET_ALL:    EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/tax/getall");
        public static TAX__GET_BY_ID : EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/tax/getbyid");
        public static TAX__SAVE:       EndPointUrl = new EndPointUrl(HttpMethodType.POST, "/tax/save");

*/
class TaxClientApi extends ApiBase_1.ApiBase {
    constructor(httpClient) {
        super(httpClient);
        this.delete = this.delete.bind(this);
        this.getAll = this.getAll.bind(this);
        this.getById = this.getById.bind(this);
        this.save = this.save.bind(this);
    }
    /**
     * Delete Entity: Tax
     * @param request
     * @returns
     */
    delete(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.TAX__DELETE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.delete.name, uuid: '1008854f-2a66-48d9-a2f2-2eebdc362865', error: error });
                throw error;
            }
        });
    }
    getAll(jsonApiQuery) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                let response_api = yield new ClientToServerApiCall_1.ClientToServerApiCall({
                    httpClient: this.getHttpClient(), endPointUrl: endpoint_1.EndPoint.TAX__GET_ALL
                })
                    .setJsonApiQuery(jsonApiQuery)
                    .request();
                return ProcessApiDataListUtil_1.ProcessApiDataListUtil.parseData(response_api.getJsonApiDocument());
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getAll.name, uuid: '25548ea0-01e0-4e34-9a22-1512d1db8b1a', error: error });
                throw error;
            }
        });
    }
    /**
     * Get Entity: Tax by Id
     * @param request
     * @returns
     */
    getById(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.TAX__GET_BY_ID }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.getById.name, uuid: 'de07f12b-2b6a-4967-8ff3-6daf67ca961e', error: error });
                throw error;
            }
        });
    }
    /**
     * Save Entity: {clientApiSentitySimpleNameimpleName}
     * @param request
     * @returns
     */
    save(request) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                return (this.addEntityAndCallEndPointUrlAndReturnEntity({ entity: request, endPointUrl: endpoint_1.EndPoint.TAX__SAVE }));
            }
            catch (error_any) {
                let error = error_any;
                LOGGER.logError({ message: error.message, methodName: this.save.name, uuid: '5998bdf4-44c8-4331-927d-6a1dd310f599', error: error });
                throw error;
            }
        });
    }
}
exports.TaxClientApi = TaxClientApi;
