// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/record/notification/EmailNotificationSelectedData.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailNotificationSelectedData = void 0;
const ClassFor_1 = require("../../../../data/ClassFor");
const JsonApiRecordBase_1 = require("../../../../jsonApi/JsonApiRecordBase");
const HashCodeUtil_1 = require("../../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._nameIsMutated = "_nameIsMutated";
        this._valueIsMutated = "_valueIsMutated";
        this.jsonType = "jsonType";
        this.name = "name";
        this.value = "value";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._nameIsMutated = "_nameIsMutated";
        this._valueIsMutated = "_valueIsMutated";
        this.jsonType = "jsonType";
        this.name = "name";
        this.value = "value";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.record.notification.EmailNotificationSelectedData";
        this.mongDataModeName = "EmailNotificationSelectedData";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class EmailNotificationSelectedData extends JsonApiRecordBase_1.JsonApiRecordBase {
    constructor(p) {
        super(p);
        this._nameIsMutated = false;
        this._valueIsMutated = false;
        this.jsonType = "com.salesstryke.entity.classes.record.notification.EmailNotificationSelectedData";
        this.name = null;
        this.value = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.record.notification.EmailNotificationSelectedData");
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsValueMutated() {
        return (this._valueIsMutated);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * name of the selectedData
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
     * value of the email
     * @returns string | null
     */
    getValue() {
        return (this.value);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsValueMutated(_valueIsMutated) {
        this._valueIsMutated = _valueIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * name of the selectedData
* ```
     */
    setName(name) {
        if (this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    /**
* ```
     * value of the email
* ```
     */
    setValue(value) {
        if (this.value !== value) {
            this._setIsValueMutated(true);
        }
        this.value = value;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._setCurrentState2({
            attributeName: 'value',
            value: this.value
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        this._setInitializedState2({
            attributeName: 'value',
            value: this.value
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.value);
        return (hashCode_n);
    }
}
exports.EmailNotificationSelectedData = EmailNotificationSelectedData;
EmailNotificationSelectedData.metaDataForEmailNotificationSelectedData = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.record.notification.EmailNotificationSelectedData", EmailNotificationSelectedData);
