// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/record/notification/EmailNotificationCustomerData.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailNotificationCustomerData = void 0;
const ClassFor_1 = require("../../../../data/ClassFor");
const JsonApiRecordBase_1 = require("../../../../jsonApi/JsonApiRecordBase");
const HashCodeUtil_1 = require("../../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._addressIsMutated = "_addressIsMutated";
        this._emailIsMutated = "_emailIsMutated";
        this._firstNameIsMutated = "_firstNameIsMutated";
        this._inServiceAreaIsMutated = "_inServiceAreaIsMutated";
        this._lastNameIsMutated = "_lastNameIsMutated";
        this._noteIsMutated = "_noteIsMutated";
        this._phoneIsMutated = "_phoneIsMutated";
        this.address = "address";
        this.email = "email";
        this.firstName = "firstName";
        this.inServiceArea = "inServiceArea";
        this.jsonType = "jsonType";
        this.lastName = "lastName";
        this.note = "note";
        this.phone = "phone";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._addressIsMutated = "_addressIsMutated";
        this._emailIsMutated = "_emailIsMutated";
        this._firstNameIsMutated = "_firstNameIsMutated";
        this._inServiceAreaIsMutated = "_inServiceAreaIsMutated";
        this._lastNameIsMutated = "_lastNameIsMutated";
        this._noteIsMutated = "_noteIsMutated";
        this._phoneIsMutated = "_phoneIsMutated";
        this.address = "address";
        this.email = "email";
        this.firstName = "firstName";
        this.inServiceArea = "inServiceArea";
        this.jsonType = "jsonType";
        this.lastName = "lastName";
        this.note = "note";
        this.phone = "phone";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.record.notification.EmailNotificationCustomerData";
        this.mongDataModeName = "EmailNotificationCustomerData";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class EmailNotificationCustomerData extends JsonApiRecordBase_1.JsonApiRecordBase {
    constructor(p) {
        super(p);
        this._addressIsMutated = false;
        this._emailIsMutated = false;
        this._firstNameIsMutated = false;
        this._inServiceAreaIsMutated = false;
        this._lastNameIsMutated = false;
        this._noteIsMutated = false;
        this._phoneIsMutated = false;
        this.address = null;
        this.email = null;
        this.firstName = null;
        this.inServiceArea = null;
        this.jsonType = "com.salesstryke.entity.classes.record.notification.EmailNotificationCustomerData";
        this.lastName = null;
        this.note = null;
        this.phone = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.record.notification.EmailNotificationCustomerData");
    }
    /**
     *
     * @returns boolean
     */
    _getIsAddressMutated() {
        return (this._addressIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsEmailMutated() {
        return (this._emailIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsFirstNameMutated() {
        return (this._firstNameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsInServiceAreaMutated() {
        return (this._inServiceAreaIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsLastNameMutated() {
        return (this._lastNameIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNoteMutated() {
        return (this._noteIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsPhoneMutated() {
        return (this._phoneIsMutated);
    }
    /**
     * address of the customer data
     * @returns string | null
     */
    getAddress() {
        return (this.address);
    }
    /**
     * email of the customer data
     * @returns string | null
     */
    getEmail() {
        return (this.email);
    }
    /**
     * firstName of the customer data
     * @returns string | null
     */
    getFirstName() {
        return (this.firstName);
    }
    /**
     * inServiceArea of the customer data
     * @returns string | null
     */
    getInServiceArea() {
        return (this.inServiceArea);
    }
    /**
     *
     * @returns string | null
     */
    getJsonType() {
        return (this.jsonType);
    }
    /**
     * lastName of the customer data
     * @returns string | null
     */
    getLastName() {
        return (this.lastName);
    }
    /**
     * note of the customer data
     * @returns string | null
     */
    getNote() {
        return (this.note);
    }
    /**
     * phone of the customer data
     * @returns string | null
     */
    getPhone() {
        return (this.phone);
    }
    /**
* ```
     *
* ```
     */
    _setIsAddressMutated(_addressIsMutated) {
        this._addressIsMutated = _addressIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsEmailMutated(_emailIsMutated) {
        this._emailIsMutated = _emailIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsFirstNameMutated(_firstNameIsMutated) {
        this._firstNameIsMutated = _firstNameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsInServiceAreaMutated(_inServiceAreaIsMutated) {
        this._inServiceAreaIsMutated = _inServiceAreaIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsLastNameMutated(_lastNameIsMutated) {
        this._lastNameIsMutated = _lastNameIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNoteMutated(_noteIsMutated) {
        this._noteIsMutated = _noteIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsPhoneMutated(_phoneIsMutated) {
        this._phoneIsMutated = _phoneIsMutated;
        return (this);
    }
    /**
* ```
     * address of the customer data
* ```
     */
    setAddress(address) {
        if (this.address !== address) {
            this._setIsAddressMutated(true);
        }
        this.address = address;
        return (this);
    }
    /**
* ```
     * email of the customer data
* ```
     */
    setEmail(email) {
        if (this.email !== email) {
            this._setIsEmailMutated(true);
        }
        this.email = email;
        return (this);
    }
    /**
* ```
     * firstName of the customer data
* ```
     */
    setFirstName(firstName) {
        if (this.firstName !== firstName) {
            this._setIsFirstNameMutated(true);
        }
        this.firstName = firstName;
        return (this);
    }
    /**
* ```
     * inServiceArea of the customer data
* ```
     */
    setInServiceArea(inServiceArea) {
        if (this.inServiceArea !== inServiceArea) {
            this._setIsInServiceAreaMutated(true);
        }
        this.inServiceArea = inServiceArea;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    setJsonType(jsonType) {
        this.jsonType = jsonType;
        return (this);
    }
    /**
* ```
     * lastName of the customer data
* ```
     */
    setLastName(lastName) {
        if (this.lastName !== lastName) {
            this._setIsLastNameMutated(true);
        }
        this.lastName = lastName;
        return (this);
    }
    /**
* ```
     * note of the customer data
* ```
     */
    setNote(note) {
        if (this.note !== note) {
            this._setIsNoteMutated(true);
        }
        this.note = note;
        return (this);
    }
    /**
* ```
     * phone of the customer data
* ```
     */
    setPhone(phone) {
        if (this.phone !== phone) {
            this._setIsPhoneMutated(true);
        }
        this.phone = phone;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: 'address',
            value: this.address
        });
        this._setCurrentState2({
            attributeName: 'email',
            value: this.email
        });
        this._setCurrentState2({
            attributeName: 'firstName',
            value: this.firstName
        });
        this._setCurrentState2({
            attributeName: 'inServiceArea',
            value: this.inServiceArea
        });
        this._setCurrentState2({
            attributeName: 'lastName',
            value: this.lastName
        });
        this._setCurrentState2({
            attributeName: 'note',
            value: this.note
        });
        this._setCurrentState2({
            attributeName: 'phone',
            value: this.phone
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: 'address',
            value: this.address
        });
        this._setInitializedState2({
            attributeName: 'email',
            value: this.email
        });
        this._setInitializedState2({
            attributeName: 'firstName',
            value: this.firstName
        });
        this._setInitializedState2({
            attributeName: 'inServiceArea',
            value: this.inServiceArea
        });
        this._setInitializedState2({
            attributeName: 'lastName',
            value: this.lastName
        });
        this._setInitializedState2({
            attributeName: 'note',
            value: this.note
        });
        this._setInitializedState2({
            attributeName: 'phone',
            value: this.phone
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.address);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.email);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.firstName);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.inServiceArea);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.lastName);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.note);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.phone);
        return (hashCode_n);
    }
}
exports.EmailNotificationCustomerData = EmailNotificationCustomerData;
EmailNotificationCustomerData.metaDataForEmailNotificationCustomerData = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.record.notification.EmailNotificationCustomerData", EmailNotificationCustomerData);
