// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/record/pandadocapi/Options.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Options = void 0;
const ClassFor_1 = require("../../../../data/ClassFor");
const JsonApiRecordBase_1 = require("../../../../jsonApi/JsonApiRecordBase");
const HashCodeUtil_1 = require("../../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._optionalIsMutated = "_optionalIsMutated";
        this._optionalSelectedIsMutated = "_optionalSelectedIsMutated";
        this.optional = "optional";
        this.optionalSelected = "optionalSelected";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._optionalIsMutated = "_optionalIsMutated";
        this._optionalSelectedIsMutated = "_optionalSelectedIsMutated";
        this.optional = "optional";
        this.optionalSelected = "optionalSelected";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.record.pandadocapi.Options";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class Options extends JsonApiRecordBase_1.JsonApiRecordBase {
    constructor(p) {
        super(p);
        this._optionalIsMutated = false;
        this._optionalSelectedIsMutated = false;
        this.optional = false;
        this.optionalSelected = false;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.record.pandadocapi.Options");
    }
    /**
     *
     * @returns boolean
     */
    _getIsOptionalMutated() {
        return (this._optionalIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsOptionalSelectedMutated() {
        return (this._optionalSelectedIsMutated);
    }
    /**
     * Identifies if the selected item is optional.
     * @returns boolean
     */
    getOptional() {
        return (this.optional);
    }
    /**
     * Identifies if the optional item is selected.
     * @returns boolean
     */
    getOptionalSelected() {
        return (this.optionalSelected);
    }
    /**
* ```
     *
* ```
     */
    _setIsOptionalMutated(_optionalIsMutated) {
        this._optionalIsMutated = _optionalIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsOptionalSelectedMutated(_optionalSelectedIsMutated) {
        this._optionalSelectedIsMutated = _optionalSelectedIsMutated;
        return (this);
    }
    /**
* ```
     * Identifies if the selected item is optional.
* ```
     */
    setOptional(optional) {
        if (this.optional !== optional) {
            this._setIsOptionalMutated(true);
        }
        this.optional = optional;
        return (this);
    }
    /**
* ```
     * Identifies if the optional item is selected.
* ```
     */
    setOptionalSelected(optionalSelected) {
        if (this.optionalSelected !== optionalSelected) {
            this._setIsOptionalSelectedMutated(true);
        }
        this.optionalSelected = optionalSelected;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: 'optional',
            value: this.optional
        });
        this._setCurrentState2({
            attributeName: 'optionalSelected',
            value: this.optionalSelected
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: 'optional',
            value: this.optional
        });
        this._setInitializedState2({
            attributeName: 'optionalSelected',
            value: this.optionalSelected
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.optional);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.optionalSelected);
        return (hashCode_n);
    }
}
exports.Options = Options;
Options.metaDataForOptions = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.record.pandadocapi.Options", Options);
