// @salesstryke/mobile-api/src/com/salesstryke/entity/classes/record/pandadocapi/Template.js
"use strict";
/**
 * This file is auto generated DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Template = void 0;
const ClassFor_1 = require("../../../../data/ClassFor");
const JsonApiRecordBase_1 = require("../../../../jsonApi/JsonApiRecordBase");
const HashCodeUtil_1 = require("../../../../utils/HashCodeUtil");
class EntityAttributes {
    constructor() {
        this._idIsMutated = "_idIsMutated";
        this._nameIsMutated = "_nameIsMutated";
        this.id = "id";
        this.name = "name";
    }
}
class MongoVirtual {
}
class MongoAttributes {
    constructor() {
        this.virtual = new MongoVirtual();
        this._idIsMutated = "_idIsMutated";
        this._nameIsMutated = "_nameIsMutated";
        this.id = "id";
        this.name = "name";
    }
}
class MetaData {
    constructor() {
        this.namespace = "com.salesstryke.entity.classes.record.pandadocapi.Template";
        this.entityAttributes = new EntityAttributes();
        this.mongoAttributes = new MongoAttributes();
    }
}
class Template extends JsonApiRecordBase_1.JsonApiRecordBase {
    constructor(p) {
        super(p);
        this._idIsMutated = false;
        this._nameIsMutated = false;
        this.id = null;
        this.name = null;
    }
    /**
     * Sets all isMutated flags to be false
     * @returns
     */
    _clearIsMutated() {
        return (this);
    }
    getClassName() {
        return ("com.salesstryke.entity.classes.record.pandadocapi.Template");
    }
    /**
     *
     * @returns boolean
     */
    _getIsIdMutated() {
        return (this._idIsMutated);
    }
    /**
     *
     * @returns boolean
     */
    _getIsNameMutated() {
        return (this._nameIsMutated);
    }
    /**
     * The id of the template.
     * @returns string | null
     */
    getId() {
        return (this.id);
    }
    /**
     * The name of the template.
     * @returns string | null
     */
    getName() {
        return (this.name);
    }
    /**
* ```
     *
* ```
     */
    _setIsIdMutated(_idIsMutated) {
        this._idIsMutated = _idIsMutated;
        return (this);
    }
    /**
* ```
     *
* ```
     */
    _setIsNameMutated(_nameIsMutated) {
        this._nameIsMutated = _nameIsMutated;
        return (this);
    }
    /**
* ```
     * The id of the template.
* ```
     */
    setId(id) {
        if (this.id !== id) {
            this._setIsIdMutated(true);
        }
        this.id = id;
        return (this);
    }
    /**
* ```
     * The name of the template.
* ```
     */
    setName(name) {
        if (this.name !== name) {
            this._setIsNameMutated(true);
        }
        this.name = name;
        return (this);
    }
    _getIsMutated() {
        let isMutated = false;
        this._setCurrentState2({
            attributeName: 'id',
            value: this.id
        });
        this._setCurrentState2({
            attributeName: 'name',
            value: this.name
        });
        this._currentHashcode2 = this.getHashCode();
        isMutated = super._getIsMutated();
        return (isMutated);
    }
    _trackMutations() {
        if (this._isTrackingingMutations) {
            return (this);
        }
        this._setInitializedState2({
            attributeName: 'id',
            value: this.id
        });
        this._setInitializedState2({
            attributeName: 'name',
            value: this.name
        });
        super._trackMutations();
        if (!this._hasInitializedHashCode2) {
            this._initializedHashcode2 = this.getHashCode();
        }
        this._hasInitializedHashCode2 = true;
        return (this);
    }
    getHashCode() {
        let hashCode_n = -1;
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.id);
        hashCode_n += HashCodeUtil_1.HashCodeUtil.getHashCodeFromValue(this.name);
        return (hashCode_n);
    }
}
exports.Template = Template;
Template.metaDataForTemplate = new MetaData();
ClassFor_1.ClassFor.getInstance().addClass("com.salesstryke.entity.classes.record.pandadocapi.Template", Template);
