// @salesstryke/mobile-api/src/com/salesstryke/module/ModuleBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModuleBase = void 0;
const EntityManager_1 = require("../data/EntityManager");
class ModuleBase {
    constructor() {
        this.initializeModuleData_ary = new Array();
        this.moduleName_s = "";
        this.entityModelBaseAry = new Array();
    }
    addEntityModel(entityModelBase) {
        this.entityModelBaseAry.push(entityModelBase);
        return (this);
    }
    addInitializeModuleData(jsonApiPersistableBase) {
        this.initializeModuleData_ary.push(jsonApiPersistableBase);
        return (this);
    }
    getEntityModels() {
        return (this.entityModelBaseAry);
    }
    getInitializeModuleData() {
        return (this.initializeModuleData_ary);
    }
    getModuleName() {
        return (this.moduleName_s);
    }
    initializeModule() {
        let size_i = this.entityModelBaseAry.length;
        for (let index_i = 0; index_i < size_i; index_i++) {
            let entityModelBase = this.entityModelBaseAry[index_i];
            entityModelBase.doPack();
        }
        this.getEntityModels().forEach((entityModelBase) => {
            EntityManager_1.EntityManager.getInstance().addEntityModel(entityModelBase.getEntityModel());
        });
    }
    setModuleName(moduleName_s) {
        this.moduleName_s = moduleName_s;
        return (this);
    }
}
exports.ModuleBase = ModuleBase;
