// @salesstryke/mobile-api/src/com/salesstryke/utils/log/LogManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogManager = void 0;
const env_manager_1 = require("@salesstryke/env-manager");
const Logger_1 = require("./Logger");
const LogLevel_1 = require("./LogLevel");
class LogManager {
    constructor() {
        this.loggers = new Map();
        this.logLevel = LogLevel_1.Level.ALL;
        const logLevelName = (0, env_manager_1.envManager)('LOG_LEVEL').asString();
        const logLevel = LogLevel_1.Level.getLevelByName({ name: logLevelName });
        if (logLevelName && logLevel) {
            this.logLevel = logLevel;
        }
    }
    static getInstance() {
        return LogManager.INSTANCE;
    }
    getLogger(namespace) {
        if (this.loggers.has(namespace)) {
            return this.loggers.get(namespace);
        }
        const newLogger = new Logger_1.Logger(namespace, this.logLevel);
        this.loggers.set(namespace, newLogger);
        return newLogger;
    }
    getLoggingLevel() {
        return this.logLevel;
    }
    setLogLevel({ level, applyToAllLoggers }) {
        this.logLevel = level;
        if (applyToAllLoggers) {
            for (const logger of this.loggers.values()) {
                logger.setLogLevel(this.logLevel);
            }
        }
        return this;
    }
}
LogManager.INSTANCE = new LogManager();
const logManager = LogManager.getInstance();
exports.LogManager = logManager;
