// @salesstryke/mobile-api/src/com/salesstryke/jsonApi/JsonApiErrorMeta.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JsonApiErrorMeta = void 0;
const ErrorBase_1 = require("../lang/ErrorBase");
const JsonApiErrorMetaErrorType_1 = require("./JsonApiErrorMetaErrorType");
const NAMESPACE = "com.salesstryke.jsonapi.JsonApiErrorMeta";
class JsonAttributes {
    constructor() {
        this.endPoint = "endPoint";
        this.errorType = "errorType";
        this.i18nKey = "i18nKey";
        this.methodName = "methodName";
        this.securityAttributeUUID = "securityAttributeUUID";
        this.userUUID = "userUUID";
        this.uiUUID = "uiUUID";
    }
}
class MetaData {
    constructor() {
        this.jsonAttribute = new JsonAttributes();
        this.namespace = NAMESPACE;
    }
}
class JsonApiErrorMeta extends ErrorBase_1.ErrorBase {
    constructor() {
        super("");
        this.endPointUrl_s = '';
        this.errorType = JsonApiErrorMetaErrorType_1.JsonApiErrorMetaErrorType.NONE;
        this.i18nKey_s = "";
        //private methodName_s:               string                      = "";
        //private namespace_s:                string                      = "";
        this.securityAttributeUUID_s = "";
        this.uiUUID_s = "";
        this.userUUID_s = "";
        Object.setPrototypeOf(this, new.target.prototype);
    }
    getEndPoint() {
        return (this.endPointUrl_s);
    }
    getErrorType() {
        return (this.errorType);
    }
    getI18nKey() {
        return (this.i18nKey_s);
    }
    // getMethodName(): string {
    //     return(this.methodName_s);
    // }
    // getNamespace(): string {
    //     return(this.namespace_s);
    // }
    getSecurityAttributeUUID() {
        return (this.securityAttributeUUID_s);
    }
    getUserUUID() {
        return (this.userUUID_s);
    }
    getUIUUID() {
        return (this.uiUUID_s);
    }
    setEndPoint(endPoint_s) {
        this.endPointUrl_s = endPoint_s;
        return (this);
    }
    setErrorType(errorType) {
        this.errorType = errorType;
        return (this);
    }
    setI18nKey(i18nKey_s) {
        this.i18nKey_s = i18nKey_s;
        return (this);
    }
    setMethodName(methodName_s) {
        this.methodName_s = methodName_s;
        return (this);
    }
    setNamespace(namespace_s) {
        this.namespace_s = namespace_s;
        return (this);
    }
    setSecurityAttibuteUUID(securityAttributeUUID_s) {
        this.securityAttributeUUID_s = securityAttributeUUID_s;
        return (this);
    }
    setUIUUID(uiUUID_s) {
        this.uiUUID_s = uiUUID_s;
        return (this);
    }
    setUserUUID(userUUID_s) {
        this.userUUID_s = userUUID_s;
        return (this);
    }
}
exports.JsonApiErrorMeta = JsonApiErrorMeta;
JsonApiErrorMeta.metaData = new MetaData();
