// @salesstryke/mobile-api/src/com/salesstryke/jsonApi/JsonApiError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JsonApiError = void 0;
class JsonApiError {
    constructor() {
        /**
         * The error code
         */
        this.code_s = null;
        this.detail_s = null;
        /**
         * The ID, use UUID if available.
         */
        this.id_s = null;
        this.jsonApiSource_c = null;
        this.meta = null;
        this.status_s = null;
        /**
         * Short description that SHOULD NOT change from occurance to occurance.
         */
        this.title_s = "";
    }
    /**
     * @return the code
     */
    getCode() {
        return (this.code_s);
    }
    /**
     * @return the detail
     */
    getDetail() {
        return (this.detail_s);
    }
    /**
     * The ID, use UUID if available.
     * @return the id
     */
    getId() {
        return (this.id_s);
    }
    getMeta() {
        return (this.meta);
    }
    getMessage() {
        let message_s = '';
        if (this.getDetail() !== null) {
            message_s += `code:${this.getDetail()}`;
        }
        if (this.getCode() !== null) {
            message_s += `code:${this.getCode()}`;
        }
        message_s += `detail:${this.getDetail() || ""} code:${this.getCode() || ""}`;
        return (message_s);
    }
    /**
     * @return the source
     */
    getSource() {
        return (this.jsonApiSource_c);
    }
    /**
     * @return the status
     */
    getStatus() {
        return (this.status_s);
    }
    /**
     * Short description that SHOULD NOT change from occurance to occurance.
     * @returns
     */
    getTitle() {
        return (this.title_s);
    }
    /**
     * @param code_s the code to set, use UUID if available
     */
    setCode(code_s) {
        this.code_s = code_s;
        return (this);
    }
    /**
     * @param detail_s the detail to set
     */
    setDetail(detail_s) {
        this.detail_s = detail_s;
        return (this);
    }
    /**
     * The ID, use UUID if available.
     * @param id_s the id to set
     */
    setId(id_s) {
        this.id_s = id_s;
        return (this);
    }
    setMeta(meta) {
        this.meta = meta;
        return (this);
    }
    /**
     * @param jsonApiSource_p the source to set
     */
    setSource(jsonApiSource_p) {
        this.jsonApiSource_c = jsonApiSource_p;
        return (this);
    }
    /**
     * @param status_s the status to set
     */
    setStatus(status_s) {
        this.status_s = status_s;
        return (this);
    }
    /**
     * Short description that SHOULD NOT change from occurance to occurance.
     * @param title_s
     * @returns
     */
    setTitle(title_s) {
        this.title_s = title_s;
        return (this);
    }
}
exports.JsonApiError = JsonApiError;
