// @salesstryke/mobile-api/src/com/salesstryke/jsonApi/ResourceObject.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceObject = void 0;
const object_1 = require("../utils/object");
class ResourceObject {
    constructor(resourceObject) {
        this.attributes = null;
        this.id = "";
        this.type = "";
        if ((0, object_1.Exists)(resourceObject)) {
            if ((0, object_1.Exists)(resourceObject.attributes)) {
                this.attributes = resourceObject.attributes;
            }
            if ((0, object_1.Exists)(resourceObject.id)) {
                this.id = resourceObject.id;
            }
            if ((0, object_1.Exists)(resourceObject.type)) {
                this.type = resourceObject.type;
            }
        }
    }
    getAttributes() {
        return (this.attributes);
    }
    getId() {
        return (this.id);
    }
    getType() {
        return (this.type);
    }
    setAttribute(attributeBase) {
        this.attributes = attributeBase;
        return (this);
    }
    setId(id) {
        this.id = id;
        return (this);
    }
    setType(type) {
        this.type = type;
        return (this);
    }
}
exports.ResourceObject = ResourceObject;
